<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="James Howe" />


<title>Behavioral analysis: genes</title>

<script src="site_libs/header-attrs-2.25/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="site_libs/pagedtable-1.1/js/pagedtable.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">CorticalAmygdala_pipelines</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jamesrhowe/CorticalAmygdala_pipelines">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Behavioral analysis: genes</h1>
<h4 class="author">James Howe</h4>
<h4 class="date">21 January, 2024</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-01-21
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong>
<code>CorticalAmygdala_pipelines/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R
Markdown file created these results, you’ll want to first commit it to
the Git repo. If you’re still working on the analysis, you can ignore
this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220906code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20220906)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220906code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220906)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomjamesrhoweCorticalAmygdalapipelinestreefba58a9f1c88c60b5879aeba11fcd3fbb4823bfdtargetblankfba58a9a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/jamesrhowe/CorticalAmygdala_pipelines/tree/fba58a9f1c88c60b5879aeba11fcd3fbb4823bfd" target="_blank">fba58a9</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomjamesrhoweCorticalAmygdalapipelinestreefba58a9f1c88c60b5879aeba11fcd3fbb4823bfdtargetblankfba58a9a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/jamesrhowe/CorticalAmygdala_pipelines/tree/fba58a9f1c88c60b5879aeba11fcd3fbb4823bfd" target="_blank">fba58a9</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    data/behavior/
    Ignored:    data/projections/
    Ignored:    data/rnascope/
    Ignored:    data/sequencing/
    Ignored:    output/.DS_Store
    Ignored:    renv/library/
    Ignored:    renv/staging/

Untracked files:
    Untracked:  analysis/08A-merge_rd1_ATAC.Rmd
    Untracked:  analysis/behavior_03-projections.Rmd
    Untracked:  analysis/behavior_04-projections_silencing_tmt.Rmd
    Untracked:  analysis/behavior_05-projections_silencing_2PE.Rmd
    Untracked:  analysis/behavior_06-genes_silencing_tmt.Rmd
    Untracked:  analysis/behavior_07-epm_genes_silence.Rmd
    Untracked:  analysis/behavior_09-epm_topography.Rmd
    Untracked:  analysis/behavior_10-epm_projections.Rmd
    Untracked:  analysis/behavior_12-openfield_projections.Rmd
    Untracked:  analysis/behavior_13-openfield_genes_silence.Rmd
    Untracked:  analysis/behavior_14-openfield_projections_silence.Rmd
    Untracked:  analysis/behavior_15-epm_genes_silence.Rmd
    Untracked:  analysis/behavior_16-epm_projections_silence.Rmd
    Untracked:  analysis/projections.Rmd
    Untracked:  analysis/sequencing-02B-initial_classification_rd1_RNA.Rmd
    Untracked:  analysis/sequencing-02C-reprocess_Neuron_rd1_RNA.Rmd
    Untracked:  analysis/sequencing-02D-Neuron_classification_rd1_RNA.Rmd
    Untracked:  analysis/sequencing-02E-reprocess_Glut_rd1_RNA.Rmd
    Untracked:  analysis/sequencing-02F-reprocess_GABA_rd1_RNA.Rmd
    Untracked:  analysis/sequencing-02G-Glut_classification_rd1_RNA.Rmd
    Untracked:  analysis/sequencing-03A-merge_filtered_RNA.Rmd
    Untracked:  analysis/sequencing-03B-initial_classification_RNA.Rmd
    Untracked:  analysis/sequencing-04-initial_deg.Rmd
    Untracked:  analysis/sequencing-04A-reprocess_Neuron_RNA.Rmd
    Untracked:  analysis/sequencing-04B-reprocess_Nonneuron_RNA.Rmd
    Untracked:  analysis/sequencing-04C-Neuron_classification_RNA.Rmd
    Untracked:  analysis/sequencing-04D-Nonneuron_classification_RNA.Rmd
    Untracked:  analysis/sequencing-05C-reprocess_Astro_RNA.Rmd
    Untracked:  analysis/sequencing-05L-Glut_classification_RNA.Rmd
    Untracked:  analysis/sequencing-05M-GABA_classification_RNA.Rmd
    Untracked:  analysis/sequencing-05N-Astro_classification_RNA.Rmd
    Untracked:  analysis/sequencing-05O-Macro_classification_RNA.Rmd
    Untracked:  analysis/sequencing-05P-NFOL_classification_RNA.Rmd
    Untracked:  analysis/sequencing-05Q-MOL_classification_RNA.Rmd
    Untracked:  analysis/sequencing-05R-ABC_classification_RNA.Rmd
    Untracked:  analysis/sequencing-05S-VLMC_classification_RNA.Rmd
    Untracked:  analysis/sequencing-05T-Immune_classification+labeling_RNA.Rmd
    Untracked:  analysis/sequencing-05X-reprocess_Astro.Rmd
    Untracked:  analysis/sequencing-05X-reprocess_GABAergic.Rmd
    Untracked:  analysis/sequencing-05X-reprocess_Glutamatergic.Rmd
    Untracked:  analysis/sequencing-05X-reprocess_Microglia.Rmd
    Untracked:  analysis/sequencing-05X-reprocess_Neuron.Rmd
    Untracked:  analysis/sequencing-05X-reprocess_Nonneuron.Rmd
    Untracked:  analysis/sequencing-05X-reprocess_OLG.Rmd
    Untracked:  analysis/sequencing-05X-reprocess_OPC.Rmd
    Untracked:  analysis/sequencing-05X-reprocess_VLMC.Rmd
    Untracked:  analysis/sequencing-06A-merged_classification_RNA.Rmd
    Untracked:  analysis/sequencing-06B-annotate_GABAergic.Rmd
    Untracked:  analysis/sequencing-06B-merged_deg_discovery_RNA.Rmd
    Untracked:  analysis/sequencing-07A-preprocess_aplCoA-1_P1_ATAC.Rmd
    Untracked:  analysis/sequencing-07B-preprocess_aplCoA-1_P2_ATAC.Rmd
    Untracked:  analysis/sequencing-07C-preprocess_aplCoA-2_ATAC.Rmd
    Untracked:  analysis/sequencing-07D-preprocess_pplCoA-1_P1_ATAC.Rmd
    Untracked:  analysis/sequencing-07E-preprocess_pplCoA-1_P2_ATAC.Rmd
    Untracked:  analysis/sequencing-07F-preprocess_pplCoA-2_ATAC.Rmd
    Untracked:  analysis/spatial-01A-preprocess_039-A_S18.Rmd
    Untracked:  analysis/spatial-01B-preprocess_039-B_S19.Rmd
    Untracked:  analysis/spatial-01C-preprocess_042-A_S22.Rmd
    Untracked:  analysis/spatial-01D-preprocess_061-D_S1.Rmd
    Untracked:  analysis/spatial-01E-preprocess_062-B_S2.Rmd
    Untracked:  analysis/spatial-01F-preprocess_063-A_S4.Rmd
    Untracked:  analysis/spatial-01G-preprocess_063-B_S5.Rmd
    Untracked:  analysis/spatial-01H-preprocess_063-D_S7.Rmd
    Untracked:  analysis/spatial-01I-preprocess_105-A_S8.Rmd
    Untracked:  analysis/spatial-01J-preprocess_105-B_S9.Rmd
    Untracked:  analysis/spatial-01K-preprocess_106-B_S11.Rmd
    Untracked:  analysis/spatial-01L-preprocess_106-C_S12.Rmd
    Untracked:  analysis/spatial-01M-preprocess_107-A_S15.Rmd
    Untracked:  analysis/spatial-01N-preprocess_107-B_S16.Rmd
    Untracked:  analysis/spatial-01O-preprocess_108-A_S13.Rmd
    Untracked:  analysis/spatial-01P-preprocess_122-A_S23.Rmd
    Untracked:  analysis/spatial-01Q-preprocess_122-B_S24.Rmd
    Untracked:  analysis/spatial-01R-preprocess_122-C_S25.Rmd
    Untracked:  analysis/spatial-01S-preprocess_123-A_S30.Rmd
    Untracked:  analysis/spatial-01T-preprocess_063-C_S6.Rmd
    Untracked:  analysis/spatial-01U-preprocess_108-C_S14.Rmd
    Untracked:  analysis/spatial-02-merge.Rmd
    Untracked:  analysis/spatial-03-classify.Rmd
    Untracked:  output/sequencing/
    Untracked:  output/spatial/

Unstaged changes:
    Modified:   analysis/sequencing-02A-merge_rd1_RNA.Rmd
    Modified:   analysis/sequencing-03-initial_classification.Rmd
    Modified:   analysis/sequencing-05A-reprocess_Glut_RNA.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with
<code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>This is a notebook that processes 4-quad data and outputs analyses
and figures for publication.</p>
<pre class="r"><code># set up paths
gene_id &lt;- params$dataset
gene_path &lt;- params$path
start_min &lt;- params$start_min
end_min &lt;- params$end_min

source(&quot;code/behavior-refs.R&quot;)</code></pre>
<pre><code>Loading required package: MASS</code></pre>
<pre><code>Loading required package: tidyverse</code></pre>
<pre><code>── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.3     ✔ readr     2.1.4
✔ forcats   1.0.0     ✔ stringr   1.5.0
✔ ggplot2   3.4.3     ✔ tibble    3.2.1
✔ lubridate 1.9.3     ✔ tidyr     1.3.0
✔ purrr     1.0.2     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
✖ dplyr::select() masks MASS::select()
ℹ Use the conflicted package (&lt;http://conflicted.r-lib.org/&gt;) to force all conflicts to become errors
Loading required package: colorRamps

Loading required package: RColorBrewer

Loading required package: spatstat

Loading required package: spatstat.data

Loading required package: spatstat.geom

spatstat.geom 3.2-5


Attaching package: &#39;spatstat.geom&#39;


The following object is masked from &#39;package:MASS&#39;:

    area


Loading required package: spatstat.random

spatstat.random 3.1-6

Loading required package: spatstat.explore

Loading required package: nlme


Attaching package: &#39;nlme&#39;


The following object is masked from &#39;package:dplyr&#39;:

    collapse


spatstat.explore 3.2-3

Loading required package: spatstat.model

Loading required package: rpart

spatstat.model 3.2-6

Loading required package: spatstat.linnet

spatstat.linnet 3.1-1


spatstat 3.0-6 
For an introduction to spatstat, type &#39;beginner&#39; 


Loading required package: viridis

Loading required package: viridisLite

Loading required package: cowplot


Attaching package: &#39;cowplot&#39;


The following object is masked from &#39;package:lubridate&#39;:

    stamp


Loading required package: grid


Attaching package: &#39;grid&#39;


The following object is masked from &#39;package:spatstat.geom&#39;:

    as.mask


Loading required package: gridExtra


Attaching package: &#39;gridExtra&#39;


The following object is masked from &#39;package:dplyr&#39;:

    combine


Loading required package: colorspace


Attaching package: &#39;colorspace&#39;


The following object is masked from &#39;package:spatstat.geom&#39;:

    coords


Loading required package: scales


Attaching package: &#39;scales&#39;


The following object is masked from &#39;package:viridis&#39;:

    viridis_pal


The following object is masked from &#39;package:spatstat.geom&#39;:

    rescale


The following object is masked from &#39;package:purrr&#39;:

    discard


The following object is masked from &#39;package:readr&#39;:

    col_factor



Attaching package: &#39;zoo&#39;


The following objects are masked from &#39;package:base&#39;:

    as.Date, as.Date.numeric


Loading required package: carData


Attaching package: &#39;car&#39;


The following object is masked from &#39;package:spatstat.model&#39;:

    bc


The following object is masked from &#39;package:spatstat.geom&#39;:

    ellipse


The following object is masked from &#39;package:dplyr&#39;:

    recode


The following object is masked from &#39;package:purrr&#39;:

    some


Loading required package: mvtnorm

Loading required package: survival

Loading required package: TH.data


Attaching package: &#39;TH.data&#39;


The following object is masked from &#39;package:MASS&#39;:

    geyser



Attaching package: &#39;rstatix&#39;


The following object is masked from &#39;package:MASS&#39;:

    select


The following object is masked from &#39;package:stats&#39;:

    filter



Attaching package: &#39;ggpubr&#39;


The following object is masked from &#39;package:cowplot&#39;:

    get_legend


The following objects are masked from &#39;package:spatstat.geom&#39;:

    border, rotate</code></pre>
<pre class="r"><code>source(&quot;code/behavior-misc_processes.R&quot;)
source(&quot;code/behavior-plot_functions.R&quot;)

gene_groups &lt;- unlist(strsplit(Sys.glob(paste0(gene_path, &quot;SAL/*&quot;)),
                                        paste0(gene_path, &quot;SAL/&quot;)))[c(FALSE, TRUE)]</code></pre>
</div>
<div id="load-the-data" class="section level1">
<h1>Load the data</h1>
<p>Names have to be given manually in most cases. Also for now copy into
console because it does not load due to odd workspace issues. This loads
all files in a directory and puts it into the correct format
automatically.</p>
<pre class="r"><code># set up the object with pointers
data_gene_sal &lt;- vector(mode = &quot;list&quot;, length = 4)
data_gene_sal &lt;- lapply(gene_groups, 
                        function(x) lapply(Sys.glob(paste0(gene_path, &quot;SAL/&quot;, x, &quot;/*&quot;)),
                                           function(y) process_series_epm(y, &quot;silence&quot;)))

data_gene_cno &lt;- vector(mode = &quot;list&quot;, length = 4)
data_gene_cno &lt;- lapply(gene_groups, 
                        function(x) lapply(Sys.glob(paste0(gene_path, &quot;CNO/&quot;, x, &quot;/*&quot;)),
                                           function(y) process_series_epm(y, &quot;silence&quot;)))

# add names to each
names(data_gene_sal) &lt;- Sys.glob(paste0(gene_path, &quot;SAL/&quot;, &quot;*&quot;))
names(data_gene_sal) &lt;- unlist(strsplit(names(data_gene_sal), 
                                                  paste0(gene_path, &quot;SAL/&quot;)))[c(FALSE, TRUE)]

names(data_gene_cno) &lt;- Sys.glob(paste0(gene_path, &quot;CNO/&quot;, &quot;*&quot;))
names(data_gene_cno) &lt;- unlist(strsplit(names(data_gene_cno), 
                                                  paste0(gene_path, &quot;CNO/&quot;)))[c(FALSE, TRUE)]

for (i in 1:length(gene_groups)){
  data_gene_sal &lt;- add_custom_names(data_gene_sal, paste0(gene_path, &quot;SAL/&quot;), 
                                              gene_groups[i], gene_labels[i])
  data_gene_cno &lt;- add_custom_names(data_gene_cno, paste0(gene_path, &quot;CNO/&quot;), 
                                              gene_groups[i], gene_labels[i])
}

data_gene_sal &lt;- list(data_gene_sal$`VGlut1-control`, data_gene_sal$`VGlut1-Gi`, 
                                data_gene_sal$`VGlut2-control`, data_gene_sal$`VGlut2-Gi`)
names(data_gene_sal) &lt;- c(&quot;VGlut1-control&quot;, &quot;VGlut1-Gi&quot;, &quot;VGlut2-control&quot;, &quot;VGlut2-Gi&quot;)

data_gene_cno &lt;- list(data_gene_cno$`VGlut1-control`, data_gene_cno$`VGlut1-Gi`, 
                                data_gene_cno$`VGlut2-control`, data_gene_cno$`VGlut2-Gi`)
names(data_gene_cno) &lt;- c(&quot;VGlut1-control&quot;, &quot;VGlut1-Gi&quot;, &quot;VGlut2-control&quot;, &quot;VGlut2-Gi&quot;)</code></pre>
</div>
<div id="open-time-based-metrics" class="section level1">
<h1>Open Time-based metrics</h1>
<div id="summary-statistics" class="section level3">
<h3>Summary statistics</h3>
<pre class="r"><code>gene_sal_array &lt;- transform_epm_opentime(data_gene_sal, &quot;full&quot;, &quot;Region&quot;, 1, 2400)

gene_cno_array &lt;- transform_epm_opentime(data_gene_cno, &quot;full&quot;, &quot;Region&quot;, 1, 2400)

opentime_group_gene &lt;- group_array(data_gene_sal, gene_sal_array, gene_cno_array, &quot;OpenTime&quot;)</code></pre>
<div id="vglut1" class="section level4">
<h4>VGlut1</h4>
<pre class="r"><code>VGlut1_group_gene &lt;- opentime_group_gene[opentime_group_gene$group 
                                     %in% c(&quot;VGlut1 Gi&quot;, &quot;VGlut1 mCh&quot;),] %&gt;%
  gather(key = &quot;time&quot;, value = &quot;OpenTime&quot;, baseline_OpenTime, treatment_OpenTime) %&gt;%
  convert_as_factor(group, time)
levels(VGlut1_group_gene$time) &lt;- c(&quot;SAL&quot;, &quot;CNO&quot;)

VGlut1_group_gene$group &lt;- factor(VGlut1_group_gene$group, 
                                  levels = c(&quot;VGlut1 mCh&quot;, &quot;VGlut1 Gi&quot;))
levels(VGlut1_group_gene$group) &lt;- c(&quot;mCh&quot;, &quot;Gi&quot;)

VGlut1_group_gene %&gt;%
  group_by(group, time) %&gt;%
  get_summary_stats(OpenTime, type = &quot;mean_se&quot;)</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["group"],"name":[1],"type":["fct"],"align":["left"]},{"label":["time"],"name":[2],"type":["fct"],"align":["left"]},{"label":["variable"],"name":[3],"type":["fct"],"align":["left"]},{"label":["n"],"name":[4],"type":["dbl"],"align":["right"]},{"label":["mean"],"name":[5],"type":["dbl"],"align":["right"]},{"label":["se"],"name":[6],"type":["dbl"],"align":["right"]}],"data":[{"1":"mCh","2":"SAL","3":"OpenTime","4":"12","5":"0.201","6":"0.043"},{"1":"mCh","2":"CNO","3":"OpenTime","4":"12","5":"0.154","6":"0.033"},{"1":"Gi","2":"SAL","3":"OpenTime","4":"11","5":"0.226","6":"0.035"},{"1":"Gi","2":"CNO","3":"OpenTime","4":"11","5":"0.163","6":"0.032"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
<div id="vglut2" class="section level4">
<h4>VGlut2</h4>
<pre class="r"><code>VGlut2_group_gene &lt;- opentime_group_gene[opentime_group_gene$group 
                                     %in% c(&quot;VGlut2 Gi&quot;, &quot;VGlut2 mCh&quot;),] %&gt;%
  gather(key = &quot;time&quot;, value = &quot;OpenTime&quot;, baseline_OpenTime, treatment_OpenTime) %&gt;%
  convert_as_factor(group, time)
levels(VGlut2_group_gene$time) &lt;- c(&quot;SAL&quot;, &quot;CNO&quot;)

VGlut2_group_gene$group &lt;- factor(VGlut2_group_gene$group, 
                                  levels = c(&quot;VGlut2 mCh&quot;, &quot;VGlut2 Gi&quot;))
levels(VGlut2_group_gene$group) &lt;- c(&quot;mCh&quot;, &quot;Gi&quot;)

VGlut2_group_gene %&gt;%
  group_by(group, time) %&gt;%
  get_summary_stats(OpenTime, type = &quot;mean_se&quot;)</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["group"],"name":[1],"type":["fct"],"align":["left"]},{"label":["time"],"name":[2],"type":["fct"],"align":["left"]},{"label":["variable"],"name":[3],"type":["fct"],"align":["left"]},{"label":["n"],"name":[4],"type":["dbl"],"align":["right"]},{"label":["mean"],"name":[5],"type":["dbl"],"align":["right"]},{"label":["se"],"name":[6],"type":["dbl"],"align":["right"]}],"data":[{"1":"mCh","2":"SAL","3":"OpenTime","4":"14","5":"0.144","6":"0.034"},{"1":"mCh","2":"CNO","3":"OpenTime","4":"14","5":"0.170","6":"0.031"},{"1":"Gi","2":"SAL","3":"OpenTime","4":"13","5":"0.192","6":"0.033"},{"1":"Gi","2":"CNO","3":"OpenTime","4":"13","5":"0.147","6":"0.038"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
<div id="stats" class="section level3 tabset">
<h3 class="tabset">Stats</h3>
<div id="vglut1-1" class="section level4 tabset">
<h4 class="tabset">VGlut1</h4>
<div id="anova" class="section level5">
<h5>ANOVA</h5>
<pre class="r"><code>display_anova(VGlut1_group_gene, &quot;OpenTime&quot;)</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":[""],"name":["_rn_"],"type":[""],"align":["left"]},{"label":["Effect"],"name":[1],"type":["chr"],"align":["left"]},{"label":["DFn"],"name":[2],"type":["dbl"],"align":["right"]},{"label":["DFd"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["F"],"name":[4],"type":["dbl"],"align":["right"]},{"label":["p"],"name":[5],"type":["dbl"],"align":["right"]},{"label":["p<.05"],"name":[6],"type":["chr"],"align":["left"]},{"label":["ges"],"name":[7],"type":["dbl"],"align":["right"]}],"data":[{"1":"group","2":"1","3":"21","4":"0.288","5":"0.597","6":"","7":"0.005","_rn_":"1"},{"1":"time","2":"1","3":"21","4":"1.840","5":"0.189","6":"","7":"0.051","_rn_":"2"},{"1":"group:time","2":"1","3":"21","4":"0.038","5":"0.848","6":"","7":"0.001","_rn_":"3"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
<div id="pairwise-comparisons" class="section level5">
<h5>Pairwise comparisons</h5>
<pre class="r"><code>pairwise_comp_anova(VGlut1_group_gene, &quot;OpenTime&quot;)</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":[""],"name":["_rn_"],"type":[""],"align":["left"]},{"label":["group"],"name":[1],"type":["fct"],"align":["left"]},{"label":["term"],"name":[2],"type":["chr"],"align":["left"]},{"label":["group1"],"name":[3],"type":["chr"],"align":["left"]},{"label":["group2"],"name":[4],"type":["chr"],"align":["left"]},{"label":["null.value"],"name":[5],"type":["dbl"],"align":["right"]},{"label":["estimate"],"name":[6],"type":["dbl"],"align":["right"]},{"label":["conf.low"],"name":[7],"type":["dbl"],"align":["right"]},{"label":["conf.high"],"name":[8],"type":["dbl"],"align":["right"]},{"label":["p.adj"],"name":[9],"type":["dbl"],"align":["right"]},{"label":["p.adj.signif"],"name":[10],"type":["chr"],"align":["left"]}],"data":[{"1":"mCh","2":"time","3":"SAL","4":"CNO","5":"0","6":"-0.04687500","7":"-0.1593976","8":"0.06564760","9":"0.397","10":"ns","_rn_":"1"},{"1":"Gi","2":"time","3":"SAL","4":"CNO","5":"0","6":"-0.06257576","7":"-0.1617182","8":"0.03656671","9":"0.203","10":"ns","_rn_":"2"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
<div id="t-test-each" class="section level5">
<h5>T-test each</h5>
<pre class="r"><code># SAL
t.test(x = VGlut1_group_gene$OpenTime[VGlut1_group_gene$group == &quot;mCh&quot; &amp; 
                                              VGlut1_group_gene$time == &quot;SAL&quot;], 
       y = VGlut1_group_gene$OpenTime[VGlut1_group_gene$group == &quot;Gi&quot; &amp; 
                                         VGlut1_group_gene$time == &quot;SAL&quot;])</code></pre>
<pre><code>
    Welch Two Sample t-test

data:  VGlut1_group_gene$OpenTime[VGlut1_group_gene$group == &quot;mCh&quot; &amp; VGlut1_group_gene$time == &quot;SAL&quot;] and VGlut1_group_gene$OpenTime[VGlut1_group_gene$group == &quot;Gi&quot; &amp; VGlut1_group_gene$time == &quot;SAL&quot;]
t = -0.45025, df = 20.46, p-value = 0.6573
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -0.14031744  0.09043739
sample estimates:
mean of x mean of y 
0.2005903 0.2255303 </code></pre>
<pre class="r"><code># CNO
t.test(x = VGlut1_group_gene$OpenTime[VGlut1_group_gene$group == &quot;mCh&quot; &amp; 
                                              VGlut1_group_gene$time == &quot;CNO&quot;], 
       y = VGlut1_group_gene$OpenTime[VGlut1_group_gene$group == &quot;Gi&quot; &amp; 
                                         VGlut1_group_gene$time == &quot;CNO&quot;])</code></pre>
<pre><code>
    Welch Two Sample t-test

data:  VGlut1_group_gene$OpenTime[VGlut1_group_gene$group == &quot;mCh&quot; &amp; VGlut1_group_gene$time == &quot;CNO&quot;] and VGlut1_group_gene$OpenTime[VGlut1_group_gene$group == &quot;Gi&quot; &amp; VGlut1_group_gene$time == &quot;CNO&quot;]
t = -0.19998, df = 20.983, p-value = 0.8434
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -0.10532647  0.08684794
sample estimates:
mean of x mean of y 
0.1537153 0.1629545 </code></pre>
</div>
</div>
<div id="vglut2-1" class="section level4 tabset">
<h4 class="tabset">VGlut2</h4>
<div id="anova-1" class="section level5">
<h5>ANOVA</h5>
<pre class="r"><code>display_anova(VGlut2_group_gene, &quot;OpenTime&quot;)</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":[""],"name":["_rn_"],"type":[""],"align":["left"]},{"label":["Effect"],"name":[1],"type":["chr"],"align":["left"]},{"label":["DFn"],"name":[2],"type":["dbl"],"align":["right"]},{"label":["DFd"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["F"],"name":[4],"type":["dbl"],"align":["right"]},{"label":["p"],"name":[5],"type":["dbl"],"align":["right"]},{"label":["p<.05"],"name":[6],"type":["chr"],"align":["left"]},{"label":["ges"],"name":[7],"type":["dbl"],"align":["right"]}],"data":[{"1":"group","2":"1","3":"25","4":"0.211","5":"0.650","6":"","7":"0.003","_rn_":"1"},{"1":"time","2":"1","3":"25","4":"0.053","5":"0.821","6":"","7":"0.001","_rn_":"2"},{"1":"group:time","2":"1","3":"25","4":"0.826","5":"0.372","6":"","7":"0.022","_rn_":"3"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
<div id="pairwise-comparisons-1" class="section level5">
<h5>Pairwise comparisons</h5>
<pre class="r"><code>pairwise_comp_anova(VGlut2_group_gene, &quot;OpenTime&quot;)</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":[""],"name":["_rn_"],"type":[""],"align":["left"]},{"label":["group"],"name":[1],"type":["fct"],"align":["left"]},{"label":["term"],"name":[2],"type":["chr"],"align":["left"]},{"label":["group1"],"name":[3],"type":["chr"],"align":["left"]},{"label":["group2"],"name":[4],"type":["chr"],"align":["left"]},{"label":["null.value"],"name":[5],"type":["dbl"],"align":["right"]},{"label":["estimate"],"name":[6],"type":["dbl"],"align":["right"]},{"label":["conf.low"],"name":[7],"type":["dbl"],"align":["right"]},{"label":["conf.high"],"name":[8],"type":["dbl"],"align":["right"]},{"label":["p.adj"],"name":[9],"type":["dbl"],"align":["right"]},{"label":["p.adj.signif"],"name":[10],"type":["chr"],"align":["left"]}],"data":[{"1":"mCh","2":"time","3":"SAL","4":"CNO","5":"0","6":"0.02660714","7":"-0.06748156","8":"0.12069585","9":"0.566","10":"ns","_rn_":"1"},{"1":"Gi","2":"time","3":"SAL","4":"CNO","5":"0","6":"-0.04455128","7":"-0.14817744","8":"0.05907487","9":"0.384","10":"ns","_rn_":"2"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
<div id="t-test-each-1" class="section level5">
<h5>T-test each</h5>
<pre class="r"><code># SAL
t.test(x = VGlut2_group_gene$OpenTime[VGlut2_group_gene$group == &quot;mCh&quot; &amp; 
                                              VGlut2_group_gene$time == &quot;SAL&quot;], 
       y = VGlut2_group_gene$OpenTime[VGlut2_group_gene$group == &quot;Gi&quot; &amp; 
                                         VGlut2_group_gene$time == &quot;SAL&quot;])</code></pre>
<pre><code>
    Welch Two Sample t-test

data:  VGlut2_group_gene$OpenTime[VGlut2_group_gene$group == &quot;mCh&quot; &amp; VGlut2_group_gene$time == &quot;SAL&quot;] and VGlut2_group_gene$OpenTime[VGlut2_group_gene$group == &quot;Gi&quot; &amp; VGlut2_group_gene$time == &quot;SAL&quot;]
t = -1.0249, df = 25, p-value = 0.3152
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -0.14534507  0.04875167
sample estimates:
mean of x mean of y 
0.1436905 0.1919872 </code></pre>
<pre class="r"><code># CNO
t.test(x = VGlut2_group_gene$OpenTime[VGlut2_group_gene$group == &quot;mCh&quot; &amp; 
                                              VGlut2_group_gene$time == &quot;CNO&quot;], 
       y = VGlut2_group_gene$OpenTime[VGlut2_group_gene$group == &quot;Gi&quot; &amp; 
                                         VGlut2_group_gene$time == &quot;CNO&quot;])</code></pre>
<pre><code>
    Welch Two Sample t-test

data:  VGlut2_group_gene$OpenTime[VGlut2_group_gene$group == &quot;mCh&quot; &amp; VGlut2_group_gene$time == &quot;CNO&quot;] and VGlut2_group_gene$OpenTime[VGlut2_group_gene$group == &quot;Gi&quot; &amp; VGlut2_group_gene$time == &quot;CNO&quot;]
t = 0.46708, df = 23.513, p-value = 0.6447
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -0.07826886  0.12399230
sample estimates:
mean of x mean of y 
0.1702976 0.1474359 </code></pre>
</div>
</div>
</div>
<div id="plots" class="section level3 tabset">
<h3 class="tabset">Plots</h3>
<div id="vglut1-2" class="section level4">
<h4>VGlut1</h4>
<pre class="r"><code>bg_twogroup(VGlut1_group_gene, &quot;Proportion Time in Open Arms&quot;, 
            c(control_color, VGlut1_color), topo_corner_axes, &quot;white&quot;)</code></pre>
<embed src="figure/behavior_07-epm_genes_silence.Rmd/11-VGlut1_open_prepost-1.pdf" width="230.4" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="vglut2-2" class="section level4">
<h4>VGlut2</h4>
<pre class="r"><code>bg_twogroup(VGlut2_group_gene, &quot;Proportion Time in Open Arms&quot;, 
            c(control_color, VGlut2_color), topo_corner_axes, &quot;white&quot;)</code></pre>
<embed src="figure/behavior_07-epm_genes_silence.Rmd/12-VGlut2_open_prepost-1.pdf" width="230.4" style="display: block; margin: auto;" type="application/pdf" />
</div>
</div>
</div>
<div id="open-entry-based-metrics" class="section level1">
<h1>Open Entry-based metrics</h1>
<div id="summary-statistics-1" class="section level3">
<h3>Summary statistics</h3>
<pre class="r"><code>gene_sal_array &lt;- transform_epm_openentry(data_gene_sal, &quot;full&quot;, &quot;OpenEntry&quot;, 1, 2400)

gene_cno_array &lt;- transform_epm_openentry(data_gene_cno, &quot;full&quot;, &quot;OpenEntry&quot;, 1, 2400)

openentry_group_gene &lt;- group_array(data_gene_sal, gene_sal_array, gene_cno_array, &quot;OpenEntry&quot;)</code></pre>
<div id="vglut1-3" class="section level4">
<h4>VGlut1</h4>
<pre class="r"><code>VGlut1_group_gene &lt;- openentry_group_gene[openentry_group_gene$group 
                                     %in% c(&quot;VGlut1 Gi&quot;, &quot;VGlut1 mCh&quot;),] %&gt;%
  gather(key = &quot;time&quot;, value = &quot;OpenEntry&quot;, baseline_OpenEntry, treatment_OpenEntry) %&gt;%
  convert_as_factor(group, time)
levels(VGlut1_group_gene$time) &lt;- c(&quot;SAL&quot;, &quot;CNO&quot;)

VGlut1_group_gene$group &lt;- factor(VGlut1_group_gene$group, 
                                  levels = c(&quot;VGlut1 mCh&quot;, &quot;VGlut1 Gi&quot;))
levels(VGlut1_group_gene$group) &lt;- c(&quot;mCh&quot;, &quot;Gi&quot;)

VGlut1_group_gene %&gt;%
  group_by(group, time) %&gt;%
  get_summary_stats(OpenEntry, type = &quot;mean_se&quot;)</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["group"],"name":[1],"type":["fct"],"align":["left"]},{"label":["time"],"name":[2],"type":["fct"],"align":["left"]},{"label":["variable"],"name":[3],"type":["fct"],"align":["left"]},{"label":["n"],"name":[4],"type":["dbl"],"align":["right"]},{"label":["mean"],"name":[5],"type":["dbl"],"align":["right"]},{"label":["se"],"name":[6],"type":["dbl"],"align":["right"]}],"data":[{"1":"mCh","2":"SAL","3":"OpenEntry","4":"12","5":"14.833","6":"2.471"},{"1":"mCh","2":"CNO","3":"OpenEntry","4":"12","5":"16.500","6":"3.283"},{"1":"Gi","2":"SAL","3":"OpenEntry","4":"11","5":"15.091","6":"2.201"},{"1":"Gi","2":"CNO","3":"OpenEntry","4":"11","5":"16.000","6":"1.991"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
<div id="vglut2-3" class="section level4">
<h4>VGlut2</h4>
<pre class="r"><code>VGlut2_group_gene &lt;- openentry_group_gene[openentry_group_gene$group 
                                     %in% c(&quot;VGlut2 Gi&quot;, &quot;VGlut2 mCh&quot;),] %&gt;%
  gather(key = &quot;time&quot;, value = &quot;OpenEntry&quot;, baseline_OpenEntry, treatment_OpenEntry) %&gt;%
  convert_as_factor(group, time)
levels(VGlut2_group_gene$time) &lt;- c(&quot;SAL&quot;, &quot;CNO&quot;)

VGlut2_group_gene$group &lt;- factor(VGlut2_group_gene$group, 
                                  levels = c(&quot;VGlut2 mCh&quot;, &quot;VGlut2 Gi&quot;))
levels(VGlut2_group_gene$group) &lt;- c(&quot;mCh&quot;, &quot;Gi&quot;)

VGlut2_group_gene %&gt;%
  group_by(group, time) %&gt;%
  get_summary_stats(OpenEntry, type = &quot;mean_se&quot;)</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["group"],"name":[1],"type":["fct"],"align":["left"]},{"label":["time"],"name":[2],"type":["fct"],"align":["left"]},{"label":["variable"],"name":[3],"type":["fct"],"align":["left"]},{"label":["n"],"name":[4],"type":["dbl"],"align":["right"]},{"label":["mean"],"name":[5],"type":["dbl"],"align":["right"]},{"label":["se"],"name":[6],"type":["dbl"],"align":["right"]}],"data":[{"1":"mCh","2":"SAL","3":"OpenEntry","4":"14","5":"13.929","6":"3.111"},{"1":"mCh","2":"CNO","3":"OpenEntry","4":"14","5":"15.571","6":"3.596"},{"1":"Gi","2":"SAL","3":"OpenEntry","4":"13","5":"17.308","6":"2.398"},{"1":"Gi","2":"CNO","3":"OpenEntry","4":"13","5":"12.692","6":"2.558"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
<div id="stats-1" class="section level3 tabset">
<h3 class="tabset">Stats</h3>
<div id="vglut1-4" class="section level4 tabset">
<h4 class="tabset">VGlut1</h4>
<div id="anova-2" class="section level5">
<h5>ANOVA</h5>
<pre class="r"><code>display_anova(VGlut1_group_gene, &quot;OpenEntry&quot;)</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":[""],"name":["_rn_"],"type":[""],"align":["left"]},{"label":["Effect"],"name":[1],"type":["chr"],"align":["left"]},{"label":["DFn"],"name":[2],"type":["dbl"],"align":["right"]},{"label":["DFd"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["F"],"name":[4],"type":["dbl"],"align":["right"]},{"label":["p"],"name":[5],"type":["dbl"],"align":["right"]},{"label":["p<.05"],"name":[6],"type":["chr"],"align":["left"]},{"label":["ges"],"name":[7],"type":["dbl"],"align":["right"]}],"data":[{"1":"group","2":"1","3":"21","4":"0.003","5":"0.957","6":"","7":"5.27e-05","_rn_":"1"},{"1":"time","2":"1","3":"21","4":"0.198","5":"0.661","6":"","7":"6.00e-03","_rn_":"2"},{"1":"group:time","2":"1","3":"21","4":"0.017","5":"0.897","6":"","7":"5.15e-04","_rn_":"3"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
<div id="pairwise-comparisons-2" class="section level5">
<h5>Pairwise comparisons</h5>
<pre class="r"><code>pairwise_comp_anova(VGlut1_group_gene, &quot;OpenEntry&quot;)</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":[""],"name":["_rn_"],"type":[""],"align":["left"]},{"label":["group"],"name":[1],"type":["fct"],"align":["left"]},{"label":["term"],"name":[2],"type":["chr"],"align":["left"]},{"label":["group1"],"name":[3],"type":["chr"],"align":["left"]},{"label":["group2"],"name":[4],"type":["chr"],"align":["left"]},{"label":["null.value"],"name":[5],"type":["dbl"],"align":["right"]},{"label":["estimate"],"name":[6],"type":["dbl"],"align":["right"]},{"label":["conf.low"],"name":[7],"type":["dbl"],"align":["right"]},{"label":["conf.high"],"name":[8],"type":["dbl"],"align":["right"]},{"label":["p.adj"],"name":[9],"type":["dbl"],"align":["right"]},{"label":["p.adj.signif"],"name":[10],"type":["chr"],"align":["left"]}],"data":[{"1":"mCh","2":"time","3":"SAL","4":"CNO","5":"0","6":"1.6666667","7":"-6.854866","8":"10.188200","9":"0.689","10":"ns","_rn_":"1"},{"1":"Gi","2":"time","3":"SAL","4":"CNO","5":"0","6":"0.9090909","7":"-5.281781","8":"7.099963","9":"0.763","10":"ns","_rn_":"2"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
<div id="t-test-each-2" class="section level5">
<h5>T-test each</h5>
<pre class="r"><code># SAL
t.test(x = VGlut1_group_gene$OpenEntry[VGlut1_group_gene$group == &quot;mCh&quot; &amp; 
                                              VGlut1_group_gene$time == &quot;SAL&quot;], 
       y = VGlut1_group_gene$OpenEntry[VGlut1_group_gene$group == &quot;Gi&quot; &amp; 
                                         VGlut1_group_gene$time == &quot;SAL&quot;])</code></pre>
<pre><code>
    Welch Two Sample t-test

data:  VGlut1_group_gene$OpenEntry[VGlut1_group_gene$group == &quot;mCh&quot; &amp; VGlut1_group_gene$time == &quot;SAL&quot;] and VGlut1_group_gene$OpenEntry[VGlut1_group_gene$group == &quot;Gi&quot; &amp; VGlut1_group_gene$time == &quot;SAL&quot;]
t = -0.077846, df = 20.905, p-value = 0.9387
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -7.140510  6.625359
sample estimates:
mean of x mean of y 
 14.83333  15.09091 </code></pre>
<pre class="r"><code># CNO
t.test(x = VGlut1_group_gene$OpenEntry[VGlut1_group_gene$group == &quot;mCh&quot; &amp; 
                                              VGlut1_group_gene$time == &quot;CNO&quot;], 
       y = VGlut1_group_gene$OpenEntry[VGlut1_group_gene$group == &quot;Gi&quot; &amp; 
                                         VGlut1_group_gene$time == &quot;CNO&quot;])</code></pre>
<pre><code>
    Welch Two Sample t-test

data:  VGlut1_group_gene$OpenEntry[VGlut1_group_gene$group == &quot;mCh&quot; &amp; VGlut1_group_gene$time == &quot;CNO&quot;] and VGlut1_group_gene$OpenEntry[VGlut1_group_gene$group == &quot;Gi&quot; &amp; VGlut1_group_gene$time == &quot;CNO&quot;]
t = 0.13022, df = 17.912, p-value = 0.8978
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -7.56992  8.56992
sample estimates:
mean of x mean of y 
     16.5      16.0 </code></pre>
</div>
</div>
<div id="vglut2-4" class="section level4 tabset">
<h4 class="tabset">VGlut2</h4>
<div id="anova-3" class="section level5">
<h5>ANOVA</h5>
<pre class="r"><code>display_anova(VGlut2_group_gene, &quot;OpenEntry&quot;)</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":[""],"name":["_rn_"],"type":[""],"align":["left"]},{"label":["Effect"],"name":[1],"type":["chr"],"align":["left"]},{"label":["DFn"],"name":[2],"type":["dbl"],"align":["right"]},{"label":["DFd"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["F"],"name":[4],"type":["dbl"],"align":["right"]},{"label":["p"],"name":[5],"type":["dbl"],"align":["right"]},{"label":["p<.05"],"name":[6],"type":["chr"],"align":["left"]},{"label":["ges"],"name":[7],"type":["dbl"],"align":["right"]}],"data":[{"1":"group","2":"1","3":"25","4":"0.007","5":"0.935","6":"","7":"0.00014","_rn_":"1"},{"1":"time","2":"1","3":"25","4":"0.251","5":"0.621","6":"","7":"0.00500","_rn_":"2"},{"1":"group:time","2":"1","3":"25","4":"1.113","5":"0.302","6":"","7":"0.02100","_rn_":"3"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
<div id="pairwise-comparisons-3" class="section level5">
<h5>Pairwise comparisons</h5>
<pre class="r"><code>pairwise_comp_anova(VGlut2_group_gene, &quot;OpenEntry&quot;)</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":[""],"name":["_rn_"],"type":[""],"align":["left"]},{"label":["group"],"name":[1],"type":["fct"],"align":["left"]},{"label":["term"],"name":[2],"type":["chr"],"align":["left"]},{"label":["group1"],"name":[3],"type":["chr"],"align":["left"]},{"label":["group2"],"name":[4],"type":["chr"],"align":["left"]},{"label":["null.value"],"name":[5],"type":["dbl"],"align":["right"]},{"label":["estimate"],"name":[6],"type":["dbl"],"align":["right"]},{"label":["conf.low"],"name":[7],"type":["dbl"],"align":["right"]},{"label":["conf.high"],"name":[8],"type":["dbl"],"align":["right"]},{"label":["p.adj"],"name":[9],"type":["dbl"],"align":["right"]},{"label":["p.adj.signif"],"name":[10],"type":["chr"],"align":["left"]}],"data":[{"1":"mCh","2":"time","3":"SAL","4":"CNO","5":"0","6":"1.642857","7":"-8.130393","8":"11.416107","9":"0.732","10":"ns","_rn_":"1"},{"1":"Gi","2":"time","3":"SAL","4":"CNO","5":"0","6":"-4.615385","7":"-11.851376","8":"2.620607","9":"0.200","10":"ns","_rn_":"2"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
<div id="t-test-each-3" class="section level5">
<h5>T-test each</h5>
<pre class="r"><code># SAL
t.test(x = VGlut2_group_gene$OpenEntry[VGlut2_group_gene$group == &quot;mCh&quot; &amp; 
                                              VGlut2_group_gene$time == &quot;SAL&quot;], 
       y = VGlut2_group_gene$OpenEntry[VGlut2_group_gene$group == &quot;Gi&quot; &amp; 
                                         VGlut2_group_gene$time == &quot;SAL&quot;])</code></pre>
<pre><code>
    Welch Two Sample t-test

data:  VGlut2_group_gene$OpenEntry[VGlut2_group_gene$group == &quot;mCh&quot; &amp; VGlut2_group_gene$time == &quot;SAL&quot;] and VGlut2_group_gene$OpenEntry[VGlut2_group_gene$group == &quot;Gi&quot; &amp; VGlut2_group_gene$time == &quot;SAL&quot;]
t = -0.86041, df = 23.898, p-value = 0.3981
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -11.486577   4.728335
sample estimates:
mean of x mean of y 
 13.92857  17.30769 </code></pre>
<pre class="r"><code># CNO
t.test(x = VGlut2_group_gene$OpenEntry[VGlut2_group_gene$group == &quot;mCh&quot; &amp; 
                                              VGlut2_group_gene$time == &quot;CNO&quot;], 
       y = VGlut2_group_gene$OpenEntry[VGlut2_group_gene$group == &quot;Gi&quot; &amp; 
                                         VGlut2_group_gene$time == &quot;CNO&quot;])</code></pre>
<pre><code>
    Welch Two Sample t-test

data:  VGlut2_group_gene$OpenEntry[VGlut2_group_gene$group == &quot;mCh&quot; &amp; VGlut2_group_gene$time == &quot;CNO&quot;] and VGlut2_group_gene$OpenEntry[VGlut2_group_gene$group == &quot;Gi&quot; &amp; VGlut2_group_gene$time == &quot;CNO&quot;]
t = 0.65242, df = 23.082, p-value = 0.5206
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -6.248003 12.006245
sample estimates:
mean of x mean of y 
 15.57143  12.69231 </code></pre>
</div>
</div>
</div>
<div id="plots-1" class="section level3 tabset">
<h3 class="tabset">Plots</h3>
<div id="vglut1-5" class="section level4">
<h4>VGlut1</h4>
<pre class="r"><code>bg_twogroup(VGlut1_group_gene, &quot;Open Arm Entries&quot;, 
            c(control_color, VGlut1_color), epm_axes_entry_genes, &quot;white&quot;)</code></pre>
<embed src="figure/behavior_07-epm_genes_silence.Rmd/22-VGlut1_Corner_prepost-1.pdf" width="230.4" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="vglut2-5" class="section level4">
<h4>VGlut2</h4>
<pre class="r"><code>bg_twogroup(VGlut2_group_gene, &quot;Open Arm Entries&quot;, 
            c(control_color, VGlut2_color), epm_axes_entry_genes, &quot;white&quot;)</code></pre>
<embed src="figure/behavior_07-epm_genes_silence.Rmd/23-VGlut2_Corner_prepost-1.pdf" width="230.4" style="display: block; margin: auto;" type="application/pdf" />
</div>
</div>
</div>
<div id="distance-based-metrics" class="section level1">
<h1>Distance-based metrics</h1>
<div id="summary-statistics-2" class="section level3">
<h3>Summary statistics</h3>
<pre class="r"><code>gene_sal_array &lt;- transform_epm_openentry(data_gene_sal, &quot;full&quot;, &quot;Distance&quot;, 1, 2400)

gene_cno_array &lt;- transform_epm_openentry(data_gene_cno, &quot;full&quot;, &quot;Distance&quot;, 1, 2400)

distance_group_gene &lt;- group_array(data_gene_sal, gene_sal_array, gene_cno_array, &quot;Distance&quot;)</code></pre>
<div id="vglut1-6" class="section level4">
<h4>VGlut1</h4>
<pre class="r"><code>VGlut1_group_gene &lt;- distance_group_gene[distance_group_gene$group 
                                     %in% c(&quot;VGlut1 Gi&quot;, &quot;VGlut1 mCh&quot;),] %&gt;%
  gather(key = &quot;time&quot;, value = &quot;Distance&quot;, baseline_Distance, treatment_Distance) %&gt;%
  convert_as_factor(group, time)
levels(VGlut1_group_gene$time) &lt;- c(&quot;SAL&quot;, &quot;CNO&quot;)

VGlut1_group_gene$group &lt;- factor(VGlut1_group_gene$group, 
                                  levels = c(&quot;VGlut1 mCh&quot;, &quot;VGlut1 Gi&quot;))
levels(VGlut1_group_gene$group) &lt;- c(&quot;mCh&quot;, &quot;Gi&quot;)

VGlut1_group_gene$Distance &lt;- VGlut1_group_gene$Distance / 100

VGlut1_group_gene %&gt;%
  group_by(group, time) %&gt;%
  get_summary_stats(Distance, type = &quot;mean_se&quot;)</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["group"],"name":[1],"type":["fct"],"align":["left"]},{"label":["time"],"name":[2],"type":["fct"],"align":["left"]},{"label":["variable"],"name":[3],"type":["fct"],"align":["left"]},{"label":["n"],"name":[4],"type":["dbl"],"align":["right"]},{"label":["mean"],"name":[5],"type":["dbl"],"align":["right"]},{"label":["se"],"name":[6],"type":["dbl"],"align":["right"]}],"data":[{"1":"mCh","2":"SAL","3":"Distance","4":"12","5":"23.576","6":"2.081"},{"1":"mCh","2":"CNO","3":"Distance","4":"12","5":"22.773","6":"2.721"},{"1":"Gi","2":"SAL","3":"Distance","4":"11","5":"22.281","6":"2.168"},{"1":"Gi","2":"CNO","3":"Distance","4":"11","5":"22.791","6":"1.426"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
<div id="vglut2-6" class="section level4">
<h4>VGlut2</h4>
<pre class="r"><code>VGlut2_group_gene &lt;- distance_group_gene[distance_group_gene$group 
                                     %in% c(&quot;VGlut2 Gi&quot;, &quot;VGlut2 mCh&quot;),] %&gt;%
  gather(key = &quot;time&quot;, value = &quot;Distance&quot;, baseline_Distance, treatment_Distance) %&gt;%
  convert_as_factor(group, time)
levels(VGlut2_group_gene$time) &lt;- c(&quot;SAL&quot;, &quot;CNO&quot;)

VGlut2_group_gene$group &lt;- factor(VGlut2_group_gene$group, 
                                  levels = c(&quot;VGlut2 mCh&quot;, &quot;VGlut2 Gi&quot;))
levels(VGlut2_group_gene$group) &lt;- c(&quot;mCh&quot;, &quot;Gi&quot;)

VGlut2_group_gene$Distance &lt;- VGlut2_group_gene$Distance / 100

VGlut2_group_gene %&gt;%
  group_by(group, time) %&gt;%
  get_summary_stats(Distance, type = &quot;mean_se&quot;)</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["group"],"name":[1],"type":["fct"],"align":["left"]},{"label":["time"],"name":[2],"type":["fct"],"align":["left"]},{"label":["variable"],"name":[3],"type":["fct"],"align":["left"]},{"label":["n"],"name":[4],"type":["dbl"],"align":["right"]},{"label":["mean"],"name":[5],"type":["dbl"],"align":["right"]},{"label":["se"],"name":[6],"type":["dbl"],"align":["right"]}],"data":[{"1":"mCh","2":"SAL","3":"Distance","4":"14","5":"18.938","6":"2.389"},{"1":"mCh","2":"CNO","3":"Distance","4":"14","5":"21.083","6":"2.595"},{"1":"Gi","2":"SAL","3":"Distance","4":"13","5":"21.996","6":"1.795"},{"1":"Gi","2":"CNO","3":"Distance","4":"13","5":"18.893","6":"1.858"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
<div id="stats-2" class="section level3 tabset">
<h3 class="tabset">Stats</h3>
<div id="vglut1-7" class="section level4 tabset">
<h4 class="tabset">VGlut1</h4>
<div id="anova-4" class="section level5">
<h5>ANOVA</h5>
<pre class="r"><code>display_anova(VGlut1_group_gene, &quot;Distance&quot;)</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":[""],"name":["_rn_"],"type":[""],"align":["left"]},{"label":["Effect"],"name":[1],"type":["chr"],"align":["left"]},{"label":["DFn"],"name":[2],"type":["dbl"],"align":["right"]},{"label":["DFd"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["F"],"name":[4],"type":["dbl"],"align":["right"]},{"label":["p"],"name":[5],"type":["dbl"],"align":["right"]},{"label":["p<.05"],"name":[6],"type":["chr"],"align":["left"]},{"label":["ges"],"name":[7],"type":["dbl"],"align":["right"]}],"data":[{"1":"group","2":"1","3":"21","4":"0.077","5":"0.784","6":"","7":"0.002000","_rn_":"1"},{"1":"time","2":"1","3":"21","4":"0.005","5":"0.944","6":"","7":"0.000108","_rn_":"2"},{"1":"group:time","2":"1","3":"21","4":"0.102","5":"0.752","6":"","7":"0.002000","_rn_":"3"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
<div id="pairwise-comparisons-4" class="section level5">
<h5>Pairwise comparisons</h5>
<pre class="r"><code>pairwise_comp_anova(VGlut1_group_gene, &quot;Distance&quot;)</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":[""],"name":["_rn_"],"type":[""],"align":["left"]},{"label":["group"],"name":[1],"type":["fct"],"align":["left"]},{"label":["term"],"name":[2],"type":["chr"],"align":["left"]},{"label":["group1"],"name":[3],"type":["chr"],"align":["left"]},{"label":["group2"],"name":[4],"type":["chr"],"align":["left"]},{"label":["null.value"],"name":[5],"type":["dbl"],"align":["right"]},{"label":["estimate"],"name":[6],"type":["dbl"],"align":["right"]},{"label":["conf.low"],"name":[7],"type":["dbl"],"align":["right"]},{"label":["conf.high"],"name":[8],"type":["dbl"],"align":["right"]},{"label":["p.adj"],"name":[9],"type":["dbl"],"align":["right"]},{"label":["p.adj.signif"],"name":[10],"type":["chr"],"align":["left"]}],"data":[{"1":"mCh","2":"time","3":"SAL","4":"CNO","5":"0","6":"-0.8036672","7":"-7.908595","8":"6.301261","9":"0.817","10":"ns","_rn_":"1"},{"1":"Gi","2":"time","3":"SAL","4":"CNO","5":"0","6":"0.5099058","7":"-4.902388","8":"5.922200","9":"0.846","10":"ns","_rn_":"2"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
<div id="t-test-each-4" class="section level5">
<h5>T-test each</h5>
<pre class="r"><code># SAL
t.test(x = VGlut1_group_gene$Distance[VGlut1_group_gene$group == &quot;mCh&quot; &amp; 
                                              VGlut1_group_gene$time == &quot;SAL&quot;], 
       y = VGlut1_group_gene$Distance[VGlut1_group_gene$group == &quot;Gi&quot; &amp; 
                                         VGlut1_group_gene$time == &quot;SAL&quot;])</code></pre>
<pre><code>
    Welch Two Sample t-test

data:  VGlut1_group_gene$Distance[VGlut1_group_gene$group == &quot;mCh&quot; &amp; VGlut1_group_gene$time == &quot;SAL&quot;] and VGlut1_group_gene$Distance[VGlut1_group_gene$group == &quot;Gi&quot; &amp; VGlut1_group_gene$time == &quot;SAL&quot;]
t = 0.43106, df = 20.837, p-value = 0.6708
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -4.957100  7.547908
sample estimates:
mean of x mean of y 
 23.57632  22.28092 </code></pre>
<pre class="r"><code># CNO
t.test(x = VGlut1_group_gene$Distance[VGlut1_group_gene$group == &quot;mCh&quot; &amp; 
                                              VGlut1_group_gene$time == &quot;CNO&quot;], 
       y = VGlut1_group_gene$Distance[VGlut1_group_gene$group == &quot;Gi&quot; &amp; 
                                         VGlut1_group_gene$time == &quot;CNO&quot;])</code></pre>
<pre><code>
    Welch Two Sample t-test

data:  VGlut1_group_gene$Distance[VGlut1_group_gene$group == &quot;mCh&quot; &amp; VGlut1_group_gene$time == &quot;CNO&quot;] and VGlut1_group_gene$Distance[VGlut1_group_gene$group == &quot;Gi&quot; &amp; VGlut1_group_gene$time == &quot;CNO&quot;]
t = -0.005914, df = 16.5, p-value = 0.9954
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -6.514862  6.478524
sample estimates:
mean of x mean of y 
 22.77266  22.79082 </code></pre>
</div>
</div>
<div id="vglut2-7" class="section level4 tabset">
<h4 class="tabset">VGlut2</h4>
<div id="anova-5" class="section level5">
<h5>ANOVA</h5>
<pre class="r"><code>display_anova(VGlut2_group_gene, &quot;Distance&quot;)</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":[""],"name":["_rn_"],"type":[""],"align":["left"]},{"label":["Effect"],"name":[1],"type":["chr"],"align":["left"]},{"label":["DFn"],"name":[2],"type":["dbl"],"align":["right"]},{"label":["DFd"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["F"],"name":[4],"type":["dbl"],"align":["right"]},{"label":["p"],"name":[5],"type":["dbl"],"align":["right"]},{"label":["p<.05"],"name":[6],"type":["chr"],"align":["left"]},{"label":["ges"],"name":[7],"type":["dbl"],"align":["right"]}],"data":[{"1":"group","2":"1","3":"25","4":"0.027","5":"0.871","6":"","7":"0.000768","_rn_":"1"},{"1":"time","2":"1","3":"25","4":"0.083","5":"0.776","6":"","7":"0.000934","_rn_":"2"},{"1":"group:time","2":"1","3":"25","4":"2.483","5":"0.128","6":"","7":"0.027000","_rn_":"3"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
<div id="pairwise-comparisons-5" class="section level5">
<h5>Pairwise comparisons</h5>
<pre class="r"><code>pairwise_comp_anova(VGlut2_group_gene, &quot;Distance&quot;)</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":[""],"name":["_rn_"],"type":[""],"align":["left"]},{"label":["group"],"name":[1],"type":["fct"],"align":["left"]},{"label":["term"],"name":[2],"type":["chr"],"align":["left"]},{"label":["group1"],"name":[3],"type":["chr"],"align":["left"]},{"label":["group2"],"name":[4],"type":["chr"],"align":["left"]},{"label":["null.value"],"name":[5],"type":["dbl"],"align":["right"]},{"label":["estimate"],"name":[6],"type":["dbl"],"align":["right"]},{"label":["conf.low"],"name":[7],"type":["dbl"],"align":["right"]},{"label":["conf.high"],"name":[8],"type":["dbl"],"align":["right"]},{"label":["p.adj"],"name":[9],"type":["dbl"],"align":["right"]},{"label":["p.adj.signif"],"name":[10],"type":["chr"],"align":["left"]}],"data":[{"1":"mCh","2":"time","3":"SAL","4":"CNO","5":"0","6":"2.145474","7":"-5.105111","8":"9.396059","9":"0.548","10":"ns","_rn_":"1"},{"1":"Gi","2":"time","3":"SAL","4":"CNO","5":"0","6":"-3.103098","7":"-8.435321","8":"2.229125","9":"0.241","10":"ns","_rn_":"2"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
<div id="t-test-each-5" class="section level5">
<h5>T-test each</h5>
<pre class="r"><code># SAL
t.test(x = VGlut2_group_gene$Distance[VGlut2_group_gene$group == &quot;mCh&quot; &amp; 
                                              VGlut2_group_gene$time == &quot;SAL&quot;], 
       y = VGlut2_group_gene$Distance[VGlut2_group_gene$group == &quot;Gi&quot; &amp; 
                                         VGlut2_group_gene$time == &quot;SAL&quot;])</code></pre>
<pre><code>
    Welch Two Sample t-test

data:  VGlut2_group_gene$Distance[VGlut2_group_gene$group == &quot;mCh&quot; &amp; VGlut2_group_gene$time == &quot;SAL&quot;] and VGlut2_group_gene$Distance[VGlut2_group_gene$group == &quot;Gi&quot; &amp; VGlut2_group_gene$time == &quot;SAL&quot;]
t = -1.0234, df = 23.656, p-value = 0.3165
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -9.231397  3.114469
sample estimates:
mean of x mean of y 
 18.93787  21.99634 </code></pre>
<pre class="r"><code># CNO
t.test(x = VGlut2_group_gene$Distance[VGlut2_group_gene$group == &quot;mCh&quot; &amp; 
                                              VGlut2_group_gene$time == &quot;CNO&quot;], 
       y = VGlut2_group_gene$Distance[VGlut2_group_gene$group == &quot;Gi&quot; &amp; 
                                         VGlut2_group_gene$time == &quot;CNO&quot;])</code></pre>
<pre><code>
    Welch Two Sample t-test

data:  VGlut2_group_gene$Distance[VGlut2_group_gene$group == &quot;mCh&quot; &amp; VGlut2_group_gene$time == &quot;CNO&quot;] and VGlut2_group_gene$Distance[VGlut2_group_gene$group == &quot;Gi&quot; &amp; VGlut2_group_gene$time == &quot;CNO&quot;]
t = 0.68624, df = 23.152, p-value = 0.4994
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -4.409495  8.789711
sample estimates:
mean of x mean of y 
 21.08335  18.89324 </code></pre>
</div>
</div>
</div>
<div id="plots-2" class="section level3 tabset">
<h3 class="tabset">Plots</h3>
<div id="vglut1-8" class="section level4">
<h4>VGlut1</h4>
<pre class="r"><code>bg_twogroup(VGlut1_group_gene, &quot;Distance (m)&quot;, 
            c(control_color, VGlut1_color), openfield_axes_dist_silence_epm, &quot;white&quot;)</code></pre>
<embed src="figure/behavior_07-epm_genes_silence.Rmd/33-VGlut1_distance_prepost-1.pdf" width="230.4" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="vglut2-8" class="section level4">
<h4>VGlut2</h4>
<pre class="r"><code>bg_twogroup(VGlut2_group_gene, &quot;Distance (m)&quot;, 
            c(control_color, VGlut2_color), openfield_axes_dist_silence_epm, &quot;white&quot;)</code></pre>
<embed src="figure/behavior_07-epm_genes_silence.Rmd/34-VGlut2_distance_prepost-1.pdf" width="230.4" style="display: block; margin: auto;" type="application/pdf" />
</div>
</div>
</div>
<div id="heatmaps" class="section level1 tabset">
<h1 class="tabset">Heatmaps</h1>
<pre class="r"><code>sal_heatmap &lt;- get_heatmap_epm(data_gene_sal, &quot;full&quot;)
cno_heatmap &lt;- get_heatmap_epm(data_gene_cno, &quot;full&quot;)</code></pre>
<div id="vglut1-9" class="section level2 tabset">
<h2 class="tabset">VGlut1</h2>
<div id="mch" class="section level3 tabset">
<h3 class="tabset">mCh</h3>
<div id="sal" class="section level4">
<h4>SAL</h4>
<pre class="r"><code>plot_heatmap_epm(sal_heatmap$`VGlut1-control`)</code></pre>
<p><img src="figure/behavior_07-epm_genes_silence.Rmd/36-VGlut1_mCh_sal_heatmap-1.png" width="288" style="display: block; margin: auto;" /></p>
</div>
<div id="cno" class="section level4">
<h4>CNO</h4>
<pre class="r"><code>plot_heatmap_epm(cno_heatmap$`VGlut1-control`)</code></pre>
<p><img src="figure/behavior_07-epm_genes_silence.Rmd/37-VGlut1_mCh_cno_heatmap-1.png" width="288" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="gi" class="section level3 tabset">
<h3 class="tabset">Gi</h3>
<div id="sal-1" class="section level4">
<h4>SAL</h4>
<pre class="r"><code>plot_heatmap_epm(sal_heatmap$`VGlut1-Gi`)</code></pre>
<p><img src="figure/behavior_07-epm_genes_silence.Rmd/38-VGlut1_Gi_sal_heatmap-1.png" width="288" style="display: block; margin: auto;" /></p>
</div>
<div id="cno-1" class="section level4">
<h4>CNO</h4>
<pre class="r"><code>plot_heatmap_epm(cno_heatmap$`VGlut1-Gi`)</code></pre>
<p><img src="figure/behavior_07-epm_genes_silence.Rmd/39-VGlut1_Gi_cno_heatmap-1.png" width="288" style="display: block; margin: auto;" /></p>
</div>
</div>
</div>
<div id="vglut2-9" class="section level2 tabset">
<h2 class="tabset">VGlut2</h2>
<div id="mch-1" class="section level3 tabset">
<h3 class="tabset">mCh</h3>
<div id="sal-2" class="section level4">
<h4>SAL</h4>
<pre class="r"><code>plot_heatmap_epm(sal_heatmap$`VGlut2-control`)</code></pre>
<p><img src="figure/behavior_07-epm_genes_silence.Rmd/40-VGlut2_mCh_sal_heatmap-1.png" width="288" style="display: block; margin: auto;" /></p>
</div>
<div id="cno-2" class="section level4">
<h4>CNO</h4>
<pre class="r"><code>plot_heatmap_epm(cno_heatmap$`VGlut2-control`)</code></pre>
<p><img src="figure/behavior_07-epm_genes_silence.Rmd/41-VGlut2_mCh_cno_heatmap-1.png" width="288" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="gi-1" class="section level3 tabset">
<h3 class="tabset">Gi</h3>
<div id="sal-3" class="section level4">
<h4>SAL</h4>
<pre class="r"><code>plot_heatmap_epm(sal_heatmap$`VGlut2-Gi`)</code></pre>
<p><img src="figure/behavior_07-epm_genes_silence.Rmd/42-VGlut2_Gi_sal_heatmap-1.png" width="288" style="display: block; margin: auto;" /></p>
</div>
<div id="cno-3" class="section level4">
<h4>CNO</h4>
<pre class="r"><code>plot_heatmap_epm(cno_heatmap$`VGlut2-Gi`)</code></pre>
<p><img src="figure/behavior_07-epm_genes_silence.Rmd/43-VGlut2_Gi_cno_heatmap-1.png" width="288" style="display: block; margin: auto;" /></p>
</div>
</div>
</div>
<div id="scale-bars" class="section level2 tabset">
<h2 class="tabset">Scale bars</h2>
<div id="horizontal" class="section level3">
<h3>Horizontal</h3>
<pre class="r"><code>plot_legend &lt;- plot_heatmap_epm_legend(sal_heatmap$`VGlut1-control`, &quot;horizontal&quot;)

grid.newpage()
grid.draw(plot_legend)</code></pre>
<p><img src="figure/behavior_07-epm_genes_silence.Rmd/44-scale_bar_horizontal-1.png" width="288" style="display: block; margin: auto;" /></p>
</div>
<div id="vertical" class="section level3">
<h3>Vertical</h3>
<pre class="r"><code>plot_legend &lt;- plot_heatmap_epm_legend(sal_heatmap$`VGlut1-control`, &quot;vertical&quot;)

grid.newpage()
grid.draw(plot_legend)</code></pre>
<p><img src="figure/behavior_07-epm_genes_silence.Rmd/45-scale_bar_vertical-1.png" width="288" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.1 (2023-06-16)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS Sonoma 14.1.2

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: America/Los_Angeles
tzcode source: internal

attached base packages:
[1] grid      stats     graphics  grDevices datasets  utils     methods  
[8] base     

other attached packages:
 [1] knitr_1.44             ggpubr_0.6.0           emmeans_1.8.8         
 [4] rstatix_0.7.2          multcomp_1.4-25        TH.data_1.1-2         
 [7] survival_3.5-7         mvtnorm_1.2-3          car_3.1-2             
[10] carData_3.0-5          gmodels_2.18.1.1       zoo_1.8-12            
[13] scales_1.2.1           colorspace_2.1-0       gridExtra_2.3         
[16] cowplot_1.1.1          viridis_0.6.4          viridisLite_0.4.2     
[19] spatstat_3.0-6         spatstat.linnet_3.1-1  spatstat.model_3.2-6  
[22] rpart_4.1.19           spatstat.explore_3.2-3 nlme_3.1-163          
[25] spatstat.random_3.1-6  spatstat.geom_3.2-5    spatstat.data_3.0-1   
[28] RColorBrewer_1.1-3     colorRamps_2.3.1       lubridate_1.9.3       
[31] forcats_1.0.0          stringr_1.5.0          dplyr_1.1.3           
[34] purrr_1.0.2            readr_2.1.4            tidyr_1.3.0           
[37] tibble_3.2.1           ggplot2_3.4.3          tidyverse_2.0.0       
[40] MASS_7.3-60            workflowr_1.7.1       

loaded via a namespace (and not attached):
 [1] deldir_1.0-9          sandwich_3.0-2        rlang_1.1.1          
 [4] magrittr_2.0.3        git2r_0.32.0          compiler_4.3.1       
 [7] getPass_0.2-2         mgcv_1.9-0            gdata_2.19.0         
[10] reshape2_1.4.4        callr_3.7.3           vctrs_0.6.3          
[13] pkgconfig_2.0.3       fastmap_1.1.1         backports_1.4.1      
[16] labeling_0.4.3        utf8_1.2.3            promises_1.2.1       
[19] rmarkdown_2.25        tzdb_0.4.0            ps_1.7.5             
[22] xfun_0.40             cachem_1.0.8          jsonlite_1.8.7       
[25] goftest_1.2-3         later_1.3.1           spatstat.utils_3.0-3 
[28] broom_1.0.5           R6_2.5.1              bslib_0.5.1          
[31] stringi_1.7.12        estimability_1.4.1    jquerylib_0.1.4      
[34] Rcpp_1.0.11           tensor_1.5            httpuv_1.6.11        
[37] Matrix_1.6-1.1        splines_4.3.1         timechange_0.2.0     
[40] tidyselect_1.2.0      rstudioapi_0.15.0     abind_1.4-5          
[43] yaml_2.3.7            codetools_0.2-19      processx_3.8.2       
[46] plyr_1.8.9            lattice_0.21-9        withr_2.5.1          
[49] evaluate_0.22         polyclip_1.10-6       pillar_1.9.0         
[52] BiocManager_1.30.22   whisker_0.4.1         renv_0.15.4          
[55] generics_0.1.3        rprojroot_2.0.3       hms_1.1.3            
[58] munsell_0.5.0         xtable_1.8-4          gtools_3.9.4         
[61] glue_1.6.2            tools_4.3.1           ggsignif_0.6.4       
[64] fs_1.6.3              cli_3.6.1             spatstat.sparse_3.0-2
[67] fansi_1.0.4           gtable_0.3.4          sass_0.4.7           
[70] digest_0.6.33         farver_2.1.1          htmltools_0.5.6      
[73] lifecycle_1.0.3       httr_1.4.7           </code></pre>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
