<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="James Howe" />


<title>Broad cell type classification: plCoA</title>

<script src="site_libs/header-attrs-2.17/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="site_libs/pagedtable-1.1/js/pagedtable.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">CorticalAmygdala_pipelines</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jamesrhowe/CorticalAmygdala_pipelines">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Broad cell type classification: plCoA</h1>
<h4 class="author">James Howe</h4>
<h4 class="date">19 October, 2022</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-10-19
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>CorticalAmygdala_pipelines/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version 1.7.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220906code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20220906)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220906code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220906)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomjamesrhoweCorticalAmygdalapipelinestree9faa91b8f8517c07ac67b7ad435b0a888641baa3targetblank9faa91ba"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/jamesrhowe/CorticalAmygdala_pipelines/tree/9faa91b8f8517c07ac67b7ad435b0a888641baa3" target="_blank">9faa91b</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomjamesrhoweCorticalAmygdalapipelinestree9faa91b8f8517c07ac67b7ad435b0a888641baa3targetblank9faa91ba" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/jamesrhowe/CorticalAmygdala_pipelines/tree/9faa91b8f8517c07ac67b7ad435b0a888641baa3" target="_blank">9faa91b</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    data/behavior/
    Ignored:    data/projections/
    Ignored:    data/sequencing/
    Ignored:    renv/library/
    Ignored:    renv/staging/

Untracked files:
    Untracked:  analysis/behavior_01-topography-testing.Rmd
    Untracked:  analysis/behavior_01-topography.Rmd
    Untracked:  analysis/behavior_02-genetics.Rmd
    Untracked:  analysis/behavior_03-projections.Rmd
    Untracked:  analysis/projections.Rmd
    Untracked:  analysis/sequencing-03-initial_classification.Rmd
    Untracked:  code/behavior-color_scales.R
    Untracked:  code/behavior_misc.R
    Untracked:  code/sequencing-color_scales.R
    Untracked:  code/sequencing-plot_functions.R
    Untracked:  code/sequencing_misc.R
    Untracked:  output/sequencing/

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with <code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>This notebook assigns broad identities based on well-known marker genes, consolidates them into legible identities, and then exports them for subclustering within their broad identities.</p>
<pre class="r"><code>id &lt;- params$id
input &lt;- paste0(params$input_path, id)
output &lt;- params$output_path
dimn &lt;- params$dimn

source(&quot;code/sequencing_misc.R&quot;)
source(&quot;code/sequencing-color_scales.R&quot;)
source(&quot;code/sequencing-plot_functions.R&quot;)

require(Seurat)
require(tidyverse)
require(knitr)
require(scCustomize)
require(clustree)
require(chisq.posthoc.test)
require(speckle)
require(ggtree)
require(pvclust)
require(dendextend)
require(heatmap3)
require(colorRamps)

# all images are PDFs, avoids pixelation
knitr::opts_chunk$set(dev = &#39;pdf&#39;)</code></pre>
<div id="read-in-data-from-step-02" class="section level2">
<h2>Read in data from step 02</h2>
<p>Imports the merged and transformed Seurat object from step 02. This object should contain SCT data, all relevant metadata, and it should be barcode- and feature-filtered already.</p>
<pre class="r"><code>array &lt;- readRDS(paste0(input, &quot;_merged.rds&quot;))</code></pre>
</div>
</div>
<div id="cluster-nuclei" class="section level1 tabset">
<h1 class="tabset">Cluster nuclei</h1>
<p>Clusters nuclei using Seurat’s implementation of the Leiden clustering algorithm (see methods for references), which improves upon the more traditional Louvain algorithm. Clustering is performed at a variety of resolutions so that the best can be chosen from among the different results, ensuring optimal cluster assignments prior to downstream segmentation.</p>
<pre class="r"><code>array &lt;- FindNeighbors(array, dims = 1:dimn, k.param = 30, verbose = FALSE)

res_index &lt;- c(0.01, seq(0.1, 2, 0.1))
for (i in 1:length(res_index)){
  array &lt;- FindClusters(array, method = 4, resolution = res_index[i], verbose = FALSE)
}</code></pre>
<div id="clustree" class="section level2 tabset">
<h2 class="tabset">Clustree</h2>
<div id="standard" class="section level3 unnumbered">
<h3 class="unnumbered">Standard</h3>
<pre class="r"><code>clustree(array, prefix = &quot;SCT_snn_res.&quot;)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/3-clustree_standard-1.pdf" width="1344" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="stability" class="section level3 unnumbered">
<h3 class="unnumbered">Stability</h3>
<pre class="r"><code>clustree(array, prefix = &quot;SCT_snn_res.&quot;, node_colour = &quot;sc3_stability&quot;)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/4-clustree_stability-1.pdf" width="1344" style="display: block; margin: auto;" type="application/pdf" />
</div>
</div>
<div id="scatter" class="section level2 tabset">
<h2 class="tabset">Scatter</h2>
<div id="non-integrated" class="section level3 tabset">
<h3 class="tabset">Non-integrated</h3>
<div id="pca" class="section level4 tabset">
<h4 class="tabset">PCA</h4>
<pre class="r"><code>src_list &lt;- lapply(res_index, function(res) {
    src &lt;- c(
        &quot;##### Res {{res}} {.unnumbered}&quot;,
        &quot;```{r pca-res-{{res}}, message = FALSE, warning = FALSE}&quot;,
        &quot;resolution_plot_formatted(array, &#39;pca&#39;, {{res}}, &#39;SCT&#39;)&quot;,  
        &quot;```&quot;,
        &quot;&quot;
    )
    knit_expand(text = src)
})
out &lt;- knit_child(text = unlist(src_list), options = list(cache = FALSE))</code></pre>
<div id="res-0.01" class="section level5 unnumbered">
<h5 class="unnumbered">Res 0.01</h5>
<pre class="r"><code>resolution_plot_formatted(array, &#39;pca&#39;, 0.01, &#39;SCT&#39;)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/pca-res-0.01-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="res-0.1" class="section level5 unnumbered">
<h5 class="unnumbered">Res 0.1</h5>
<pre class="r"><code>resolution_plot_formatted(array, &#39;pca&#39;, 0.1, &#39;SCT&#39;)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/pca-res-0.1-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="res-0.2" class="section level5 unnumbered">
<h5 class="unnumbered">Res 0.2</h5>
<pre class="r"><code>resolution_plot_formatted(array, &#39;pca&#39;, 0.2, &#39;SCT&#39;)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/pca-res-0.2-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="res-0.3" class="section level5 unnumbered">
<h5 class="unnumbered">Res 0.3</h5>
<pre class="r"><code>resolution_plot_formatted(array, &#39;pca&#39;, 0.3, &#39;SCT&#39;)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/pca-res-0.3-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="res-0.4" class="section level5 unnumbered">
<h5 class="unnumbered">Res 0.4</h5>
<pre class="r"><code>resolution_plot_formatted(array, &#39;pca&#39;, 0.4, &#39;SCT&#39;)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/pca-res-0.4-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="res-0.5" class="section level5 unnumbered">
<h5 class="unnumbered">Res 0.5</h5>
<pre class="r"><code>resolution_plot_formatted(array, &#39;pca&#39;, 0.5, &#39;SCT&#39;)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/pca-res-0.5-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="res-0.6" class="section level5 unnumbered">
<h5 class="unnumbered">Res 0.6</h5>
<pre class="r"><code>resolution_plot_formatted(array, &#39;pca&#39;, 0.6, &#39;SCT&#39;)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/pca-res-0.6-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="res-0.7" class="section level5 unnumbered">
<h5 class="unnumbered">Res 0.7</h5>
<pre class="r"><code>resolution_plot_formatted(array, &#39;pca&#39;, 0.7, &#39;SCT&#39;)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/pca-res-0.7-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="res-0.8" class="section level5 unnumbered">
<h5 class="unnumbered">Res 0.8</h5>
<pre class="r"><code>resolution_plot_formatted(array, &#39;pca&#39;, 0.8, &#39;SCT&#39;)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/pca-res-0.8-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="res-0.9" class="section level5 unnumbered">
<h5 class="unnumbered">Res 0.9</h5>
<pre class="r"><code>resolution_plot_formatted(array, &#39;pca&#39;, 0.9, &#39;SCT&#39;)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/pca-res-0.9-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="res-1" class="section level5 unnumbered">
<h5 class="unnumbered">Res 1</h5>
<pre class="r"><code>resolution_plot_formatted(array, &#39;pca&#39;, 1, &#39;SCT&#39;)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/pca-res-1-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="res-1.1" class="section level5 unnumbered">
<h5 class="unnumbered">Res 1.1</h5>
<pre class="r"><code>resolution_plot_formatted(array, &#39;pca&#39;, 1.1, &#39;SCT&#39;)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/pca-res-1.1-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="res-1.2" class="section level5 unnumbered">
<h5 class="unnumbered">Res 1.2</h5>
<pre class="r"><code>resolution_plot_formatted(array, &#39;pca&#39;, 1.2, &#39;SCT&#39;)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/pca-res-1.2-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="res-1.3" class="section level5 unnumbered">
<h5 class="unnumbered">Res 1.3</h5>
<pre class="r"><code>resolution_plot_formatted(array, &#39;pca&#39;, 1.3, &#39;SCT&#39;)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/pca-res-1.3-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="res-1.4" class="section level5 unnumbered">
<h5 class="unnumbered">Res 1.4</h5>
<pre class="r"><code>resolution_plot_formatted(array, &#39;pca&#39;, 1.4, &#39;SCT&#39;)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/pca-res-1.4-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="res-1.5" class="section level5 unnumbered">
<h5 class="unnumbered">Res 1.5</h5>
<pre class="r"><code>resolution_plot_formatted(array, &#39;pca&#39;, 1.5, &#39;SCT&#39;)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/pca-res-1.5-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="res-1.6" class="section level5 unnumbered">
<h5 class="unnumbered">Res 1.6</h5>
<pre class="r"><code>resolution_plot_formatted(array, &#39;pca&#39;, 1.6, &#39;SCT&#39;)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/pca-res-1.6-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="res-1.7" class="section level5 unnumbered">
<h5 class="unnumbered">Res 1.7</h5>
<pre class="r"><code>resolution_plot_formatted(array, &#39;pca&#39;, 1.7, &#39;SCT&#39;)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/pca-res-1.7-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="res-1.8" class="section level5 unnumbered">
<h5 class="unnumbered">Res 1.8</h5>
<pre class="r"><code>resolution_plot_formatted(array, &#39;pca&#39;, 1.8, &#39;SCT&#39;)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/pca-res-1.8-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="res-1.9" class="section level5 unnumbered">
<h5 class="unnumbered">Res 1.9</h5>
<pre class="r"><code>resolution_plot_formatted(array, &#39;pca&#39;, 1.9, &#39;SCT&#39;)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/pca-res-1.9-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="res-2" class="section level5 unnumbered">
<h5 class="unnumbered">Res 2</h5>
<pre class="r"><code>resolution_plot_formatted(array, &#39;pca&#39;, 2, &#39;SCT&#39;)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/pca-res-2-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
</div>
<div id="tsne" class="section level4 tabset">
<h4 class="tabset">tSNE</h4>
<pre class="r"><code>src_list &lt;- lapply(res_index, function(res) {
    src &lt;- c(
        &quot;##### Res {{res}} {.unnumbered}&quot;,
        &quot;```{r tsne-res-{{res}}, message = FALSE, warning = FALSE}&quot;,
        &quot;resolution_plot_formatted(array, &#39;tsne&#39;, {{res}}, &#39;SCT&#39;)&quot;,  
        &quot;```&quot;,
        &quot;&quot;
    )
    knit_expand(text = src)
})
out &lt;- knit_child(text = unlist(src_list), options = list(cache = FALSE))</code></pre>
<div id="res-0.01-1" class="section level5 unnumbered">
<h5 class="unnumbered">Res 0.01</h5>
<pre class="r"><code>resolution_plot_formatted(array, &#39;tsne&#39;, 0.01, &#39;SCT&#39;)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/tsne-res-0.01-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="res-0.1-1" class="section level5 unnumbered">
<h5 class="unnumbered">Res 0.1</h5>
<pre class="r"><code>resolution_plot_formatted(array, &#39;tsne&#39;, 0.1, &#39;SCT&#39;)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/tsne-res-0.1-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="res-0.2-1" class="section level5 unnumbered">
<h5 class="unnumbered">Res 0.2</h5>
<pre class="r"><code>resolution_plot_formatted(array, &#39;tsne&#39;, 0.2, &#39;SCT&#39;)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/tsne-res-0.2-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="res-0.3-1" class="section level5 unnumbered">
<h5 class="unnumbered">Res 0.3</h5>
<pre class="r"><code>resolution_plot_formatted(array, &#39;tsne&#39;, 0.3, &#39;SCT&#39;)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/tsne-res-0.3-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="res-0.4-1" class="section level5 unnumbered">
<h5 class="unnumbered">Res 0.4</h5>
<pre class="r"><code>resolution_plot_formatted(array, &#39;tsne&#39;, 0.4, &#39;SCT&#39;)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/tsne-res-0.4-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="res-0.5-1" class="section level5 unnumbered">
<h5 class="unnumbered">Res 0.5</h5>
<pre class="r"><code>resolution_plot_formatted(array, &#39;tsne&#39;, 0.5, &#39;SCT&#39;)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/tsne-res-0.5-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="res-0.6-1" class="section level5 unnumbered">
<h5 class="unnumbered">Res 0.6</h5>
<pre class="r"><code>resolution_plot_formatted(array, &#39;tsne&#39;, 0.6, &#39;SCT&#39;)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/tsne-res-0.6-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="res-0.7-1" class="section level5 unnumbered">
<h5 class="unnumbered">Res 0.7</h5>
<pre class="r"><code>resolution_plot_formatted(array, &#39;tsne&#39;, 0.7, &#39;SCT&#39;)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/tsne-res-0.7-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="res-0.8-1" class="section level5 unnumbered">
<h5 class="unnumbered">Res 0.8</h5>
<pre class="r"><code>resolution_plot_formatted(array, &#39;tsne&#39;, 0.8, &#39;SCT&#39;)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/tsne-res-0.8-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="res-0.9-1" class="section level5 unnumbered">
<h5 class="unnumbered">Res 0.9</h5>
<pre class="r"><code>resolution_plot_formatted(array, &#39;tsne&#39;, 0.9, &#39;SCT&#39;)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/tsne-res-0.9-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="res-1-1" class="section level5 unnumbered">
<h5 class="unnumbered">Res 1</h5>
<pre class="r"><code>resolution_plot_formatted(array, &#39;tsne&#39;, 1, &#39;SCT&#39;)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/tsne-res-1-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="res-1.1-1" class="section level5 unnumbered">
<h5 class="unnumbered">Res 1.1</h5>
<pre class="r"><code>resolution_plot_formatted(array, &#39;tsne&#39;, 1.1, &#39;SCT&#39;)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/tsne-res-1.1-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="res-1.2-1" class="section level5 unnumbered">
<h5 class="unnumbered">Res 1.2</h5>
<pre class="r"><code>resolution_plot_formatted(array, &#39;tsne&#39;, 1.2, &#39;SCT&#39;)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/tsne-res-1.2-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="res-1.3-1" class="section level5 unnumbered">
<h5 class="unnumbered">Res 1.3</h5>
<pre class="r"><code>resolution_plot_formatted(array, &#39;tsne&#39;, 1.3, &#39;SCT&#39;)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/tsne-res-1.3-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="res-1.4-1" class="section level5 unnumbered">
<h5 class="unnumbered">Res 1.4</h5>
<pre class="r"><code>resolution_plot_formatted(array, &#39;tsne&#39;, 1.4, &#39;SCT&#39;)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/tsne-res-1.4-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="res-1.5-1" class="section level5 unnumbered">
<h5 class="unnumbered">Res 1.5</h5>
<pre class="r"><code>resolution_plot_formatted(array, &#39;tsne&#39;, 1.5, &#39;SCT&#39;)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/tsne-res-1.5-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="res-1.6-1" class="section level5 unnumbered">
<h5 class="unnumbered">Res 1.6</h5>
<pre class="r"><code>resolution_plot_formatted(array, &#39;tsne&#39;, 1.6, &#39;SCT&#39;)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/tsne-res-1.6-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="res-1.7-1" class="section level5 unnumbered">
<h5 class="unnumbered">Res 1.7</h5>
<pre class="r"><code>resolution_plot_formatted(array, &#39;tsne&#39;, 1.7, &#39;SCT&#39;)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/tsne-res-1.7-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="res-1.8-1" class="section level5 unnumbered">
<h5 class="unnumbered">Res 1.8</h5>
<pre class="r"><code>resolution_plot_formatted(array, &#39;tsne&#39;, 1.8, &#39;SCT&#39;)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/tsne-res-1.8-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="res-1.9-1" class="section level5 unnumbered">
<h5 class="unnumbered">Res 1.9</h5>
<pre class="r"><code>resolution_plot_formatted(array, &#39;tsne&#39;, 1.9, &#39;SCT&#39;)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/tsne-res-1.9-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="res-2-1" class="section level5 unnumbered">
<h5 class="unnumbered">Res 2</h5>
<pre class="r"><code>resolution_plot_formatted(array, &#39;tsne&#39;, 2, &#39;SCT&#39;)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/tsne-res-2-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
</div>
<div id="umap" class="section level4 tabset">
<h4 class="tabset">UMAP</h4>
<pre class="r"><code>src_list &lt;- lapply(res_index, function(res) {
    src &lt;- c(
        &quot;##### Res {{res}} {.unnumbered}&quot;,
        &quot;```{r umap-res-{{res}}, message = FALSE, warning = FALSE}&quot;,
        &quot;resolution_plot_formatted(array, &#39;umap&#39;, {{res}}, &#39;SCT&#39;)&quot;,  
        &quot;```&quot;,
        &quot;&quot;
    )
    knit_expand(text = src)
})
out &lt;- knit_child(text = unlist(src_list), options = list(cache = FALSE))</code></pre>
<div id="res-0.01-2" class="section level5 unnumbered">
<h5 class="unnumbered">Res 0.01</h5>
<pre class="r"><code>resolution_plot_formatted(array, &#39;umap&#39;, 0.01, &#39;SCT&#39;)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/umap-res-0.01-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="res-0.1-2" class="section level5 unnumbered">
<h5 class="unnumbered">Res 0.1</h5>
<pre class="r"><code>resolution_plot_formatted(array, &#39;umap&#39;, 0.1, &#39;SCT&#39;)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/umap-res-0.1-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="res-0.2-2" class="section level5 unnumbered">
<h5 class="unnumbered">Res 0.2</h5>
<pre class="r"><code>resolution_plot_formatted(array, &#39;umap&#39;, 0.2, &#39;SCT&#39;)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/umap-res-0.2-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="res-0.3-2" class="section level5 unnumbered">
<h5 class="unnumbered">Res 0.3</h5>
<pre class="r"><code>resolution_plot_formatted(array, &#39;umap&#39;, 0.3, &#39;SCT&#39;)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/umap-res-0.3-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="res-0.4-2" class="section level5 unnumbered">
<h5 class="unnumbered">Res 0.4</h5>
<pre class="r"><code>resolution_plot_formatted(array, &#39;umap&#39;, 0.4, &#39;SCT&#39;)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/umap-res-0.4-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="res-0.5-2" class="section level5 unnumbered">
<h5 class="unnumbered">Res 0.5</h5>
<pre class="r"><code>resolution_plot_formatted(array, &#39;umap&#39;, 0.5, &#39;SCT&#39;)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/umap-res-0.5-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="res-0.6-2" class="section level5 unnumbered">
<h5 class="unnumbered">Res 0.6</h5>
<pre class="r"><code>resolution_plot_formatted(array, &#39;umap&#39;, 0.6, &#39;SCT&#39;)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/umap-res-0.6-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="res-0.7-2" class="section level5 unnumbered">
<h5 class="unnumbered">Res 0.7</h5>
<pre class="r"><code>resolution_plot_formatted(array, &#39;umap&#39;, 0.7, &#39;SCT&#39;)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/umap-res-0.7-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="res-0.8-2" class="section level5 unnumbered">
<h5 class="unnumbered">Res 0.8</h5>
<pre class="r"><code>resolution_plot_formatted(array, &#39;umap&#39;, 0.8, &#39;SCT&#39;)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/umap-res-0.8-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="res-0.9-2" class="section level5 unnumbered">
<h5 class="unnumbered">Res 0.9</h5>
<pre class="r"><code>resolution_plot_formatted(array, &#39;umap&#39;, 0.9, &#39;SCT&#39;)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/umap-res-0.9-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="res-1-2" class="section level5 unnumbered">
<h5 class="unnumbered">Res 1</h5>
<pre class="r"><code>resolution_plot_formatted(array, &#39;umap&#39;, 1, &#39;SCT&#39;)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/umap-res-1-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="res-1.1-2" class="section level5 unnumbered">
<h5 class="unnumbered">Res 1.1</h5>
<pre class="r"><code>resolution_plot_formatted(array, &#39;umap&#39;, 1.1, &#39;SCT&#39;)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/umap-res-1.1-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="res-1.2-2" class="section level5 unnumbered">
<h5 class="unnumbered">Res 1.2</h5>
<pre class="r"><code>resolution_plot_formatted(array, &#39;umap&#39;, 1.2, &#39;SCT&#39;)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/umap-res-1.2-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="res-1.3-2" class="section level5 unnumbered">
<h5 class="unnumbered">Res 1.3</h5>
<pre class="r"><code>resolution_plot_formatted(array, &#39;umap&#39;, 1.3, &#39;SCT&#39;)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/umap-res-1.3-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="res-1.4-2" class="section level5 unnumbered">
<h5 class="unnumbered">Res 1.4</h5>
<pre class="r"><code>resolution_plot_formatted(array, &#39;umap&#39;, 1.4, &#39;SCT&#39;)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/umap-res-1.4-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="res-1.5-2" class="section level5 unnumbered">
<h5 class="unnumbered">Res 1.5</h5>
<pre class="r"><code>resolution_plot_formatted(array, &#39;umap&#39;, 1.5, &#39;SCT&#39;)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/umap-res-1.5-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="res-1.6-2" class="section level5 unnumbered">
<h5 class="unnumbered">Res 1.6</h5>
<pre class="r"><code>resolution_plot_formatted(array, &#39;umap&#39;, 1.6, &#39;SCT&#39;)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/umap-res-1.6-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="res-1.7-2" class="section level5 unnumbered">
<h5 class="unnumbered">Res 1.7</h5>
<pre class="r"><code>resolution_plot_formatted(array, &#39;umap&#39;, 1.7, &#39;SCT&#39;)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/umap-res-1.7-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="res-1.8-2" class="section level5 unnumbered">
<h5 class="unnumbered">Res 1.8</h5>
<pre class="r"><code>resolution_plot_formatted(array, &#39;umap&#39;, 1.8, &#39;SCT&#39;)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/umap-res-1.8-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="res-1.9-2" class="section level5 unnumbered">
<h5 class="unnumbered">Res 1.9</h5>
<pre class="r"><code>resolution_plot_formatted(array, &#39;umap&#39;, 1.9, &#39;SCT&#39;)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/umap-res-1.9-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="res-2-2" class="section level5 unnumbered">
<h5 class="unnumbered">Res 2</h5>
<pre class="r"><code>resolution_plot_formatted(array, &#39;umap&#39;, 2, &#39;SCT&#39;)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/umap-res-2-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
</div>
</div>
</div>
</div>
<div id="annotate-clusters" class="section level1">
<h1>Annotate clusters</h1>
<div id="select-optimal-resolution" class="section level2">
<h2>Select optimal resolution</h2>
<p>Optimal resolution can be selected by examining clustering on dimension-reduced data to determine if they match visual clustering. Further, clustree (and its packaged stability metric) can be used to find the highest resolution that does not lose stability, providing an optimal balance between over- and under-clustering. It is difficult to automate this without some degree of supervision or hand-curation. This is mostly to identify large clusters, so there is little need to get extremely specific so long as clusters are coherent.</p>
<pre class="r"><code>optimal_res &lt;- 0.5

clusters &lt;- array[[paste0(&quot;SCT_snn_res.&quot;, optimal_res)]]
array[[&quot;seurat_clusters&quot;]] &lt;- NULL
for (i in 1:21){
  array[[paste0(&quot;SCT_snn_res.&quot;, res_index[i])]] &lt;- NULL 
}
array &lt;- AddMetaData(array, clusters, col.name = &quot;clusters&quot;)
Idents(array) &lt;- &quot;clusters&quot;

# helps downstrean formatting to regenerate cluster vector from the array vector
tissue_clusters &lt;- as.character(array$clusters)
class_clusters &lt;- tissue_clusters

paste(optimal_res, &quot;is optimal clustering resolution&quot;)</code></pre>
<pre><code>[1] &quot;0.5 is optimal clustering resolution&quot;</code></pre>
</div>
<div id="dot-plot" class="section level2">
<h2>Dot Plot</h2>
<pre class="r"><code>array &lt;- PrepSCTFindMarkers(array)

DotPlot_scCustom(array, features = marker_features, colors_use = dotplot_colors)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/9-dot_plot_markers-1.pdf" width="1920" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="scatter-1" class="section level2 tabset">
<h2 class="tabset">Scatter</h2>
<div id="pca-1" class="section level3 tabset">
<h3 class="tabset">PCA</h3>
<pre class="r"><code>src_list &lt;- lapply(marker_features, function(marker) {
    src &lt;- c(
        &quot;#### {{marker}} {.unnumbered}&quot;,
        &quot;```{r pca-{{marker}}, message = FALSE, warning = FALSE}&quot;,
        &quot;feature_plot_formatted(array, &#39;pca&#39;, &#39;{{marker}}&#39;, feature_intensity_colors, NA)&quot;,  
        &quot;```&quot;,
        &quot;&quot;
    )
    knit_expand(text = src)
})
out &lt;- knit_child(text = unlist(src_list), options = list(cache = FALSE))</code></pre>
<div id="syp" class="section level4 unnumbered">
<h4 class="unnumbered">Syp</h4>
<pre class="r"><code>feature_plot_formatted(array, &#39;pca&#39;, &#39;Syp&#39;, feature_intensity_colors, NA)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/pca-Syp-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="eno2" class="section level4 unnumbered">
<h4 class="unnumbered">Eno2</h4>
<pre class="r"><code>feature_plot_formatted(array, &#39;pca&#39;, &#39;Eno2&#39;, feature_intensity_colors, NA)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/pca-Eno2-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="slc17a7" class="section level4 unnumbered">
<h4 class="unnumbered">Slc17a7</h4>
<pre class="r"><code>feature_plot_formatted(array, &#39;pca&#39;, &#39;Slc17a7&#39;, feature_intensity_colors, NA)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/pca-Slc17a7-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="slc17a6" class="section level4 unnumbered">
<h4 class="unnumbered">Slc17a6</h4>
<pre class="r"><code>feature_plot_formatted(array, &#39;pca&#39;, &#39;Slc17a6&#39;, feature_intensity_colors, NA)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/pca-Slc17a6-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="gad1" class="section level4 unnumbered">
<h4 class="unnumbered">Gad1</h4>
<pre class="r"><code>feature_plot_formatted(array, &#39;pca&#39;, &#39;Gad1&#39;, feature_intensity_colors, NA)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/pca-Gad1-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="gad2" class="section level4 unnumbered">
<h4 class="unnumbered">Gad2</h4>
<pre class="r"><code>feature_plot_formatted(array, &#39;pca&#39;, &#39;Gad2&#39;, feature_intensity_colors, NA)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/pca-Gad2-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="aldh1l1" class="section level4 unnumbered">
<h4 class="unnumbered">Aldh1l1</h4>
<pre class="r"><code>feature_plot_formatted(array, &#39;pca&#39;, &#39;Aldh1l1&#39;, feature_intensity_colors, NA)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/pca-Aldh1l1-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="aqp4" class="section level4 unnumbered">
<h4 class="unnumbered">Aqp4</h4>
<pre class="r"><code>feature_plot_formatted(array, &#39;pca&#39;, &#39;Aqp4&#39;, feature_intensity_colors, NA)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/pca-Aqp4-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="tmem119" class="section level4 unnumbered">
<h4 class="unnumbered">Tmem119</h4>
<pre class="r"><code>feature_plot_formatted(array, &#39;pca&#39;, &#39;Tmem119&#39;, feature_intensity_colors, NA)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/pca-Tmem119-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="ptprc" class="section level4 unnumbered">
<h4 class="unnumbered">Ptprc</h4>
<pre class="r"><code>feature_plot_formatted(array, &#39;pca&#39;, &#39;Ptprc&#39;, feature_intensity_colors, NA)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/pca-Ptprc-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="pdgfra" class="section level4 unnumbered">
<h4 class="unnumbered">Pdgfra</h4>
<pre class="r"><code>feature_plot_formatted(array, &#39;pca&#39;, &#39;Pdgfra&#39;, feature_intensity_colors, NA)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/pca-Pdgfra-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="cspg4" class="section level4 unnumbered">
<h4 class="unnumbered">Cspg4</h4>
<pre class="r"><code>feature_plot_formatted(array, &#39;pca&#39;, &#39;Cspg4&#39;, feature_intensity_colors, NA)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/pca-Cspg4-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="bmp4" class="section level4 unnumbered">
<h4 class="unnumbered">Bmp4</h4>
<pre class="r"><code>feature_plot_formatted(array, &#39;pca&#39;, &#39;Bmp4&#39;, feature_intensity_colors, NA)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/pca-Bmp4-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="enpp6" class="section level4 unnumbered">
<h4 class="unnumbered">Enpp6</h4>
<pre class="r"><code>feature_plot_formatted(array, &#39;pca&#39;, &#39;Enpp6&#39;, feature_intensity_colors, NA)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/pca-Enpp6-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="mog" class="section level4 unnumbered">
<h4 class="unnumbered">Mog</h4>
<pre class="r"><code>feature_plot_formatted(array, &#39;pca&#39;, &#39;Mog&#39;, feature_intensity_colors, NA)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/pca-Mog-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="mal" class="section level4 unnumbered">
<h4 class="unnumbered">Mal</h4>
<pre class="r"><code>feature_plot_formatted(array, &#39;pca&#39;, &#39;Mal&#39;, feature_intensity_colors, NA)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/pca-Mal-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="cldn5" class="section level4 unnumbered">
<h4 class="unnumbered">Cldn5</h4>
<pre class="r"><code>feature_plot_formatted(array, &#39;pca&#39;, &#39;Cldn5&#39;, feature_intensity_colors, NA)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/pca-Cldn5-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="adgrl4" class="section level4 unnumbered">
<h4 class="unnumbered">Adgrl4</h4>
<pre class="r"><code>feature_plot_formatted(array, &#39;pca&#39;, &#39;Adgrl4&#39;, feature_intensity_colors, NA)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/pca-Adgrl4-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="kcnj8" class="section level4 unnumbered">
<h4 class="unnumbered">Kcnj8</h4>
<pre class="r"><code>feature_plot_formatted(array, &#39;pca&#39;, &#39;Kcnj8&#39;, feature_intensity_colors, NA)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/pca-Kcnj8-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="abcc9" class="section level4 unnumbered">
<h4 class="unnumbered">Abcc9</h4>
<pre class="r"><code>feature_plot_formatted(array, &#39;pca&#39;, &#39;Abcc9&#39;, feature_intensity_colors, NA)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/pca-Abcc9-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="dapl1" class="section level4 unnumbered">
<h4 class="unnumbered">Dapl1</h4>
<pre class="r"><code>feature_plot_formatted(array, &#39;pca&#39;, &#39;Dapl1&#39;, feature_intensity_colors, NA)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/pca-Dapl1-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="slc47a1" class="section level4 unnumbered">
<h4 class="unnumbered">Slc47a1</h4>
<pre class="r"><code>feature_plot_formatted(array, &#39;pca&#39;, &#39;Slc47a1&#39;, feature_intensity_colors, NA)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/pca-Slc47a1-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
</div>
<div id="tsne-1" class="section level3 tabset">
<h3 class="tabset">tSNE</h3>
<pre class="r"><code>src_list &lt;- lapply(marker_features, function(marker) {
    src &lt;- c(
        &quot;#### {{marker}} {.unnumbered}&quot;,
        &quot;```{r tsne-{{marker}}, message = FALSE, warning = FALSE}&quot;,
        &quot;feature_plot_formatted(array, &#39;tsne&#39;, &#39;{{marker}}&#39;, feature_intensity_colors, NA)&quot;,  
        &quot;```&quot;,
        &quot;&quot;
    )
    knit_expand(text = src)
})
out &lt;- knit_child(text = unlist(src_list), options = list(cache = FALSE))</code></pre>
<div id="syp-1" class="section level4 unnumbered">
<h4 class="unnumbered">Syp</h4>
<pre class="r"><code>feature_plot_formatted(array, &#39;tsne&#39;, &#39;Syp&#39;, feature_intensity_colors, NA)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/tsne-Syp-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="eno2-1" class="section level4 unnumbered">
<h4 class="unnumbered">Eno2</h4>
<pre class="r"><code>feature_plot_formatted(array, &#39;tsne&#39;, &#39;Eno2&#39;, feature_intensity_colors, NA)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/tsne-Eno2-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="slc17a7-1" class="section level4 unnumbered">
<h4 class="unnumbered">Slc17a7</h4>
<pre class="r"><code>feature_plot_formatted(array, &#39;tsne&#39;, &#39;Slc17a7&#39;, feature_intensity_colors, NA)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/tsne-Slc17a7-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="slc17a6-1" class="section level4 unnumbered">
<h4 class="unnumbered">Slc17a6</h4>
<pre class="r"><code>feature_plot_formatted(array, &#39;tsne&#39;, &#39;Slc17a6&#39;, feature_intensity_colors, NA)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/tsne-Slc17a6-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="gad1-1" class="section level4 unnumbered">
<h4 class="unnumbered">Gad1</h4>
<pre class="r"><code>feature_plot_formatted(array, &#39;tsne&#39;, &#39;Gad1&#39;, feature_intensity_colors, NA)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/tsne-Gad1-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="gad2-1" class="section level4 unnumbered">
<h4 class="unnumbered">Gad2</h4>
<pre class="r"><code>feature_plot_formatted(array, &#39;tsne&#39;, &#39;Gad2&#39;, feature_intensity_colors, NA)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/tsne-Gad2-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="aldh1l1-1" class="section level4 unnumbered">
<h4 class="unnumbered">Aldh1l1</h4>
<pre class="r"><code>feature_plot_formatted(array, &#39;tsne&#39;, &#39;Aldh1l1&#39;, feature_intensity_colors, NA)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/tsne-Aldh1l1-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="aqp4-1" class="section level4 unnumbered">
<h4 class="unnumbered">Aqp4</h4>
<pre class="r"><code>feature_plot_formatted(array, &#39;tsne&#39;, &#39;Aqp4&#39;, feature_intensity_colors, NA)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/tsne-Aqp4-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="tmem119-1" class="section level4 unnumbered">
<h4 class="unnumbered">Tmem119</h4>
<pre class="r"><code>feature_plot_formatted(array, &#39;tsne&#39;, &#39;Tmem119&#39;, feature_intensity_colors, NA)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/tsne-Tmem119-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="ptprc-1" class="section level4 unnumbered">
<h4 class="unnumbered">Ptprc</h4>
<pre class="r"><code>feature_plot_formatted(array, &#39;tsne&#39;, &#39;Ptprc&#39;, feature_intensity_colors, NA)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/tsne-Ptprc-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="pdgfra-1" class="section level4 unnumbered">
<h4 class="unnumbered">Pdgfra</h4>
<pre class="r"><code>feature_plot_formatted(array, &#39;tsne&#39;, &#39;Pdgfra&#39;, feature_intensity_colors, NA)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/tsne-Pdgfra-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="cspg4-1" class="section level4 unnumbered">
<h4 class="unnumbered">Cspg4</h4>
<pre class="r"><code>feature_plot_formatted(array, &#39;tsne&#39;, &#39;Cspg4&#39;, feature_intensity_colors, NA)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/tsne-Cspg4-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="bmp4-1" class="section level4 unnumbered">
<h4 class="unnumbered">Bmp4</h4>
<pre class="r"><code>feature_plot_formatted(array, &#39;tsne&#39;, &#39;Bmp4&#39;, feature_intensity_colors, NA)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/tsne-Bmp4-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="enpp6-1" class="section level4 unnumbered">
<h4 class="unnumbered">Enpp6</h4>
<pre class="r"><code>feature_plot_formatted(array, &#39;tsne&#39;, &#39;Enpp6&#39;, feature_intensity_colors, NA)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/tsne-Enpp6-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="mog-1" class="section level4 unnumbered">
<h4 class="unnumbered">Mog</h4>
<pre class="r"><code>feature_plot_formatted(array, &#39;tsne&#39;, &#39;Mog&#39;, feature_intensity_colors, NA)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/tsne-Mog-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="mal-1" class="section level4 unnumbered">
<h4 class="unnumbered">Mal</h4>
<pre class="r"><code>feature_plot_formatted(array, &#39;tsne&#39;, &#39;Mal&#39;, feature_intensity_colors, NA)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/tsne-Mal-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="cldn5-1" class="section level4 unnumbered">
<h4 class="unnumbered">Cldn5</h4>
<pre class="r"><code>feature_plot_formatted(array, &#39;tsne&#39;, &#39;Cldn5&#39;, feature_intensity_colors, NA)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/tsne-Cldn5-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="adgrl4-1" class="section level4 unnumbered">
<h4 class="unnumbered">Adgrl4</h4>
<pre class="r"><code>feature_plot_formatted(array, &#39;tsne&#39;, &#39;Adgrl4&#39;, feature_intensity_colors, NA)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/tsne-Adgrl4-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="kcnj8-1" class="section level4 unnumbered">
<h4 class="unnumbered">Kcnj8</h4>
<pre class="r"><code>feature_plot_formatted(array, &#39;tsne&#39;, &#39;Kcnj8&#39;, feature_intensity_colors, NA)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/tsne-Kcnj8-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="abcc9-1" class="section level4 unnumbered">
<h4 class="unnumbered">Abcc9</h4>
<pre class="r"><code>feature_plot_formatted(array, &#39;tsne&#39;, &#39;Abcc9&#39;, feature_intensity_colors, NA)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/tsne-Abcc9-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="dapl1-1" class="section level4 unnumbered">
<h4 class="unnumbered">Dapl1</h4>
<pre class="r"><code>feature_plot_formatted(array, &#39;tsne&#39;, &#39;Dapl1&#39;, feature_intensity_colors, NA)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/tsne-Dapl1-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="slc47a1-1" class="section level4 unnumbered">
<h4 class="unnumbered">Slc47a1</h4>
<pre class="r"><code>feature_plot_formatted(array, &#39;tsne&#39;, &#39;Slc47a1&#39;, feature_intensity_colors, NA)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/tsne-Slc47a1-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
</div>
<div id="umap-1" class="section level3 tabset">
<h3 class="tabset">UMAP</h3>
<pre class="r"><code>src_list &lt;- lapply(marker_features, function(marker) {
    src &lt;- c(
        &quot;#### {{marker}} {.unnumbered}&quot;,
        &quot;```{r umap-{{marker}}, message = FALSE, warning = FALSE}&quot;,
        &quot;feature_plot_formatted(array, &#39;umap&#39;, &#39;{{marker}}&#39;, feature_intensity_colors, NA)&quot;,  
        &quot;```&quot;,
        &quot;&quot;
    )
    knit_expand(text = src)
})
out &lt;- knit_child(text = unlist(src_list), options = list(cache = FALSE))</code></pre>
<div id="syp-2" class="section level4 unnumbered">
<h4 class="unnumbered">Syp</h4>
<pre class="r"><code>feature_plot_formatted(array, &#39;umap&#39;, &#39;Syp&#39;, feature_intensity_colors, NA)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/umap-Syp-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="eno2-2" class="section level4 unnumbered">
<h4 class="unnumbered">Eno2</h4>
<pre class="r"><code>feature_plot_formatted(array, &#39;umap&#39;, &#39;Eno2&#39;, feature_intensity_colors, NA)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/umap-Eno2-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="slc17a7-2" class="section level4 unnumbered">
<h4 class="unnumbered">Slc17a7</h4>
<pre class="r"><code>feature_plot_formatted(array, &#39;umap&#39;, &#39;Slc17a7&#39;, feature_intensity_colors, NA)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/umap-Slc17a7-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="slc17a6-2" class="section level4 unnumbered">
<h4 class="unnumbered">Slc17a6</h4>
<pre class="r"><code>feature_plot_formatted(array, &#39;umap&#39;, &#39;Slc17a6&#39;, feature_intensity_colors, NA)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/umap-Slc17a6-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="gad1-2" class="section level4 unnumbered">
<h4 class="unnumbered">Gad1</h4>
<pre class="r"><code>feature_plot_formatted(array, &#39;umap&#39;, &#39;Gad1&#39;, feature_intensity_colors, NA)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/umap-Gad1-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="gad2-2" class="section level4 unnumbered">
<h4 class="unnumbered">Gad2</h4>
<pre class="r"><code>feature_plot_formatted(array, &#39;umap&#39;, &#39;Gad2&#39;, feature_intensity_colors, NA)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/umap-Gad2-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="aldh1l1-2" class="section level4 unnumbered">
<h4 class="unnumbered">Aldh1l1</h4>
<pre class="r"><code>feature_plot_formatted(array, &#39;umap&#39;, &#39;Aldh1l1&#39;, feature_intensity_colors, NA)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/umap-Aldh1l1-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="aqp4-2" class="section level4 unnumbered">
<h4 class="unnumbered">Aqp4</h4>
<pre class="r"><code>feature_plot_formatted(array, &#39;umap&#39;, &#39;Aqp4&#39;, feature_intensity_colors, NA)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/umap-Aqp4-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="tmem119-2" class="section level4 unnumbered">
<h4 class="unnumbered">Tmem119</h4>
<pre class="r"><code>feature_plot_formatted(array, &#39;umap&#39;, &#39;Tmem119&#39;, feature_intensity_colors, NA)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/umap-Tmem119-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="ptprc-2" class="section level4 unnumbered">
<h4 class="unnumbered">Ptprc</h4>
<pre class="r"><code>feature_plot_formatted(array, &#39;umap&#39;, &#39;Ptprc&#39;, feature_intensity_colors, NA)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/umap-Ptprc-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="pdgfra-2" class="section level4 unnumbered">
<h4 class="unnumbered">Pdgfra</h4>
<pre class="r"><code>feature_plot_formatted(array, &#39;umap&#39;, &#39;Pdgfra&#39;, feature_intensity_colors, NA)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/umap-Pdgfra-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="cspg4-2" class="section level4 unnumbered">
<h4 class="unnumbered">Cspg4</h4>
<pre class="r"><code>feature_plot_formatted(array, &#39;umap&#39;, &#39;Cspg4&#39;, feature_intensity_colors, NA)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/umap-Cspg4-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="bmp4-2" class="section level4 unnumbered">
<h4 class="unnumbered">Bmp4</h4>
<pre class="r"><code>feature_plot_formatted(array, &#39;umap&#39;, &#39;Bmp4&#39;, feature_intensity_colors, NA)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/umap-Bmp4-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="enpp6-2" class="section level4 unnumbered">
<h4 class="unnumbered">Enpp6</h4>
<pre class="r"><code>feature_plot_formatted(array, &#39;umap&#39;, &#39;Enpp6&#39;, feature_intensity_colors, NA)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/umap-Enpp6-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="mog-2" class="section level4 unnumbered">
<h4 class="unnumbered">Mog</h4>
<pre class="r"><code>feature_plot_formatted(array, &#39;umap&#39;, &#39;Mog&#39;, feature_intensity_colors, NA)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/umap-Mog-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="mal-2" class="section level4 unnumbered">
<h4 class="unnumbered">Mal</h4>
<pre class="r"><code>feature_plot_formatted(array, &#39;umap&#39;, &#39;Mal&#39;, feature_intensity_colors, NA)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/umap-Mal-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="cldn5-2" class="section level4 unnumbered">
<h4 class="unnumbered">Cldn5</h4>
<pre class="r"><code>feature_plot_formatted(array, &#39;umap&#39;, &#39;Cldn5&#39;, feature_intensity_colors, NA)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/umap-Cldn5-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="adgrl4-2" class="section level4 unnumbered">
<h4 class="unnumbered">Adgrl4</h4>
<pre class="r"><code>feature_plot_formatted(array, &#39;umap&#39;, &#39;Adgrl4&#39;, feature_intensity_colors, NA)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/umap-Adgrl4-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="kcnj8-2" class="section level4 unnumbered">
<h4 class="unnumbered">Kcnj8</h4>
<pre class="r"><code>feature_plot_formatted(array, &#39;umap&#39;, &#39;Kcnj8&#39;, feature_intensity_colors, NA)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/umap-Kcnj8-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="abcc9-2" class="section level4 unnumbered">
<h4 class="unnumbered">Abcc9</h4>
<pre class="r"><code>feature_plot_formatted(array, &#39;umap&#39;, &#39;Abcc9&#39;, feature_intensity_colors, NA)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/umap-Abcc9-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="dapl1-2" class="section level4 unnumbered">
<h4 class="unnumbered">Dapl1</h4>
<pre class="r"><code>feature_plot_formatted(array, &#39;umap&#39;, &#39;Dapl1&#39;, feature_intensity_colors, NA)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/umap-Dapl1-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="slc47a1-2" class="section level4 unnumbered">
<h4 class="unnumbered">Slc47a1</h4>
<pre class="r"><code>feature_plot_formatted(array, &#39;umap&#39;, &#39;Slc47a1&#39;, feature_intensity_colors, NA)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/umap-Slc47a1-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
</div>
</div>
</div>
<div id="consolidate-clusters" class="section level1">
<h1>Consolidate clusters</h1>
<p>Based on known marker gene expression, assign broad cellular identities to clusters at the optimal resolution selected. These identities are hand-assigned, and multiple clusters are in many cases combined into broad cellular identities due to uneven differences between cell types.</p>
<pre class="r"><code>glut_clusters &lt;- c(0, 5:8, 10, 17, 23, 25)
gaba_clusters &lt;- c(9, 13, 15, 20, 22)

astro_clusters &lt;- c(1, 3, 14)
micro_clusters &lt;- c(4, 19)
opc_clusters &lt;- c(2, 18)
olg_clusters &lt;- 12
endo_clusters &lt;- 24
mural_clusters &lt;- 11
vlmc_clusters &lt;- c(16, 21)

# this part is automated and doesn&#39;t need hand curation
neuron_clusters &lt;- c(glut_clusters, gaba_clusters)
glia_clusters &lt;- c(astro_clusters, micro_clusters, opc_clusters, olg_clusters, 
                   endo_clusters, mural_clusters, vlmc_clusters)

tissue_clusters[tissue_clusters %in% neuron_clusters] &lt;- &quot;Neuron&quot;
tissue_clusters[tissue_clusters %in% glia_clusters] &lt;- &quot;Nonneuron&quot;

class_clusters[class_clusters %in% glut_clusters] &lt;- &quot;Glutamatergic&quot;
class_clusters[class_clusters %in% gaba_clusters] &lt;- &quot;GABAergic&quot;
class_clusters[class_clusters %in% astro_clusters] &lt;- &quot;Astro&quot;
class_clusters[class_clusters %in% micro_clusters] &lt;- &quot;Microglia&quot;
class_clusters[class_clusters %in% opc_clusters] &lt;- &quot;OPC&quot;
class_clusters[class_clusters %in% olg_clusters] &lt;- &quot;OLG&quot;
class_clusters[class_clusters %in% endo_clusters] &lt;- &quot;Endothelial&quot;
class_clusters[class_clusters %in% mural_clusters] &lt;- &quot;Mural&quot;
class_clusters[class_clusters %in% vlmc_clusters] &lt;- &quot;VLMC&quot;

# replace clusters with curated types, reorder for plotting
array &lt;- AddMetaData(array, as.factor(tissue_clusters), col.name = &quot;Tissue&quot;)
array$Tissue &lt;- factor(array$Tissue, levels = c(&quot;Neuron&quot;, &quot;Nonneuron&quot;))

array &lt;- AddMetaData(array, as.factor(class_clusters), col.name = &quot;Class&quot;)
array$Class &lt;- factor(array$Class, levels = c(&quot;Glutamatergic&quot;, &quot;GABAergic&quot;,
                                              &quot;Astro&quot;, &quot;Microglia&quot;,
                                              &quot;OPC&quot;, &quot;OLG&quot;, 
                                              &quot;Endothelial&quot;, &quot;Mural&quot;, 
                                              &quot;VLMC&quot;))

plot_group_array &lt;- cbind.data.frame(array$Tissue,
                                     array$Class,
                                     array$orig.ident,
                                     array$Region)
colnames(plot_group_array) &lt;- c(&quot;Tissue&quot;, &quot;Class&quot;, &quot;Batch&quot;, &quot;Region&quot;)

table(array$Tissue)</code></pre>
<pre><code>
   Neuron Nonneuron 
    25582     21836 </code></pre>
<pre class="r"><code>table(array$Class)</code></pre>
<pre><code>
Glutamatergic     GABAergic         Astro     Microglia           OPC 
        20433          5149          9607          3603          3921 
          OLG   Endothelial         Mural          VLMC 
         1564           114          1939          1088 </code></pre>
</div>
<div id="visualize-consolidated-clusters" class="section level1 tabset">
<h1 class="tabset">Visualize consolidated clusters</h1>
<div id="non-integrated-1" class="section level2 tabset">
<h2 class="tabset">Non-integrated</h2>
<div id="pca-2" class="section level3 tabset">
<h3 class="tabset">PCA</h3>
<pre class="r"><code>src_list &lt;- mapply(function(tg, tc) {
    src &lt;- c(
        &quot;#### {{tg}} {.unnumbered}&quot;,
        &quot;```{r pca-{{tg}}, message = FALSE, warning = FALSE}&quot;,
        &quot;cluster_plot_formatted(array, &#39;pca&#39;, &#39;{{tg}}&#39;, {{tc}})&quot;,  
        &quot;```&quot;,
        &quot;&quot;
    )
    knit_expand(text = src)
}, tissue_class, tissue_color_list)
out &lt;- knit_child(text = unlist(src_list), options = list(cache = FALSE))</code></pre>
<div id="tissue" class="section level4 unnumbered">
<h4 class="unnumbered">Tissue</h4>
<pre class="r"><code>cluster_plot_formatted(array, &#39;pca&#39;, &#39;Tissue&#39;, tissue_colors)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/pca-Tissue-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="class" class="section level4 unnumbered">
<h4 class="unnumbered">Class</h4>
<pre class="r"><code>cluster_plot_formatted(array, &#39;pca&#39;, &#39;Class&#39;, class_colors)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/pca-Class-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
</div>
<div id="tsne-2" class="section level3 tabset">
<h3 class="tabset">tSNE</h3>
<pre class="r"><code>src_list &lt;- mapply(function(tg, tc) {
    src &lt;- c(
        &quot;##### {{tg}} {.unnumbered}&quot;,
        &quot;```{r tsne-{{tg}}, message = FALSE, warning = FALSE}&quot;,
        &quot;cluster_plot_formatted(array, &#39;tsne&#39;, &#39;{{tg}}&#39;, {{tc}})&quot;,  
        &quot;```&quot;,
        &quot;&quot;
    )
    knit_expand(text = src)
}, tissue_class, tissue_color_list)
out &lt;- knit_child(text = unlist(src_list), options = list(cache = FALSE))</code></pre>
<div id="tissue-1" class="section level5 unnumbered">
<h5 class="unnumbered">Tissue</h5>
<pre class="r"><code>cluster_plot_formatted(array, &#39;tsne&#39;, &#39;Tissue&#39;, tissue_colors)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/tsne-Tissue-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="class-1" class="section level5 unnumbered">
<h5 class="unnumbered">Class</h5>
<pre class="r"><code>cluster_plot_formatted(array, &#39;tsne&#39;, &#39;Class&#39;, class_colors)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/tsne-Class-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
</div>
<div id="umap-2" class="section level3 tabset">
<h3 class="tabset">UMAP</h3>
<pre class="r"><code>src_list &lt;- mapply(function(tg, tc) {
    src &lt;- c(
        &quot;#### {{tg}} {.unnumbered}&quot;,
        &quot;```{r umap-{{tg}}, message = FALSE, warning = FALSE}&quot;,
        &quot;cluster_plot_formatted(array, &#39;umap&#39;, &#39;{{tg}}&#39;, {{tc}})&quot;,  
        &quot;```&quot;,
        &quot;&quot;
    )
    knit_expand(text = src)
}, tissue_class, tissue_color_list)
out &lt;- knit_child(text = unlist(src_list), options = list(cache = FALSE))</code></pre>
<div id="tissue-2" class="section level4 unnumbered">
<h4 class="unnumbered">Tissue</h4>
<pre class="r"><code>cluster_plot_formatted(array, &#39;umap&#39;, &#39;Tissue&#39;, tissue_colors)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/umap-Tissue-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="class-2" class="section level4 unnumbered">
<h4 class="unnumbered">Class</h4>
<pre class="r"><code>cluster_plot_formatted(array, &#39;umap&#39;, &#39;Class&#39;, class_colors)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/umap-Class-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
</div>
</div>
</div>
<div id="visualize-consolidated-cluster-composition" class="section level1">
<h1>Visualize consolidated cluster composition</h1>
<div id="nested-composition" class="section level2">
<h2>Nested Composition</h2>
<pre class="r"><code>absolute_bar_plot_formatted(plot_group_array, &quot;Tissue&quot;, &quot;Class&quot;, class_colors)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/17-bar_nested-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="absolute-composition" class="section level2 tabset">
<h2 class="tabset">Absolute Composition</h2>
<div id="tissue-3" class="section level3 tabset">
<h3 class="tabset">Tissue</h3>
<pre class="r"><code>src_list &lt;- mapply(function(br, br_colors) {
    src &lt;- c(
        &quot;#### {{br}} {.unnumbered}&quot;,
        &quot;```{r tissue-absolute-{{br}}, message = FALSE, warning = FALSE}&quot;,
        &quot;absolute_bar_plot_formatted(plot_group_array, &#39;Tissue&#39;, &#39;{{br}}&#39;, {{br_colors}})&quot;,  
        &quot;```&quot;,
        &quot;&quot;
    )
    knit_expand(text = src)
}, batch_region, batch_color_list)
out &lt;- knit_child(text = unlist(src_list), options = list(cache = FALSE))</code></pre>
<div id="batch" class="section level4 unnumbered">
<h4 class="unnumbered">Batch</h4>
<pre class="r"><code>absolute_bar_plot_formatted(plot_group_array, &#39;Tissue&#39;, &#39;Batch&#39;, batch_colors_rna)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/tissue-absolute-Batch-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="region" class="section level4 unnumbered">
<h4 class="unnumbered">Region</h4>
<pre class="r"><code>absolute_bar_plot_formatted(plot_group_array, &#39;Tissue&#39;, &#39;Region&#39;, region_colors)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/tissue-absolute-Region-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
</div>
<div id="class-3" class="section level3 tabset">
<h3 class="tabset">Class</h3>
<pre class="r"><code>src_list &lt;- mapply(function(br, br_colors) {
    src &lt;- c(
        &quot;#### {{br}} {.unnumbered}&quot;,
        &quot;```{r class-absolute-{{br}}, message = FALSE, warning = FALSE}&quot;,
        &quot;absolute_bar_plot_formatted(plot_group_array, &#39;Class&#39;, &#39;{{br}}&#39;, {{br_colors}})&quot;,  
        &quot;```&quot;,
        &quot;&quot;
    )
    knit_expand(text = src)
}, batch_region, batch_color_list)
out &lt;- knit_child(text = unlist(src_list), options = list(cache = FALSE))</code></pre>
<div id="batch-1" class="section level4 unnumbered">
<h4 class="unnumbered">Batch</h4>
<pre class="r"><code>absolute_bar_plot_formatted(plot_group_array, &#39;Class&#39;, &#39;Batch&#39;, batch_colors_rna)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/class-absolute-Batch-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="region-1" class="section level4 unnumbered">
<h4 class="unnumbered">Region</h4>
<pre class="r"><code>absolute_bar_plot_formatted(plot_group_array, &#39;Class&#39;, &#39;Region&#39;, region_colors)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/class-absolute-Region-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
</div>
</div>
<div id="relative-composition" class="section level2 tabset">
<h2 class="tabset">Relative Composition</h2>
<div id="tissue-4" class="section level3 tabset">
<h3 class="tabset">Tissue</h3>
<div id="bar" class="section level4 tabset">
<h4 class="tabset">Bar</h4>
<div id="batch-2" class="section level5">
<h5>Batch</h5>
<pre class="r"><code>relative_bar_plot_formatted(plot_group_array, &#39;Tissue&#39;, &#39;Batch&#39;, batch_colors_rna)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/20-tissue_relative_batch-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="region-2" class="section level5">
<h5>Region</h5>
<pre class="r"><code>tissue_region_bar &lt;- relative_bar_plot_formatted(plot_group_array, &#39;Tissue&#39;, &#39;Region&#39;, region_colors)

tissue_region_bar + geom_hline(yintercept = table(array$Region)[2] / length(array$Region), linetype = &quot;dashed&quot;)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/21-tissue_relative_region-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
</div>
<div id="chi-square-results" class="section level4">
<h4>Chi-square results</h4>
<pre class="r"><code>chisq_table &lt;- table(array$Region, array$Tissue)

chisq.test(chisq_table)</code></pre>
<pre><code>
    Pearson&#39;s Chi-squared test with Yates&#39; continuity correction

data:  chisq_table
X-squared = 0.96506, df = 1, p-value = 0.3259</code></pre>
<pre class="r"><code>chisq.posthoc.test(chisq_table, method = &quot;bonferroni&quot;)</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["Dimension"],"name":[1],"type":["chr"],"align":["left"]},{"label":["Value"],"name":[2],"type":["chr"],"align":["left"]},{"label":["Neuron"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["Nonneuron"],"name":[4],"type":["dbl"],"align":["right"]}],"data":[{"1":"aplCoA","2":"Residuals","3":"0.9917337","4":"-0.9917337"},{"1":"aplCoA","2":"p values","3":"1.0000000","4":"1.0000000"},{"1":"pplCoA","2":"Residuals","3":"-0.9917337","4":"0.9917337"},{"1":"pplCoA","2":"p values","3":"1.0000000","4":"1.0000000"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<pre class="r"><code># can&#39;t do propeller w/ only two groups, so skip here (if it doesn&#39;t pass chi-sq though, likely won&#39;t pass propeller)</code></pre>
</div>
</div>
<div id="class-4" class="section level3 tabset">
<h3 class="tabset">Class</h3>
<div id="bar-1" class="section level4 tabset">
<h4 class="tabset">Bar</h4>
<div id="batch-3" class="section level5">
<h5>Batch</h5>
<pre class="r"><code>relative_bar_plot_formatted(plot_group_array, &#39;Class&#39;, &#39;Batch&#39;, batch_colors_rna)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/23-tissue_relative_batch-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="region-3" class="section level5">
<h5>Region</h5>
<pre class="r"><code>class_region_bar &lt;- relative_bar_plot_formatted(plot_group_array, &#39;Class&#39;, &#39;Region&#39;, region_colors)

class_region_bar + geom_hline(yintercept = table(array$Region)[2] / length(array$Region), linetype = &quot;dashed&quot;)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/24-tissue_relative_region-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
</div>
<div id="chi-square-results-1" class="section level4">
<h4>Chi-square results</h4>
<pre class="r"><code># don&#39;t take this seriously for more complex comparisons, massively distorts p-values, mostly included for consistency

chisq_table &lt;- table(array$Region, array$Class)

chisq.test(chisq_table)</code></pre>
<pre><code>
    Pearson&#39;s Chi-squared test

data:  chisq_table
X-squared = 793.81, df = 8, p-value &lt; 2.2e-16</code></pre>
<pre class="r"><code>chisq.posthoc.test(chisq_table, method = &quot;bonferroni&quot;)</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["Dimension"],"name":[1],"type":["chr"],"align":["left"]},{"label":["Value"],"name":[2],"type":["chr"],"align":["left"]},{"label":["Glutamatergic"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["GABAergic"],"name":[4],"type":["dbl"],"align":["right"]},{"label":["Astro"],"name":[5],"type":["dbl"],"align":["right"]},{"label":["Microglia"],"name":[6],"type":["dbl"],"align":["right"]},{"label":["OPC"],"name":[7],"type":["dbl"],"align":["right"]},{"label":["OLG"],"name":[8],"type":["dbl"],"align":["right"]},{"label":["Endothelial"],"name":[9],"type":["dbl"],"align":["right"]},{"label":["Mural"],"name":[10],"type":["dbl"],"align":["right"]},{"label":["VLMC"],"name":[11],"type":["dbl"],"align":["right"]}],"data":[{"1":"aplCoA","2":"Residuals","3":"3.047376","4":"-3.261611","5":"1.686581","6":"-8.871931","7":"-16.83268","8":"4.629739","9":"-0.7645524","10":"13.05828","11":"16.29018"},{"1":"aplCoA","2":"p values","3":"0.041553","4":"0.019941","5":"1.000000","6":"0.000000","7":"0.00000","8":"0.000066","9":"1.0000000","10":"0.00000","11":"0.00000"},{"1":"pplCoA","2":"Residuals","3":"-3.047376","4":"3.261611","5":"-1.686581","6":"8.871931","7":"16.83268","8":"-4.629739","9":"0.7645524","10":"-13.05828","11":"-16.29018"},{"1":"pplCoA","2":"p values","3":"0.041553","4":"0.019941","5":"1.000000","6":"0.000000","7":"0.00000","8":"0.000066","9":"1.0000000","10":"0.00000","11":"0.00000"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
<div id="propeller-results" class="section level4">
<h4>Propeller results</h4>
<pre class="r"><code>props &lt;- propeller(array, clusters = array$Class, sample = array$orig.ident, group = array$Region)

prop_table &lt;- cbind.data.frame(rownames(props), props$BaselineProp.Freq*100, props$Tstatistic, props$FDR,
                               ifelse(props$FDR &gt; 0.05, &quot;ns&quot;, ifelse(
                                 props$FDR &gt; 0.01, &quot;*&quot;, ifelse(
                                   props$FDR &gt; 0.001, &quot;**&quot;, ifelse(
                                     props$FDR &gt; 0.0001, &quot;***&quot;, &quot;****&quot;
                               )))))
colnames(prop_table) &lt;- c(&quot;Tissue&quot;, &quot;% Freq&quot;, &quot;F-statistic&quot;, &quot;FDR&quot;, &quot;Significance&quot;)

prop_table</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["Tissue"],"name":[1],"type":["chr"],"align":["left"]},{"label":["% Freq"],"name":[2],"type":["dbl"],"align":["right"]},{"label":["F-statistic"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["FDR"],"name":[4],"type":["dbl"],"align":["right"]},{"label":["Significance"],"name":[5],"type":["chr"],"align":["left"]}],"data":[{"1":"VLMC","2":"2.294487","3":"6.6518136","4":"0.005123556","5":"**"},{"1":"OPC","2":"8.269012","3":"-4.1994030","4":"0.025884776","5":"*"},{"1":"Microglia","2":"7.598380","3":"-3.6477194","4":"0.032486600","5":"*"},{"1":"Mural","2":"4.089164","3":"2.7442170","4":"0.075865253","5":"ns"},{"1":"OLG","2":"3.298326","3":"1.7011758","4":"0.252104054","5":"ns"},{"1":"GABAergic","2":"10.858746","3":"-1.3246182","4":"0.350611267","5":"ns"},{"1":"Astro","2":"20.260239","3":"0.8376107","4":"0.558632004","5":"ns"},{"1":"Endothelial","2":"0.240415","3":"0.2634313","4":"0.875803240","5":"ns"},{"1":"Glutamatergic","2":"43.091231","3":"-0.1631226","4":"0.875803240","5":"ns"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
<div id="region-4" class="section level3 tabset">
<h3 class="tabset">Region</h3>
<div id="absolute" class="section level4 tabset">
<h4 class="tabset">Absolute</h4>
<pre class="r"><code>src_list &lt;- mapply(function(tc, tc_colors) {
    src &lt;- c(
        &quot;##### {{tc}} {.unnumbered}&quot;,
        &quot;```{r region-absolute-{{tc}}, message = FALSE, warning = FALSE}&quot;,
        &quot;absolute_bar_plot_formatted(plot_group_array, &#39;Region&#39;, &#39;{{tc}}&#39;, {{tc_colors}})&quot;,  
        &quot;```&quot;,
        &quot;&quot;
    )
    knit_expand(text = src)
}, tissue_class, tissue_color_list)
out &lt;- knit_child(text = unlist(src_list), options = list(cache = FALSE))</code></pre>
<div id="tissue-5" class="section level5 unnumbered">
<h5 class="unnumbered">Tissue</h5>
<pre class="r"><code>absolute_bar_plot_formatted(plot_group_array, &#39;Region&#39;, &#39;Tissue&#39;, tissue_colors)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/region-absolute-Tissue-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="class-5" class="section level5 unnumbered">
<h5 class="unnumbered">Class</h5>
<pre class="r"><code>absolute_bar_plot_formatted(plot_group_array, &#39;Region&#39;, &#39;Class&#39;, class_colors)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/region-absolute-Class-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
</div>
<div id="relative" class="section level4 tabset">
<h4 class="tabset">Relative</h4>
<pre class="r"><code>src_list &lt;- mapply(function(tc, tc_colors) {
    src &lt;- c(
        &quot;##### {{tc}} {.unnumbered}&quot;,
        &quot;```{r region-relative-{{tc}}, message = FALSE, warning = FALSE}&quot;,
        &quot;relative_bar_plot_formatted(plot_group_array, &#39;Region&#39;, &#39;{{tc}}&#39;, {{tc_colors}})&quot;,  
        &quot;```&quot;,
        &quot;&quot;
    )
    knit_expand(text = src)
}, tissue_class, tissue_color_list)
out &lt;- knit_child(text = unlist(src_list), options = list(cache = FALSE))</code></pre>
<div id="tissue-6" class="section level5 unnumbered">
<h5 class="unnumbered">Tissue</h5>
<pre class="r"><code>relative_bar_plot_formatted(plot_group_array, &#39;Region&#39;, &#39;Tissue&#39;, tissue_colors)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/region-relative-Tissue-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="class-6" class="section level5 unnumbered">
<h5 class="unnumbered">Class</h5>
<pre class="r"><code>relative_bar_plot_formatted(plot_group_array, &#39;Region&#39;, &#39;Class&#39;, class_colors)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/region-relative-Class-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
</div>
<div id="chi-square-results-2" class="section level4">
<h4>Chi-square results</h4>
<pre class="r"><code>chisq_table &lt;- table(array$Class, array$Region)

chisq.test(chisq_table)</code></pre>
<pre><code>
    Pearson&#39;s Chi-squared test

data:  chisq_table
X-squared = 793.81, df = 8, p-value &lt; 2.2e-16</code></pre>
<pre class="r"><code>chisq.posthoc.test(chisq_table, method = &quot;bonferroni&quot;)</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["Dimension"],"name":[1],"type":["chr"],"align":["left"]},{"label":["Value"],"name":[2],"type":["chr"],"align":["left"]},{"label":["aplCoA"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["pplCoA"],"name":[4],"type":["dbl"],"align":["right"]}],"data":[{"1":"Glutamatergic","2":"Residuals","3":"3.0473763","4":"-3.0473763"},{"1":"Glutamatergic","2":"p values","3":"0.0415530","4":"0.0415530"},{"1":"GABAergic","2":"Residuals","3":"-3.2616114","4":"3.2616114"},{"1":"GABAergic","2":"p values","3":"0.0199410","4":"0.0199410"},{"1":"Astro","2":"Residuals","3":"1.6865810","4":"-1.6865810"},{"1":"Astro","2":"p values","3":"1.0000000","4":"1.0000000"},{"1":"Microglia","2":"Residuals","3":"-8.8719309","4":"8.8719309"},{"1":"Microglia","2":"p values","3":"0.0000000","4":"0.0000000"},{"1":"OPC","2":"Residuals","3":"-16.8326770","4":"16.8326770"},{"1":"OPC","2":"p values","3":"0.0000000","4":"0.0000000"},{"1":"OLG","2":"Residuals","3":"4.6297386","4":"-4.6297386"},{"1":"OLG","2":"p values","3":"0.0000660","4":"0.0000660"},{"1":"Endothelial","2":"Residuals","3":"-0.7645524","4":"0.7645524"},{"1":"Endothelial","2":"p values","3":"1.0000000","4":"1.0000000"},{"1":"Mural","2":"Residuals","3":"13.0582770","4":"-13.0582770"},{"1":"Mural","2":"p values","3":"0.0000000","4":"0.0000000"},{"1":"VLMC","2":"Residuals","3":"16.2901823","4":"-16.2901823"},{"1":"VLMC","2":"p values","3":"0.0000000","4":"0.0000000"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
</div>
<div id="examine-metrics-for-consolidated-clusters" class="section level2 tabset">
<h2 class="tabset">Examine metrics for consolidated clusters</h2>
<div id="tissue-7" class="section level3 tabset">
<h3 class="tabset">Tissue</h3>
<pre class="r"><code>src_list &lt;- mapply(function(metric, label) {
    src &lt;- c(
        &quot;#### {{label}} {.unnumbered}&quot;,
        &quot;```{r vln-tissue-{{metric}}, message = FALSE, warning = FALSE}&quot;,
        &quot;violin_metrics_formatted(array, &#39;Tissue&#39;, &#39;{{metric}}&#39;, &#39;{{label}}&#39;, tissue_colors)&quot;,  
        &quot;```&quot;,
        &quot;&quot;
    )
    knit_expand(text = src)
}, vln_metrics$id, vln_metrics$label)
out &lt;- knit_child(text = unlist(src_list), options = list(cache = FALSE))</code></pre>
<div id="umis" class="section level4 unnumbered">
<h4 class="unnumbered">UMIs</h4>
<pre class="r"><code>violin_metrics_formatted(array, &#39;Tissue&#39;, &#39;nCount_RNA&#39;, &#39;UMIs&#39;, tissue_colors)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/vln-tissue-nCount_RNA-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="genes" class="section level4 unnumbered">
<h4 class="unnumbered">Genes</h4>
<pre class="r"><code>violin_metrics_formatted(array, &#39;Tissue&#39;, &#39;nFeature_RNA&#39;, &#39;Genes&#39;, tissue_colors)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/vln-tissue-nFeature_RNA-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="mito" class="section level4 unnumbered">
<h4 class="unnumbered">% Mito</h4>
<pre class="r"><code>violin_metrics_formatted(array, &#39;Tissue&#39;, &#39;Mito_proportion&#39;, &#39;% Mito&#39;, tissue_colors)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/vln-tissue-Mito_proportion-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="ribo" class="section level4 unnumbered">
<h4 class="unnumbered">% Ribo</h4>
<pre class="r"><code>violin_metrics_formatted(array, &#39;Tissue&#39;, &#39;Ribo_proportion&#39;, &#39;% Ribo&#39;, tissue_colors)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/vln-tissue-Ribo_proportion-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
</div>
<div id="group" class="section level3 tabset">
<h3 class="tabset">Group</h3>
<pre class="r"><code>src_list &lt;- mapply(function(metric, label) {
    src &lt;- c(
        &quot;#### {{label}} {.unnumbered}&quot;,
        &quot;```{r vln-class-{{metric}}, message = FALSE, warning = FALSE}&quot;,
        &quot;violin_metrics_formatted(array, &#39;Class&#39;, &#39;{{metric}}&#39;, &#39;{{label}}&#39;, class_colors)&quot;,  
        &quot;```&quot;,
        &quot;&quot;
    )
    knit_expand(text = src)
}, vln_metrics$id, vln_metrics$label)
out &lt;- knit_child(text = unlist(src_list), options = list(cache = FALSE))</code></pre>
<div id="umis-1" class="section level4 unnumbered">
<h4 class="unnumbered">UMIs</h4>
<pre class="r"><code>violin_metrics_formatted(array, &#39;Class&#39;, &#39;nCount_RNA&#39;, &#39;UMIs&#39;, class_colors)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/vln-class-nCount_RNA-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="genes-1" class="section level4 unnumbered">
<h4 class="unnumbered">Genes</h4>
<pre class="r"><code>violin_metrics_formatted(array, &#39;Class&#39;, &#39;nFeature_RNA&#39;, &#39;Genes&#39;, class_colors)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/vln-class-nFeature_RNA-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="mito-1" class="section level4 unnumbered">
<h4 class="unnumbered">% Mito</h4>
<pre class="r"><code>violin_metrics_formatted(array, &#39;Class&#39;, &#39;Mito_proportion&#39;, &#39;% Mito&#39;, class_colors)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/vln-class-Mito_proportion-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="ribo-1" class="section level4 unnumbered">
<h4 class="unnumbered">% Ribo</h4>
<pre class="r"><code>violin_metrics_formatted(array, &#39;Class&#39;, &#39;Ribo_proportion&#39;, &#39;% Ribo&#39;, class_colors)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/vln-class-Ribo_proportion-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
</div>
</div>
<div id="visualize-marker-genes-in-consolidated-clusters" class="section level2 tabset">
<h2 class="tabset">Visualize marker genes in consolidated clusters</h2>
<div id="dot-plot-1" class="section level3 tabset">
<h3 class="tabset">Dot Plot</h3>
<pre class="r"><code>src_list &lt;- lapply(tissue_class, function(tg) {
    src &lt;- c(
        &quot;#### {{tg}} {.unnumbered}&quot;,
        &quot;```{r dot-{{tg}}, message = FALSE, warning = FALSE, fig.height = 4, fig.width = 20}&quot;,
        &quot;DotPlot(array, features = marker_features, group.by = &#39;{{tg}}&#39;) +
                 scale_colour_gradientn(colours = dotplot_colors)&quot;,  
        &quot;```&quot;,
        &quot;&quot;
    )
    knit_expand(text = src)
})
out &lt;- knit_child(text = unlist(src_list), options = list(cache = FALSE))</code></pre>
<div id="tissue-8" class="section level4 unnumbered">
<h4 class="unnumbered">Tissue</h4>
<pre class="r"><code>DotPlot(array, features = marker_features, group.by = &#39;Tissue&#39;) +
                 scale_colour_gradientn(colours = dotplot_colors)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/dot-Tissue-1.pdf" width="1920" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="class-7" class="section level4 unnumbered">
<h4 class="unnumbered">Class</h4>
<pre class="r"><code>DotPlot(array, features = marker_features, group.by = &#39;Class&#39;) +
                 scale_colour_gradientn(colours = dotplot_colors)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/dot-Class-1.pdf" width="1920" style="display: block; margin: auto;" type="application/pdf" />
</div>
</div>
<div id="stacked-violin" class="section level3 tabset">
<h3 class="tabset">Stacked Violin</h3>
<pre class="r"><code>src_list &lt;- mapply(function(tg, tc) {
    src &lt;- c(
        &quot;#### {{tg}} {.unnumbered}&quot;,
        &quot;```{r stacked_vln_marker-{{tg}}, message = FALSE, warning = FALSE, fig.height = 12, fig.width = 12}&quot;,
        &quot;VlnPlot(array, features = marker_features, 
                 pt.size = 0, stack = TRUE, flip = TRUE, 
                 group.by = &#39;{{tg}}&#39;) + 
                 theme(legend.position = &#39;none&#39;)&quot;,  
        &quot;```&quot;,
        &quot;&quot;
    )
    knit_expand(text = src)
}, tissue_class, tissue_color_list)
out &lt;- knit_child(text = unlist(src_list), options = list(cache = FALSE))</code></pre>
<div id="tissue-9" class="section level4 unnumbered">
<h4 class="unnumbered">Tissue</h4>
<pre class="r"><code>VlnPlot(array, features = marker_features, 
                 pt.size = 0, stack = TRUE, flip = TRUE, 
                 group.by = &#39;Tissue&#39;) + 
                 theme(legend.position = &#39;none&#39;)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/stacked_vln_marker-Tissue-1.pdf" width="1152" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="class-8" class="section level4 unnumbered">
<h4 class="unnumbered">Class</h4>
<pre class="r"><code>VlnPlot(array, features = marker_features, 
                 pt.size = 0, stack = TRUE, flip = TRUE, 
                 group.by = &#39;Class&#39;) + 
                 theme(legend.position = &#39;none&#39;)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/stacked_vln_marker-Class-1.pdf" width="1152" style="display: block; margin: auto;" type="application/pdf" />
</div>
</div>
<div id="heatmap" class="section level3 tabset">
<h3 class="tabset">Heatmap</h3>
<pre class="r"><code>src_list &lt;- mapply(function(tg, tc) {
    src &lt;- c(
        &quot;#### {{tg}} {.unnumbered}&quot;,
        &quot;```{r heatmap_marker-{{tg}}, message = FALSE, warning = FALSE, dev = &#39;png&#39;, fig.height = 8, fig.width = 16}&quot;,
        &quot;DoHeatmap(array, features = marker_features, group.by = &#39;{{tg}}&#39;, size = 0,
                   group.colors = {{tc}}) +
                   scale_fill_gradientn(colours = heatmap_colors) +  
                   scale_colour_manual(values = {{tc}})&quot;,  
        &quot;```&quot;,
        &quot;&quot;
    )
    knit_expand(text = src)
}, tissue_class, tissue_color_list)
out &lt;- knit_child(text = unlist(src_list), options = list(cache = FALSE))</code></pre>
<div id="tissue-10" class="section level4 unnumbered">
<h4 class="unnumbered">Tissue</h4>
<pre class="r"><code>DoHeatmap(array, features = marker_features, group.by = &#39;Tissue&#39;, size = 0,
                   group.colors = tissue_colors) +
                   scale_fill_gradientn(colours = heatmap_colors) +  
                   scale_colour_manual(values = tissue_colors)</code></pre>
<p><img src="figure/sequencing-03-initial_classification.Rmd/heatmap_marker-Tissue-1.png" width="1536" style="display: block; margin: auto;" /></p>
</div>
<div id="class-9" class="section level4 unnumbered">
<h4 class="unnumbered">Class</h4>
<pre class="r"><code>DoHeatmap(array, features = marker_features, group.by = &#39;Class&#39;, size = 0,
                   group.colors = class_colors) +
                   scale_fill_gradientn(colours = heatmap_colors) +  
                   scale_colour_manual(values = class_colors)</code></pre>
<p><img src="figure/sequencing-03-initial_classification.Rmd/heatmap_marker-Class-1.png" width="1536" style="display: block; margin: auto;" /></p>
</div>
</div>
</div>
<div id="visualize-cluster-distances" class="section level2 tabset">
<h2 class="tabset">Visualize cluster distances</h2>
<div id="dendrogram" class="section level3 tabset">
<h3 class="tabset">Dendrogram</h3>
<div id="class-10" class="section level4">
<h4>Class</h4>
<pre class="r"><code>class_dend &lt;- create_dendrogram(array, array$Class, class_colors)

plot(class_dend[[2]], axes = FALSE)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/35-class_dendrogram-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="batch-4" class="section level4">
<h4>Batch</h4>
<pre class="r"><code>batch_dend &lt;- create_dendrogram(array, as.factor(array$orig.ident), batch_colors_rna)

plot(batch_dend[[2]], axes = FALSE)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/36-library_dendrogram-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="class-with-region-id" class="section level4">
<h4>Class with Region ID</h4>
<pre class="r"><code># create region+class convolution ID
array &lt;- AddMetaData(array, metadata = paste(array$Region, array$Class), col.name = &quot;Region_Class&quot;)

class_region_dend &lt;- create_dendrogram(array, as.factor(array$Region_Class), region_class_colors)

plot(class_region_dend[[2]], axes = FALSE)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/37-class_dendrogram_region-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
</div>
<div id="cluster-distance-heatmap" class="section level3 tabset">
<h3 class="tabset">Cluster distance heatmap</h3>
<div id="class-11" class="section level4 tabset">
<h4 class="tabset">Class</h4>
<div id="distance" class="section level5">
<h5>Distance</h5>
<pre class="r"><code>dendrogram_heatmap(class_dend, method_id = &quot;distance&quot;)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/38-cluster_heatmap_noregion_dist-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="correlation" class="section level5">
<h5>Correlation</h5>
<pre class="r"><code>dendrogram_heatmap(class_dend, method_id = &quot;correlation&quot;)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/39-cluster_heatmap_noregion_cor-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
</div>
<div id="batch-5" class="section level4 tabset">
<h4 class="tabset">Batch</h4>
<div id="distance-1" class="section level5">
<h5>Distance</h5>
<pre class="r"><code>dendrogram_heatmap(batch_dend, method_id = &quot;distance&quot;)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/40-cluster_heatmap_batch_dist-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="correlation-1" class="section level5">
<h5>Correlation</h5>
<pre class="r"><code>dendrogram_heatmap(batch_dend, method_id = &quot;correlation&quot;)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/41-cluster_heatmap_batch_cor-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
</div>
<div id="class-with-region-id-1" class="section level4 tabset">
<h4 class="tabset">Class with Region ID</h4>
<div id="distance-2" class="section level5">
<h5>Distance</h5>
<pre class="r"><code>dendrogram_heatmap(class_region_dend, method_id = &quot;distance&quot;)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/42-cluster_heatmap_region_dist-1.pdf" width="2880" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="correlation-2" class="section level5">
<h5>Correlation</h5>
<pre class="r"><code>dendrogram_heatmap(class_region_dend, method_id = &quot;correlation&quot;)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/43-cluster_heatmap_region_cor-1.pdf" width="2880" style="display: block; margin: auto;" type="application/pdf" />
</div>
</div>
</div>
<div id="pseudobulk-pca" class="section level3 tabset">
<h3 class="tabset">Pseudobulk PCA</h3>
<div id="class-12" class="section level4">
<h4>Class</h4>
<pre class="r"><code>pseudobulk_pca(array, &quot;Class&quot;, &quot;Class&quot;, class_colors)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/44-pseudobulk_pca_class-1.pdf" width="960" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="batch-6" class="section level4">
<h4>Batch</h4>
<pre class="r"><code>pseudobulk_pca(array, &quot;orig.ident&quot;, &quot;Batch&quot;, batch_colors_rna)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/45-pseudobulk_pca_batch-1.pdf" width="960" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="region-class" class="section level4">
<h4>Region-Class</h4>
<pre class="r"><code>pseudobulk_pca(array, &quot;Region_Class&quot;, &quot;Region_Class&quot;, region_class_colors)</code></pre>
<embed src="figure/sequencing-03-initial_classification.Rmd/46-pseudobulk_pca_region-class-1.pdf" width="960" style="display: block; margin: auto;" type="application/pdf" />
</div>
</div>
</div>
</div>
<div id="split-matrix-into-constituent-parts-and-save" class="section level1">
<h1>Split matrix into constituent parts and save</h1>
<pre class="r"><code># save cluster assignments, do not need whole array
# save the subgroups/classes
for (i in 1:length(levels(array$Class))){
  subset_array &lt;- subset(array, subset = Class == levels(array$Class)[i])
  metadata_array &lt;- cbind.data.frame(subset_array$Region,
                                     subset_array$Mito_proportion, subset_array$Ribo_proportion,
                                     subset_array$Tissue,
                                     subset_array$Class)
  colnames(metadata_array) &lt;- c(&quot;Region&quot;, 
                                &quot;Mito_proportion&quot;, &quot;Ribo_proportion&quot;, 
                                &quot;Tissue&quot;, &quot;Class&quot;)
  subset_array &lt;- CreateSeuratObject(subset_array@assays$RNA@counts)
  for (j in 1:length(colnames(metadata_array))){
    subset_array &lt;- AddMetaData(subset_array, 
                                metadata = metadata_array[,j], 
                                col.name = colnames(metadata_array)[j])
  }
  saveRDS(subset_array, file = paste0(output, id, &quot;_&quot;, levels(array$Class)[i], &quot;_initial.rds&quot;))
  gc()
}

# need to loop separately due to subsetting quirks
# save the overgroups/tissues
for (i in 1:length(levels(array$Tissue))){
  subset_array &lt;- subset(array, subset = Tissue == levels(array$Tissue)[i])
  metadata_array &lt;- cbind.data.frame(subset_array$Region,
                                     subset_array$Mito_proportion, subset_array$Ribo_proportion,
                                     subset_array$Tissue,
                                     subset_array$Class)
  colnames(metadata_array) &lt;- c(&quot;Region&quot;, 
                                &quot;Mito_proportion&quot;, &quot;Ribo_proportion&quot;, 
                                &quot;Tissue&quot;, &quot;Class&quot;)
  subset_array &lt;- CreateSeuratObject(subset_array@assays$RNA@counts)
  for (j in 1:length(colnames(metadata_array))){
    subset_array &lt;- AddMetaData(subset_array, 
                                metadata = metadata_array[,j], 
                                col.name = colnames(metadata_array)[j])
  }
  saveRDS(subset_array, file = paste0(output, id, &quot;_&quot;, levels(array$Tissue)[i], &quot;_initial.rds&quot;))
  gc()
}

saveRDS(array, file = paste0(output, id, &quot;_all_initial.rds&quot;))
gc()

# save cluster assignments, do not need whole array
cluster_file &lt;- cbind.data.frame(array$Tissue, array$Class)
colnames(cluster_file) &lt;- c(&quot;Tissue&quot;, &quot;Class&quot;)
write.csv(cluster_file, file = &quot;output/sequencing/cluster_lists/03-initial_clusters.csv&quot;, col.names = TRUE)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.1 (2022-06-23)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur ... 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices datasets  utils     methods   base     

other attached packages:
 [1] colorRamps_2.3.1         heatmap3_1.1.9           dendextend_1.16.0       
 [4] pvclust_2.2-0            speckle_0.99.1           chisq.posthoc.test_0.1.2
 [7] clustree_0.5.0           ggraph_2.1.0             scCustomize_0.7.0       
[10] knitr_1.40               forcats_0.5.2            stringr_1.4.1           
[13] dplyr_1.0.10             purrr_0.3.5              readr_2.1.3             
[16] tidyr_1.2.1              tibble_3.1.8             ggplot2_3.3.6           
[19] tidyverse_1.3.2          sp_1.5-0                 SeuratObject_4.1.2      
[22] Seurat_4.2.0             matrixStats_0.62.0       colorspace_2.0-3        
[25] farver_2.1.1             colorjam_0.0.23.900      jamba_0.0.87.900        
[28] viridis_0.6.2            viridisLite_0.4.1        workflowr_1.7.0         

loaded via a namespace (and not attached):
  [1] ggprism_1.0.3.9000          scattermore_0.8            
  [3] irlba_2.3.5.1               DelayedArray_0.22.0        
  [5] data.table_1.14.4           rpart_4.1.16               
  [7] RCurl_1.98-1.9              generics_0.1.3             
  [9] BiocGenerics_0.42.0         callr_3.7.2                
 [11] cowplot_1.1.1               RANN_2.6.1                 
 [13] future_1.28.0               tzdb_0.3.0                 
 [15] spatstat.data_2.2-0         xml2_1.3.3                 
 [17] lubridate_1.8.0             httpuv_1.6.6               
 [19] SummarizedExperiment_1.26.1 assertthat_0.2.1           
 [21] gargle_1.2.1                xfun_0.34                  
 [23] hms_1.1.2                   jquerylib_0.1.4            
 [25] evaluate_0.17               promises_1.2.0.1           
 [27] fansi_1.0.3                 dbplyr_2.2.1               
 [29] readxl_1.4.1                igraph_1.3.5               
 [31] DBI_1.1.3                   htmlwidgets_1.5.4          
 [33] spatstat.geom_2.4-0         googledrive_2.0.0          
 [35] stats4_4.2.1                paletteer_1.4.1            
 [37] ellipsis_0.3.2              ggpubr_0.4.0               
 [39] backports_1.4.1             prismatic_1.1.1            
 [41] deldir_1.0-6                MatrixGenerics_1.8.1       
 [43] vctrs_0.4.2                 SingleCellExperiment_1.18.1
 [45] Biobase_2.56.0              ROCR_1.0-11                
 [47] abind_1.4-5                 cachem_1.0.6               
 [49] withr_2.5.0                 ggforce_0.4.1              
 [51] progressr_0.11.0            checkmate_2.1.0            
 [53] sctransform_0.3.5           dittoSeq_1.9.3             
 [55] goftest_1.2-3               cluster_2.1.4              
 [57] lazyeval_0.2.2              crayon_1.5.2               
 [59] labeling_0.4.2              edgeR_3.38.4               
 [61] pkgconfig_2.0.3             tweenr_2.0.2               
 [63] GenomeInfoDb_1.32.4         nlme_3.1-160               
 [65] vipor_0.4.5                 rlang_1.0.6                
 [67] globals_0.16.1              lifecycle_1.0.3            
 [69] miniUI_0.1.1.1              modelr_0.1.9               
 [71] ggrastr_1.0.1               cellranger_1.1.0           
 [73] rprojroot_2.0.3             polyclip_1.10-0            
 [75] lmtest_0.9-40               Matrix_1.5-1               
 [77] carData_3.0-5               zoo_1.8-11                 
 [79] reprex_2.0.2                beeswarm_0.4.0             
 [81] whisker_0.4                 ggridges_0.5.4             
 [83] GlobalOptions_0.1.2         processx_3.7.0             
 [85] googlesheets4_1.0.1         pheatmap_1.0.12            
 [87] png_0.1-7                   bitops_1.0-7               
 [89] getPass_0.2-2               KernSmooth_2.23-20         
 [91] shape_1.4.6                 parallelly_1.32.1          
 [93] spatstat.random_2.2-0       rstatix_0.7.0              
 [95] S4Vectors_0.34.0            ggsignif_0.6.4             
 [97] scales_1.2.1                magrittr_2.0.3             
 [99] plyr_1.8.7                  ica_1.0-3                  
[101] zlibbioc_1.42.0             compiler_4.2.1             
[103] RColorBrewer_1.1-3          fitdistrplus_1.1-8         
[105] snakecase_0.11.0            cli_3.4.1                  
[107] XVector_0.36.0              listenv_0.8.0              
[109] patchwork_1.1.2             pbapply_1.5-0              
[111] ps_1.7.1                    MASS_7.3-58.1              
[113] mgcv_1.8-40                 tidyselect_1.2.0           
[115] stringi_1.7.8               highr_0.9                  
[117] yaml_2.3.6                  locfit_1.5-9.6             
[119] ggrepel_0.9.1               grid_4.2.1                 
[121] sass_0.4.2                  tools_4.2.1                
[123] future.apply_1.9.1          parallel_4.2.1             
[125] circlize_0.4.15             rstudioapi_0.14            
[127] git2r_0.30.1                janitor_2.1.0              
[129] gridExtra_2.3               Rtsne_0.16                 
[131] digest_0.6.30               BiocManager_1.30.18        
[133] rgeos_0.5-9                 shiny_1.7.2                
[135] Rcpp_1.0.9                  GenomicRanges_1.48.0       
[137] car_3.1-0                   broom_1.0.1                
[139] later_1.3.0                 RcppAnnoy_0.0.19           
[141] httr_1.4.4                  rvest_1.0.3                
[143] fs_1.5.2                    tensor_1.5                 
[145] reticulate_1.26             IRanges_2.30.1             
[147] splines_4.2.1               uwot_0.1.14                
[149] rematch2_2.1.2              spatstat.utils_2.3-1       
[151] graphlayouts_0.8.2          renv_0.15.4                
[153] plotly_4.10.0               xtable_1.8-4               
[155] jsonlite_1.8.2              tidygraph_1.2.2            
[157] R6_2.5.1                    pillar_1.8.1               
[159] htmltools_0.5.3             mime_0.12                  
[161] glue_1.6.2                  fastmap_1.1.0              
[163] codetools_0.2-18            utf8_1.2.2                 
[165] lattice_0.20-45             bslib_0.4.0                
[167] spatstat.sparse_2.1-1       ggbeeswarm_0.6.0           
[169] leiden_0.4.3                colorway_0.2.0             
[171] survival_3.4-0              limma_3.52.4               
[173] rmarkdown_2.17              munsell_0.5.0              
[175] fastcluster_1.2.3           GenomeInfoDbData_1.2.8     
[177] haven_2.5.1                 reshape2_1.4.4             
[179] gtable_0.3.1                spatstat.core_2.4-4        </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
