<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="James Howe" />


<title>NFOL cell type classification (final): plCoA</title>

<script src="site_libs/header-attrs-2.17/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="site_libs/pagedtable-1.1/js/pagedtable.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">CorticalAmygdala_pipelines</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jamesrhowe/CorticalAmygdala_pipelines">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">NFOL cell type classification (final):
plCoA</h1>
<h4 class="author">James Howe</h4>
<h4 class="date">13 December, 2022</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-12-13
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong>
<code>CorticalAmygdala_pipelines/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R
Markdown file created these results, you’ll want to first commit it to
the Git repo. If you’re still working on the analysis, you can ignore
this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220906code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20220906)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220906code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220906)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomjamesrhoweCorticalAmygdalapipelinestree47f3cfa02d05e518b29d3c64427e8b76d3844624targetblank47f3cfaa">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/jamesrhowe/CorticalAmygdala_pipelines/tree/47f3cfa02d05e518b29d3c64427e8b76d3844624" target="_blank">47f3cfa</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomjamesrhoweCorticalAmygdalapipelinestree47f3cfa02d05e518b29d3c64427e8b76d3844624targetblank47f3cfaa"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/jamesrhowe/CorticalAmygdala_pipelines/tree/47f3cfa02d05e518b29d3c64427e8b76d3844624" target="_blank">47f3cfa</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    data/behavior/
    Ignored:    data/projections/
    Ignored:    data/sequencing/
    Ignored:    renv/library/
    Ignored:    renv/staging/

Untracked files:
    Untracked:  analysis/behavior_01-topography-testing.Rmd
    Untracked:  analysis/behavior_01-topography.Rmd
    Untracked:  analysis/behavior_02-genetics.Rmd
    Untracked:  analysis/behavior_03-projections.Rmd
    Untracked:  analysis/behavior_04-projections_silencing_tmt.Rmd
    Untracked:  analysis/behavior_05-projections_silencing_2PE.Rmd
    Untracked:  analysis/projections.Rmd
    Untracked:  analysis/sequencing-02B-initial_classification_rd1_RNA.Rmd
    Untracked:  analysis/sequencing-02C-reprocess_Neuron_rd1_RNA.Rmd
    Untracked:  analysis/sequencing-02D-Neuron_classification_rd1_RNA.Rmd
    Untracked:  analysis/sequencing-02E-reprocess_Glut_rd1_RNA.Rmd
    Untracked:  analysis/sequencing-02F-reprocess_GABA_rd1_RNA.Rmd
    Untracked:  analysis/sequencing-02G-Glut_classification_rd1_RNA.Rmd
    Untracked:  analysis/sequencing-03A-merge_filtered_RNA.Rmd
    Untracked:  analysis/sequencing-03B-initial_classification_RNA.Rmd
    Untracked:  analysis/sequencing-04-initial_deg.Rmd
    Untracked:  analysis/sequencing-04A-reprocess_Neuron_RNA.Rmd
    Untracked:  analysis/sequencing-04B-reprocess_Nonneuron_RNA.Rmd
    Untracked:  analysis/sequencing-04C-Neuron_classification_RNA.Rmd
    Untracked:  analysis/sequencing-04D-Nonneuron_classification_RNA.Rmd
    Untracked:  analysis/sequencing-05C-reprocess_Astro_RNA.Rmd
    Untracked:  analysis/sequencing-05L-Glut_classification_RNA.Rmd
    Untracked:  analysis/sequencing-05M-GABA_classification_RNA.Rmd
    Untracked:  analysis/sequencing-05N-Astro_classification_RNA.Rmd
    Untracked:  analysis/sequencing-05O-Macro_classification_RNA.Rmd
    Untracked:  analysis/sequencing-05P-NFOL_classification_RNA.Rmd
    Untracked:  analysis/sequencing-05Q-MOL_classification_RNA.Rmd
    Untracked:  analysis/sequencing-05R-ABC_classification_RNA.Rmd
    Untracked:  analysis/sequencing-05S-VLMC_classification_RNA.Rmd
    Untracked:  analysis/sequencing-05X-reprocess_Astro.Rmd
    Untracked:  analysis/sequencing-05X-reprocess_GABAergic.Rmd
    Untracked:  analysis/sequencing-05X-reprocess_Glutamatergic.Rmd
    Untracked:  analysis/sequencing-05X-reprocess_Microglia.Rmd
    Untracked:  analysis/sequencing-05X-reprocess_Neuron.Rmd
    Untracked:  analysis/sequencing-05X-reprocess_Nonneuron.Rmd
    Untracked:  analysis/sequencing-05X-reprocess_OLG.Rmd
    Untracked:  analysis/sequencing-05X-reprocess_OPC.Rmd
    Untracked:  analysis/sequencing-05X-reprocess_VLMC.Rmd
    Untracked:  analysis/sequencing-06A-merged_classification_RNA.Rmd
    Untracked:  analysis/sequencing-06B-annotate_GABAergic.Rmd
    Untracked:  behav.csv
    Untracked:  code/behavior-color_scales.R
    Untracked:  code/behavior_misc.R
    Untracked:  code/sequencing-color_scales.R
    Untracked:  code/sequencing-plot_functions.R
    Untracked:  code/sequencing_misc.R
    Untracked:  output/sequencing/

Unstaged changes:
    Modified:   analysis/sequencing-02A-merge_rd1_RNA.Rmd
    Modified:   analysis/sequencing-03-initial_classification.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with
<code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>This notebook assigns broad identities based on well-known marker
genes, consolidates them into legible identities, and then exports them
for subclustering within their broad identities.</p>
<pre class="r"><code>id &lt;- params$id
cell_class &lt;- params$class
path &lt;- params$path
dimn &lt;- params$dimn

source(&quot;code/sequencing_misc.R&quot;)
source(&quot;code/sequencing-color_scales.R&quot;)
source(&quot;code/sequencing-plot_functions.R&quot;)

require(Seurat)
require(tidyverse)
require(knitr)
require(scCustomize)
require(clustree)
require(chisq.posthoc.test)
require(speckle)
require(ggtree)
require(pvclust)
require(dendextend)
require(heatmap3)
require(colorRamps)
require(Libra)
require(enrichR)
require(EnhancedVolcano)

# all images are PDFs, avoids pixelation
knitr::opts_chunk$set(dev = &#39;pdf&#39;)</code></pre>
<div id="read-in-data-from-step-02" class="section level2">
<h2>Read in data from step 02</h2>
<p>Imports the merged and transformed Seurat object from step 02. This
object should contain SCT data, all relevant metadata, and it should be
barcode- and feature-filtered already.</p>
<pre class="r"><code>array &lt;- readRDS(paste0(path, id, &quot;_&quot;, cell_class, &quot;_reprocessed_RNA.rds&quot;))</code></pre>
</div>
</div>
<div id="cluster-nuclei" class="section level1 tabset">
<h1 class="tabset">Cluster nuclei</h1>
<p>Clusters nuclei using Seurat’s implementation of the Leiden
clustering algorithm (see methods for references), which improves upon
the more traditional Louvain algorithm. Clustering is performed at a
variety of resolutions so that the best can be chosen from among the
different results, ensuring optimal cluster assignments prior to
downstream segmentation.</p>
<pre class="r"><code>array &lt;- FindNeighbors(array, dims = 1:dimn, k.param = 30, verbose = FALSE)

res_index &lt;- c(0.01, seq(0.1, 1, 0.1))
for (i in 1:length(res_index)){
  array &lt;- FindClusters(array, method = 4, resolution = res_index[i], verbose = FALSE)
}</code></pre>
<div id="clustree" class="section level2 tabset">
<h2 class="tabset">Clustree</h2>
<div id="standard" class="section level3">
<h3>Standard</h3>
<pre class="r"><code>clustree(array, prefix = &quot;SCT_snn_res.&quot;)</code></pre>
<embed src="figure/sequencing-05P-NFOL_classification_RNA.Rmd/3-clustree_standard-1.pdf" width="1344" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="stability" class="section level3">
<h3>Stability</h3>
<pre class="r"><code>clustree(array, prefix = &quot;SCT_snn_res.&quot;, node_colour = &quot;sc3_stability&quot;)</code></pre>
<embed src="figure/sequencing-05P-NFOL_classification_RNA.Rmd/4-clustree_stability-1.pdf" width="1344" style="display: block; margin: auto;" type="application/pdf" />
</div>
</div>
<div id="scatter" class="section level2 tabset">
<h2 class="tabset">Scatter</h2>
<div id="pca" class="section level3 tabset">
<h3 class="tabset">PCA</h3>
<pre class="r"><code>src_list &lt;- lapply(res_index, function(res) {
    src &lt;- c(
        &quot;#### Res {{res}} {.unnumbered}&quot;,
        &quot;```{r pca-res-{{res}}, message = FALSE, warning = FALSE}&quot;,
        &quot;resolution_plot_formatted(array, &#39;pca&#39;, {{res}}, &#39;SCT&#39;)&quot;,  
        &quot;```&quot;,
        &quot;&quot;
    )
    knit_expand(text = src)
})
out &lt;- knit_child(text = unlist(src_list), options = list(cache = FALSE))</code></pre>
<div id="res-0.01" class="section level4 unnumbered">
<h4 class="unnumbered">Res 0.01</h4>
<pre class="r"><code>resolution_plot_formatted(array, &#39;pca&#39;, 0.01, &#39;SCT&#39;)</code></pre>
<embed src="figure/sequencing-05P-NFOL_classification_RNA.Rmd/pca-res-0.01-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="res-0.1" class="section level4 unnumbered">
<h4 class="unnumbered">Res 0.1</h4>
<pre class="r"><code>resolution_plot_formatted(array, &#39;pca&#39;, 0.1, &#39;SCT&#39;)</code></pre>
<embed src="figure/sequencing-05P-NFOL_classification_RNA.Rmd/pca-res-0.1-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="res-0.2" class="section level4 unnumbered">
<h4 class="unnumbered">Res 0.2</h4>
<pre class="r"><code>resolution_plot_formatted(array, &#39;pca&#39;, 0.2, &#39;SCT&#39;)</code></pre>
<embed src="figure/sequencing-05P-NFOL_classification_RNA.Rmd/pca-res-0.2-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="res-0.3" class="section level4 unnumbered">
<h4 class="unnumbered">Res 0.3</h4>
<pre class="r"><code>resolution_plot_formatted(array, &#39;pca&#39;, 0.3, &#39;SCT&#39;)</code></pre>
<embed src="figure/sequencing-05P-NFOL_classification_RNA.Rmd/pca-res-0.3-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="res-0.4" class="section level4 unnumbered">
<h4 class="unnumbered">Res 0.4</h4>
<pre class="r"><code>resolution_plot_formatted(array, &#39;pca&#39;, 0.4, &#39;SCT&#39;)</code></pre>
<embed src="figure/sequencing-05P-NFOL_classification_RNA.Rmd/pca-res-0.4-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="res-0.5" class="section level4 unnumbered">
<h4 class="unnumbered">Res 0.5</h4>
<pre class="r"><code>resolution_plot_formatted(array, &#39;pca&#39;, 0.5, &#39;SCT&#39;)</code></pre>
<embed src="figure/sequencing-05P-NFOL_classification_RNA.Rmd/pca-res-0.5-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="res-0.6" class="section level4 unnumbered">
<h4 class="unnumbered">Res 0.6</h4>
<pre class="r"><code>resolution_plot_formatted(array, &#39;pca&#39;, 0.6, &#39;SCT&#39;)</code></pre>
<embed src="figure/sequencing-05P-NFOL_classification_RNA.Rmd/pca-res-0.6-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="res-0.7" class="section level4 unnumbered">
<h4 class="unnumbered">Res 0.7</h4>
<pre class="r"><code>resolution_plot_formatted(array, &#39;pca&#39;, 0.7, &#39;SCT&#39;)</code></pre>
<embed src="figure/sequencing-05P-NFOL_classification_RNA.Rmd/pca-res-0.7-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="res-0.8" class="section level4 unnumbered">
<h4 class="unnumbered">Res 0.8</h4>
<pre class="r"><code>resolution_plot_formatted(array, &#39;pca&#39;, 0.8, &#39;SCT&#39;)</code></pre>
<embed src="figure/sequencing-05P-NFOL_classification_RNA.Rmd/pca-res-0.8-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="res-0.9" class="section level4 unnumbered">
<h4 class="unnumbered">Res 0.9</h4>
<pre class="r"><code>resolution_plot_formatted(array, &#39;pca&#39;, 0.9, &#39;SCT&#39;)</code></pre>
<embed src="figure/sequencing-05P-NFOL_classification_RNA.Rmd/pca-res-0.9-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="res-1" class="section level4 unnumbered">
<h4 class="unnumbered">Res 1</h4>
<pre class="r"><code>resolution_plot_formatted(array, &#39;pca&#39;, 1, &#39;SCT&#39;)</code></pre>
<embed src="figure/sequencing-05P-NFOL_classification_RNA.Rmd/pca-res-1-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
</div>
<div id="tsne" class="section level3 tabset">
<h3 class="tabset">tSNE</h3>
<pre class="r"><code>src_list &lt;- lapply(res_index, function(res) {
    src &lt;- c(
        &quot;#### Res {{res}} {.unnumbered}&quot;,
        &quot;```{r tsne-res-{{res}}, message = FALSE, warning = FALSE}&quot;,
        &quot;resolution_plot_formatted(array, &#39;tsne&#39;, {{res}}, &#39;SCT&#39;)&quot;,  
        &quot;```&quot;,
        &quot;&quot;
    )
    knit_expand(text = src)
})
out &lt;- knit_child(text = unlist(src_list), options = list(cache = FALSE))</code></pre>
<div id="res-0.01-1" class="section level4 unnumbered">
<h4 class="unnumbered">Res 0.01</h4>
<pre class="r"><code>resolution_plot_formatted(array, &#39;tsne&#39;, 0.01, &#39;SCT&#39;)</code></pre>
<embed src="figure/sequencing-05P-NFOL_classification_RNA.Rmd/tsne-res-0.01-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="res-0.1-1" class="section level4 unnumbered">
<h4 class="unnumbered">Res 0.1</h4>
<pre class="r"><code>resolution_plot_formatted(array, &#39;tsne&#39;, 0.1, &#39;SCT&#39;)</code></pre>
<embed src="figure/sequencing-05P-NFOL_classification_RNA.Rmd/tsne-res-0.1-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="res-0.2-1" class="section level4 unnumbered">
<h4 class="unnumbered">Res 0.2</h4>
<pre class="r"><code>resolution_plot_formatted(array, &#39;tsne&#39;, 0.2, &#39;SCT&#39;)</code></pre>
<embed src="figure/sequencing-05P-NFOL_classification_RNA.Rmd/tsne-res-0.2-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="res-0.3-1" class="section level4 unnumbered">
<h4 class="unnumbered">Res 0.3</h4>
<pre class="r"><code>resolution_plot_formatted(array, &#39;tsne&#39;, 0.3, &#39;SCT&#39;)</code></pre>
<embed src="figure/sequencing-05P-NFOL_classification_RNA.Rmd/tsne-res-0.3-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="res-0.4-1" class="section level4 unnumbered">
<h4 class="unnumbered">Res 0.4</h4>
<pre class="r"><code>resolution_plot_formatted(array, &#39;tsne&#39;, 0.4, &#39;SCT&#39;)</code></pre>
<embed src="figure/sequencing-05P-NFOL_classification_RNA.Rmd/tsne-res-0.4-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="res-0.5-1" class="section level4 unnumbered">
<h4 class="unnumbered">Res 0.5</h4>
<pre class="r"><code>resolution_plot_formatted(array, &#39;tsne&#39;, 0.5, &#39;SCT&#39;)</code></pre>
<embed src="figure/sequencing-05P-NFOL_classification_RNA.Rmd/tsne-res-0.5-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="res-0.6-1" class="section level4 unnumbered">
<h4 class="unnumbered">Res 0.6</h4>
<pre class="r"><code>resolution_plot_formatted(array, &#39;tsne&#39;, 0.6, &#39;SCT&#39;)</code></pre>
<embed src="figure/sequencing-05P-NFOL_classification_RNA.Rmd/tsne-res-0.6-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="res-0.7-1" class="section level4 unnumbered">
<h4 class="unnumbered">Res 0.7</h4>
<pre class="r"><code>resolution_plot_formatted(array, &#39;tsne&#39;, 0.7, &#39;SCT&#39;)</code></pre>
<embed src="figure/sequencing-05P-NFOL_classification_RNA.Rmd/tsne-res-0.7-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="res-0.8-1" class="section level4 unnumbered">
<h4 class="unnumbered">Res 0.8</h4>
<pre class="r"><code>resolution_plot_formatted(array, &#39;tsne&#39;, 0.8, &#39;SCT&#39;)</code></pre>
<embed src="figure/sequencing-05P-NFOL_classification_RNA.Rmd/tsne-res-0.8-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="res-0.9-1" class="section level4 unnumbered">
<h4 class="unnumbered">Res 0.9</h4>
<pre class="r"><code>resolution_plot_formatted(array, &#39;tsne&#39;, 0.9, &#39;SCT&#39;)</code></pre>
<embed src="figure/sequencing-05P-NFOL_classification_RNA.Rmd/tsne-res-0.9-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="res-1-1" class="section level4 unnumbered">
<h4 class="unnumbered">Res 1</h4>
<pre class="r"><code>resolution_plot_formatted(array, &#39;tsne&#39;, 1, &#39;SCT&#39;)</code></pre>
<embed src="figure/sequencing-05P-NFOL_classification_RNA.Rmd/tsne-res-1-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
</div>
<div id="umap" class="section level3 tabset">
<h3 class="tabset">UMAP</h3>
<pre class="r"><code>src_list &lt;- lapply(res_index, function(res) {
    src &lt;- c(
        &quot;#### Res {{res}} {.unnumbered}&quot;,
        &quot;```{r umap-res-{{res}}, message = FALSE, warning = FALSE}&quot;,
        &quot;resolution_plot_formatted(array, &#39;umap&#39;, {{res}}, &#39;SCT&#39;)&quot;,  
        &quot;```&quot;,
        &quot;&quot;
    )
    knit_expand(text = src)
})
out &lt;- knit_child(text = unlist(src_list), options = list(cache = FALSE))</code></pre>
<div id="res-0.01-2" class="section level4 unnumbered">
<h4 class="unnumbered">Res 0.01</h4>
<pre class="r"><code>resolution_plot_formatted(array, &#39;umap&#39;, 0.01, &#39;SCT&#39;)</code></pre>
<embed src="figure/sequencing-05P-NFOL_classification_RNA.Rmd/umap-res-0.01-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="res-0.1-2" class="section level4 unnumbered">
<h4 class="unnumbered">Res 0.1</h4>
<pre class="r"><code>resolution_plot_formatted(array, &#39;umap&#39;, 0.1, &#39;SCT&#39;)</code></pre>
<embed src="figure/sequencing-05P-NFOL_classification_RNA.Rmd/umap-res-0.1-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="res-0.2-2" class="section level4 unnumbered">
<h4 class="unnumbered">Res 0.2</h4>
<pre class="r"><code>resolution_plot_formatted(array, &#39;umap&#39;, 0.2, &#39;SCT&#39;)</code></pre>
<embed src="figure/sequencing-05P-NFOL_classification_RNA.Rmd/umap-res-0.2-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="res-0.3-2" class="section level4 unnumbered">
<h4 class="unnumbered">Res 0.3</h4>
<pre class="r"><code>resolution_plot_formatted(array, &#39;umap&#39;, 0.3, &#39;SCT&#39;)</code></pre>
<embed src="figure/sequencing-05P-NFOL_classification_RNA.Rmd/umap-res-0.3-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="res-0.4-2" class="section level4 unnumbered">
<h4 class="unnumbered">Res 0.4</h4>
<pre class="r"><code>resolution_plot_formatted(array, &#39;umap&#39;, 0.4, &#39;SCT&#39;)</code></pre>
<embed src="figure/sequencing-05P-NFOL_classification_RNA.Rmd/umap-res-0.4-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="res-0.5-2" class="section level4 unnumbered">
<h4 class="unnumbered">Res 0.5</h4>
<pre class="r"><code>resolution_plot_formatted(array, &#39;umap&#39;, 0.5, &#39;SCT&#39;)</code></pre>
<embed src="figure/sequencing-05P-NFOL_classification_RNA.Rmd/umap-res-0.5-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="res-0.6-2" class="section level4 unnumbered">
<h4 class="unnumbered">Res 0.6</h4>
<pre class="r"><code>resolution_plot_formatted(array, &#39;umap&#39;, 0.6, &#39;SCT&#39;)</code></pre>
<embed src="figure/sequencing-05P-NFOL_classification_RNA.Rmd/umap-res-0.6-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="res-0.7-2" class="section level4 unnumbered">
<h4 class="unnumbered">Res 0.7</h4>
<pre class="r"><code>resolution_plot_formatted(array, &#39;umap&#39;, 0.7, &#39;SCT&#39;)</code></pre>
<embed src="figure/sequencing-05P-NFOL_classification_RNA.Rmd/umap-res-0.7-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="res-0.8-2" class="section level4 unnumbered">
<h4 class="unnumbered">Res 0.8</h4>
<pre class="r"><code>resolution_plot_formatted(array, &#39;umap&#39;, 0.8, &#39;SCT&#39;)</code></pre>
<embed src="figure/sequencing-05P-NFOL_classification_RNA.Rmd/umap-res-0.8-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="res-0.9-2" class="section level4 unnumbered">
<h4 class="unnumbered">Res 0.9</h4>
<pre class="r"><code>resolution_plot_formatted(array, &#39;umap&#39;, 0.9, &#39;SCT&#39;)</code></pre>
<embed src="figure/sequencing-05P-NFOL_classification_RNA.Rmd/umap-res-0.9-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="res-1-2" class="section level4 unnumbered">
<h4 class="unnumbered">Res 1</h4>
<pre class="r"><code>resolution_plot_formatted(array, &#39;umap&#39;, 1, &#39;SCT&#39;)</code></pre>
<embed src="figure/sequencing-05P-NFOL_classification_RNA.Rmd/umap-res-1-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
</div>
</div>
</div>
<div id="annotate-clusters" class="section level1">
<h1>Annotate clusters</h1>
<div id="select-optimal-resolution" class="section level2">
<h2>Select optimal resolution</h2>
<p>Optimal resolution can be selected by examining clustering on
dimension-reduced data to determine if they match visual clustering.
Further, clustree (and its packaged stability metric) can be used to
find the highest resolution that does not lose stability, providing an
optimal balance between over- and under-clustering. It is difficult to
automate this without some degree of supervision or hand-curation. This
is mostly to identify large clusters, so there is little need to get
extremely specific so long as clusters are coherent.</p>
<pre class="r"><code>optimal_res &lt;- 0.2

clusters &lt;- array[[paste0(&quot;SCT_snn_res.&quot;, optimal_res)]]
array[[&quot;seurat_clusters&quot;]] &lt;- NULL
for (i in 1:11){
  array[[paste0(&quot;SCT_snn_res.&quot;, res_index[i])]] &lt;- NULL 
}
array &lt;- AddMetaData(array, clusters, col.name = &quot;clusters&quot;)
Idents(array) &lt;- &quot;clusters&quot;

# helps downstrean formatting to regenerate cluster vector from the array vector
type_clusters_marker &lt;- type_clusters &lt;- as.character(array$clusters)

paste(optimal_res, &quot;is optimal clustering resolution&quot;)</code></pre>
<pre><code>[1] &quot;0.2 is optimal clustering resolution&quot;</code></pre>
</div>
<div id="identify-marker-genes" class="section level2">
<h2>Identify marker genes</h2>
<pre class="r"><code>pct_expressed_aplCoA &lt;- do.call(cbind.data.frame, 
                         (lapply(levels(array$clusters), 
                 function(x) rowSums(array@assays$RNA@counts[, array$clusters == x &amp; array$Region == &quot;aplCoA&quot;] != 0) 
                             / length(array$clusters[array$clusters == x &amp; array$Region == &quot;aplCoA&quot;]))))

pct_expressed_pplCoA &lt;- do.call(cbind.data.frame, 
                         (lapply(levels(array$clusters), 
                 function(x) rowSums(array@assays$RNA@counts[, array$clusters == x &amp; array$Region == &quot;pplCoA&quot;] != 0) 
                             / length(array$clusters[array$clusters == x &amp; array$Region == &quot;pplCoA&quot;]))))

pct_expressed &lt;- pmax(pct_expressed_aplCoA, pct_expressed_pplCoA)

colnames(pct_expressed) &lt;- levels(array$clusters)

rm(pct_expressed_aplCoA, pct_expressed_pplCoA)


# find de novo markers, clusters
marker_type &lt;- markers_sc_wilcox(array, &quot;clusters&quot;)

# find regional markers within cell types
marker_type_region &lt;- markers_pb_edgeRLRT(array, &quot;clusters&quot;, 0.1)</code></pre>
</div>
<div id="select-marker-genes" class="section level2">
<h2>Select marker genes</h2>
<pre class="r"><code># region-specific
marker_type_region_aplCoA &lt;- lapply(marker_type_region, 
                                    function(x) {x[x$p_val_adj &lt; .05 &amp; x$avg_logFC &gt; 0.585,]})
marker_type_region_pplCoA &lt;- lapply(marker_type_region, 
                                    function(x) {x[x$p_val_adj &lt; .05 &amp; x$avg_logFC &lt; -0.585,]})

# based on numbers for max of each one
deg_range_type &lt;- c(0, 50)

markers_list &lt;- c(&quot;Pdgfra&quot;, &quot;Cspg4&quot;, #OPC
                  &quot;Bmp4&quot;, &quot;Enpp6&quot;, # NFOL
                  &quot;Mog&quot;, &quot;Mal&quot;, # oligodendrocyte/OLG
                  &quot;Frmd4a&quot;, &quot;Mbp&quot;, # 0
                  &quot;Sgcd&quot;, &quot;Pcdh15&quot;) # 1 </code></pre>
</div>
</div>
<div id="consolidate-clusters" class="section level1">
<h1>Consolidate clusters</h1>
<p>Based on known marker gene expression, assign broad cellular
identities to clusters at the optimal resolution selected. These
identities are hand-assigned, and multiple clusters are in many cases
combined into broad cellular identities due to uneven differences
between cell types.</p>
<pre class="r"><code>type_clusters[type_clusters %in% &quot;0&quot;] &lt;- &quot;NFOL.1&quot;
type_clusters_marker[type_clusters_marker %in% &quot;0&quot;] &lt;- &quot;NFOL.Frmd4a&quot;
names(marker_type)[1] &lt;- &quot;NFOL.1 | NFOL.Frmd4a&quot;

type_clusters[type_clusters %in% &quot;1&quot;] &lt;- &quot;NFOL.2&quot;
type_clusters_marker[type_clusters_marker %in% &quot;1&quot;] &lt;- &quot;NFOL.Sgcd&quot;
names(marker_type)[2] &lt;- &quot;NFOL.2 | NFOL.Sgcd&quot;

# annotate all of these
names(marker_type_region_aplCoA) &lt;- names(marker_type_region_pplCoA) &lt;- names(marker_type)

# replace clusters with curated types
array &lt;- AddMetaData(array, as.factor(type_clusters), col.name = &quot;Type&quot;)
array &lt;- AddMetaData(array, as.factor(type_clusters_marker), col.name = &quot;Type.Marker&quot;)

array$Type &lt;- factor(array$Type, levels = c(&quot;NFOL.1&quot;, &quot;NFOL.2&quot;))
array$Type.Marker &lt;- factor(array$Type.Marker, levels = c(&quot;NFOL.Frmd4a&quot;, &quot;NFOL.Sgcd&quot;))

Idents(array) &lt;- &quot;Type&quot;

plot_group_array &lt;- cbind.data.frame(array$Type,
                                     array$orig.ident,
                                     array$Region)
colnames(plot_group_array) &lt;- c(&quot;Type&quot;, &quot;Batch&quot;, &quot;Region&quot;)

table(array$Type)</code></pre>
<pre><code>
NFOL.1 NFOL.2 
   209    109 </code></pre>
</div>
<div id="visualize-consolidated-clusters" class="section level1 tabset">
<h1 class="tabset">Visualize consolidated clusters</h1>
<pre class="r"><code>src_list &lt;- mapply(function(tab_name, dr_type) {
    src &lt;- c(
        &quot;## {{tab_name}} {.unnumbered}&quot;,
        &quot;```{r type-{{tab_name}}, message = FALSE, warning = FALSE}&quot;,
        &quot;cluster_plot_formatted(array, &#39;{{dr_type}}&#39;, &#39;Type&#39;, nfol_type_colors)&quot;,  
        &quot;```&quot;,
        &quot;&quot;
    )
    knit_expand(text = src)
}, dr_params$name, dr_params$type)
out &lt;- knit_child(text = unlist(src_list), options = list(cache = FALSE))</code></pre>
<div id="pca-1" class="section level2 unnumbered">
<h2 class="unnumbered">PCA</h2>
<pre class="r"><code>cluster_plot_formatted(array, &#39;pca&#39;, &#39;Type&#39;, nfol_type_colors)</code></pre>
<embed src="figure/sequencing-05P-NFOL_classification_RNA.Rmd/type-PCA-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="tsne-1" class="section level2 unnumbered">
<h2 class="unnumbered">tSNE</h2>
<pre class="r"><code>cluster_plot_formatted(array, &#39;tsne&#39;, &#39;Type&#39;, nfol_type_colors)</code></pre>
<embed src="figure/sequencing-05P-NFOL_classification_RNA.Rmd/type-tSNE-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="umap-1" class="section level2 unnumbered">
<h2 class="unnumbered">UMAP</h2>
<pre class="r"><code>cluster_plot_formatted(array, &#39;umap&#39;, &#39;Type&#39;, nfol_type_colors)</code></pre>
<embed src="figure/sequencing-05P-NFOL_classification_RNA.Rmd/type-UMAP-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
</div>
<div id="show-gene-expression-information" class="section level1">
<h1>Show gene expression information</h1>
<div id="volcano-region-markers" class="section level2 tabset">
<h2 class="tabset">Volcano (region markers)</h2>
<div id="unlabeled" class="section level3 tabset">
<h3 class="tabset">Unlabeled</h3>
<pre class="r"><code>src_list &lt;- lapply(names(marker_type_region), function(clusters) {
    src &lt;- c(
        &quot;#### {{clusters}} {.unnumbered}&quot;,
        &quot;```{r volcano-unlabeled-{{clusters}}, message = FALSE, warning = FALSE, fig.height = 12, fig.width = 12}&quot;,
        &quot;enhanced_volcano_plot(marker_type_region[[&#39;{{clusters}}&#39;]], 
                               title_id = paste(&#39;{{clusters}}&#39;), label_size = 0)&quot;,  
        &quot;```&quot;,
        &quot;&quot;
    )
    knit_expand(text = src)
})
out &lt;- knit_child(text = unlist(src_list), options = list(cache = FALSE))</code></pre>
<div id="section" class="section level4 unnumbered">
<h4 class="unnumbered">0</h4>
<pre class="r"><code>enhanced_volcano_plot(marker_type_region[[&#39;0&#39;]], 
                               title_id = paste(&#39;0&#39;), label_size = 0)</code></pre>
<embed src="figure/sequencing-05P-NFOL_classification_RNA.Rmd/volcano-unlabeled-0-1.pdf" width="1152" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="section-1" class="section level4 unnumbered">
<h4 class="unnumbered">1</h4>
<pre class="r"><code>enhanced_volcano_plot(marker_type_region[[&#39;1&#39;]], 
                               title_id = paste(&#39;1&#39;), label_size = 0)</code></pre>
<embed src="figure/sequencing-05P-NFOL_classification_RNA.Rmd/volcano-unlabeled-1-1.pdf" width="1152" style="display: block; margin: auto;" type="application/pdf" />
</div>
</div>
<div id="labeled" class="section level3 tabset">
<h3 class="tabset">Labeled</h3>
<pre class="r"><code>src_list &lt;- lapply(names(marker_type_region), function(clusters) {
    src &lt;- c(
        &quot;#### {{clusters}} {.unnumbered}&quot;,
        &quot;```{r volcano-labeled-{{clusters}}, message = FALSE, warning = FALSE, fig.height = 12, fig.width = 12}&quot;,
        &quot;enhanced_volcano_plot(marker_type_region[[&#39;{{clusters}}&#39;]], 
                               title_id = paste(&#39;{{clusters}}&#39;), label_size = 1)&quot;,  
        &quot;```&quot;,
        &quot;&quot;
    )
    knit_expand(text = src)
})
out &lt;- knit_child(text = unlist(src_list), options = list(cache = FALSE))</code></pre>
<div id="section-2" class="section level4 unnumbered">
<h4 class="unnumbered">0</h4>
<pre class="r"><code>enhanced_volcano_plot(marker_type_region[[&#39;0&#39;]], 
                               title_id = paste(&#39;0&#39;), label_size = 1)</code></pre>
<embed src="figure/sequencing-05P-NFOL_classification_RNA.Rmd/volcano-labeled-0-1.pdf" width="1152" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="section-3" class="section level4 unnumbered">
<h4 class="unnumbered">1</h4>
<pre class="r"><code>enhanced_volcano_plot(marker_type_region[[&#39;1&#39;]], 
                               title_id = paste(&#39;1&#39;), label_size = 1)</code></pre>
<embed src="figure/sequencing-05P-NFOL_classification_RNA.Rmd/volcano-labeled-1-1.pdf" width="1152" style="display: block; margin: auto;" type="application/pdf" />
</div>
</div>
</div>
<div id="list" class="section level2 tabset">
<h2 class="tabset">List</h2>
<div id="type" class="section level3 tabset">
<h3 class="tabset">Type</h3>
<pre class="r"><code>src_list &lt;- lapply(names(marker_type), function(group) {
    src &lt;- c(
        &quot;#### {{group}} {.unnumbered}&quot;,
        &quot;```{r marker-type-{{group}}, message = FALSE, warning = FALSE}&quot;,
        &quot;marker_type[[&#39;{{group}}&#39;]]&quot;,  
        &quot;```&quot;,
        &quot;&quot;
    )
    knit_expand(text = src)
})
out &lt;- knit_child(text = unlist(src_list), options = list(cache = FALSE))</code></pre>
<div id="nfol.1-nfol.frmd4a" class="section level4 unnumbered">
<h4 class="unnumbered">NFOL.1 | NFOL.Frmd4a</h4>
<pre class="r"><code>marker_type[[&#39;NFOL.1 | NFOL.Frmd4a&#39;]]</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":[""],"name":["_rn_"],"type":[""],"align":["left"]},{"label":["gene"],"name":[1],"type":["chr"],"align":["left"]},{"label":["combined_p_val"],"name":[2],"type":["dbl"],"align":["right"]}],"data":[{"1":"Frmd4a","2":"3.588493e-36","_rn_":"1"},{"1":"Nfasc","2":"6.964570e-34","_rn_":"2"},{"1":"Tmem108","2":"2.125491e-33","_rn_":"3"},{"1":"Mbp","2":"2.630549e-32","_rn_":"4"},{"1":"Gm13187","2":"5.948385e-30","_rn_":"5"},{"1":"Pdcd4","2":"1.415253e-28","_rn_":"6"},{"1":"Gjc3","2":"1.570210e-26","_rn_":"7"},{"1":"Plekhg1","2":"2.618461e-26","_rn_":"8"},{"1":"Map1b","2":"7.234495e-26","_rn_":"9"},{"1":"Gm36988","2":"1.931619e-25","_rn_":"10"},{"1":"Pik3r3","2":"8.657190e-24","_rn_":"11"},{"1":"Fam13c","2":"8.339933e-23","_rn_":"12"},{"1":"Nhsl2","2":"4.297154e-20","_rn_":"13"},{"1":"Ephb1","2":"1.610935e-19","_rn_":"14"},{"1":"Parvb","2":"2.515785e-19","_rn_":"15"},{"1":"9130019P16Rik","2":"4.661049e-19","_rn_":"16"},{"1":"Dpysl2","2":"1.008756e-18","_rn_":"17"},{"1":"Bbip1","2":"1.649820e-18","_rn_":"18"},{"1":"Tmem163","2":"1.294985e-17","_rn_":"19"},{"1":"Specc1","2":"1.742408e-17","_rn_":"20"},{"1":"Mical3","2":"2.750902e-17","_rn_":"21"},{"1":"Cldn11","2":"6.300789e-17","_rn_":"22"},{"1":"Cerk","2":"5.204092e-16","_rn_":"23"},{"1":"Mag","2":"4.701395e-14","_rn_":"24"},{"1":"Frmd4b","2":"5.891920e-14","_rn_":"25"},{"1":"Prickle1","2":"5.660380e-13","_rn_":"26"},{"1":"Igsf11","2":"1.138743e-12","_rn_":"27"},{"1":"Tmeff2","2":"2.437355e-12","_rn_":"28"},{"1":"Ppfibp2","2":"2.864360e-12","_rn_":"29"},{"1":"Cnksr3","2":"3.904649e-12","_rn_":"30"},{"1":"5830418P13Rik","2":"7.577048e-12","_rn_":"31"},{"1":"Ablim2","2":"2.216366e-11","_rn_":"32"},{"1":"Sema4d","2":"1.792928e-10","_rn_":"33"},{"1":"Tcf7l1","2":"2.396076e-10","_rn_":"34"},{"1":"Nkain2","2":"8.115707e-08","_rn_":"35"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
<div id="nfol.2-nfol.sgcd" class="section level4 unnumbered">
<h4 class="unnumbered">NFOL.2 | NFOL.Sgcd</h4>
<pre class="r"><code>marker_type[[&#39;NFOL.2 | NFOL.Sgcd&#39;]]</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":[""],"name":["_rn_"],"type":[""],"align":["left"]},{"label":["gene"],"name":[1],"type":["chr"],"align":["left"]},{"label":["combined_p_val"],"name":[2],"type":["dbl"],"align":["right"]}],"data":[{"1":"Sgcd","2":"5.454850e-37","_rn_":"36"},{"1":"Pcdh15","2":"1.542581e-31","_rn_":"37"},{"1":"Nrxn1","2":"5.558894e-31","_rn_":"38"},{"1":"Ntm","2":"7.009039e-30","_rn_":"39"},{"1":"Fgf14","2":"6.919062e-28","_rn_":"40"},{"1":"Brinp3","2":"1.425889e-27","_rn_":"41"},{"1":"Ptprt","2":"1.844470e-27","_rn_":"42"},{"1":"Lhfpl3","2":"4.493210e-27","_rn_":"43"},{"1":"Rora","2":"6.723278e-26","_rn_":"44"},{"1":"Dcc","2":"6.291290e-23","_rn_":"45"},{"1":"Ncald","2":"1.499677e-22","_rn_":"46"},{"1":"Astn2","2":"2.354016e-22","_rn_":"47"},{"1":"Dpyd","2":"1.104652e-21","_rn_":"48"},{"1":"Nxph1","2":"1.127355e-21","_rn_":"49"},{"1":"Sntg1","2":"5.003038e-21","_rn_":"50"},{"1":"Edil3","2":"1.284669e-20","_rn_":"51"},{"1":"Fam155a","2":"5.212461e-18","_rn_":"52"},{"1":"Npas3","2":"1.450779e-17","_rn_":"53"},{"1":"Kcnd2","2":"1.485333e-17","_rn_":"54"},{"1":"Nav3","2":"1.482761e-16","_rn_":"55"},{"1":"Syn2","2":"2.649603e-16","_rn_":"56"},{"1":"Cntn6","2":"6.574371e-14","_rn_":"57"},{"1":"Gpc5","2":"2.253577e-13","_rn_":"58"},{"1":"C430049B03Rik","2":"5.547865e-13","_rn_":"59"},{"1":"Plppr1","2":"1.303766e-12","_rn_":"60"},{"1":"Robo1","2":"4.887045e-11","_rn_":"61"},{"1":"Nrxn3","2":"8.911409e-11","_rn_":"62"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
<div id="region" class="section level3 tabset">
<h3 class="tabset">Region</h3>
<div id="aplcoa" class="section level4 tabset">
<h4 class="tabset">aplCoA</h4>
<pre class="r"><code>src_list &lt;- lapply(names(marker_type_region_aplCoA), function(group) {
    src &lt;- c(
        &quot;##### {{group}} {.unnumbered}&quot;,
        &quot;```{r marker-type-aplCoA-{{group}}, message = FALSE, warning = FALSE}&quot;,
        &quot;marker_type_region_aplCoA[[&#39;{{group}}&#39;]]&quot;,  
        &quot;```&quot;,
        &quot;&quot;
    )
    knit_expand(text = src)
})
out &lt;- knit_child(text = unlist(src_list), options = list(cache = FALSE))</code></pre>
<div id="nfol.1-nfol.frmd4a-1" class="section level5 unnumbered">
<h5 class="unnumbered">NFOL.1 | NFOL.Frmd4a</h5>
<pre class="r"><code>marker_type_region_aplCoA[[&#39;NFOL.1 | NFOL.Frmd4a&#39;]]</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":[""],"name":["_rn_"],"type":[""],"align":["left"]},{"label":["avg_logFC"],"name":[1],"type":["dbl"],"align":["right"]},{"label":["p_val"],"name":[2],"type":["dbl"],"align":["right"]},{"label":["p_val_adj"],"name":[3],"type":["dbl"],"align":["right"]}],"data":[{"1":"0.816863","2":"3.987127e-05","3":"0.04532638","_rn_":"Cd9"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
<div id="nfol.2-nfol.sgcd-1" class="section level5 unnumbered">
<h5 class="unnumbered">NFOL.2 | NFOL.Sgcd</h5>
<pre class="r"><code>marker_type_region_aplCoA[[&#39;NFOL.2 | NFOL.Sgcd&#39;]]</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["avg_logFC"],"name":[1],"type":["dbl"],"align":["right"]},{"label":["p_val"],"name":[2],"type":["dbl"],"align":["right"]},{"label":["p_val_adj"],"name":[3],"type":["dbl"],"align":["right"]}],"data":[],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
<div id="pplcoa" class="section level4 tabset">
<h4 class="tabset">pplCoA</h4>
<pre class="r"><code>src_list &lt;- lapply(names(marker_type_region_pplCoA), function(group) {
    src &lt;- c(
        &quot;##### {{group}} {.unnumbered}&quot;,
        &quot;```{r marker-type-pplCoA-{{group}}, message = FALSE, warning = FALSE}&quot;,
        &quot;marker_type_region_pplCoA[[&#39;{{group}}&#39;]]&quot;,  
        &quot;```&quot;,
        &quot;&quot;
    )
    knit_expand(text = src)
})
out &lt;- knit_child(text = unlist(src_list), options = list(cache = FALSE))</code></pre>
<div id="nfol.1-nfol.frmd4a-2" class="section level5 unnumbered">
<h5 class="unnumbered">NFOL.1 | NFOL.Frmd4a</h5>
<pre class="r"><code>marker_type_region_pplCoA[[&#39;NFOL.1 | NFOL.Frmd4a&#39;]]</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":[""],"name":["_rn_"],"type":[""],"align":["left"]},{"label":["avg_logFC"],"name":[1],"type":["dbl"],"align":["right"]},{"label":["p_val"],"name":[2],"type":["dbl"],"align":["right"]},{"label":["p_val_adj"],"name":[3],"type":["dbl"],"align":["right"]}],"data":[{"1":"-1.0000043","2":"1.829855e-07","3":"0.001185405","_rn_":"9130019P16Rik"},{"1":"-0.8300091","2":"5.273004e-06","3":"0.010989820","_rn_":"Bcas1os2"},{"1":"-0.9717289","2":"2.024974e-05","3":"0.036174713","_rn_":"Etl4"},{"1":"-0.8867036","2":"1.863867e-06","3":"0.007769221","_rn_":"Gm13187"},{"1":"-1.4364706","2":"2.926290e-05","3":"0.038172220","_rn_":"Prss48"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
<div id="nfol.2-nfol.sgcd-2" class="section level5 unnumbered">
<h5 class="unnumbered">NFOL.2 | NFOL.Sgcd</h5>
<pre class="r"><code>marker_type_region_pplCoA[[&#39;NFOL.2 | NFOL.Sgcd&#39;]]</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["avg_logFC"],"name":[1],"type":["dbl"],"align":["right"]},{"label":["p_val"],"name":[2],"type":["dbl"],"align":["right"]},{"label":["p_val_adj"],"name":[3],"type":["dbl"],"align":["right"]}],"data":[],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
</div>
</div>
<div id="scatter-1" class="section level2 tabset">
<h2 class="tabset">Scatter</h2>
<div id="pca-2" class="section level3 tabset">
<h3 class="tabset">PCA</h3>
<pre class="r"><code>src_list &lt;- lapply(markers_list, function(marker) {
    src &lt;- c(
        &quot;#### {{marker}} {.unnumbered}&quot;,
        &quot;```{r pca-{{marker}}, message = FALSE, warning = FALSE}&quot;,
        &quot;feature_plot_formatted(array, &#39;pca&#39;, &#39;{{marker}}&#39;, feature_intensity_colors, NA)&quot;,  
        &quot;```&quot;,
        &quot;&quot;
    )
    knit_expand(text = src)
})
out &lt;- knit_child(text = unlist(src_list), options = list(cache = FALSE))</code></pre>
<div id="pdgfra" class="section level4 unnumbered">
<h4 class="unnumbered">Pdgfra</h4>
<pre class="r"><code>feature_plot_formatted(array, &#39;pca&#39;, &#39;Pdgfra&#39;, feature_intensity_colors, NA)</code></pre>
<embed src="figure/sequencing-05P-NFOL_classification_RNA.Rmd/pca-Pdgfra-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="cspg4" class="section level4 unnumbered">
<h4 class="unnumbered">Cspg4</h4>
<pre class="r"><code>feature_plot_formatted(array, &#39;pca&#39;, &#39;Cspg4&#39;, feature_intensity_colors, NA)</code></pre>
<embed src="figure/sequencing-05P-NFOL_classification_RNA.Rmd/pca-Cspg4-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="bmp4" class="section level4 unnumbered">
<h4 class="unnumbered">Bmp4</h4>
<pre class="r"><code>feature_plot_formatted(array, &#39;pca&#39;, &#39;Bmp4&#39;, feature_intensity_colors, NA)</code></pre>
<embed src="figure/sequencing-05P-NFOL_classification_RNA.Rmd/pca-Bmp4-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="enpp6" class="section level4 unnumbered">
<h4 class="unnumbered">Enpp6</h4>
<pre class="r"><code>feature_plot_formatted(array, &#39;pca&#39;, &#39;Enpp6&#39;, feature_intensity_colors, NA)</code></pre>
<embed src="figure/sequencing-05P-NFOL_classification_RNA.Rmd/pca-Enpp6-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="mog" class="section level4 unnumbered">
<h4 class="unnumbered">Mog</h4>
<pre class="r"><code>feature_plot_formatted(array, &#39;pca&#39;, &#39;Mog&#39;, feature_intensity_colors, NA)</code></pre>
<embed src="figure/sequencing-05P-NFOL_classification_RNA.Rmd/pca-Mog-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="mal" class="section level4 unnumbered">
<h4 class="unnumbered">Mal</h4>
<pre class="r"><code>feature_plot_formatted(array, &#39;pca&#39;, &#39;Mal&#39;, feature_intensity_colors, NA)</code></pre>
<embed src="figure/sequencing-05P-NFOL_classification_RNA.Rmd/pca-Mal-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="frmd4a" class="section level4 unnumbered">
<h4 class="unnumbered">Frmd4a</h4>
<pre class="r"><code>feature_plot_formatted(array, &#39;pca&#39;, &#39;Frmd4a&#39;, feature_intensity_colors, NA)</code></pre>
<embed src="figure/sequencing-05P-NFOL_classification_RNA.Rmd/pca-Frmd4a-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="mbp" class="section level4 unnumbered">
<h4 class="unnumbered">Mbp</h4>
<pre class="r"><code>feature_plot_formatted(array, &#39;pca&#39;, &#39;Mbp&#39;, feature_intensity_colors, NA)</code></pre>
<embed src="figure/sequencing-05P-NFOL_classification_RNA.Rmd/pca-Mbp-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="sgcd" class="section level4 unnumbered">
<h4 class="unnumbered">Sgcd</h4>
<pre class="r"><code>feature_plot_formatted(array, &#39;pca&#39;, &#39;Sgcd&#39;, feature_intensity_colors, NA)</code></pre>
<embed src="figure/sequencing-05P-NFOL_classification_RNA.Rmd/pca-Sgcd-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="pcdh15" class="section level4 unnumbered">
<h4 class="unnumbered">Pcdh15</h4>
<pre class="r"><code>feature_plot_formatted(array, &#39;pca&#39;, &#39;Pcdh15&#39;, feature_intensity_colors, NA)</code></pre>
<embed src="figure/sequencing-05P-NFOL_classification_RNA.Rmd/pca-Pcdh15-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
</div>
<div id="tsne-2" class="section level3 tabset">
<h3 class="tabset">tSNE</h3>
<pre class="r"><code>src_list &lt;- lapply(markers_list, function(marker) {
    src &lt;- c(
        &quot;#### {{marker}} {.unnumbered}&quot;,
        &quot;```{r tsne-{{marker}}, message = FALSE, warning = FALSE}&quot;,
        &quot;feature_plot_formatted(array, &#39;tsne&#39;, &#39;{{marker}}&#39;, feature_intensity_colors, NA)&quot;,  
        &quot;```&quot;,
        &quot;&quot;
    )
    knit_expand(text = src)
})
out &lt;- knit_child(text = unlist(src_list), options = list(cache = FALSE))</code></pre>
<div id="pdgfra-1" class="section level4 unnumbered">
<h4 class="unnumbered">Pdgfra</h4>
<pre class="r"><code>feature_plot_formatted(array, &#39;tsne&#39;, &#39;Pdgfra&#39;, feature_intensity_colors, NA)</code></pre>
<embed src="figure/sequencing-05P-NFOL_classification_RNA.Rmd/tsne-Pdgfra-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="cspg4-1" class="section level4 unnumbered">
<h4 class="unnumbered">Cspg4</h4>
<pre class="r"><code>feature_plot_formatted(array, &#39;tsne&#39;, &#39;Cspg4&#39;, feature_intensity_colors, NA)</code></pre>
<embed src="figure/sequencing-05P-NFOL_classification_RNA.Rmd/tsne-Cspg4-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="bmp4-1" class="section level4 unnumbered">
<h4 class="unnumbered">Bmp4</h4>
<pre class="r"><code>feature_plot_formatted(array, &#39;tsne&#39;, &#39;Bmp4&#39;, feature_intensity_colors, NA)</code></pre>
<embed src="figure/sequencing-05P-NFOL_classification_RNA.Rmd/tsne-Bmp4-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="enpp6-1" class="section level4 unnumbered">
<h4 class="unnumbered">Enpp6</h4>
<pre class="r"><code>feature_plot_formatted(array, &#39;tsne&#39;, &#39;Enpp6&#39;, feature_intensity_colors, NA)</code></pre>
<embed src="figure/sequencing-05P-NFOL_classification_RNA.Rmd/tsne-Enpp6-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="mog-1" class="section level4 unnumbered">
<h4 class="unnumbered">Mog</h4>
<pre class="r"><code>feature_plot_formatted(array, &#39;tsne&#39;, &#39;Mog&#39;, feature_intensity_colors, NA)</code></pre>
<embed src="figure/sequencing-05P-NFOL_classification_RNA.Rmd/tsne-Mog-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="mal-1" class="section level4 unnumbered">
<h4 class="unnumbered">Mal</h4>
<pre class="r"><code>feature_plot_formatted(array, &#39;tsne&#39;, &#39;Mal&#39;, feature_intensity_colors, NA)</code></pre>
<embed src="figure/sequencing-05P-NFOL_classification_RNA.Rmd/tsne-Mal-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="frmd4a-1" class="section level4 unnumbered">
<h4 class="unnumbered">Frmd4a</h4>
<pre class="r"><code>feature_plot_formatted(array, &#39;tsne&#39;, &#39;Frmd4a&#39;, feature_intensity_colors, NA)</code></pre>
<embed src="figure/sequencing-05P-NFOL_classification_RNA.Rmd/tsne-Frmd4a-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="mbp-1" class="section level4 unnumbered">
<h4 class="unnumbered">Mbp</h4>
<pre class="r"><code>feature_plot_formatted(array, &#39;tsne&#39;, &#39;Mbp&#39;, feature_intensity_colors, NA)</code></pre>
<embed src="figure/sequencing-05P-NFOL_classification_RNA.Rmd/tsne-Mbp-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="sgcd-1" class="section level4 unnumbered">
<h4 class="unnumbered">Sgcd</h4>
<pre class="r"><code>feature_plot_formatted(array, &#39;tsne&#39;, &#39;Sgcd&#39;, feature_intensity_colors, NA)</code></pre>
<embed src="figure/sequencing-05P-NFOL_classification_RNA.Rmd/tsne-Sgcd-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="pcdh15-1" class="section level4 unnumbered">
<h4 class="unnumbered">Pcdh15</h4>
<pre class="r"><code>feature_plot_formatted(array, &#39;tsne&#39;, &#39;Pcdh15&#39;, feature_intensity_colors, NA)</code></pre>
<embed src="figure/sequencing-05P-NFOL_classification_RNA.Rmd/tsne-Pcdh15-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
</div>
<div id="umap-2" class="section level3 tabset">
<h3 class="tabset">UMAP</h3>
<pre class="r"><code>src_list &lt;- lapply(markers_list, function(marker) {
    src &lt;- c(
        &quot;#### {{marker}} {.unnumbered}&quot;,
        &quot;```{r umap-{{marker}}, message = FALSE, warning = FALSE}&quot;,
        &quot;feature_plot_formatted(array, &#39;umap&#39;, &#39;{{marker}}&#39;, feature_intensity_colors, NA)&quot;,  
        &quot;```&quot;,
        &quot;&quot;
    )
    knit_expand(text = src)
})
out &lt;- knit_child(text = unlist(src_list), options = list(cache = FALSE))</code></pre>
<div id="pdgfra-2" class="section level4 unnumbered">
<h4 class="unnumbered">Pdgfra</h4>
<pre class="r"><code>feature_plot_formatted(array, &#39;umap&#39;, &#39;Pdgfra&#39;, feature_intensity_colors, NA)</code></pre>
<embed src="figure/sequencing-05P-NFOL_classification_RNA.Rmd/umap-Pdgfra-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="cspg4-2" class="section level4 unnumbered">
<h4 class="unnumbered">Cspg4</h4>
<pre class="r"><code>feature_plot_formatted(array, &#39;umap&#39;, &#39;Cspg4&#39;, feature_intensity_colors, NA)</code></pre>
<embed src="figure/sequencing-05P-NFOL_classification_RNA.Rmd/umap-Cspg4-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="bmp4-2" class="section level4 unnumbered">
<h4 class="unnumbered">Bmp4</h4>
<pre class="r"><code>feature_plot_formatted(array, &#39;umap&#39;, &#39;Bmp4&#39;, feature_intensity_colors, NA)</code></pre>
<embed src="figure/sequencing-05P-NFOL_classification_RNA.Rmd/umap-Bmp4-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="enpp6-2" class="section level4 unnumbered">
<h4 class="unnumbered">Enpp6</h4>
<pre class="r"><code>feature_plot_formatted(array, &#39;umap&#39;, &#39;Enpp6&#39;, feature_intensity_colors, NA)</code></pre>
<embed src="figure/sequencing-05P-NFOL_classification_RNA.Rmd/umap-Enpp6-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="mog-2" class="section level4 unnumbered">
<h4 class="unnumbered">Mog</h4>
<pre class="r"><code>feature_plot_formatted(array, &#39;umap&#39;, &#39;Mog&#39;, feature_intensity_colors, NA)</code></pre>
<embed src="figure/sequencing-05P-NFOL_classification_RNA.Rmd/umap-Mog-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="mal-2" class="section level4 unnumbered">
<h4 class="unnumbered">Mal</h4>
<pre class="r"><code>feature_plot_formatted(array, &#39;umap&#39;, &#39;Mal&#39;, feature_intensity_colors, NA)</code></pre>
<embed src="figure/sequencing-05P-NFOL_classification_RNA.Rmd/umap-Mal-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="frmd4a-2" class="section level4 unnumbered">
<h4 class="unnumbered">Frmd4a</h4>
<pre class="r"><code>feature_plot_formatted(array, &#39;umap&#39;, &#39;Frmd4a&#39;, feature_intensity_colors, NA)</code></pre>
<embed src="figure/sequencing-05P-NFOL_classification_RNA.Rmd/umap-Frmd4a-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="mbp-2" class="section level4 unnumbered">
<h4 class="unnumbered">Mbp</h4>
<pre class="r"><code>feature_plot_formatted(array, &#39;umap&#39;, &#39;Mbp&#39;, feature_intensity_colors, NA)</code></pre>
<embed src="figure/sequencing-05P-NFOL_classification_RNA.Rmd/umap-Mbp-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="sgcd-2" class="section level4 unnumbered">
<h4 class="unnumbered">Sgcd</h4>
<pre class="r"><code>feature_plot_formatted(array, &#39;umap&#39;, &#39;Sgcd&#39;, feature_intensity_colors, NA)</code></pre>
<embed src="figure/sequencing-05P-NFOL_classification_RNA.Rmd/umap-Sgcd-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="pcdh15-2" class="section level4 unnumbered">
<h4 class="unnumbered">Pcdh15</h4>
<pre class="r"><code>feature_plot_formatted(array, &#39;umap&#39;, &#39;Pcdh15&#39;, feature_intensity_colors, NA)</code></pre>
<embed src="figure/sequencing-05P-NFOL_classification_RNA.Rmd/umap-Pcdh15-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
</div>
</div>
<div id="deg-abundance" class="section level2 tabset">
<h2 class="tabset">DEG Abundance</h2>
<div id="aplcoa-1" class="section level3">
<h3>aplCoA</h3>
<pre class="r"><code>deg_abundance_plot_magnitude(marker_type_region_aplCoA, deg_intensity_colors_aplCoA, deg_range_type, &quot;aplCoA&quot;)</code></pre>
<embed src="figure/sequencing-05P-NFOL_classification_RNA.Rmd/21-deg_abundance_magnitude_type_aplCoA-1.pdf" width="192" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="pplcoa-1" class="section level3">
<h3>pplCoA</h3>
<pre class="r"><code>deg_abundance_plot_magnitude(marker_type_region_pplCoA, deg_intensity_colors_pplCoA, deg_range_type, &quot;pplCoA&quot;)</code></pre>
<embed src="figure/sequencing-05P-NFOL_classification_RNA.Rmd/22-deg_abundance_magnitude_type_pplCoA-1.pdf" width="192" style="display: block; margin: auto;" type="application/pdf" />
</div>
</div>
</div>
<div id="visualize-consolidated-cluster-composition"
class="section level1">
<h1>Visualize consolidated cluster composition</h1>
<div id="absolute-composition" class="section level2 tabset">
<h2 class="tabset">Absolute Composition</h2>
<pre class="r"><code>src_list &lt;- mapply(function(br, br_colors) {
    src &lt;- c(
        &quot;### {{br}} {.unnumbered}&quot;,
        &quot;```{r class-absolute-{{br}}, message = FALSE, warning = FALSE}&quot;,
        &quot;absolute_bar_plot_formatted(plot_group_array, &#39;Type&#39;, &#39;{{br}}&#39;, {{br_colors}})&quot;,  
        &quot;```&quot;,
        &quot;&quot;
    )
    knit_expand(text = src)
}, batch_region, batch_color_list)
out &lt;- knit_child(text = unlist(src_list), options = list(cache = FALSE))</code></pre>
<div id="batch" class="section level3 unnumbered">
<h3 class="unnumbered">Batch</h3>
<pre class="r"><code>absolute_bar_plot_formatted(plot_group_array, &#39;Type&#39;, &#39;Batch&#39;, batch_colors_rna)</code></pre>
<embed src="figure/sequencing-05P-NFOL_classification_RNA.Rmd/class-absolute-Batch-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="region-1" class="section level3 unnumbered">
<h3 class="unnumbered">Region</h3>
<pre class="r"><code>absolute_bar_plot_formatted(plot_group_array, &#39;Type&#39;, &#39;Region&#39;, region_colors)</code></pre>
<embed src="figure/sequencing-05P-NFOL_classification_RNA.Rmd/class-absolute-Region-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
</div>
<div id="relative-composition" class="section level2 tabset">
<h2 class="tabset">Relative Composition</h2>
<div id="bar" class="section level3 tabset">
<h3 class="tabset">Bar</h3>
<div id="batch-1" class="section level4">
<h4>Batch</h4>
<pre class="r"><code>relative_bar_plot_formatted(plot_group_array, &#39;Type&#39;, &#39;Batch&#39;, batch_colors_rna)</code></pre>
<embed src="figure/sequencing-05P-NFOL_classification_RNA.Rmd/24-type_relative_batch-1.pdf" width="288" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="region-2" class="section level4">
<h4>Region</h4>
<pre class="r"><code>relative_bar_plot_formatted(plot_group_array, &#39;Type&#39;, &#39;Region&#39;, region_colors) + 
  geom_hline(yintercept = table(array$Region)[2] / length(array$Region), linetype = &quot;dashed&quot;)</code></pre>
<embed src="figure/sequencing-05P-NFOL_classification_RNA.Rmd/25-type_relative_region-1.pdf" width="288" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="chi-square-results" class="section level4">
<h4>Chi-square results</h4>
<pre class="r"><code>chisq_table &lt;- table(array$Region, array$Type)

chisq.test(chisq_table)</code></pre>
<pre><code>
    Pearson&#39;s Chi-squared test with Yates&#39; continuity correction

data:  chisq_table
X-squared = 1.3298, df = 1, p-value = 0.2488</code></pre>
<pre class="r"><code>chisq.posthoc.test(chisq_table, method = &quot;bonferroni&quot;)</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["Dimension"],"name":[1],"type":["chr"],"align":["left"]},{"label":["Value"],"name":[2],"type":["chr"],"align":["left"]},{"label":["NFOL.1"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["NFOL.2"],"name":[4],"type":["dbl"],"align":["right"]}],"data":[{"1":"aplCoA","2":"Residuals","3":"1.271865","4":"-1.271865"},{"1":"aplCoA","2":"p values","3":"0.813685","4":"0.813685"},{"1":"pplCoA","2":"Residuals","3":"-1.271865","4":"1.271865"},{"1":"pplCoA","2":"p values","3":"0.813685","4":"0.813685"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<pre class="r"><code># can&#39;t do propeller w/ only two groups, so skip here (if it doesn&#39;t pass chi-sq though, likely won&#39;t pass propeller)</code></pre>
</div>
</div>
<div id="region-3" class="section level3 tabset">
<h3 class="tabset">Region</h3>
<div id="absolute" class="section level4">
<h4>Absolute</h4>
<pre class="r"><code>absolute_bar_plot_formatted(plot_group_array, &#39;Region&#39;, &#39;Type&#39;, nfol_type_colors)</code></pre>
<embed src="figure/sequencing-05P-NFOL_classification_RNA.Rmd/27-region_relative-1.pdf" width="288" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="relative" class="section level4">
<h4>Relative</h4>
<pre class="r"><code>relative_bar_plot_formatted(plot_group_array, &#39;Region&#39;, &#39;Type&#39;, nfol_type_colors) +
  geom_hline(yintercept = table(array$Type)[2] / length(array$Type), linetype = &quot;dashed&quot;)</code></pre>
<embed src="figure/sequencing-05P-NFOL_classification_RNA.Rmd/28-region_relative-1.pdf" width="288" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="chi-square-results-1" class="section level4">
<h4>Chi-square results</h4>
<pre class="r"><code>chisq_table &lt;- table(array$Type, array$Region)

chisq.test(chisq_table)</code></pre>
<pre><code>
    Pearson&#39;s Chi-squared test with Yates&#39; continuity correction

data:  chisq_table
X-squared = 1.3298, df = 1, p-value = 0.2488</code></pre>
<pre class="r"><code>chisq.posthoc.test(chisq_table, method = &quot;bonferroni&quot;)</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["Dimension"],"name":[1],"type":["chr"],"align":["left"]},{"label":["Value"],"name":[2],"type":["chr"],"align":["left"]},{"label":["aplCoA"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["pplCoA"],"name":[4],"type":["dbl"],"align":["right"]}],"data":[{"1":"NFOL.1","2":"Residuals","3":"1.271865","4":"-1.271865"},{"1":"NFOL.1","2":"p values","3":"0.813685","4":"0.813685"},{"1":"NFOL.2","2":"Residuals","3":"-1.271865","4":"1.271865"},{"1":"NFOL.2","2":"p values","3":"0.813685","4":"0.813685"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
<div id="batch-2" class="section level3 tabset">
<h3 class="tabset">Batch</h3>
<div id="absolute-1" class="section level4">
<h4>Absolute</h4>
<pre class="r"><code>absolute_bar_plot_formatted(plot_group_array, &#39;Batch&#39;, &#39;Type&#39;, nfol_type_colors)</code></pre>
<embed src="figure/sequencing-05P-NFOL_classification_RNA.Rmd/30-region_relative-1.pdf" width="384" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="relative-1" class="section level4">
<h4>Relative</h4>
<pre class="r"><code>relative_bar_plot_formatted(plot_group_array, &#39;Batch&#39;, &#39;Type&#39;, nfol_type_colors) +
  geom_hline(yintercept = table(array$Type)[2] / length(array$Type), linetype = &quot;dashed&quot;)</code></pre>
<embed src="figure/sequencing-05P-NFOL_classification_RNA.Rmd/31-region_relative-1.pdf" width="384" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="chi-square-results-2" class="section level4">
<h4>Chi-square results</h4>
<pre class="r"><code>chisq_table &lt;- table(array$Type, array$orig.ident)

chisq.test(chisq_table)</code></pre>
<pre><code>
    Pearson&#39;s Chi-squared test

data:  chisq_table
X-squared = 5.5115, df = 5, p-value = 0.3567</code></pre>
<pre class="r"><code>chisq.posthoc.test(chisq_table, method = &quot;bonferroni&quot;)</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["Dimension"],"name":[1],"type":["chr"],"align":["left"]},{"label":["Value"],"name":[2],"type":["chr"],"align":["left"]},{"label":["aplCoA-1"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["aplCoA-2"],"name":[4],"type":["dbl"],"align":["right"]},{"label":["aplCoA-3"],"name":[5],"type":["dbl"],"align":["right"]},{"label":["pplCoA-1"],"name":[6],"type":["dbl"],"align":["right"]},{"label":["pplCoA-2"],"name":[7],"type":["dbl"],"align":["right"]},{"label":["pplCoA-3"],"name":[8],"type":["dbl"],"align":["right"]}],"data":[{"1":"NFOL.1","2":"Residuals","3":"-0.5106105","4":"1.618193","5":"0.7748754","6":"0.4727207","7":"-0.5340443","8":"-1.60237"},{"1":"NFOL.1","2":"p values","3":"1.0000000","4":"1.000000","5":"1.0000000","6":"1.0000000","7":"1.0000000","8":"1.00000"},{"1":"NFOL.2","2":"Residuals","3":"0.5106105","4":"-1.618193","5":"-0.7748754","6":"-0.4727207","7":"0.5340443","8":"1.60237"},{"1":"NFOL.2","2":"p values","3":"1.0000000","4":"1.000000","5":"1.0000000","6":"1.0000000","7":"1.0000000","8":"1.00000"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
</div>
<div id="examine-metrics-for-consolidated-clusters"
class="section level2 tabset">
<h2 class="tabset">Examine metrics for consolidated clusters</h2>
<pre class="r"><code>src_list &lt;- mapply(function(metric, label) {
    src &lt;- c(
        &quot;### {{label}} {.unnumbered}&quot;,
        &quot;```{r vln-type-{{metric}}, message = FALSE, warning = FALSE}&quot;,
        &quot;violin_metrics_formatted(array, &#39;Type&#39;, &#39;{{metric}}&#39;, &#39;{{label}}&#39;, mol_type_colors)&quot;,  
        &quot;```&quot;,
        &quot;&quot;
    )
    knit_expand(text = src)
}, vln_metrics$id, vln_metrics$label)
out &lt;- knit_child(text = unlist(src_list), options = list(cache = FALSE))</code></pre>
<div id="umis" class="section level3 unnumbered">
<h3 class="unnumbered">UMIs</h3>
<pre class="r"><code>violin_metrics_formatted(array, &#39;Type&#39;, &#39;nCount_RNA&#39;, &#39;UMIs&#39;, mol_type_colors)</code></pre>
<embed src="figure/sequencing-05P-NFOL_classification_RNA.Rmd/vln-type-nCount_RNA-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="genes" class="section level3 unnumbered">
<h3 class="unnumbered">Genes</h3>
<pre class="r"><code>violin_metrics_formatted(array, &#39;Type&#39;, &#39;nFeature_RNA&#39;, &#39;Genes&#39;, mol_type_colors)</code></pre>
<embed src="figure/sequencing-05P-NFOL_classification_RNA.Rmd/vln-type-nFeature_RNA-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="mito" class="section level3 unnumbered">
<h3 class="unnumbered">% Mito</h3>
<pre class="r"><code>violin_metrics_formatted(array, &#39;Type&#39;, &#39;Mito_proportion&#39;, &#39;% Mito&#39;, mol_type_colors)</code></pre>
<embed src="figure/sequencing-05P-NFOL_classification_RNA.Rmd/vln-type-Mito_proportion-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="ribo" class="section level3 unnumbered">
<h3 class="unnumbered">% Ribo</h3>
<pre class="r"><code>violin_metrics_formatted(array, &#39;Type&#39;, &#39;Ribo_proportion&#39;, &#39;% Ribo&#39;, mol_type_colors)</code></pre>
<embed src="figure/sequencing-05P-NFOL_classification_RNA.Rmd/vln-type-Ribo_proportion-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
</div>
</div>
<div id="visualize-marker-genes-in-consolidated-clusters"
class="section level1 tabset">
<h1 class="tabset">Visualize marker genes in consolidated clusters</h1>
<div id="dot-plot" class="section level2 tabset">
<h2 class="tabset">Dot plot</h2>
<pre class="r"><code>array &lt;- PrepSCTFindMarkers(array)

DotPlot_scCustom(array, features = markers_list, colors_use = dotplot_colors) + 
  theme(axis.ticks.x = element_blank(),
        axis.text.x = element_text(angle = 45, vjust = 1, hjust=1))</code></pre>
<embed src="figure/sequencing-05P-NFOL_classification_RNA.Rmd/34-dot_tg-1.pdf" width="1152" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="stacked-violin" class="section level2 tabset">
<h2 class="tabset">Stacked Violin</h2>
<div id="all-markers" class="section level3">
<h3>All markers</h3>
<pre class="r"><code>VlnPlot(array, features = markers_list, 
        pt.size = 0, stack = TRUE, flip = TRUE, 
        group.by = &#39;Type&#39;) + 
        theme(legend.position = &#39;none&#39;)</code></pre>
<embed src="figure/sequencing-05P-NFOL_classification_RNA.Rmd/35-stacked_vln_marker-1.pdf" width="1152" style="display: block; margin: auto;" type="application/pdf" />
</div>
</div>
<div id="heatmap" class="section level2">
<h2>Heatmap</h2>
<pre class="r"><code>DoHeatmap(array, features = markers_list, group.by = &#39;Type&#39;, size = 0,
          group.colors = nfol_type_colors) +
          scale_fill_gradientn(colours = heatmap_colors) +  
          scale_colour_manual(values = nfol_type_colors)</code></pre>
<p><img src="figure/sequencing-05P-NFOL_classification_RNA.Rmd/36-heatmap_marker-1.png" width="1536" style="display: block; margin: auto;" /></p>
<div id="umis-1" class="section level3 unnumbered">
<h3 class="unnumbered">UMIs</h3>
<pre class="r"><code>violin_metrics_formatted(array, &#39;Type&#39;, &#39;nCount_RNA&#39;, &#39;UMIs&#39;, mol_type_colors)</code></pre>
<embed src="figure/sequencing-05P-NFOL_classification_RNA.Rmd/vln-type-nCount_RNA-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="genes-1" class="section level3 unnumbered">
<h3 class="unnumbered">Genes</h3>
<pre class="r"><code>violin_metrics_formatted(array, &#39;Type&#39;, &#39;nFeature_RNA&#39;, &#39;Genes&#39;, mol_type_colors)</code></pre>
<embed src="figure/sequencing-05P-NFOL_classification_RNA.Rmd/vln-type-nFeature_RNA-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="mito-1" class="section level3 unnumbered">
<h3 class="unnumbered">% Mito</h3>
<pre class="r"><code>violin_metrics_formatted(array, &#39;Type&#39;, &#39;Mito_proportion&#39;, &#39;% Mito&#39;, mol_type_colors)</code></pre>
<embed src="figure/sequencing-05P-NFOL_classification_RNA.Rmd/vln-type-Mito_proportion-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="ribo-1" class="section level3 unnumbered">
<h3 class="unnumbered">% Ribo</h3>
<pre class="r"><code>violin_metrics_formatted(array, &#39;Type&#39;, &#39;Ribo_proportion&#39;, &#39;% Ribo&#39;, mol_type_colors)</code></pre>
<embed src="figure/sequencing-05P-NFOL_classification_RNA.Rmd/vln-type-Ribo_proportion-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
</div>
</div>
<div id="visualize-cluster-distances" class="section level1 tabset">
<h1 class="tabset">Visualize cluster distances</h1>
<div id="dendrogram" class="section level2 tabset">
<h2 class="tabset">Dendrogram</h2>
<div id="region-4" class="section level3">
<h3>Region</h3>
<pre class="r"><code>region_dend &lt;- create_dendrogram(array, as.factor(array$Region), region_colors)

plot(region_dend[[2]], axes = FALSE)</code></pre>
<embed src="figure/sequencing-05P-NFOL_classification_RNA.Rmd/37-region_dendrogram-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="batch-3" class="section level3">
<h3>Batch</h3>
<pre class="r"><code>batch_dend &lt;- create_dendrogram(array, as.factor(array$orig.ident), batch_colors_rna)

plot(batch_dend[[2]], axes = FALSE)</code></pre>
<embed src="figure/sequencing-05P-NFOL_classification_RNA.Rmd/38-library_dendrogram-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="type-1" class="section level3">
<h3>Type</h3>
<pre class="r"><code>type_dend &lt;- create_dendrogram(array, as.factor(array$Type), nfol_type_colors)

plot(type_dend[[2]], axes = FALSE)</code></pre>
<embed src="figure/sequencing-05P-NFOL_classification_RNA.Rmd/39-type_dendrogram-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="type-with-region-id" class="section level3">
<h3>Type with Region ID</h3>
<pre class="r"><code># create region+class convolution ID
array &lt;- AddMetaData(array, metadata = paste(array$Region, array$Type), col.name = &quot;Region_Type&quot;)

type_region_dend &lt;- create_dendrogram(array, as.factor(array$Region_Type), region_nfol_type_colors)

plot(type_region_dend[[2]], axes = FALSE)</code></pre>
<embed src="figure/sequencing-05P-NFOL_classification_RNA.Rmd/40-type_dendrogram_region-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
</div>
<div id="cluster-distance-heatmap" class="section level2 tabset">
<h2 class="tabset">Cluster distance heatmap</h2>
<div id="region-5" class="section level3 tabset">
<h3 class="tabset">Region</h3>
<div id="distance" class="section level4">
<h4>Distance</h4>
<pre class="r"><code>dendrogram_heatmap(region_dend, method_id = &quot;distance&quot;)</code></pre>
<embed src="figure/sequencing-05P-NFOL_classification_RNA.Rmd/41-cluster_heatmap_region_dist-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="correlation" class="section level4">
<h4>Correlation</h4>
<pre class="r"><code>dendrogram_heatmap(region_dend, method_id = &quot;correlation&quot;)</code></pre>
<embed src="figure/sequencing-05P-NFOL_classification_RNA.Rmd/42-cluster_heatmap_region_cor-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
</div>
<div id="batch-4" class="section level3 tabset">
<h3 class="tabset">Batch</h3>
<div id="distance-1" class="section level4">
<h4>Distance</h4>
<pre class="r"><code>dendrogram_heatmap(batch_dend, method_id = &quot;distance&quot;)</code></pre>
<embed src="figure/sequencing-05P-NFOL_classification_RNA.Rmd/43-cluster_heatmap_batch_dist-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="correlation-1" class="section level4">
<h4>Correlation</h4>
<pre class="r"><code>dendrogram_heatmap(batch_dend, method_id = &quot;correlation&quot;)</code></pre>
<embed src="figure/sequencing-05P-NFOL_classification_RNA.Rmd/44-cluster_heatmap_batch_cor-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
</div>
<div id="type-2" class="section level3 tabset">
<h3 class="tabset">Type</h3>
<div id="distance-2" class="section level4">
<h4>Distance</h4>
<pre class="r"><code>dendrogram_heatmap(type_dend, method_id = &quot;distance&quot;)</code></pre>
<embed src="figure/sequencing-05P-NFOL_classification_RNA.Rmd/45-cluster_heatmap_type_dist-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="correlation-2" class="section level4">
<h4>Correlation</h4>
<pre class="r"><code>dendrogram_heatmap(type_dend, method_id = &quot;correlation&quot;)</code></pre>
<embed src="figure/sequencing-05P-NFOL_classification_RNA.Rmd/46-cluster_heatmap_type_cor-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
</div>
<div id="class-with-region-id" class="section level3 tabset">
<h3 class="tabset">Class with Region ID</h3>
<div id="distance-3" class="section level4">
<h4>Distance</h4>
<pre class="r"><code>dendrogram_heatmap(type_region_dend, method_id = &quot;distance&quot;)</code></pre>
<embed src="figure/sequencing-05P-NFOL_classification_RNA.Rmd/47-cluster_heatmap_region_dist-1.pdf" width="2880" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="correlation-3" class="section level4">
<h4>Correlation</h4>
<pre class="r"><code>dendrogram_heatmap(type_region_dend, method_id = &quot;correlation&quot;)</code></pre>
<embed src="figure/sequencing-05P-NFOL_classification_RNA.Rmd/45-cluster_heatmap_region_cor-1.pdf" width="2880" style="display: block; margin: auto;" type="application/pdf" />
</div>
</div>
</div>
<div id="pseudobulk-pca" class="section level2 tabset">
<h2 class="tabset">Pseudobulk PCA</h2>
<div id="batch-5" class="section level3">
<h3>Batch</h3>
<pre class="r"><code>pseudobulk_pca(array, &quot;orig.ident&quot;, &quot;Batch&quot;, batch_colors_rna)</code></pre>
<embed src="figure/sequencing-05P-NFOL_classification_RNA.Rmd/46-pseudobulk_pca_batch-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="region-class" class="section level3">
<h3>Region-Class</h3>
<pre class="r"><code>pseudobulk_pca(array, &quot;Region_Type&quot;, &quot;Region_Type&quot;, region_nfol_type_colors)</code></pre>
<embed src="figure/sequencing-05P-NFOL_classification_RNA.Rmd/47-pseudobulk_pca_region-type-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
</div>
</div>
<div id="enrichr" class="section level1 tabset">
<h1 class="tabset">enrichR</h1>
<pre class="r"><code>dbs &lt;- c(&quot;GO_Molecular_Function_2018&quot;, &quot;GO_Cellular_Component_2018&quot;, &quot;GO_Biological_Process_2018&quot;, &quot;KEGG_2016&quot;)</code></pre>
<div id="type-3" class="section level2 tabset">
<h2 class="tabset">Type</h2>
<pre class="r"><code>enrichr_type &lt;- marker_type

for (i in 1:length(enrichr_type)){
  enrichr_type[[i]] &lt;- enrichr(marker_type[[i]]$gene, databases = dbs) 
}</code></pre>
<div id="nfol.1-nfol.frmd4a-3" class="section level3 tabset">
<h3 class="tabset">NFOL.1 | NFOL.Frmd4a</h3>
<pre class="r"><code>src_list &lt;- lapply(names(enrichr_type[[1]]), function(set) {
    src &lt;- c(
        &quot;#### {{set}} {.unnumbered}&quot;,
        &quot;```{r enrichr-NFOL.1|NFOL.Frmd4a2-{{set}}, message = FALSE, warning = FALSE}&quot;,
        &quot;enrichr_type[[1]]${{set}}&quot;,  
        &quot;```&quot;,
        &quot;&quot;
    )
    knit_expand(text = src)
})
out &lt;- knit_child(text = unlist(src_list), options = list(cache = FALSE))</code></pre>
<div id="go_molecular_function_2018" class="section level4 unnumbered">
<h4 class="unnumbered">GO_Molecular_Function_2018</h4>
<pre class="r"><code>enrichr_type[[1]]$GO_Molecular_Function_2018</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["Term"],"name":[1],"type":["chr"],"align":["left"]},{"label":["Overlap"],"name":[2],"type":["chr"],"align":["left"]},{"label":["P.value"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["Adjusted.P.value"],"name":[4],"type":["dbl"],"align":["right"]},{"label":["Old.P.value"],"name":[5],"type":["int"],"align":["right"]},{"label":["Old.Adjusted.P.value"],"name":[6],"type":["int"],"align":["right"]},{"label":["Odds.Ratio"],"name":[7],"type":["dbl"],"align":["right"]},{"label":["Combined.Score"],"name":[8],"type":["dbl"],"align":["right"]},{"label":["Genes"],"name":[9],"type":["chr"],"align":["left"]}],"data":[{"1":"actin binding (GO:0003779)","2":"4/255","3":"0.0009904088","4":"0.03664512","5":"0","6":"0","7":"10.1344300","8":"70.1038333","9":"ABLIM2;MAP1B;MICAL3;PARVB"},{"1":"1-phosphatidylinositol-3-kinase regulator activity (GO:0046935)","2":"1/6","3":"0.0104553653","4":"0.12873082","5":"0","6":"0","7":"117.4117647","8":"535.4727919","9":"PIK3R3"},{"1":"hydrolase activity, acting on carbon-nitrogen (but not peptide) bonds, in cyclic amides (GO:0016812)","2":"1/6","3":"0.0104553653","4":"0.12873082","5":"0","6":"0","7":"117.4117647","8":"535.4727919","9":"DPYSL2"},{"1":"phosphatidylinositol 3-kinase regulator activity (GO:0035014)","2":"1/8","3":"0.0139168458","4":"0.12873082","5":"0","6":"0","7":"83.8571429","8":"358.4603753","9":"PIK3R3"},{"1":"ephrin receptor activity (GO:0005003)","2":"1/17","3":"0.0293485146","4":"0.13462154","5":"0","6":"0","7":"36.6709559","8":"129.3939572","9":"EPHB1"},{"1":"transmembrane-ephrin receptor activity (GO:0005005)","2":"1/17","3":"0.0293485146","4":"0.13462154","5":"0","6":"0","7":"36.6709559","8":"129.3939572","9":"EPHB1"},{"1":"semaphorin receptor binding (GO:0030215)","2":"1/18","3":"0.0310485942","4":"0.13462154","5":"0","6":"0","7":"34.5121107","8":"119.8330112","9":"SEMA4D"},{"1":"FAD binding (GO:0071949)","2":"1/19","3":"0.0327457812","4":"0.13462154","5":"0","6":"0","7":"32.5931373","8":"111.4353217","9":"MICAL3"},{"1":"neuropilin binding (GO:0038191)","2":"1/19","3":"0.0327457812","4":"0.13462154","5":"0","6":"0","7":"32.5931373","8":"111.4353217","9":"SEMA4D"},{"1":"actinin binding (GO:0042805)","2":"1/22","3":"0.0378200346","4":"0.13993413","5":"0","6":"0","7":"27.9327731","8":"91.4774940","9":"CNKSR3"},{"1":"phosphotyrosine residue binding (GO:0001784)","2":"1/30","3":"0.0512252878","4":"0.15794464","5":"0","6":"0","7":"20.2190669","8":"60.0814016","9":"PIK3R3"},{"1":"alpha-actinin binding (GO:0051393)","2":"1/30","3":"0.0512252878","4":"0.15794464","5":"0","6":"0","7":"20.2190669","8":"60.0814016","9":"CNKSR3"},{"1":"protein phosphorylated amino acid binding (GO:0045309)","2":"1/35","3":"0.0595112534","4":"0.16937818","5":"0","6":"0","7":"17.2413495","8":"48.6480167","9":"PIK3R3"},{"1":"oxidoreductase activity, acting on paired donors, with incorporation or reduction of molecular oxygen, NAD(P)H as one donor, and incorporation of one atom of oxygen (GO:0016709)","2":"1/42","3":"0.0709936098","4":"0.18314801","5":"0","6":"0","7":"14.2926829","8":"37.8065105","9":"MICAL3"},{"1":"1-phosphatidylinositol-3-kinase activity (GO:0016303)","2":"1/44","3":"0.0742491927","4":"0.18314801","5":"0","6":"0","7":"13.6265390","8":"35.4334760","9":"PIK3R3"},{"1":"flavin adenine dinucleotide binding (GO:0050660)","2":"1/49","3":"0.0823397387","4":"0.18970347","5":"0","6":"0","7":"12.2040441","8":"30.4722953","9":"MICAL3"},{"1":"phosphatidylinositol kinase activity (GO:0052742)","2":"1/52","3":"0.0871610525","4":"0.18970347","5":"0","6":"0","7":"11.4844291","8":"28.0219804","9":"PIK3R3"},{"1":"transmembrane receptor protein tyrosine kinase activity (GO:0004714)","2":"1/62","3":"0.1030550394","4":"0.20227359","5":"0","6":"0","7":"9.5969142","8":"21.8089114","9":"EPHB1"},{"1":"transmembrane receptor protein kinase activity (GO:0019199)","2":"1/63","3":"0.1046295715","4":"0.20227359","5":"0","6":"0","7":"9.4416509","8":"21.3129128","9":"EPHB1"},{"1":"mitogen-activated protein kinase kinase binding (GO:0031434)","2":"1/66","3":"0.1093370749","4":"0.20227359","5":"0","6":"0","7":"9.0045249","8":"19.9298927","9":"EPHB1"},{"1":"phosphatidylinositol 3-kinase activity (GO:0035004)","2":"1/77","3":"0.1263930938","4":"0.21772803","5":"0","6":"0","7":"7.6969814","8":"15.9201165","9":"PIK3R3"},{"1":"MAP kinase kinase kinase activity (GO:0004709)","2":"1/79","3":"0.1294599114","4":"0.21772803","5":"0","6":"0","7":"7.4988688","8":"15.3305674","9":"EPHB1"},{"1":"protein kinase regulator activity (GO:0019887)","2":"1/108","3":"0.1727707098","4":"0.27793549","5":"0","6":"0","7":"5.4584937","8":"9.5839683","9":"CNKSR3"},{"1":"magnesium ion binding (GO:0000287)","2":"1/128","3":"0.2014128573","4":"0.29809103","5":"0","6":"0","7":"4.5942566","8":"7.3618297","9":"CERK"},{"1":"actin filament binding (GO:0051015)","2":"1/128","3":"0.2014128573","4":"0.29809103","5":"0","6":"0","7":"4.5942566","8":"7.3618297","9":"ABLIM2"},{"1":"protein tyrosine kinase activity (GO:0004713)","2":"1/148","3":"0.2290906712","4":"0.32601365","5":"0","6":"0","7":"3.9651861","8":"5.8432465","9":"EPHB1"},{"1":"microtubule binding (GO:0008017)","2":"1/196","3":"0.2917696079","4":"0.39983243","5":"0","6":"0","7":"2.9819005","8":"3.6730776","9":"MAP1B"},{"1":"regulatory region DNA binding (GO:0000975)","2":"1/225","3":"0.3272084240","4":"0.43238256","5":"0","6":"0","7":"2.5920431","8":"2.8957215","9":"TCF7L1"},{"1":"tubulin binding (GO:0015631)","2":"1/256","3":"0.3631857102","4":"0.46337487","5":"0","6":"0","7":"2.2733564","8":"2.3025485","9":"MAP1B"},{"1":"zinc ion binding (GO:0008270)","2":"1/303","3":"0.4141929569","4":"0.51083798","5":"0","6":"0","7":"1.9149786","8":"1.6879068","9":"TMEM163"},{"1":"transcription regulatory region DNA binding (GO:0044212)","2":"1/375","3":"0.4847217237","4":"0.56870605","5":"0","6":"0","7":"1.5406574","8":"1.1157138","9":"TCF7L1"},{"1":"sequence-specific DNA binding (GO:0043565)","2":"1/395","3":"0.5028015534","4":"0.56870605","5":"0","6":"0","7":"1.4609585","8":"1.0044962","9":"TCF7L1"},{"1":"transition metal ion binding (GO:0046914)","2":"1/400","3":"0.5072243142","4":"0.56870605","5":"0","6":"0","7":"1.4422822","8":"0.9790239","9":"TMEM163"},{"1":"metal ion binding (GO:0046872)","2":"1/443","3":"0.5437151112","4":"0.59168997","5":"0","6":"0","7":"1.2991083","8":"0.7915855","9":"CERK"},{"1":"protein kinase binding (GO:0019901)","2":"1/496","3":"0.5850926851","4":"0.61491717","5":"0","6":"0","7":"1.1568627","8":"0.6200611","9":"EPHB1"},{"1":"protein kinase activity (GO:0004672)","2":"1/514","3":"0.5982977828","4":"0.61491717","5":"0","6":"0","7":"1.1152391","8":"0.5728612","9":"EPHB1"},{"1":"DNA binding (GO:0003677)","2":"1/894","3":"0.7984933904","4":"0.79849339","5":"0","6":"0","7":"0.6281536","8":"0.1413525","9":"TCF7L1"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
<div id="go_cellular_component_2018" class="section level4 unnumbered">
<h4 class="unnumbered">GO_Cellular_Component_2018</h4>
<pre class="r"><code>enrichr_type[[1]]$GO_Cellular_Component_2018</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["Term"],"name":[1],"type":["chr"],"align":["left"]},{"label":["Overlap"],"name":[2],"type":["chr"],"align":["left"]},{"label":["P.value"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["Adjusted.P.value"],"name":[4],"type":["dbl"],"align":["right"]},{"label":["Old.P.value"],"name":[5],"type":["int"],"align":["right"]},{"label":["Old.Adjusted.P.value"],"name":[6],"type":["int"],"align":["right"]},{"label":["Odds.Ratio"],"name":[7],"type":["dbl"],"align":["right"]},{"label":["Combined.Score"],"name":[8],"type":["dbl"],"align":["right"]},{"label":["Genes"],"name":[9],"type":["chr"],"align":["left"]}],"data":[{"1":"main axon (GO:0044304)","2":"2/34","3":"0.001611348","4":"0.03222697","5":"0","6":"0","7":"37.7518939","8":"242.7704971","9":"NFASC;MBP"},{"1":"early endosome (GO:0005769)","2":"3/223","3":"0.006882166","4":"0.04950420","5":"0","6":"0","7":"8.4140625","8":"41.8921186","9":"TMEM108;TMEM163;EPHB1"},{"1":"early endosome membrane (GO:0031901)","2":"2/74","3":"0.007425630","4":"0.04950420","5":"0","6":"0","7":"16.7449495","8":"82.0974355","9":"TMEM163;EPHB1"},{"1":"apical dendrite (GO:0097440)","2":"1/8","3":"0.013916846","4":"0.06257269","5":"0","6":"0","7":"83.8571429","8":"358.4603753","9":"MAP1B"},{"1":"axon initial segment (GO:0043194)","2":"1/9","3":"0.015643173","4":"0.06257269","5":"0","6":"0","7":"73.3713235","8":"305.0574683","9":"NFASC"},{"1":"node of Ranvier (GO:0033268)","2":"1/12","3":"0.020804561","4":"0.06934854","5":"0","6":"0","7":"53.3529412","8":"206.6136938","9":"NFASC"},{"1":"axon (GO:0030424)","2":"2/142","3":"0.025558496","4":"0.07302427","5":"0","6":"0","7":"8.5822511","8":"31.4692738","9":"MAP1B;TMEM108"},{"1":"microtubule cytoskeleton (GO:0015630)","2":"3/389","3":"0.030184698","4":"0.07546175","5":"0","6":"0","7":"4.7552623","8":"16.6454161","9":"SPECC1;DPYSL2;MAP1B"},{"1":"microtubule (GO:0005874)","2":"2/211","3":"0.052498575","4":"0.11488987","5":"0","6":"0","7":"5.7288676","8":"16.8827967","9":"DPYSL2;MAP1B"},{"1":"polymeric cytoskeletal fiber (GO:0099513)","2":"2/222","3":"0.057444936","4":"0.11488987","5":"0","6":"0","7":"5.4393939","8":"15.5399592","9":"DPYSL2;MAP1B"},{"1":"actin cytoskeleton (GO:0015629)","2":"2/295","3":"0.093914941","4":"0.17075444","5":"0","6":"0","7":"4.0690868","8":"9.6248786","9":"ABLIM2;PARVB"},{"1":"ficolin-1-rich granule membrane (GO:0101003)","2":"1/62","3":"0.103055039","4":"0.17175840","5":"0","6":"0","7":"9.5969142","8":"21.8089114","9":"NFASC"},{"1":"bicellular tight junction (GO:0005923)","2":"1/73","3":"0.120227941","4":"0.18407248","5":"0","6":"0","7":"8.1262255","8":"17.2143184","9":"CLDN11"},{"1":"focal adhesion (GO:0005925)","2":"2/357","3":"0.128850735","4":"0.18407248","5":"0","6":"0","7":"3.3478446","8":"6.8600706","9":"NFASC;PARVB"},{"1":"cytoskeleton (GO:0005856)","2":"2/521","3":"0.231329023","4":"0.30843870","5":"0","6":"0","7":"2.2708005","8":"3.3242572","9":"ABLIM2;PARVB"},{"1":"tertiary granule (GO:0070820)","2":"1/165","3":"0.251882195","4":"0.31485274","5":"0","6":"0","7":"3.5511119","8":"4.8962510","9":"NFASC"},{"1":"ficolin-1-rich granule (GO:0101002)","2":"1/185","3":"0.277858400","4":"0.32689223","5":"0","6":"0","7":"3.1619246","8":"4.0492986","9":"NFASC"},{"1":"dendrite (GO:0030425)","2":"1/216","3":"0.316398705","4":"0.35155412","5":"0","6":"0","7":"2.7017784","8":"3.1090772","9":"MAP1B"},{"1":"microtubule organizing center (GO:0005815)","2":"1/508","3":"0.593942105","4":"0.62520222","5":"0","6":"0","7":"1.1287852","8":"0.5880671","9":"SPECC1"},{"1":"integral component of plasma membrane (GO:0005887)","2":"2/1464","3":"0.737114252","4":"0.73711425","5":"0","6":"0","7":"0.7670273","8":"0.2339528","9":"SEMA4D;EPHB1"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
<div id="go_biological_process_2018" class="section level4 unnumbered">
<h4 class="unnumbered">GO_Biological_Process_2018</h4>
<pre class="r"><code>enrichr_type[[1]]$GO_Biological_Process_2018</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["Term"],"name":[1],"type":["chr"],"align":["left"]},{"label":["Overlap"],"name":[2],"type":["chr"],"align":["left"]},{"label":["P.value"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["Adjusted.P.value"],"name":[4],"type":["dbl"],"align":["right"]},{"label":["Old.P.value"],"name":[5],"type":["int"],"align":["right"]},{"label":["Old.Adjusted.P.value"],"name":[6],"type":["int"],"align":["right"]},{"label":["Odds.Ratio"],"name":[7],"type":["dbl"],"align":["right"]},{"label":["Combined.Score"],"name":[8],"type":["dbl"],"align":["right"]},{"label":["Genes"],"name":[9],"type":["chr"],"align":["left"]}],"data":[{"1":"lamellipodium assembly (GO:0030032)","2":"2/23","3":"0.0007355116","4":"0.09606642","5":"0","6":"0","7":"57.5584416","8":"415.28094306","9":"ABLIM2;PARVB"},{"1":"lamellipodium organization (GO:0097581)","2":"2/26","3":"0.0009417208","4":"0.09606642","5":"0","6":"0","7":"50.3560606","8":"350.87104396","9":"ABLIM2;PARVB"},{"1":"nervous system development (GO:0007399)","2":"5/456","3":"0.0011127384","4":"0.09606642","5":"0","6":"0","7":"7.2113821","8":"49.04411385","9":"TCF7L1;NFASC;DPYSL2;MBP;EPHB1"},{"1":"regulation of sodium ion transport (GO:0002028)","2":"2/36","3":"0.0018055708","4":"0.10413736","5":"0","6":"0","7":"35.5276292","8":"224.42371840","9":"NKAIN2;CNKSR3"},{"1":"myelination (GO:0042552)","2":"2/38","3":"0.0020103738","4":"0.10413736","5":"0","6":"0","7":"33.5505051","8":"208.32966662","9":"NFASC;MBP"},{"1":"neurogenesis (GO:0022008)","2":"2/51","3":"0.0035945543","4":"0.14509485","5":"0","6":"0","7":"24.6332715","8":"138.64431104","9":"TCF7L1;EPHB1"},{"1":"regulation of signal transduction (GO:0009966)","2":"3/234","3":"0.0078533490","4":"0.14509485","5":"0","6":"0","7":"8.0089286","8":"38.81779685","9":"TCF7L1;TMEM108;EPHB1"},{"1":"plasma membrane bounded cell projection assembly (GO:0120031)","2":"3/242","3":"0.0086081507","4":"0.14509485","5":"0","6":"0","7":"7.7377092","8":"36.79316143","9":"ABLIM2;PARVB;BBIP1"},{"1":"negative regulation of smooth muscle cell differentiation (GO:0051151)","2":"1/6","3":"0.0104553653","4":"0.14509485","5":"0","6":"0","7":"117.4117647","8":"535.47279185","9":"PDCD4"},{"1":"regulation of skeletal muscle satellite cell proliferation (GO:0014842)","2":"1/6","3":"0.0104553653","4":"0.14509485","5":"0","6":"0","7":"117.4117647","8":"535.47279185","9":"EPHB1"},{"1":"protein localization to non-motile cilium (GO:0097499)","2":"1/6","3":"0.0104553653","4":"0.14509485","5":"0","6":"0","7":"117.4117647","8":"535.47279185","9":"BBIP1"},{"1":"dendrite extension (GO:0097484)","2":"1/6","3":"0.0104553653","4":"0.14509485","5":"0","6":"0","7":"117.4117647","8":"535.47279185","9":"TMEM108"},{"1":"negative regulation of vasculature development (GO:1901343)","2":"1/6","3":"0.0104553653","4":"0.14509485","5":"0","6":"0","7":"117.4117647","8":"535.47279185","9":"PDCD4"},{"1":"regulation of protein depolymerization (GO:1901879)","2":"1/6","3":"0.0104553653","4":"0.14509485","5":"0","6":"0","7":"117.4117647","8":"535.47279185","9":"MAP1B"},{"1":"retinal ganglion cell axon guidance (GO:0031290)","2":"1/6","3":"0.0104553653","4":"0.14509485","5":"0","6":"0","7":"117.4117647","8":"535.47279185","9":"EPHB1"},{"1":"regulation of collateral sprouting (GO:0048670)","2":"1/7","3":"0.0121875781","4":"0.14509485","5":"0","6":"0","7":"97.8382353","8":"431.20617589","9":"SEMA4D"},{"1":"protein localization to ciliary membrane (GO:1903441)","2":"1/7","3":"0.0121875781","4":"0.14509485","5":"0","6":"0","7":"97.8382353","8":"431.20617589","9":"BBIP1"},{"1":"central nervous system neuron axonogenesis (GO:0021955)","2":"1/7","3":"0.0121875781","4":"0.14509485","5":"0","6":"0","7":"97.8382353","8":"431.20617589","9":"EPHB1"},{"1":"leukocyte aggregation (GO:0070486)","2":"1/7","3":"0.0121875781","4":"0.14509485","5":"0","6":"0","7":"97.8382353","8":"431.20617589","9":"SEMA4D"},{"1":"regulation of cell projection organization (GO:0031344)","2":"1/7","3":"0.0121875781","4":"0.14509485","5":"0","6":"0","7":"97.8382353","8":"431.20617589","9":"SEMA4D"},{"1":"retrograde axonal transport (GO:0008090)","2":"1/7","3":"0.0121875781","4":"0.14509485","5":"0","6":"0","7":"97.8382353","8":"431.20617589","9":"TMEM108"},{"1":"positive regulation of smooth muscle cell apoptotic process (GO:0034393)","2":"1/8","3":"0.0139168458","4":"0.14509485","5":"0","6":"0","7":"83.8571429","8":"358.46037530","9":"PDCD4"},{"1":"positive regulation of chemokine secretion (GO:0090197)","2":"1/8","3":"0.0139168458","4":"0.14509485","5":"0","6":"0","7":"83.8571429","8":"358.46037530","9":"MBP"},{"1":"postsynaptic density organization (GO:0097106)","2":"1/9","3":"0.0156431733","4":"0.14509485","5":"0","6":"0","7":"73.3713235","8":"305.05746832","9":"TMEM108"},{"1":"negative regulation of cytokine production involved in inflammatory response (GO:1900016)","2":"1/9","3":"0.0156431733","4":"0.14509485","5":"0","6":"0","7":"73.3713235","8":"305.05746832","9":"PDCD4"},{"1":"negative regulation of heterotypic cell-cell adhesion (GO:0034115)","2":"1/9","3":"0.0156431733","4":"0.14509485","5":"0","6":"0","7":"73.3713235","8":"305.05746832","9":"MBP"},{"1":"regulation of cell-cell adhesion involved in gastrulation (GO:0070587)","2":"1/9","3":"0.0156431733","4":"0.14509485","5":"0","6":"0","7":"73.3713235","8":"305.05746832","9":"MBP"},{"1":"regulation of phospholipid metabolic process (GO:1903725)","2":"1/10","3":"0.0173665652","4":"0.14509485","5":"0","6":"0","7":"65.2156863","8":"264.33277144","9":"PIK3R3"},{"1":"central nervous system projection neuron axonogenesis (GO:0021952)","2":"1/10","3":"0.0173665652","4":"0.14509485","5":"0","6":"0","7":"65.2156863","8":"264.33277144","9":"EPHB1"},{"1":"positive regulation of glucose import in response to insulin stimulus (GO:2001275)","2":"1/10","3":"0.0173665652","4":"0.14509485","5":"0","6":"0","7":"65.2156863","8":"264.33277144","9":"PIK3R3"},{"1":"actin filament depolymerization (GO:0030042)","2":"1/10","3":"0.0173665652","4":"0.14509485","5":"0","6":"0","7":"65.2156863","8":"264.33277144","9":"MICAL3"},{"1":"positive regulation of endothelial cell apoptotic process (GO:2000353)","2":"1/11","3":"0.0190870263","4":"0.14539823","5":"0","6":"0","7":"58.6911765","8":"232.34348502","9":"PDCD4"},{"1":"regulation of lipid kinase activity (GO:0043550)","2":"1/11","3":"0.0190870263","4":"0.14539823","5":"0","6":"0","7":"58.6911765","8":"232.34348502","9":"PIK3R3"},{"1":"negative regulation of JUN kinase activity (GO:0043508)","2":"1/11","3":"0.0190870263","4":"0.14539823","5":"0","6":"0","7":"58.6911765","8":"232.34348502","9":"PDCD4"},{"1":"cytoskeleton organization (GO:0007010)","2":"2/127","3":"0.0207622404","4":"0.14563193","5":"0","6":"0","7":"9.6193939","8":"37.27148948","9":"DPYSL2;MICAL3"},{"1":"negative regulation of vascular smooth muscle cell proliferation (GO:1904706)","2":"1/12","3":"0.0208045615","4":"0.14563193","5":"0","6":"0","7":"53.3529412","8":"206.61369383","9":"PDCD4"},{"1":"negative regulation of myoblast differentiation (GO:0045662)","2":"1/12","3":"0.0208045615","4":"0.14563193","5":"0","6":"0","7":"53.3529412","8":"206.61369383","9":"PRICKLE1"},{"1":"regulation of glucose import in response to insulin stimulus (GO:2001273)","2":"1/13","3":"0.0225191754","4":"0.14854148","5":"0","6":"0","7":"48.9044118","8":"185.51341333","9":"PIK3R3"},{"1":"positive regulation of sodium ion transport (GO:0010765)","2":"1/13","3":"0.0225191754","4":"0.14854148","5":"0","6":"0","7":"48.9044118","8":"185.51341333","9":"CNKSR3"},{"1":"epiboly involved in wound healing (GO:0090505)","2":"1/14","3":"0.0242308729","4":"0.14854148","5":"0","6":"0","7":"45.1402715","8":"167.92757530","9":"TMEFF2"},{"1":"negative regulation of axon extension involved in axon guidance (GO:0048843)","2":"1/15","3":"0.0259396587","4":"0.14854148","5":"0","6":"0","7":"41.9138655","8":"153.06869330","9":"SEMA4D"},{"1":"positive regulation of cellular response to insulin stimulus (GO:1900078)","2":"1/15","3":"0.0259396587","4":"0.14854148","5":"0","6":"0","7":"41.9138655","8":"153.06869330","9":"PIK3R3"},{"1":"positive regulation of epithelial cell apoptotic process (GO:1904037)","2":"1/16","3":"0.0276455377","4":"0.14854148","5":"0","6":"0","7":"39.1176471","8":"140.36549891","9":"PDCD4"},{"1":"cardiac epithelial to mesenchymal transition (GO:0060317)","2":"1/16","3":"0.0276455377","4":"0.14854148","5":"0","6":"0","7":"39.1176471","8":"140.36549891","9":"PDCD4"},{"1":"dendritic spine morphogenesis (GO:0060997)","2":"1/16","3":"0.0276455377","4":"0.14854148","5":"0","6":"0","7":"39.1176471","8":"140.36549891","9":"EPHB1"},{"1":"negative regulation of cell differentiation (GO:0045596)","2":"2/151","3":"0.0286324354","4":"0.14854148","5":"0","6":"0","7":"8.0601993","8":"28.63962188","9":"SEMA4D;PDCD4"},{"1":"positive regulation of interleukin-6 secretion (GO:2000778)","2":"1/17","3":"0.0293485146","4":"0.14854148","5":"0","6":"0","7":"36.6709559","8":"129.39395719","9":"MBP"},{"1":"negative regulation of axon guidance (GO:1902668)","2":"1/17","3":"0.0293485146","4":"0.14854148","5":"0","6":"0","7":"36.6709559","8":"129.39395719","9":"SEMA4D"},{"1":"negative regulation of stress fiber assembly (GO:0051497)","2":"1/17","3":"0.0293485146","4":"0.14854148","5":"0","6":"0","7":"36.6709559","8":"129.39395719","9":"TMEFF2"},{"1":"regulation of microtubule polymerization or depolymerization (GO:0031110)","2":"1/18","3":"0.0310485942","4":"0.14854148","5":"0","6":"0","7":"34.5121107","8":"119.83301121","9":"MAP1B"},{"1":"negative regulation of peptidyl-serine phosphorylation (GO:0033137)","2":"1/18","3":"0.0310485942","4":"0.14854148","5":"0","6":"0","7":"34.5121107","8":"119.83301121","9":"CNKSR3"},{"1":"regulation of heterotypic cell-cell adhesion (GO:0034114)","2":"1/19","3":"0.0327457812","4":"0.14854148","5":"0","6":"0","7":"32.5931373","8":"111.43532170","9":"MBP"},{"1":"neuron projection morphogenesis (GO:0048812)","2":"2/164","3":"0.0333202074","4":"0.14854148","5":"0","6":"0","7":"7.4085297","8":"25.20078984","9":"MAP1B;EPHB1"},{"1":"cell projection assembly (GO:0030031)","2":"1/20","3":"0.0344400804","4":"0.14854148","5":"0","6":"0","7":"30.8761610","8":"104.00740625","9":"PARVB"},{"1":"calcium-independent cell-cell adhesion via plasma membrane cell-adhesion molecules (GO:0016338)","2":"1/20","3":"0.0344400804","4":"0.14854148","5":"0","6":"0","7":"30.8761610","8":"104.00740625","9":"CLDN11"},{"1":"negative regulation of macromolecule biosynthetic process (GO:0010558)","2":"1/20","3":"0.0344400804","4":"0.14854148","5":"0","6":"0","7":"30.8761610","8":"104.00740625","9":"TMEFF2"},{"1":"negative regulation of actin filament bundle assembly (GO:0032232)","2":"1/20","3":"0.0344400804","4":"0.14854148","5":"0","6":"0","7":"30.8761610","8":"104.00740625","9":"TMEFF2"},{"1":"wound healing, spreading of cells (GO:0044319)","2":"1/20","3":"0.0344400804","4":"0.14854148","5":"0","6":"0","7":"30.8761610","8":"104.00740625","9":"TMEFF2"},{"1":"neuron projection development (GO:0031175)","2":"2/168","3":"0.0348195034","4":"0.14854148","5":"0","6":"0","7":"7.2285506","8":"24.27041951","9":"TMEM108;EPHB1"},{"1":"regulation of microtubule depolymerization (GO:0031114)","2":"1/22","3":"0.0378200346","4":"0.14854148","5":"0","6":"0","7":"27.9327731","8":"91.47749401","9":"MAP1B"},{"1":"peripheral nervous system development (GO:0007422)","2":"1/22","3":"0.0378200346","4":"0.14854148","5":"0","6":"0","7":"27.9327731","8":"91.47749401","9":"NFASC"},{"1":"cellular response to nerve growth factor stimulus (GO:1990090)","2":"1/22","3":"0.0378200346","4":"0.14854148","5":"0","6":"0","7":"27.9327731","8":"91.47749401","9":"TMEM108"},{"1":"regulation of cytokine production involved in inflammatory response (GO:1900015)","2":"1/22","3":"0.0378200346","4":"0.14854148","5":"0","6":"0","7":"27.9327731","8":"91.47749401","9":"PDCD4"},{"1":"negative regulation of protein phosphorylation (GO:0001933)","2":"2/179","3":"0.0390756059","4":"0.14854148","5":"0","6":"0","7":"6.7755521","8":"21.96808063","9":"SEMA4D;CNKSR3"},{"1":"regulation of axon extension involved in axon guidance (GO:0048841)","2":"1/23","3":"0.0395056990","4":"0.14854148","5":"0","6":"0","7":"26.6617647","8":"86.15243598","9":"SEMA4D"},{"1":"negative regulation of axon extension (GO:0030517)","2":"1/23","3":"0.0395056990","4":"0.14854148","5":"0","6":"0","7":"26.6617647","8":"86.15243598","9":"SEMA4D"},{"1":"dendritic spine organization (GO:0097061)","2":"1/25","3":"0.0428684261","4":"0.14854148","5":"0","6":"0","7":"24.4375000","8":"76.96883173","9":"EPHB1"},{"1":"positive regulation of endopeptidase activity (GO:0010950)","2":"1/25","3":"0.0428684261","4":"0.14854148","5":"0","6":"0","7":"24.4375000","8":"76.96883173","9":"MBP"},{"1":"sensory perception of pain (GO:0019233)","2":"1/25","3":"0.0428684261","4":"0.14854148","5":"0","6":"0","7":"24.4375000","8":"76.96883173","9":"EPHB1"},{"1":"regulation of vascular smooth muscle cell proliferation (GO:1904705)","2":"1/26","3":"0.0445454983","4":"0.14854148","5":"0","6":"0","7":"23.4588235","8":"72.98612816","9":"PDCD4"},{"1":"protein depolymerization (GO:0051261)","2":"1/26","3":"0.0445454983","4":"0.14854148","5":"0","6":"0","7":"23.4588235","8":"72.98612816","9":"MICAL3"},{"1":"modulation of excitatory postsynaptic potential (GO:0098815)","2":"1/26","3":"0.0445454983","4":"0.14854148","5":"0","6":"0","7":"23.4588235","8":"72.98612816","9":"TMEM108"},{"1":"positive regulation of glucose import (GO:0046326)","2":"1/27","3":"0.0462197158","4":"0.14854148","5":"0","6":"0","7":"22.5554299","8":"69.34325925","9":"PIK3R3"},{"1":"regulation of supramolecular fiber organization (GO:1902903)","2":"1/27","3":"0.0462197158","4":"0.14854148","5":"0","6":"0","7":"22.5554299","8":"69.34325925","9":"MAP1B"},{"1":"positive regulation of synapse assembly (GO:0051965)","2":"1/27","3":"0.0462197158","4":"0.14854148","5":"0","6":"0","7":"22.5554299","8":"69.34325925","9":"EPHB1"},{"1":"negative chemotaxis (GO:0050919)","2":"1/27","3":"0.0462197158","4":"0.14854148","5":"0","6":"0","7":"22.5554299","8":"69.34325925","9":"SEMA4D"},{"1":"negative regulation of smooth muscle cell proliferation (GO:0048662)","2":"1/28","3":"0.0478910835","4":"0.14854148","5":"0","6":"0","7":"21.7189542","8":"66.00012158","9":"PDCD4"},{"1":"leukocyte cell-cell adhesion (GO:0007159)","2":"1/28","3":"0.0478910835","4":"0.14854148","5":"0","6":"0","7":"21.7189542","8":"66.00012158","9":"SEMA4D"},{"1":"negative regulation of osteoblast differentiation (GO:0045668)","2":"1/28","3":"0.0478910835","4":"0.14854148","5":"0","6":"0","7":"21.7189542","8":"66.00012158","9":"SEMA4D"},{"1":"phosphatidylinositol phosphorylation (GO:0046854)","2":"1/28","3":"0.0478910835","4":"0.14854148","5":"0","6":"0","7":"21.7189542","8":"66.00012158","9":"PIK3R3"},{"1":"negative regulation of peptidyl-tyrosine phosphorylation (GO:0050732)","2":"1/28","3":"0.0478910835","4":"0.14854148","5":"0","6":"0","7":"21.7189542","8":"66.00012158","9":"SEMA4D"},{"1":"neural crest cell migration (GO:0001755)","2":"1/29","3":"0.0495596059","4":"0.14854148","5":"0","6":"0","7":"20.9422269","8":"62.92257879","9":"SEMA4D"},{"1":"neuron projection extension (GO:1990138)","2":"1/29","3":"0.0495596059","4":"0.14854148","5":"0","6":"0","7":"20.9422269","8":"62.92257879","9":"TMEM108"},{"1":"primary neural tube formation (GO:0014020)","2":"1/29","3":"0.0495596059","4":"0.14854148","5":"0","6":"0","7":"20.9422269","8":"62.92257879","9":"PRICKLE1"},{"1":"regulation of NIK/NF-kappaB signaling (GO:1901222)","2":"1/29","3":"0.0495596059","4":"0.14854148","5":"0","6":"0","7":"20.9422269","8":"62.92257879","9":"PDCD4"},{"1":"negative regulation of JNK cascade (GO:0046329)","2":"1/29","3":"0.0495596059","4":"0.14854148","5":"0","6":"0","7":"20.9422269","8":"62.92257879","9":"PDCD4"},{"1":"positive regulation of intracellular signal transduction (GO:1902533)","2":"3/480","3":"0.0510289989","4":"0.14854148","5":"0","6":"0","7":"3.8301887","8":"11.39619479","9":"SEMA4D;PDCD4;EPHB1"},{"1":"tube closure (GO:0060606)","2":"1/30","3":"0.0512252878","4":"0.14854148","5":"0","6":"0","7":"20.2190669","8":"60.08140155","9":"PRICKLE1"},{"1":"semaphorin-plexin signaling pathway (GO:0071526)","2":"1/30","3":"0.0512252878","4":"0.14854148","5":"0","6":"0","7":"20.2190669","8":"60.08140155","9":"SEMA4D"},{"1":"positive regulation of phosphorylation (GO:0042327)","2":"2/209","3":"0.0516167301","4":"0.14854148","5":"0","6":"0","7":"5.7848046","8":"17.14563682","9":"SEMA4D;EPHB1"},{"1":"beta-catenin-TCF complex assembly (GO:1904837)","2":"1/31","3":"0.0528881339","4":"0.15052777","5":"0","6":"0","7":"19.5441176","8":"57.45142461","9":"TCF7L1"},{"1":"negative regulation of cell-cell adhesion (GO:0022408)","2":"1/32","3":"0.0545481488","4":"0.15314039","5":"0","6":"0","7":"18.9127135","8":"55.01087073","9":"MBP"},{"1":"substrate adhesion-dependent cell spreading (GO:0034446)","2":"1/33","3":"0.0562053372","4":"0.15314039","5":"0","6":"0","7":"18.3207721","8":"52.74080454","9":"PARVB"},{"1":"dendrite morphogenesis (GO:0048813)","2":"1/33","3":"0.0562053372","4":"0.15314039","5":"0","6":"0","7":"18.3207721","8":"52.74080454","9":"EPHB1"},{"1":"neural tube closure (GO:0001843)","2":"1/34","3":"0.0578597039","4":"0.15314039","5":"0","6":"0","7":"17.7647059","8":"50.62468808","9":"PRICKLE1"},{"1":"regulation of phosphatidylinositol 3-kinase activity (GO:0043551)","2":"1/34","3":"0.0578597039","4":"0.15314039","5":"0","6":"0","7":"17.7647059","8":"50.62468808","9":"PIK3R3"},{"1":"axonogenesis (GO:0007409)","2":"2/224","3":"0.0583613945","4":"0.15314039","5":"0","6":"0","7":"5.3898444","8":"15.31309045","9":"MAP1B;EPHB1"},{"1":"regulation of JUN kinase activity (GO:0043506)","2":"1/35","3":"0.0595112534","4":"0.15314039","5":"0","6":"0","7":"17.2413495","8":"48.64801673","9":"PDCD4"},{"1":"lipid phosphorylation (GO:0046834)","2":"1/36","3":"0.0611599904","4":"0.15314039","5":"0","6":"0","7":"16.7478992","8":"46.79801912","9":"PIK3R3"},{"1":"neural crest cell development (GO:0014032)","2":"1/36","3":"0.0611599904","4":"0.15314039","5":"0","6":"0","7":"16.7478992","8":"46.79801912","9":"SEMA4D"},{"1":"positive regulation of NIK/NF-kappaB signaling (GO:1901224)","2":"1/36","3":"0.0611599904","4":"0.15314039","5":"0","6":"0","7":"16.7478992","8":"46.79801912","9":"PDCD4"},{"1":"regulation of synapse assembly (GO:0051963)","2":"1/36","3":"0.0611599904","4":"0.15314039","5":"0","6":"0","7":"16.7478992","8":"46.79801912","9":"EPHB1"},{"1":"regulation of endothelial cell apoptotic process (GO:2000351)","2":"1/36","3":"0.0611599904","4":"0.15314039","5":"0","6":"0","7":"16.7478992","8":"46.79801912","9":"PDCD4"},{"1":"actin polymerization or depolymerization (GO:0008154)","2":"1/37","3":"0.0628059196","4":"0.15314039","5":"0","6":"0","7":"16.2818627","8":"45.06340839","9":"MICAL3"},{"1":"regulation of cell morphogenesis involved in differentiation (GO:0010769)","2":"1/37","3":"0.0628059196","4":"0.15314039","5":"0","6":"0","7":"16.2818627","8":"45.06340839","9":"SEMA4D"},{"1":"axonal transport (GO:0098930)","2":"1/37","3":"0.0628059196","4":"0.15314039","5":"0","6":"0","7":"16.2818627","8":"45.06340839","9":"TMEM108"},{"1":"regulation of metal ion transport (GO:0010959)","2":"1/38","3":"0.0644490455","4":"0.15314039","5":"0","6":"0","7":"15.8410175","8":"43.43417472","9":"NKAIN2"},{"1":"positive regulation of axonogenesis (GO:0050772)","2":"1/38","3":"0.0644490455","4":"0.15314039","5":"0","6":"0","7":"15.8410175","8":"43.43417472","9":"SEMA4D"},{"1":"regulation of dendrite development (GO:0050773)","2":"1/38","3":"0.0644490455","4":"0.15314039","5":"0","6":"0","7":"15.8410175","8":"43.43417472","9":"SEMA4D"},{"1":"negative regulation of ERK1 and ERK2 cascade (GO:0070373)","2":"1/41","3":"0.0693616504","4":"0.16129288","5":"0","6":"0","7":"14.6507353","8":"39.09433195","9":"CNKSR3"},{"1":"ameboidal-type cell migration (GO:0001667)","2":"1/41","3":"0.0693616504","4":"0.16129288","5":"0","6":"0","7":"14.6507353","8":"39.09433195","9":"SEMA4D"},{"1":"regulation of ERK1 and ERK2 cascade (GO:0070372)","2":"2/248","3":"0.0697482727","4":"0.16129288","5":"0","6":"0","7":"4.8580931","8":"12.93643460","9":"CNKSR3;EPHB1"},{"1":"regulation of stress-activated MAPK cascade (GO:0032872)","2":"1/42","3":"0.0709936098","4":"0.16271987","5":"0","6":"0","7":"14.2926829","8":"37.80651048","9":"EPHB1"},{"1":"ceramide metabolic process (GO:0006672)","2":"1/43","3":"0.0726227890","4":"0.16499388","5":"0","6":"0","7":"13.9516807","8":"36.58795481","9":"CERK"},{"1":"substantia nigra development (GO:0021762)","2":"1/44","3":"0.0742491927","4":"0.16578053","5":"0","6":"0","7":"13.6265390","8":"35.43347596","9":"MBP"},{"1":"positive regulation of interleukin-6 production (GO:0032755)","2":"1/44","3":"0.0742491927","4":"0.16578053","5":"0","6":"0","7":"13.6265390","8":"35.43347596","9":"MBP"},{"1":"regulation of dendrite morphogenesis (GO:0048814)","2":"1/45","3":"0.0758728255","4":"0.16795045","5":"0","6":"0","7":"13.3161765","8":"34.33838017","9":"SEMA4D"},{"1":"negative regulation of ossification (GO:0030279)","2":"1/46","3":"0.0774936919","4":"0.16795045","5":"0","6":"0","7":"13.0196078","8":"33.29841184","9":"SEMA4D"},{"1":"endocytosis (GO:0006897)","2":"2/264","3":"0.0777146851","4":"0.16795045","5":"0","6":"0","7":"4.5577146","8":"11.64364369","9":"DPYSL2;TMEM108"},{"1":"cellular response to interleukin-12 (GO:0071349)","2":"1/47","3":"0.0791117966","4":"0.16795045","5":"0","6":"0","7":"12.7359335","8":"32.30970413","9":"PDCD4"},{"1":"interleukin-12-mediated signaling pathway (GO:0035722)","2":"1/47","3":"0.0791117966","4":"0.16795045","5":"0","6":"0","7":"12.7359335","8":"32.30970413","9":"PDCD4"},{"1":"positive regulation of ion transport (GO:0043270)","2":"1/47","3":"0.0791117966","4":"0.16795045","5":"0","6":"0","7":"12.7359335","8":"32.30970413","9":"CNKSR3"},{"1":"positive regulation of developmental growth (GO:0048639)","2":"1/52","3":"0.0871610525","4":"0.18353425","5":"0","6":"0","7":"11.4844291","8":"28.02198044","9":"SEMA4D"},{"1":"positive regulation of nervous system development (GO:0051962)","2":"1/53","3":"0.0887626820","4":"0.18539947","5":"0","6":"0","7":"11.2630090","8":"27.27663103","9":"EPHB1"},{"1":"positive regulation of phosphatidylinositol 3-kinase signaling (GO:0014068)","2":"1/54","3":"0.0903615816","4":"0.18607077","5":"0","6":"0","7":"11.0499445","8":"26.56336033","9":"SEMA4D"},{"1":"negative regulation of cellular component organization (GO:0051129)","2":"1/55","3":"0.0919577557","4":"0.18607077","5":"0","6":"0","7":"10.8447712","8":"25.88024389","9":"TMEFF2"},{"1":"cell morphogenesis involved in differentiation (GO:0000904)","2":"1/55","3":"0.0919577557","4":"0.18607077","5":"0","6":"0","7":"10.8447712","8":"25.88024389","9":"PARVB"},{"1":"glycosphingolipid metabolic process (GO:0006687)","2":"1/55","3":"0.0919577557","4":"0.18607077","5":"0","6":"0","7":"10.8447712","8":"25.88024389","9":"CERK"},{"1":"protein import (GO:0017038)","2":"1/57","3":"0.0951419456","4":"0.19102143","5":"0","6":"0","7":"10.4564076","8":"24.59749985","9":"PRICKLE1"},{"1":"cell chemotaxis (GO:0060326)","2":"1/61","3":"0.1014778177","4":"0.20063172","5":"0","6":"0","7":"9.7573529","8":"22.32399464","9":"EPHB1"},{"1":"regulation of stress fiber assembly (GO:0051492)","2":"1/61","3":"0.1014778177","4":"0.20063172","5":"0","6":"0","7":"9.7573529","8":"22.32399464","9":"TMEFF2"},{"1":"actin cytoskeleton reorganization (GO:0031532)","2":"1/62","3":"0.1030550394","4":"0.20068613","5":"0","6":"0","7":"9.5969142","8":"21.80891136","9":"PARVB"},{"1":"negative regulation of MAP kinase activity (GO:0043407)","2":"1/62","3":"0.1030550394","4":"0.20068613","5":"0","6":"0","7":"9.5969142","8":"21.80891136","9":"PDCD4"},{"1":"cellular amide metabolic process (GO:0043603)","2":"1/63","3":"0.1046295715","4":"0.20073377","5":"0","6":"0","7":"9.4416509","8":"21.31291281","9":"CERK"},{"1":"establishment of protein localization to organelle (GO:0072594)","2":"1/63","3":"0.1046295715","4":"0.20073377","5":"0","6":"0","7":"9.4416509","8":"21.31291281","9":"PRICKLE1"},{"1":"regulation of cell migration (GO:0030334)","2":"2/317","3":"0.1059595140","4":"0.20179054","5":"0","6":"0","7":"3.7806638","8":"8.48644919","9":"TMEFF2;SEMA4D"},{"1":"negative regulation of cell adhesion (GO:0007162)","2":"1/66","3":"0.1093370749","4":"0.20372879","5":"0","6":"0","7":"9.0045249","8":"19.92989267","9":"SEMA4D"},{"1":"glycolipid metabolic process (GO:0006664)","2":"1/66","3":"0.1093370749","4":"0.20372879","5":"0","6":"0","7":"9.0045249","8":"19.92989267","9":"CERK"},{"1":"positive regulation of protein modification by small protein conjugation or removal (GO:1903322)","2":"1/66","3":"0.1093370749","4":"0.20372879","5":"0","6":"0","7":"9.0045249","8":"19.92989267","9":"PRICKLE1"},{"1":"insulin receptor signaling pathway (GO:0008286)","2":"1/68","3":"0.1124620438","4":"0.20805478","5":"0","6":"0","7":"8.7348551","8":"19.08687700","9":"PIK3R3"},{"1":"negative regulation of phosphatase activity (GO:0010923)","2":"1/69","3":"0.1140205317","4":"0.20944197","5":"0","6":"0","7":"8.6059689","8":"18.68680064","9":"SEMA4D"},{"1":"regulation of osteoblast differentiation (GO:0045667)","2":"1/70","3":"0.1155763609","4":"0.21026169","5":"0","6":"0","7":"8.4808184","8":"18.30011211","9":"SEMA4D"},{"1":"positive regulation of proteasomal ubiquitin-dependent protein catabolic process (GO:0032436)","2":"1/71","3":"0.1171295359","4":"0.21026169","5":"0","6":"0","7":"8.3592437","8":"17.92618756","9":"PRICKLE1"},{"1":"nuclear import (GO:0051170)","2":"1/72","3":"0.1186800611","4":"0.21026169","5":"0","6":"0","7":"8.2410936","8":"17.56444036","9":"PRICKLE1"},{"1":"regulation of neuron death (GO:1901214)","2":"1/73","3":"0.1202279410","4":"0.21026169","5":"0","6":"0","7":"8.1262255","8":"17.21431840","9":"EPHB1"},{"1":"negative regulation of cell cycle (GO:0045786)","2":"1/73","3":"0.1202279410","4":"0.21026169","5":"0","6":"0","7":"8.1262255","8":"17.21431840","9":"PDCD4"},{"1":"positive regulation of inflammatory response (GO:0050729)","2":"1/74","3":"0.1217731798","4":"0.21026169","5":"0","6":"0","7":"8.0145044","8":"16.87530164","9":"PDCD4"},{"1":"positive regulation of cytokine secretion (GO:0050715)","2":"1/74","3":"0.1217731798","4":"0.21026169","5":"0","6":"0","7":"8.0145044","8":"16.87530164","9":"MBP"},{"1":"canonical Wnt signaling pathway (GO:0060070)","2":"1/74","3":"0.1217731798","4":"0.21026169","5":"0","6":"0","7":"8.0145044","8":"16.87530164","9":"TCF7L1"},{"1":"positive regulation of proteasomal protein catabolic process (GO:1901800)","2":"1/74","3":"0.1217731798","4":"0.21026169","5":"0","6":"0","7":"8.0145044","8":"16.87530164","9":"PRICKLE1"},{"1":"positive regulation of defense response (GO:0031349)","2":"1/75","3":"0.1233157820","4":"0.21151515","5":"0","6":"0","7":"7.9058029","8":"16.54689979","9":"PDCD4"},{"1":"negative regulation of inflammatory response (GO:0050728)","2":"1/79","3":"0.1294599114","4":"0.21915109","5":"0","6":"0","7":"7.4988688","8":"15.33056742","9":"PDCD4"},{"1":"negative regulation of cell communication (GO:0010648)","2":"1/79","3":"0.1294599114","4":"0.21915109","5":"0","6":"0","7":"7.4988688","8":"15.33056742","9":"EPHB1"},{"1":"negative regulation of MAPK cascade (GO:0043409)","2":"1/81","3":"0.1325162695","4":"0.22143041","5":"0","6":"0","7":"7.3106618","8":"14.77521188","9":"CNKSR3"},{"1":"regulation of phosphatidylinositol 3-kinase signaling (GO:0014066)","2":"1/81","3":"0.1325162695","4":"0.22143041","5":"0","6":"0","7":"7.3106618","8":"14.77521188","9":"SEMA4D"},{"1":"BMP signaling pathway (GO:0030509)","2":"1/82","3":"0.1340405370","4":"0.22190031","5":"0","6":"0","7":"7.2200436","8":"14.50949350","9":"PDCD4"},{"1":"modulation of chemical synaptic transmission (GO:0050804)","2":"1/83","3":"0.1355622026","4":"0.22190031","5":"0","6":"0","7":"7.1316356","8":"14.25132342","9":"TMEM108"},{"1":"ephrin receptor signaling pathway (GO:0048013)","2":"1/83","3":"0.1355622026","4":"0.22190031","5":"0","6":"0","7":"7.1316356","8":"14.25132342","9":"EPHB1"},{"1":"negative regulation of signaling (GO:0023057)","2":"1/84","3":"0.1370812706","4":"0.22190031","5":"0","6":"0","7":"7.0453579","8":"14.00040357","9":"EPHB1"},{"1":"positive regulation of protein ubiquitination (GO:0031398)","2":"1/84","3":"0.1370812706","4":"0.22190031","5":"0","6":"0","7":"7.0453579","8":"14.00040357","9":"PRICKLE1"},{"1":"cellular response to molecule of bacterial origin (GO:0071219)","2":"1/85","3":"0.1385977453","4":"0.22296159","5":"0","6":"0","7":"6.9611345","8":"13.75645093","9":"PDCD4"},{"1":"regulation of peptidyl-tyrosine phosphorylation (GO:0050730)","2":"1/86","3":"0.1401116310","4":"0.22400563","5":"0","6":"0","7":"6.8788927","8":"13.51919664","9":"SEMA4D"},{"1":"regulation of peptidyl-serine phosphorylation (GO:0033135)","2":"1/87","3":"0.1416229320","4":"0.22431850","5":"0","6":"0","7":"6.7985636","8":"13.28838515","9":"CNKSR3"},{"1":"cellular response to BMP stimulus (GO:0071773)","2":"1/88","3":"0.1431316525","4":"0.22431850","5":"0","6":"0","7":"6.7200811","8":"13.06377338","9":"PDCD4"},{"1":"positive regulation of cellular component biogenesis (GO:0044089)","2":"1/88","3":"0.1431316525","4":"0.22431850","5":"0","6":"0","7":"6.7200811","8":"13.06377338","9":"EPHB1"},{"1":"regulation of proteasomal ubiquitin-dependent protein catabolic process (GO:0032434)","2":"1/89","3":"0.1446377969","4":"0.22431850","5":"0","6":"0","7":"6.6433824","8":"12.84513001","9":"PRICKLE1"},{"1":"Wnt signaling pathway, planar cell polarity pathway (GO:0060071)","2":"1/89","3":"0.1446377969","4":"0.22431850","5":"0","6":"0","7":"6.6433824","8":"12.84513001","9":"PRICKLE1"},{"1":"regulation of establishment of planar polarity (GO:0090175)","2":"1/90","3":"0.1461413695","4":"0.22493750","5":"0","6":"0","7":"6.5684071","8":"12.63223477","9":"PRICKLE1"},{"1":"regulation of JNK cascade (GO:0046328)","2":"1/91","3":"0.1476423744","4":"0.22493750","5":"0","6":"0","7":"6.4950980","8":"12.42487780","9":"EPHB1"},{"1":"positive regulation of response to external stimulus (GO:0032103)","2":"1/91","3":"0.1476423744","4":"0.22493750","5":"0","6":"0","7":"6.4950980","8":"12.42487780","9":"PDCD4"},{"1":"cellular response to lipopolysaccharide (GO:0071222)","2":"1/92","3":"0.1491408160","4":"0.22589165","5":"0","6":"0","7":"6.4234001","8":"12.22285908","9":"PDCD4"},{"1":"transmembrane receptor protein tyrosine kinase signaling pathway (GO:0007169)","2":"2/397","3":"0.1527988050","4":"0.22998721","5":"0","6":"0","7":"3.0026851","8":"5.64094395","9":"PIK3R3;EPHB1"},{"1":"axon development (GO:0061564)","2":"1/95","3":"0.1536208029","4":"0.22998721","5":"0","6":"0","7":"6.2174593","8":"11.64696779","9":"MAP1B"},{"1":"regulation of cell death (GO:0010941)","2":"1/97","3":"0.1565947218","4":"0.23043355","5":"0","6":"0","7":"6.0873162","8":"11.28645762","9":"EPHB1"},{"1":"protein localization to nucleus (GO:0034504)","2":"1/97","3":"0.1565947218","4":"0.23043355","5":"0","6":"0","7":"6.0873162","8":"11.28645762","9":"PRICKLE1"},{"1":"negative regulation of cell motility (GO:2000146)","2":"1/98","3":"0.1580778721","4":"0.23043355","5":"0","6":"0","7":"6.0242571","8":"11.11275137","9":"TMEFF2"},{"1":"cell morphogenesis involved in neuron differentiation (GO:0048667)","2":"1/98","3":"0.1580778721","4":"0.23043355","5":"0","6":"0","7":"6.0242571","8":"11.11275137","9":"MAP1B"},{"1":"phosphatidylinositol biosynthetic process (GO:0006661)","2":"1/99","3":"0.1595584887","4":"0.23043355","5":"0","6":"0","7":"5.9624850","8":"10.94321537","9":"PIK3R3"},{"1":"positive regulation of cell growth (GO:0030307)","2":"1/99","3":"0.1595584887","4":"0.23043355","5":"0","6":"0","7":"5.9624850","8":"10.94321537","9":"SEMA4D"},{"1":"negative regulation of programmed cell death (GO:0043069)","2":"2/409","3":"0.1601530768","4":"0.23043355","5":"0","6":"0","7":"2.9123669","8":"5.33436460","9":"SEMA4D;EPHB1"},{"1":"small GTPase mediated signal transduction (GO:0007264)","2":"1/100","3":"0.1610365758","4":"0.23043355","5":"0","6":"0","7":"5.9019608","8":"10.77771083","9":"CNKSR3"},{"1":"positive regulation of protein phosphorylation (GO:0001934)","2":"2/413","3":"0.1626198094","4":"0.23082696","5":"0","6":"0","7":"2.8834329","8":"5.23729521","9":"SEMA4D;EPHB1"},{"1":"regulation of protein ubiquitination (GO:0031396)","2":"1/102","3":"0.1639851779","4":"0.23082696","5":"0","6":"0","7":"5.7845079","8":"10.45827009","9":"PRICKLE1"},{"1":"regulation of kinase activity (GO:0043549)","2":"1/102","3":"0.1639851779","4":"0.23082696","5":"0","6":"0","7":"5.7845079","8":"10.45827009","9":"EPHB1"},{"1":"protein import into nucleus (GO:0006606)","2":"1/104","3":"0.1669237121","4":"0.23369320","5":"0","6":"0","7":"5.6716162","8":"10.15343163","9":"PRICKLE1"},{"1":"negative regulation of transcription, DNA-templated (GO:0045892)","2":"3/814","3":"0.1693241168","4":"0.23577928","5":"0","6":"0","7":"2.2141646","8":"3.93222472","9":"SEMA4D;PDCD4;PRICKLE1"},{"1":"positive regulation of cellular component organization (GO:0051130)","2":"1/107","3":"0.1713127088","4":"0.23727268","5":"0","6":"0","7":"5.5102664","8":"9.72156837","9":"EPHB1"},{"1":"regulation of Wnt signaling pathway (GO:0030111)","2":"1/110","3":"0.1756792397","4":"0.24019169","5":"0","6":"0","7":"5.3577982","8":"9.31772240","9":"TCF7L1"},{"1":"Wnt signaling pathway (GO:0016055)","2":"1/110","3":"0.1756792397","4":"0.24019169","5":"0","6":"0","7":"5.3577982","8":"9.31772240","9":"TCF7L1"},{"1":"negative regulation of cytokine production (GO:0001818)","2":"1/111","3":"0.1771297769","4":"0.24019169","5":"0","6":"0","7":"5.3088235","8":"9.18889725","9":"PDCD4"},{"1":"cellular response to insulin stimulus (GO:0032869)","2":"1/111","3":"0.1771297769","4":"0.24019169","5":"0","6":"0","7":"5.3088235","8":"9.18889725","9":"PIK3R3"},{"1":"phosphatidylinositol metabolic process (GO:0046488)","2":"1/113","3":"0.1800234165","4":"0.24284409","5":"0","6":"0","7":"5.2134979","8":"8.93941981","9":"PIK3R3"},{"1":"positive regulation of kinase activity (GO:0033674)","2":"1/114","3":"0.1814665270","4":"0.24352244","5":"0","6":"0","7":"5.1671005","8":"8.81860804","9":"EPHB1"},{"1":"negative regulation of nucleic acid-templated transcription (GO:1903507)","2":"2/445","3":"0.1825986452","4":"0.24377860","5":"0","6":"0","7":"2.6707709","8":"4.54155174","9":"PDCD4;PRICKLE1"},{"1":"sphingolipid metabolic process (GO:0006665)","2":"1/117","3":"0.1857810709","4":"0.24549642","5":"0","6":"0","7":"5.0327079","8":"8.47098519","9":"CERK"},{"1":"positive regulation of peptidyl-tyrosine phosphorylation (GO:0050731)","2":"1/117","3":"0.1857810709","4":"0.24549642","5":"0","6":"0","7":"5.0327079","8":"8.47098519","9":"SEMA4D"},{"1":"plasma membrane bounded cell projection organization (GO:0120036)","2":"1/119","3":"0.1886451513","4":"0.24801571","5":"0","6":"0","7":"4.9469093","8":"8.25088831","9":"TMEM108"},{"1":"macromolecular complex assembly (GO:0065003)","2":"1/121","3":"0.1914994439","4":"0.25049675","5":"0","6":"0","7":"4.8639706","8":"8.03951289","9":"CNKSR3"},{"1":"negative regulation of cell migration (GO:0030336)","2":"1/122","3":"0.1929229300","4":"0.25109065","5":"0","6":"0","7":"4.8235294","8":"7.93694639","9":"TMEFF2"},{"1":"non-canonical Wnt signaling pathway (GO:0035567)","2":"1/128","3":"0.2014128573","4":"0.25953199","5":"0","6":"0","7":"4.5942566","8":"7.36182971","9":"PRICKLE1"},{"1":"negative regulation of response to stimulus (GO:0048585)","2":"1/128","3":"0.2014128573","4":"0.25953199","5":"0","6":"0","7":"4.5942566","8":"7.36182971","9":"EPHB1"},{"1":"generation of neurons (GO:0048699)","2":"1/131","3":"0.2056252232","4":"0.26364818","5":"0","6":"0","7":"4.4875566","8":"7.09796853","9":"TCF7L1"},{"1":"negative regulation of apoptotic process (GO:0043066)","2":"2/486","3":"0.2087135117","4":"0.26628965","5":"0","6":"0","7":"2.4393939","8":"3.82202468","9":"SEMA4D;EPHB1"},{"1":"positive regulation of transferase activity (GO:0051347)","2":"1/138","3":"0.2153703106","4":"0.27343584","5":"0","6":"0","7":"4.2567626","8":"6.53581773","9":"EPHB1"},{"1":"regulation of cellular component organization (GO:0051128)","2":"1/141","3":"0.2195111344","4":"0.27733358","5":"0","6":"0","7":"4.1649160","8":"6.31548000","9":"SEMA4D"},{"1":"cell-cell adhesion via plasma-membrane adhesion molecules (GO:0098742)","2":"1/143","3":"0.2222598816","4":"0.27944325","5":"0","6":"0","7":"4.1058409","8":"6.17480679","9":"CLDN11"},{"1":"neuron development (GO:0048666)","2":"1/145","3":"0.2249992239","4":"0.28016730","5":"0","6":"0","7":"4.0484069","8":"6.03883980","9":"TMEM108"},{"1":"transmembrane receptor protein serine/threonine kinase signaling pathway (GO:0007178)","2":"1/145","3":"0.2249992239","4":"0.28016730","5":"0","6":"0","7":"4.0484069","8":"6.03883980","9":"PDCD4"},{"1":"negative regulation of cellular macromolecule biosynthetic process (GO:2000113)","2":"2/513","3":"0.2261400863","4":"0.28024059","5":"0","6":"0","7":"2.3073000","8":"3.43003362","9":"PDCD4;PRICKLE1"},{"1":"negative regulation of canonical Wnt signaling pathway (GO:0090090)","2":"1/149","3":"0.2304498183","4":"0.28422144","5":"0","6":"0","7":"3.9381955","8":"5.78017683","9":"PRICKLE1"},{"1":"response to lipopolysaccharide (GO:0032496)","2":"1/156","3":"0.2398988832","4":"0.29373470","5":"0","6":"0","7":"3.7590133","8":"5.36613342","9":"PDCD4"},{"1":"negative regulation of cellular process (GO:0048523)","2":"2/535","3":"0.2404314933","4":"0.29373470","5":"0","6":"0","7":"2.2095628","8":"3.14933422","9":"SEMA4D;PDCD4"},{"1":"axon guidance (GO:0007411)","2":"1/159","3":"0.2439138915","4":"0.29659013","5":"0","6":"0","7":"3.6870812","8":"5.20225037","9":"EPHB1"},{"1":"positive regulation of protein modification process (GO:0031401)","2":"1/164","3":"0.2505598334","4":"0.30324765","5":"0","6":"0","7":"3.5730783","8":"4.94534595","9":"SEMA4D"},{"1":"glycerophospholipid biosynthetic process (GO:0046474)","2":"1/165","3":"0.2518821951","4":"0.30343018","5":"0","6":"0","7":"3.5511119","8":"4.89625101","9":"PIK3R3"},{"1":"regulation of inflammatory response (GO:0050727)","2":"1/167","3":"0.2545201223","4":"0.30518848","5":"0","6":"0","7":"3.5079731","8":"4.80022398","9":"PDCD4"},{"1":"positive regulation of hydrolase activity (GO:0051345)","2":"1/173","3":"0.2623798034","4":"0.31316299","5":"0","6":"0","7":"3.3845759","8":"4.52843463","9":"SEMA4D"},{"1":"negative regulation of Wnt signaling pathway (GO:0030178)","2":"1/175","3":"0.2649817636","4":"0.31481778","5":"0","6":"0","7":"3.3453347","8":"4.44291983","9":"PRICKLE1"},{"1":"protein autophosphorylation (GO:0046777)","2":"1/176","3":"0.2662793979","4":"0.31491490","5":"0","6":"0","7":"3.3260504","8":"4.40106036","9":"EPHB1"},{"1":"cellular response to lipid (GO:0071396)","2":"1/179","3":"0.2701589622","4":"0.31805078","5":"0","6":"0","7":"3.2694977","8":"4.27893791","9":"PDCD4"},{"1":"positive regulation of cell motility (GO:2000147)","2":"1/180","3":"0.2714477165","4":"0.31812198","5":"0","6":"0","7":"3.2510680","8":"4.23934631","9":"SEMA4D"},{"1":"receptor-mediated endocytosis (GO:0006898)","2":"1/189","3":"0.2829474604","4":"0.32862275","5":"0","6":"0","7":"3.0940238","8":"3.90618662","9":"TMEM108"},{"1":"regulation of GTPase activity (GO:0043087)","2":"1/189","3":"0.2829474604","4":"0.32862275","5":"0","6":"0","7":"3.0940238","8":"3.90618662","9":"SEMA4D"},{"1":"protein complex assembly (GO:0006461)","2":"1/190","3":"0.2842142717","4":"0.32862275","5":"0","6":"0","7":"3.0774977","8":"3.87157469","9":"TCF7L1"},{"1":"positive regulation of GTPase activity (GO:0043547)","2":"1/193","3":"0.2880016751","4":"0.33005502","5":"0","6":"0","7":"3.0289522","8":"3.77040634","9":"SEMA4D"},{"1":"cilium organization (GO:0044782)","2":"1/193","3":"0.2880016751","4":"0.33005502","5":"0","6":"0","7":"3.0289522","8":"3.77040634","9":"BBIP1"},{"1":"negative regulation of gene expression (GO:0010629)","2":"2/619","3":"0.2953051221","4":"0.33693404","5":"0","6":"0","7":"1.9004960","8":"2.31812273","9":"PDCD4;PRICKLE1"},{"1":"positive regulation of ERK1 and ERK2 cascade (GO:0070374)","2":"1/202","3":"0.2992474502","4":"0.33993460","5":"0","6":"0","7":"2.8920105","8":"3.48916576","9":"EPHB1"},{"1":"regulation of MAPK cascade (GO:0043408)","2":"1/204","3":"0.3017229771","4":"0.34125000","5":"0","6":"0","7":"2.8632280","8":"3.43085149","9":"EPHB1"},{"1":"positive regulation of signal transduction (GO:0009967)","2":"1/207","3":"0.3054203458","4":"0.34392987","5":"0","6":"0","7":"2.8211022","8":"3.34601419","9":"TMEM108"},{"1":"regulation of canonical Wnt signaling pathway (GO:0060828)","2":"1/214","3":"0.3139737327","4":"0.35203115","5":"0","6":"0","7":"2.7274234","8":"3.15957255","9":"PRICKLE1"},{"1":"regulation of cell cycle (GO:0051726)","2":"1/216","3":"0.3163987049","4":"0.35322097","5":"0","6":"0","7":"2.7017784","8":"3.10907725","9":"PDCD4"},{"1":"central nervous system development (GO:0007417)","2":"1/218","3":"0.3188153494","4":"0.35439131","5":"0","6":"0","7":"2.6766061","8":"3.05974405","9":"MBP"},{"1":"positive regulation of cell migration (GO:0030335)","2":"1/222","3":"0.3236237661","4":"0.35819895","5":"0","6":"0","7":"2.6276284","8":"2.96442116","9":"SEMA4D"},{"1":"Ras protein signal transduction (GO:0007265)","2":"1/224","3":"0.3260155935","4":"0.35931080","5":"0","6":"0","7":"2.6037985","8":"2.91836354","9":"CNKSR3"},{"1":"apoptotic process (GO:0006915)","2":"1/232","3":"0.3355010042","4":"0.36819814","5":"0","6":"0","7":"2.5126050","8":"2.74409217","9":"PDCD4"},{"1":"anterograde trans-synaptic signaling (GO:0098916)","2":"1/241","3":"0.3460171063","4":"0.37813684","5":"0","6":"0","7":"2.4172794","8":"2.56537903","9":"MBP"},{"1":"regulation of protein phosphorylation (GO:0001932)","2":"1/262","3":"0.3699298502","4":"0.40257072","5":"0","6":"0","7":"2.2204192","8":"2.20807786","9":"SEMA4D"},{"1":"cellular response to oxygen-containing compound (GO:1901701)","2":"1/275","3":"0.3843049844","4":"0.41646440","5":"0","6":"0","7":"2.1136754","8":"2.02134754","9":"PDCD4"},{"1":"MAPK cascade (GO:0000165)","2":"1/279","3":"0.3886636419","4":"0.41943285","5":"0","6":"0","7":"2.0828396","8":"1.96836879","9":"MBP"},{"1":"negative regulation of signal transduction (GO:0009968)","2":"1/284","3":"0.3940698271","4":"0.42350243","5":"0","6":"0","7":"2.0455207","8":"1.90484441","9":"EPHB1"},{"1":"cilium assembly (GO:0060271)","2":"1/289","3":"0.3994295583","4":"0.42511662","5":"0","6":"0","7":"2.0094975","8":"1.84415178","9":"BBIP1"},{"1":"chemical synaptic transmission (GO:0007268)","2":"1/290","3":"0.4004959642","4":"0.42511662","5":"0","6":"0","7":"2.0024425","8":"1.83233819","9":"MBP"},{"1":"positive regulation of MAPK cascade (GO:0043410)","2":"1/290","3":"0.4004959642","4":"0.42511662","5":"0","6":"0","7":"2.0024425","8":"1.83233819","9":"EPHB1"},{"1":"regulation of apoptotic process (GO:0042981)","2":"2/816","3":"0.4209554123","4":"0.44501001","5":"0","6":"0","7":"1.4258804","8":"1.23371218","9":"SEMA4D;EPHB1"},{"1":"intracellular protein transport (GO:0006886)","2":"1/348","3":"0.4592979058","4":"0.48356975","5":"0","6":"0","7":"1.6628242","8":"1.29377076","9":"PRICKLE1"},{"1":"regulation of transcription from RNA polymerase II promoter (GO:0006357)","2":"3/1479","3":"0.4844141308","4":"0.50794842","5":"0","6":"0","7":"1.1743521","8":"0.85118815","9":"TCF7L1;SEMA4D;PDCD4"},{"1":"organelle assembly (GO:0070925)","2":"1/377","3":"0.4865580627","4":"0.50813927","5":"0","6":"0","7":"1.5323060","8":"1.10387177","9":"BBIP1"},{"1":"vesicle-mediated transport (GO:0016192)","2":"1/411","3":"0.5168202773","4":"0.53757611","5":"0","6":"0","7":"1.4027977","8":"0.92593078","9":"DPYSL2"},{"1":"regulation of transcription, DNA-templated (GO:0006355)","2":"3/1599","3":"0.5382313927","4":"0.55760772","5":"0","6":"0","7":"1.0790061","8":"0.66840837","9":"TCF7L1;PDCD4;PRICKLE1"},{"1":"protein phosphorylation (GO:0006468)","2":"1/471","3":"0.5660505760","4":"0.58409203","5":"0","6":"0","7":"1.2199625","8":"0.69424629","9":"EPHB1"},{"1":"neutrophil degranulation (GO:0043312)","2":"1/480","3":"0.5730015437","4":"0.58891825","5":"0","6":"0","7":"1.1964878","8":"0.66628440","9":"NFASC"},{"1":"neutrophil activation involved in immune response (GO:0002283)","2":"1/484","3":"0.5760560312","4":"0.58971744","5":"0","6":"0","7":"1.1863354","8":"0.65432370","9":"NFASC"},{"1":"neutrophil mediated immunity (GO:0002446)","2":"1/488","3":"0.5790892885","4":"0.59048868","5":"0","6":"0","7":"1.1763498","8":"0.64263825","9":"NFASC"},{"1":"negative regulation of transcription from RNA polymerase II promoter (GO:0000122)","2":"1/566","3":"0.6341935630","4":"0.64414170","5":"0","6":"0","7":"1.0098907","8":"0.45990529","9":"SEMA4D"},{"1":"regulation of nucleic acid-templated transcription (GO:1903506)","2":"1/608","3":"0.6608929845","4":"0.66863782","5":"0","6":"0","7":"0.9379785","8":"0.38847631","9":"TCF7L1"},{"1":"regulation of cellular macromolecule biosynthetic process (GO:2000112)","2":"1/632","3":"0.6752894452","4":"0.67908393","5":"0","6":"0","7":"0.9011839","8":"0.35381731","9":"TCF7L1"},{"1":"cytokine-mediated signaling pathway (GO:0019221)","2":"1/634","3":"0.6764619821","4":"0.67908393","5":"0","6":"0","7":"0.8982437","8":"0.35110461","9":"PDCD4"},{"1":"regulation of gene expression (GO:0010468)","2":"1/1038","3":"0.8454067928","4":"0.84540679","5":"0","6":"0","7":"0.5368427","8":"0.09015594","9":"TCF7L1"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
<div id="kegg_2016" class="section level4 unnumbered">
<h4 class="unnumbered">KEGG_2016</h4>
<pre class="r"><code>enrichr_type[[1]]$KEGG_2016</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["Term"],"name":[1],"type":["chr"],"align":["left"]},{"label":["Overlap"],"name":[2],"type":["chr"],"align":["left"]},{"label":["P.value"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["Adjusted.P.value"],"name":[4],"type":["dbl"],"align":["right"]},{"label":["Old.P.value"],"name":[5],"type":["int"],"align":["right"]},{"label":["Old.Adjusted.P.value"],"name":[6],"type":["int"],"align":["right"]},{"label":["Odds.Ratio"],"name":[7],"type":["dbl"],"align":["right"]},{"label":["Combined.Score"],"name":[8],"type":["dbl"],"align":["right"]},{"label":["Genes"],"name":[9],"type":["chr"],"align":["left"]}],"data":[{"1":"Axon guidance Homo sapiens hsa04360","2":"4/127","3":"6.970237e-05","4":"0.006064106","5":"0","6":"0","7":"20.815106","8":"199.227133","9":"ABLIM2;SEMA4D;DPYSL2;EPHB1"},{"1":"Cell adhesion molecules (CAMs) Homo sapiens hsa04514","2":"3/142","3":"1.941894e-03","4":"0.084472383","5":"0","6":"0","7":"13.371853","8":"83.495071","9":"CLDN11;NFASC;MAG"},{"1":"Endometrial cancer Homo sapiens hsa05213","2":"2/52","3":"3.734245e-03","4":"0.091653053","5":"0","6":"0","7":"24.139394","8":"134.944272","9":"TCF7L1;PIK3R3"},{"1":"Acute myeloid leukemia Homo sapiens hsa05221","2":"2/57","3":"4.470077e-03","4":"0.091653053","5":"0","6":"0","7":"21.939394","8":"118.699792","9":"TCF7L1;PIK3R3"},{"1":"Colorectal cancer Homo sapiens hsa05210","2":"2/62","3":"5.267417e-03","4":"0.091653053","5":"0","6":"0","7":"20.106061","8":"105.480721","9":"TCF7L1;PIK3R3"},{"1":"Prostate cancer Homo sapiens hsa05215","2":"2/89","3":"1.059106e-02","4":"0.153570388","5":"0","6":"0","7":"13.847440","8":"62.974624","9":"TCF7L1;PIK3R3"},{"1":"Leukocyte transendothelial migration Homo sapiens hsa04670","2":"2/118","3":"1.808908e-02","4":"0.224821471","5":"0","6":"0","7":"10.370428","8":"41.610791","9":"CLDN11;PIK3R3"},{"1":"Hepatitis C Homo sapiens hsa05160","2":"2/133","3":"2.263056e-02","4":"0.246107340","5":"0","6":"0","7":"9.176035","8":"34.762988","9":"CLDN11;PIK3R3"},{"1":"Wnt signaling pathway Homo sapiens hsa04310","2":"2/142","3":"2.555850e-02","4":"0.247065460","5":"0","6":"0","7":"8.582251","8":"31.469274","9":"TCF7L1;PRICKLE1"},{"1":"Focal adhesion Homo sapiens hsa04510","2":"2/202","3":"4.857388e-02","4":"0.280716738","5":"0","6":"0","7":"5.989394","8":"18.115936","9":"PIK3R3;PARVB"},{"1":"Proteoglycans in cancer Homo sapiens hsa05205","2":"2/203","3":"4.900437e-02","4":"0.280716738","5":"0","6":"0","7":"5.959294","8":"17.972313","9":"PDCD4;PIK3R3"},{"1":"Thyroid cancer Homo sapiens hsa05216","2":"1/29","3":"4.955961e-02","4":"0.280716738","5":"0","6":"0","7":"20.942227","8":"62.922579","9":"TCF7L1"},{"1":"Aldosterone-regulated sodium reabsorption Homo sapiens hsa04960","2":"1/39","3":"6.608937e-02","4":"0.280716738","5":"0","6":"0","7":"15.423375","8":"41.901412","9":"PIK3R3"},{"1":"Carbohydrate digestion and absorption Homo sapiens hsa04973","2":"1/45","3":"7.587283e-02","4":"0.280716738","5":"0","6":"0","7":"13.316176","8":"34.338380","9":"PIK3R3"},{"1":"Sphingolipid metabolism Homo sapiens hsa00600","2":"1/47","3":"7.911180e-02","4":"0.280716738","5":"0","6":"0","7":"12.735934","8":"32.309704","9":"CERK"},{"1":"Type II diabetes mellitus Homo sapiens hsa04930","2":"1/48","3":"8.072714e-02","4":"0.280716738","5":"0","6":"0","7":"12.464330","8":"31.368736","9":"PIK3R3"},{"1":"Basal cell carcinoma Homo sapiens hsa05217","2":"1/55","3":"9.195776e-02","4":"0.280716738","5":"0","6":"0","7":"10.844771","8":"25.880244","9":"TCF7L1"},{"1":"Regulation of lipolysis in adipocytes Homo sapiens hsa04923","2":"1/56","3":"9.355121e-02","4":"0.280716738","5":"0","6":"0","7":"10.647059","8":"25.225505","9":"PIK3R3"},{"1":"Non-small cell lung cancer Homo sapiens hsa05223","2":"1/56","3":"9.355121e-02","4":"0.280716738","5":"0","6":"0","7":"10.647059","8":"25.225505","9":"PIK3R3"},{"1":"mTOR signaling pathway Homo sapiens hsa04150","2":"1/60","3":"9.989790e-02","4":"0.280716738","5":"0","6":"0","7":"9.923230","8":"22.859219","9":"PIK3R3"},{"1":"VEGF signaling pathway Homo sapiens hsa04370","2":"1/61","3":"1.014778e-01","4":"0.280716738","5":"0","6":"0","7":"9.757353","8":"22.323995","9":"PIK3R3"},{"1":"Longevity regulating pathway - multiple species Homo sapiens hsa04213","2":"1/64","3":"1.062014e-01","4":"0.280716738","5":"0","6":"0","7":"9.291317","8":"20.835014","9":"PIK3R3"},{"1":"Glioma Homo sapiens hsa05214","2":"1/65","3":"1.077706e-01","4":"0.280716738","5":"0","6":"0","7":"9.145680","8":"20.374294","9":"PIK3R3"},{"1":"Renal cell carcinoma Homo sapiens hsa05211","2":"1/66","3":"1.093371e-01","4":"0.280716738","5":"0","6":"0","7":"9.004525","8":"19.929893","9":"PIK3R3"},{"1":"Pancreatic cancer Homo sapiens hsa05212","2":"1/66","3":"1.093371e-01","4":"0.280716738","5":"0","6":"0","7":"9.004525","8":"19.929893","9":"PIK3R3"},{"1":"Central carbon metabolism in cancer Homo sapiens hsa05230","2":"1/67","3":"1.109009e-01","4":"0.280716738","5":"0","6":"0","7":"8.867647","8":"19.501005","9":"PIK3R3"},{"1":"Fc epsilon RI signaling pathway Homo sapiens hsa04664","2":"1/68","3":"1.124620e-01","4":"0.280716738","5":"0","6":"0","7":"8.734855","8":"19.086877","9":"PIK3R3"},{"1":"Melanoma Homo sapiens hsa05218","2":"1/71","3":"1.171295e-01","4":"0.280716738","5":"0","6":"0","7":"8.359244","8":"17.926188","9":"PIK3R3"},{"1":"Prolactin signaling pathway Homo sapiens hsa04917","2":"1/72","3":"1.186801e-01","4":"0.280716738","5":"0","6":"0","7":"8.241094","8":"17.564440","9":"PIK3R3"},{"1":"Chronic myeloid leukemia Homo sapiens hsa05220","2":"1/73","3":"1.202279e-01","4":"0.280716738","5":"0","6":"0","7":"8.126225","8":"17.214318","9":"PIK3R3"},{"1":"B cell receptor signaling pathway Homo sapiens hsa04662","2":"1/73","3":"1.202279e-01","4":"0.280716738","5":"0","6":"0","7":"8.126225","8":"17.214318","9":"PIK3R3"},{"1":"Adherens junction Homo sapiens hsa04520","2":"1/74","3":"1.217732e-01","4":"0.280716738","5":"0","6":"0","7":"8.014504","8":"16.875302","9":"TCF7L1"},{"1":"Arrhythmogenic right ventricular cardiomyopathy (ARVC) Homo sapiens hsa05412","2":"1/74","3":"1.217732e-01","4":"0.280716738","5":"0","6":"0","7":"8.014504","8":"16.875302","9":"TCF7L1"},{"1":"Bacterial invasion of epithelial cells Homo sapiens hsa05100","2":"1/78","3":"1.279278e-01","4":"0.280716738","5":"0","6":"0","7":"7.596639","8":"15.620886","9":"PIK3R3"},{"1":"Small cell lung cancer Homo sapiens hsa05222","2":"1/86","3":"1.401116e-01","4":"0.280716738","5":"0","6":"0","7":"6.878893","8":"13.519197","9":"PIK3R3"},{"1":"ErbB signaling pathway Homo sapiens hsa04012","2":"1/87","3":"1.416229e-01","4":"0.280716738","5":"0","6":"0","7":"6.798564","8":"13.288385","9":"PIK3R3"},{"1":"Fc gamma R-mediated phagocytosis Homo sapiens hsa04666","2":"1/93","3":"1.506367e-01","4":"0.280716738","5":"0","6":"0","7":"6.353261","8":"12.025988","9":"PIK3R3"},{"1":"Longevity regulating pathway - mammal Homo sapiens hsa04211","2":"1/94","3":"1.521300e-01","4":"0.280716738","5":"0","6":"0","7":"6.284630","8":"11.834082","9":"PIK3R3"},{"1":"Pathways in cancer Homo sapiens hsa05200","2":"2/397","3":"1.527988e-01","4":"0.280716738","5":"0","6":"0","7":"3.002685","8":"5.640944","9":"TCF7L1;PIK3R3"},{"1":"Progesterone-mediated oocyte maturation Homo sapiens hsa04914","2":"1/98","3":"1.580779e-01","4":"0.280716738","5":"0","6":"0","7":"6.024257","8":"11.112751","9":"PIK3R3"},{"1":"Phosphatidylinositol signaling system Homo sapiens hsa04070","2":"1/98","3":"1.580779e-01","4":"0.280716738","5":"0","6":"0","7":"6.024257","8":"11.112751","9":"PIK3R3"},{"1":"Inflammatory mediator regulation of TRP channels Homo sapiens hsa04750","2":"1/98","3":"1.580779e-01","4":"0.280716738","5":"0","6":"0","7":"6.024257","8":"11.112751","9":"PIK3R3"},{"1":"Estrogen signaling pathway Homo sapiens hsa04915","2":"1/99","3":"1.595585e-01","4":"0.280716738","5":"0","6":"0","7":"5.962485","8":"10.943215","9":"PIK3R3"},{"1":"Melanogenesis Homo sapiens hsa04916","2":"1/100","3":"1.610366e-01","4":"0.280716738","5":"0","6":"0","7":"5.901961","8":"10.777711","9":"TCF7L1"},{"1":"Amoebiasis Homo sapiens hsa05146","2":"1/100","3":"1.610366e-01","4":"0.280716738","5":"0","6":"0","7":"5.901961","8":"10.777711","9":"PIK3R3"},{"1":"AGE-RAGE signaling pathway in diabetic complications Homo sapiens hsa04933","2":"1/101","3":"1.625121e-01","4":"0.280716738","5":"0","6":"0","7":"5.842647","8":"10.616105","9":"PIK3R3"},{"1":"Choline metabolism in cancer Homo sapiens hsa05231","2":"1/101","3":"1.625121e-01","4":"0.280716738","5":"0","6":"0","7":"5.842647","8":"10.616105","9":"PIK3R3"},{"1":"HIF-1 signaling pathway Homo sapiens hsa04066","2":"1/103","3":"1.654557e-01","4":"0.280716738","5":"0","6":"0","7":"5.727509","8":"10.304085","9":"PIK3R3"},{"1":"Chagas disease (American trypanosomiasis) Homo sapiens hsa05142","2":"1/104","3":"1.669237e-01","4":"0.280716738","5":"0","6":"0","7":"5.671616","8":"10.153432","9":"PIK3R3"},{"1":"T cell receptor signaling pathway Homo sapiens hsa04660","2":"1/104","3":"1.669237e-01","4":"0.280716738","5":"0","6":"0","7":"5.671616","8":"10.153432","9":"PIK3R3"},{"1":"Toll-like receptor signaling pathway Homo sapiens hsa04620","2":"1/106","3":"1.698522e-01","4":"0.280716738","5":"0","6":"0","7":"5.563025","8":"9.862279","9":"PIK3R3"},{"1":"Insulin resistance Homo sapiens hsa04931","2":"1/109","3":"1.742262e-01","4":"0.280716738","5":"0","6":"0","7":"5.407680","8":"9.449383","9":"PIK3R3"},{"1":"TNF signaling pathway Homo sapiens hsa04668","2":"1/110","3":"1.756792e-01","4":"0.280716738","5":"0","6":"0","7":"5.357798","8":"9.317722","9":"PIK3R3"},{"1":"Cholinergic synapse Homo sapiens hsa04725","2":"1/111","3":"1.771298e-01","4":"0.280716738","5":"0","6":"0","7":"5.308824","8":"9.188897","9":"PIK3R3"},{"1":"Toxoplasmosis Homo sapiens hsa05145","2":"1/118","3":"1.872143e-01","4":"0.280716738","5":"0","6":"0","7":"4.989442","8":"8.359816","9":"PIK3R3"},{"1":"Thyroid hormone signaling pathway Homo sapiens hsa04919","2":"1/118","3":"1.872143e-01","4":"0.280716738","5":"0","6":"0","7":"4.989442","8":"8.359816","9":"PIK3R3"},{"1":"Sphingolipid signaling pathway Homo sapiens hsa04071","2":"1/120","3":"1.900735e-01","4":"0.280716738","5":"0","6":"0","7":"4.905091","8":"8.144141","9":"PIK3R3"},{"1":"Neurotrophin signaling pathway Homo sapiens hsa04722","2":"1/120","3":"1.900735e-01","4":"0.280716738","5":"0","6":"0","7":"4.905091","8":"8.144141","9":"PIK3R3"},{"1":"Platelet activation Homo sapiens hsa04611","2":"1/122","3":"1.929229e-01","4":"0.280716738","5":"0","6":"0","7":"4.823529","8":"7.936946","9":"PIK3R3"},{"1":"AMPK signaling pathway Homo sapiens hsa04152","2":"1/124","3":"1.957626e-01","4":"0.280716738","5":"0","6":"0","7":"4.744620","8":"7.737775","9":"PIK3R3"},{"1":"Osteoclast differentiation Homo sapiens hsa04380","2":"1/132","3":"2.070245e-01","4":"0.280716738","5":"0","6":"0","7":"4.453076","8":"7.013229","9":"PIK3R3"},{"1":"FoxO signaling pathway Homo sapiens hsa04068","2":"1/133","3":"2.084215e-01","4":"0.280716738","5":"0","6":"0","7":"4.419118","8":"6.930029","9":"PIK3R3"},{"1":"Natural killer cell mediated cytotoxicity Homo sapiens hsa04650","2":"1/135","3":"2.112081e-01","4":"0.280716738","5":"0","6":"0","7":"4.352722","8":"6.768095","9":"PIK3R3"},{"1":"Measles Homo sapiens hsa05162","2":"1/136","3":"2.125979e-01","4":"0.280716738","5":"0","6":"0","7":"4.320261","8":"6.689288","9":"PIK3R3"},{"1":"Tight junction Homo sapiens hsa04530","2":"1/139","3":"2.167530e-01","4":"0.280716738","5":"0","6":"0","7":"4.225703","8":"6.461088","9":"CLDN11"},{"1":"Insulin signaling pathway Homo sapiens hsa04910","2":"1/139","3":"2.167530e-01","4":"0.280716738","5":"0","6":"0","7":"4.225703","8":"6.461088","9":"PIK3R3"},{"1":"Apoptosis Homo sapiens hsa04210","2":"1/140","3":"2.181332e-01","4":"0.280716738","5":"0","6":"0","7":"4.195091","8":"6.387652","9":"PIK3R3"},{"1":"Signaling pathways regulating pluripotency of stem cells Homo sapiens hsa04550","2":"1/142","3":"2.208867e-01","4":"0.280716738","5":"0","6":"0","7":"4.135169","8":"6.244541","9":"PIK3R3"},{"1":"Phospholipase D signaling pathway Homo sapiens hsa04072","2":"1/144","3":"2.236307e-01","4":"0.280716738","5":"0","6":"0","7":"4.076923","8":"6.106249","9":"PIK3R3"},{"1":"Hepatitis B Homo sapiens hsa05161","2":"1/146","3":"2.263654e-01","4":"0.280716738","5":"0","6":"0","7":"4.020284","8":"5.972553","9":"PIK3R3"},{"1":"Adrenergic signaling in cardiomyocytes Homo sapiens hsa04261","2":"1/148","3":"2.290907e-01","4":"0.280716738","5":"0","6":"0","7":"3.965186","8":"5.843247","9":"PIK3R3"},{"1":"Non-alcoholic fatty liver disease (NAFLD) Homo sapiens hsa04932","2":"1/151","3":"2.331611e-01","4":"0.281097198","5":"0","6":"0","7":"3.885294","8":"5.657087","9":"PIK3R3"},{"1":"Hippo signaling pathway Homo sapiens hsa04390","2":"1/153","3":"2.358632e-01","4":"0.281097198","5":"0","6":"0","7":"3.833785","8":"5.537915","9":"TCF7L1"},{"1":"Jak-STAT signaling pathway Homo sapiens hsa04630","2":"1/158","3":"2.425778e-01","4":"0.281390306","5":"0","6":"0","7":"3.710753","8":"5.256032","9":"PIK3R3"},{"1":"Oxytocin signaling pathway Homo sapiens hsa04921","2":"1/158","3":"2.425778e-01","4":"0.281390306","5":"0","6":"0","7":"3.710753","8":"5.256032","9":"PIK3R3"},{"1":"cGMP-PKG signaling pathway Homo sapiens hsa04022","2":"1/167","3":"2.545201e-01","4":"0.291358561","5":"0","6":"0","7":"3.507973","8":"4.800224","9":"PIK3R3"},{"1":"Influenza A Homo sapiens hsa05164","2":"1/175","3":"2.649818e-01","4":"0.299394980","5":"0","6":"0","7":"3.345335","8":"4.442920","9":"PIK3R3"},{"1":"Chemokine signaling pathway Homo sapiens hsa04062","2":"1/187","3":"2.804073e-01","4":"0.312761989","5":"0","6":"0","7":"3.127609","8":"3.976793","9":"PIK3R3"},{"1":"cAMP signaling pathway Homo sapiens hsa04024","2":"1/199","3":"2.955182e-01","4":"0.325398853","5":"0","6":"0","7":"2.936275","8":"3.579392","9":"PIK3R3"},{"1":"Epstein-Barr virus infection Homo sapiens hsa05169","2":"1/202","3":"2.992475e-01","4":"0.325398853","5":"0","6":"0","7":"2.892011","8":"3.489166","9":"PIK3R3"},{"1":"Viral carcinogenesis Homo sapiens hsa05203","2":"1/205","3":"3.029576e-01","4":"0.325398853","5":"0","6":"0","7":"2.849048","8":"3.402227","9":"PIK3R3"},{"1":"Rap1 signaling pathway Homo sapiens hsa04015","2":"1/211","3":"3.103206e-01","4":"0.329104997","5":"0","6":"0","7":"2.766807","8":"3.237577","9":"PIK3R3"},{"1":"Regulation of actin cytoskeleton Homo sapiens hsa04810","2":"1/214","3":"3.139737e-01","4":"0.329104997","5":"0","6":"0","7":"2.727423","8":"3.159573","9":"PIK3R3"},{"1":"Ras signaling pathway Homo sapiens hsa04014","2":"1/227","3":"3.295879e-01","4":"0.341358934","5":"0","6":"0","7":"2.568844","8":"2.851191","9":"PIK3R3"},{"1":"HTLV-I infection Homo sapiens hsa05166","2":"1/258","3":"3.654415e-01","4":"0.374040135","5":"0","6":"0","7":"2.255436","8":"2.270433","9":"PIK3R3"},{"1":"MicroRNAs in cancer Homo sapiens hsa05206","2":"1/297","3":"4.079095e-01","4":"0.412652603","5":"0","6":"0","7":"1.954392","8":"1.752523","9":"PDCD4"},{"1":"PI3K-Akt signaling pathway Homo sapiens hsa04151","2":"1/341","3":"4.525101e-01","4":"0.452510140","5":"0","6":"0","7":"1.697664","8":"1.346155","9":"PIK3R3"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
<div id="nfol.2-nfol.sgcd-3" class="section level3 tabset">
<h3 class="tabset">NFOL.2 | NFOL.Sgcd</h3>
<pre class="r"><code>src_list &lt;- lapply(names(enrichr_type[[2]]), function(set) {
    src &lt;- c(
        &quot;#### {{set}} {.unnumbered}&quot;,
        &quot;```{r enrichr-NFOL.2|NFOL.Sgcd-{{set}}, message = FALSE, warning = FALSE}&quot;,
        &quot;enrichr_type[[2]]${{set}}&quot;,  
        &quot;```&quot;,
        &quot;&quot;
    )
    knit_expand(text = src)
})
out &lt;- knit_child(text = unlist(src_list), options = list(cache = FALSE))</code></pre>
<div id="go_molecular_function_2018-1"
class="section level4 unnumbered">
<h4 class="unnumbered">GO_Molecular_Function_2018</h4>
<pre class="r"><code>enrichr_type[[2]]$GO_Molecular_Function_2018</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["Term"],"name":[1],"type":["chr"],"align":["left"]},{"label":["Overlap"],"name":[2],"type":["chr"],"align":["left"]},{"label":["P.value"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["Adjusted.P.value"],"name":[4],"type":["dbl"],"align":["right"]},{"label":["Old.P.value"],"name":[5],"type":["int"],"align":["right"]},{"label":["Old.Adjusted.P.value"],"name":[6],"type":["int"],"align":["right"]},{"label":["Odds.Ratio"],"name":[7],"type":["dbl"],"align":["right"]},{"label":["Combined.Score"],"name":[8],"type":["dbl"],"align":["right"]},{"label":["Genes"],"name":[9],"type":["chr"],"align":["left"]}],"data":[{"1":"mechanically-gated ion channel activity (GO:0008381)","2":"1/6","3":"0.008073627","4":"0.1171309","5":"0","6":"0","7":"153.6000000","8":"740.2218192","9":"FAM155A"},{"1":"acetylcholine receptor binding (GO:0033130)","2":"1/7","3":"0.009413121","4":"0.1171309","5":"0","6":"0","7":"127.9935897","8":"597.1733894","9":"NRXN1"},{"1":"transcription corepressor binding (GO:0001222)","2":"1/12","3":"0.016084511","4":"0.1171309","5":"0","6":"0","7":"69.7972028","8":"288.2553657","9":"RORA"},{"1":"lipid phosphatase activity (GO:0042577)","2":"1/12","3":"0.016084511","4":"0.1171309","5":"0","6":"0","7":"69.7972028","8":"288.2553657","9":"PLPPR1"},{"1":"LRR domain binding (GO:0030275)","2":"1/14","3":"0.018740936","4":"0.1171309","5":"0","6":"0","7":"59.0532544","8":"234.8574522","9":"ROBO1"},{"1":"outward rectifier potassium channel activity (GO:0015271)","2":"1/14","3":"0.018740936","4":"0.1171309","5":"0","6":"0","7":"59.0532544","8":"234.8574522","9":"KCND2"},{"1":"phosphatidate phosphatase activity (GO:0008195)","2":"1/14","3":"0.018740936","4":"0.1171309","5":"0","6":"0","7":"59.0532544","8":"234.8574522","9":"PLPPR1"},{"1":"transcription coactivator binding (GO:0001223)","2":"1/14","3":"0.018740936","4":"0.1171309","5":"0","6":"0","7":"59.0532544","8":"234.8574522","9":"RORA"},{"1":"oxidoreductase activity, acting on the CH-CH group of donors, NAD or NADP as acceptor (GO:0016628)","2":"1/17","3":"0.022712631","4":"0.1261813","5":"0","6":"0","7":"47.9735577","8":"181.5719550","9":"DPYD"},{"1":"RNA polymerase II transcription factor activity, sequence-specific transcription regulatory region DNA binding (GO:0001133)","2":"1/19","3":"0.025351824","4":"0.1267591","5":"0","6":"0","7":"42.6388889","8":"156.6938488","9":"RORA"},{"1":"voltage-gated ion channel activity (GO:0005244)","2":"1/21","3":"0.027984153","4":"0.1272007","5":"0","6":"0","7":"38.3711538","8":"137.2197311","9":"KCND2"},{"1":"voltage-gated ion channel activity involved in regulation of postsynaptic membrane potential (GO:1905030)","2":"1/23","3":"0.030609636","4":"0.1275402","5":"0","6":"0","7":"34.8793706","8":"121.6048472","9":"KCND2"},{"1":"transcription cofactor binding (GO:0001221)","2":"1/26","3":"0.034535061","4":"0.1280005","5":"0","6":"0","7":"30.6892308","8":"103.2932053","9":"RORA"},{"1":"calcium channel regulator activity (GO:0005246)","2":"1/27","3":"0.035840132","4":"0.1280005","5":"0","6":"0","7":"29.5073964","8":"98.2208874","9":"NRXN1"},{"1":"NADP binding (GO:0050661)","2":"1/31","3":"0.041043446","4":"0.1323126","5":"0","6":"0","7":"25.5679487","8":"81.6416334","9":"DPYD"},{"1":"ligand-gated channel activity (GO:0022834)","2":"1/32","3":"0.042340044","4":"0.1323126","5":"0","6":"0","7":"24.7419355","8":"78.2345437","9":"NCALD"},{"1":"tubulin binding (GO:0015631)","2":"2/256","3":"0.046439007","4":"0.1365853","5":"0","6":"0","7":"6.2107087","8":"19.0644875","9":"NAV3;NCALD"},{"1":"calcium ion binding (GO:0005509)","2":"2/285","3":"0.056229708","4":"0.1561936","5":"0","6":"0","7":"5.5660777","8":"16.0208975","9":"NRXN1;ASTN2"},{"1":"flavin adenine dinucleotide binding (GO:0050660)","2":"1/49","3":"0.064125621","4":"0.1603141","5":"0","6":"0","7":"15.9655449","8":"43.8559354","9":"DPYD"},{"1":"sterol binding (GO:0032934)","2":"1/49","3":"0.064125621","4":"0.1603141","5":"0","6":"0","7":"15.9655449","8":"43.8559354","9":"RORA"},{"1":"phosphoric ester hydrolase activity (GO:0042578)","2":"1/57","3":"0.074211767","4":"0.1766947","5":"0","6":"0","7":"13.6792582","8":"35.5774602","9":"PLPPR1"},{"1":"protein tyrosine phosphatase activity (GO:0004725)","2":"1/63","3":"0.081707599","4":"0.1856991","5":"0","6":"0","7":"12.3517370","8":"30.9362625","9":"PTPRT"},{"1":"ligand-dependent nuclear receptor transcription coactivator activity (GO:0030374)","2":"1/66","3":"0.085433565","4":"0.1857187","5":"0","6":"0","7":"11.7798817","8":"28.9787000","9":"RORA"},{"1":"ligand-gated ion channel activity (GO:0015276)","2":"1/69","3":"0.089144971","4":"0.1857187","5":"0","6":"0","7":"11.2584842","8":"27.2172881","9":"NCALD"},{"1":"core promoter sequence-specific DNA binding (GO:0001046)","2":"1/74","3":"0.095298440","4":"0.1905969","5":"0","6":"0","7":"10.4847208","8":"24.6468718","9":"RORA"},{"1":"voltage-gated potassium channel activity (GO:0005249)","2":"1/86","3":"0.109903874","4":"0.1997525","5":"0","6":"0","7":"8.9990950","8":"19.8713442","9":"KCND2"},{"1":"ion channel activity (GO:0005216)","2":"1/94","3":"0.119514402","4":"0.1997525","5":"0","6":"0","7":"8.2216708","8":"17.4654465","9":"NCALD"},{"1":"calcium channel activity (GO:0005262)","2":"1/94","3":"0.119514402","4":"0.1997525","5":"0","6":"0","7":"8.2216708","8":"17.4654465","9":"FAM155A"},{"1":"metal ion binding (GO:0046872)","2":"2/443","3":"0.119653459","4":"0.1997525","5":"0","6":"0","7":"3.5432200","8":"7.5228071","9":"NRXN1;ASTN2"},{"1":"core promoter binding (GO:0001047)","2":"1/98","3":"0.124282117","4":"0.1997525","5":"0","6":"0","7":"7.8810468","8":"16.4335679","9":"RORA"},{"1":"voltage-gated cation channel activity (GO:0022843)","2":"1/101","3":"0.127841579","4":"0.1997525","5":"0","6":"0","7":"7.6434615","8":"15.7223210","9":"KCND2"},{"1":"potassium channel activity (GO:0005267)","2":"1/101","3":"0.127841579","4":"0.1997525","5":"0","6":"0","7":"7.6434615","8":"15.7223210","9":"KCND2"},{"1":"phosphatase activity (GO:0016791)","2":"1/105","3":"0.132565862","4":"0.2008574","5":"0","6":"0","7":"7.3480030","8":"14.8479309","9":"PLPPR1"},{"1":"sequence-specific double-stranded DNA binding (GO:1990837)","2":"1/113","3":"0.141940608","4":"0.2087362","5":"0","6":"0","7":"6.8203984","8":"13.3157813","9":"RORA"},{"1":"phosphoprotein phosphatase activity (GO:0004721)","2":"1/137","3":"0.169483129","4":"0.2421188","5":"0","6":"0","7":"5.6100113","8":"9.9577807","9":"PTPRT"},{"1":"microtubule binding (GO:0008017)","2":"1/196","3":"0.233620902","4":"0.3241479","5":"0","6":"0","7":"3.9009862","8":"5.6722506","9":"NAV3"},{"1":"RNA polymerase II regulatory region DNA binding (GO:0001012)","2":"1/202","3":"0.239869459","4":"0.3241479","5":"0","6":"0","7":"3.7833907","8":"5.4013972","9":"RORA"},{"1":"actin binding (GO:0003779)","2":"1/255","3":"0.292975224","4":"0.3768488","5":"0","6":"0","7":"2.9859176","8":"3.6657132","9":"NCALD"},{"1":"ATP binding (GO:0005524)","2":"1/256","3":"0.293942032","4":"0.3768488","5":"0","6":"0","7":"2.9740573","8":"3.6413546","9":"SYN2"},{"1":"adenyl ribonucleotide binding (GO:0032559)","2":"1/280","3":"0.316767183","4":"0.3823632","5":"0","6":"0","7":"2.7149159","8":"3.1210353","9":"SYN2"},{"1":"transcriptional activator activity, RNA polymerase II transcription regulatory region sequence-specific binding (GO:0001228)","2":"1/285","3":"0.321432161","4":"0.3823632","5":"0","6":"0","7":"2.6664410","8":"3.0263272","9":"RORA"},{"1":"transcription coactivator activity (GO:0003713)","2":"1/292","3":"0.327911631","4":"0.3823632","5":"0","6":"0","7":"2.6013746","8":"2.9005616","9":"RORA"},{"1":"transcription regulatory region sequence-specific DNA binding (GO:0000976)","2":"1/293","3":"0.328832392","4":"0.3823632","5":"0","6":"0","7":"2.5923340","8":"2.8832123","9":"RORA"},{"1":"cadherin binding (GO:0045296)","2":"1/314","3":"0.347890117","4":"0.3953297","5":"0","6":"0","7":"2.4158270","8":"2.5507959","9":"PTPRT"},{"1":"transcription regulatory region DNA binding (GO:0044212)","2":"1/375","3":"0.400339089","4":"0.4448212","5":"0","6":"0","7":"2.0155286","8":"1.8451023","9":"RORA"},{"1":"sequence-specific DNA binding (GO:0043565)","2":"1/395","3":"0.416633180","4":"0.4449350","5":"0","6":"0","7":"1.9112651","8":"1.6734065","9":"RORA"},{"1":"purine ribonucleoside triphosphate binding (GO:0035639)","2":"1/397","3":"0.418238936","4":"0.4449350","5":"0","6":"0","7":"1.9014180","8":"1.6574706","9":"SYN2"},{"1":"RNA polymerase II regulatory region sequence-specific DNA binding (GO:0000977)","2":"1/461","3":"0.467433746","4":"0.4869102","5":"0","6":"0","7":"1.6315217","8":"1.2407685","9":"RORA"},{"1":"protein homodimerization activity (GO:0042803)","2":"1/665","3":"0.598928924","4":"0.6111520","5":"0","6":"0","7":"1.1184546","8":"0.5733336","9":"DPYD"},{"1":"DNA binding (GO:0003677)","2":"1/894","3":"0.709317820","4":"0.7093178","5":"0","6":"0","7":"0.8217762","8":"0.2822403","9":"RORA"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
<div id="go_cellular_component_2018-1"
class="section level4 unnumbered">
<h4 class="unnumbered">GO_Cellular_Component_2018</h4>
<pre class="r"><code>enrichr_type[[2]]$GO_Cellular_Component_2018</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["Term"],"name":[1],"type":["chr"],"align":["left"]},{"label":["Overlap"],"name":[2],"type":["chr"],"align":["left"]},{"label":["P.value"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["Adjusted.P.value"],"name":[4],"type":["dbl"],"align":["right"]},{"label":["Old.P.value"],"name":[5],"type":["int"],"align":["right"]},{"label":["Old.Adjusted.P.value"],"name":[6],"type":["int"],"align":["right"]},{"label":["Odds.Ratio"],"name":[7],"type":["dbl"],"align":["right"]},{"label":["Combined.Score"],"name":[8],"type":["dbl"],"align":["right"]},{"label":["Genes"],"name":[9],"type":["chr"],"align":["left"]}],"data":[{"1":"dystrophin-associated glycoprotein complex (GO:0016010)","2":"2/14","3":"0.0001581178","4":"0.002529884","5":"0","6":"0","7":"133.073333","8":"1164.6805074","9":"SGCD;SNTG1"},{"1":"integral component of plasma membrane (GO:0005887)","2":"7/1464","3":"0.0026802670","4":"0.021442136","5":"0","6":"0","7":"4.447907","8":"26.3397865","9":"PLPPR1;KCND2;NRXN1;PCDH15;NRXN3;GPC5;ROBO1"},{"1":"clathrin vesicle coat (GO:0030125)","2":"1/6","3":"0.0080736269","4":"0.043003491","5":"0","6":"0","7":"153.600000","8":"740.2218192","9":"NCALD"},{"1":"clathrin coat of trans-Golgi network vesicle (GO:0030130)","2":"1/8","3":"0.0107508729","4":"0.043003491","5":"0","6":"0","7":"109.703297","8":"497.2596291","9":"NCALD"},{"1":"microtubule end (GO:1990752)","2":"1/22","3":"0.0292977495","4":"0.091034976","5":"0","6":"0","7":"36.542125","8":"129.0026369","9":"NAV3"},{"1":"dendrite (GO:0030425)","2":"2/216","3":"0.0341381158","4":"0.091034976","5":"0","6":"0","7":"7.386542","8":"24.9468695","9":"KCND2;BRINP3"},{"1":"ruffle membrane (GO:0032587)","2":"1/54","3":"0.0704417902","4":"0.161009806","5":"0","6":"0","7":"14.455733","8":"38.3506053","9":"SNTG1"},{"1":"nuclear transcription factor complex (GO:0044798)","2":"1/72","3":"0.0928418711","4":"0.185683742","5":"0","6":"0","7":"10.781148","8":"25.6252540","9":"RORA"},{"1":"lysosomal lumen (GO:0043202)","2":"1/87","3":"0.1111106909","4":"0.197530117","5":"0","6":"0","7":"8.894007","8":"19.5421647","9":"GPC5"},{"1":"Golgi lumen (GO:0005796)","2":"1/99","3":"0.1254701557","4":"0.200752249","5":"0","6":"0","7":"7.800235","8":"16.1908501","9":"GPC5"},{"1":"RNA polymerase II transcription factor complex (GO:0090575)","2":"1/148","3":"0.1818203787","4":"0.263023029","5":"0","6":"0","7":"5.187336","8":"8.8430393","9":"RORA"},{"1":"vacuolar lumen (GO:0005775)","2":"1/162","3":"0.1972672719","4":"0.263023029","5":"0","6":"0","7":"4.732919","8":"7.6824545","9":"GPC5"},{"1":"microtubule (GO:0005874)","2":"1/211","3":"0.2491503941","4":"0.306646639","5":"0","6":"0","7":"3.619597","8":"5.0301489","9":"NAV3"},{"1":"lysosome (GO:0005764)","2":"1/423","3":"0.4387303371","4":"0.501406100","5":"0","6":"0","7":"1.781899","8":"1.4680540","9":"GPC5"},{"1":"cytoskeleton (GO:0005856)","2":"1/521","3":"0.5098980414","4":"0.543891244","5":"0","6":"0","7":"1.438831","8":"0.9691169","9":"SNTG1"},{"1":"nucleolus (GO:0005730)","2":"1/677","3":"0.6056002269","4":"0.605600227","5":"0","6":"0","7":"1.097918","8":"0.5506443","9":"NRXN1"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
<div id="go_biological_process_2018-1"
class="section level4 unnumbered">
<h4 class="unnumbered">GO_Biological_Process_2018</h4>
<pre class="r"><code>enrichr_type[[2]]$GO_Biological_Process_2018</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["Term"],"name":[1],"type":["chr"],"align":["left"]},{"label":["Overlap"],"name":[2],"type":["chr"],"align":["left"]},{"label":["P.value"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["Adjusted.P.value"],"name":[4],"type":["dbl"],"align":["right"]},{"label":["Old.P.value"],"name":[5],"type":["int"],"align":["right"]},{"label":["Old.Adjusted.P.value"],"name":[6],"type":["int"],"align":["right"]},{"label":["Odds.Ratio"],"name":[7],"type":["dbl"],"align":["right"]},{"label":["Combined.Score"],"name":[8],"type":["dbl"],"align":["right"]},{"label":["Genes"],"name":[9],"type":["chr"],"align":["left"]}],"data":[{"1":"axon guidance (GO:0007411)","2":"4/159","3":"5.852771e-05","4":"0.01048848","5":"0","6":"0","7":"22.2361851","8":"216.7140893","9":"DCC;NRXN1;NRXN3;ROBO1"},{"1":"neuron cell-cell adhesion (GO:0007158)","2":"2/14","3":"1.581178e-04","4":"0.01048848","5":"0","6":"0","7":"133.0733333","8":"1164.6805074","9":"NRXN1;NRXN3"},{"1":"neuromuscular process controlling balance (GO:0050885)","2":"2/14","3":"1.581178e-04","4":"0.01048848","5":"0","6":"0","7":"133.0733333","8":"1164.6805074","9":"NRXN1;PCDH15"},{"1":"axonogenesis (GO:0007409)","2":"4/224","3":"2.195775e-04","4":"0.01092398","5":"0","6":"0","7":"15.6150198","8":"131.5378875","9":"DCC;NRXN1;NRXN3;ROBO1"},{"1":"nervous system development (GO:0007399)","2":"5/456","3":"3.214075e-04","4":"0.01279202","5":"0","6":"0","7":"9.8377343","8":"79.1229375","9":"NAV3;CNTN6;NRXN1;PCDH15;ROBO1"},{"1":"muscle cell differentiation (GO:0042692)","2":"2/37","3":"1.135283e-03","4":"0.03765356","5":"0","6":"0","7":"45.5725714","8":"309.0218194","9":"SGCD;RORA"},{"1":"cellular response to decreased oxygen levels (GO:0036294)","2":"2/65","3":"3.464102e-03","4":"0.06990445","5":"0","6":"0","7":"25.2825397","8":"143.2332212","9":"KCND2;RORA"},{"1":"homophilic cell adhesion via plasma membrane adhesion molecules (GO:0007156)","2":"2/65","3":"3.464102e-03","4":"0.06990445","5":"0","6":"0","7":"25.2825397","8":"143.2332212","9":"PTPRT;ROBO1"},{"1":"anterograde trans-synaptic signaling (GO:0098916)","2":"3/241","3":"4.082049e-03","4":"0.06990445","5":"0","6":"0","7":"10.3650210","8":"57.0196002","9":"KCND2;NRXN1;SYN2"},{"1":"chemical synaptic transmission (GO:0007268)","2":"3/290","3":"6.821695e-03","4":"0.06990445","5":"0","6":"0","7":"8.5740418","8":"42.7642968","9":"KCND2;NRXN1;SYN2"},{"1":"negative regulation of cell motility (GO:2000146)","2":"2/98","3":"7.702147e-03","4":"0.06990445","5":"0","6":"0","7":"16.5641667","8":"80.6054781","9":"NAV3;ROBO1"},{"1":"vocal learning (GO:0042297)","2":"1/6","3":"8.073627e-03","4":"0.06990445","5":"0","6":"0","7":"153.6000000","8":"740.2218192","9":"NRXN1"},{"1":"regulation of chemokine-mediated signaling pathway (GO:0070099)","2":"1/6","3":"8.073627e-03","4":"0.06990445","5":"0","6":"0","7":"153.6000000","8":"740.2218192","9":"ROBO1"},{"1":"imitative learning (GO:0098596)","2":"1/6","3":"8.073627e-03","4":"0.06990445","5":"0","6":"0","7":"153.6000000","8":"740.2218192","9":"NRXN1"},{"1":"equilibrioception (GO:0050957)","2":"1/6","3":"8.073627e-03","4":"0.06990445","5":"0","6":"0","7":"153.6000000","8":"740.2218192","9":"PCDH15"},{"1":"regulation of collateral sprouting (GO:0048670)","2":"1/7","3":"9.413121e-03","4":"0.06990445","5":"0","6":"0","7":"127.9935897","8":"597.1733894","9":"DCC"},{"1":"negative regulation of dendrite development (GO:2000171)","2":"1/7","3":"9.413121e-03","4":"0.06990445","5":"0","6":"0","7":"127.9935897","8":"597.1733894","9":"DCC"},{"1":"regulation of transcription involved in cell fate commitment (GO:0060850)","2":"1/7","3":"9.413121e-03","4":"0.06990445","5":"0","6":"0","7":"127.9935897","8":"597.1733894","9":"RORA"},{"1":"postsynaptic density assembly (GO:0097107)","2":"1/7","3":"9.413121e-03","4":"0.06990445","5":"0","6":"0","7":"127.9935897","8":"597.1733894","9":"NRXN1"},{"1":"calcium ion import across plasma membrane (GO:0098703)","2":"1/8","3":"1.075087e-02","4":"0.06990445","5":"0","6":"0","7":"109.7032967","8":"497.2596291","9":"FAM155A"},{"1":"calcium ion import into cytosol (GO:1902656)","2":"1/8","3":"1.075087e-02","4":"0.06990445","5":"0","6":"0","7":"109.7032967","8":"497.2596291","9":"FAM155A"},{"1":"T-helper 17 cell differentiation (GO:0072539)","2":"1/8","3":"1.075087e-02","4":"0.06990445","5":"0","6":"0","7":"109.7032967","8":"497.2596291","9":"RORA"},{"1":"pyrimidine nucleobase metabolic process (GO:0006206)","2":"1/8","3":"1.075087e-02","4":"0.06990445","5":"0","6":"0","7":"109.7032967","8":"497.2596291","9":"DPYD"},{"1":"negative regulation of cell migration (GO:0030336)","2":"2/122","3":"1.172556e-02","4":"0.06990445","5":"0","6":"0","7":"13.2353333","8":"58.8440813","9":"NAV3;ROBO1"},{"1":"postsynaptic density organization (GO:0097106)","2":"1/9","3":"1.208689e-02","4":"0.06990445","5":"0","6":"0","7":"95.9855769","8":"423.8371982","9":"NRXN1"},{"1":"positive regulation of synapse maturation (GO:0090129)","2":"1/9","3":"1.208689e-02","4":"0.06990445","5":"0","6":"0","7":"95.9855769","8":"423.8371982","9":"NRXN1"},{"1":"postsynaptic membrane assembly (GO:0097104)","2":"1/9","3":"1.208689e-02","4":"0.06990445","5":"0","6":"0","7":"95.9855769","8":"423.8371982","9":"NRXN1"},{"1":"dephosphorylation (GO:0016311)","2":"2/125","3":"1.228126e-02","4":"0.06990445","5":"0","6":"0","7":"12.9105691","8":"56.8023811","9":"PTPRT;PLPPR1"},{"1":"regulation of synapse maturation (GO:0090128)","2":"1/10","3":"1.342116e-02","4":"0.06990445","5":"0","6":"0","7":"85.3162393","8":"367.7917020","9":"NRXN1"},{"1":"protein localization to synapse (GO:0035418)","2":"1/10","3":"1.342116e-02","4":"0.06990445","5":"0","6":"0","7":"85.3162393","8":"367.7917020","9":"NRXN1"},{"1":"regulation of mammary gland epithelial cell proliferation (GO:0033599)","2":"1/10","3":"1.342116e-02","4":"0.06990445","5":"0","6":"0","7":"85.3162393","8":"367.7917020","9":"ROBO1"},{"1":"positive regulation of circadian rhythm (GO:0042753)","2":"1/10","3":"1.342116e-02","4":"0.06990445","5":"0","6":"0","7":"85.3162393","8":"367.7917020","9":"RORA"},{"1":"cellular response to hypoxia (GO:0071456)","2":"2/133","3":"1.381865e-02","4":"0.06990445","5":"0","6":"0","7":"12.1172519","8":"51.8828790","9":"KCND2;RORA"},{"1":"regulation of vascular endothelial growth factor signaling pathway (GO:1900746)","2":"1/11","3":"1.475370e-02","4":"0.06990445","5":"0","6":"0","7":"76.7807692","8":"323.7277777","9":"ROBO1"},{"1":"T-helper cell differentiation (GO:0042093)","2":"1/11","3":"1.475370e-02","4":"0.06990445","5":"0","6":"0","7":"76.7807692","8":"323.7277777","9":"RORA"},{"1":"pyrimidine nucleoside catabolic process (GO:0046135)","2":"1/11","3":"1.475370e-02","4":"0.06990445","5":"0","6":"0","7":"76.7807692","8":"323.7277777","9":"DPYD"},{"1":"heart process (GO:0003015)","2":"1/11","3":"1.475370e-02","4":"0.06990445","5":"0","6":"0","7":"76.7807692","8":"323.7277777","9":"SGCD"},{"1":"negative regulation of chemotaxis (GO:0050922)","2":"1/11","3":"1.475370e-02","4":"0.06990445","5":"0","6":"0","7":"76.7807692","8":"323.7277777","9":"ROBO1"},{"1":"cellular response to sterol (GO:0036315)","2":"1/11","3":"1.475370e-02","4":"0.06990445","5":"0","6":"0","7":"76.7807692","8":"323.7277777","9":"RORA"},{"1":"postsynapse assembly (GO:0099068)","2":"1/11","3":"1.475370e-02","4":"0.06990445","5":"0","6":"0","7":"76.7807692","8":"323.7277777","9":"NRXN1"},{"1":"response to sterol (GO:0036314)","2":"1/11","3":"1.475370e-02","4":"0.06990445","5":"0","6":"0","7":"76.7807692","8":"323.7277777","9":"RORA"},{"1":"purine nucleobase metabolic process (GO:0006144)","2":"1/11","3":"1.475370e-02","4":"0.06990445","5":"0","6":"0","7":"76.7807692","8":"323.7277777","9":"DPYD"},{"1":"cell-cell adhesion via plasma-membrane adhesion molecules (GO:0098742)","2":"2/143","3":"1.585176e-02","4":"0.07112928","5":"0","6":"0","7":"11.2521986","8":"46.6344548","9":"PTPRT;ROBO1"},{"1":"regulation of cholesterol homeostasis (GO:2000188)","2":"1/12","3":"1.608451e-02","4":"0.07112928","5":"0","6":"0","7":"69.7972028","8":"288.2553657","9":"RORA"},{"1":"nucleoside catabolic process (GO:0009164)","2":"1/12","3":"1.608451e-02","4":"0.07112928","5":"0","6":"0","7":"69.7972028","8":"288.2553657","9":"DPYD"},{"1":"muscle cell development (GO:0055001)","2":"1/13","3":"1.741359e-02","4":"0.07372987","5":"0","6":"0","7":"63.9775641","8":"259.1414089","9":"SGCD"},{"1":"regulation of cellular carbohydrate metabolic process (GO:0010675)","2":"1/13","3":"1.741359e-02","4":"0.07372987","5":"0","6":"0","7":"63.9775641","8":"259.1414089","9":"RORA"},{"1":"pyrimidine-containing compound catabolic process (GO:0072529)","2":"1/14","3":"1.874094e-02","4":"0.07458893","5":"0","6":"0","7":"59.0532544","8":"234.8574522","9":"DPYD"},{"1":"cell migration involved in sprouting angiogenesis (GO:0002042)","2":"1/14","3":"1.874094e-02","4":"0.07458893","5":"0","6":"0","7":"59.0532544","8":"234.8574522","9":"ROBO1"},{"1":"purine-containing compound catabolic process (GO:0072523)","2":"1/14","3":"1.874094e-02","4":"0.07458893","5":"0","6":"0","7":"59.0532544","8":"234.8574522","9":"DPYD"},{"1":"negative regulation of interleukin-2 production (GO:0032703)","2":"1/15","3":"2.006656e-02","4":"0.07679318","5":"0","6":"0","7":"54.8324176","8":"214.3235039","9":"NAV3"},{"1":"pyrimidine nucleoside metabolic process (GO:0006213)","2":"1/15","3":"2.006656e-02","4":"0.07679318","5":"0","6":"0","7":"54.8324176","8":"214.3235039","9":"DPYD"},{"1":"postsynaptic membrane organization (GO:0001941)","2":"1/16","3":"2.139046e-02","4":"0.07739456","5":"0","6":"0","7":"51.1743590","8":"196.7557093","9":"NRXN1"},{"1":"regulation of steroid metabolic process (GO:0019218)","2":"1/16","3":"2.139046e-02","4":"0.07739456","5":"0","6":"0","7":"51.1743590","8":"196.7557093","9":"RORA"},{"1":"neuromuscular process (GO:0050905)","2":"1/16","3":"2.139046e-02","4":"0.07739456","5":"0","6":"0","7":"51.1743590","8":"196.7557093","9":"NRXN1"},{"1":"negative regulation of microtubule depolymerization (GO:0007026)","2":"1/18","3":"2.403309e-02","4":"0.08143960","5":"0","6":"0","7":"45.1493213","8":"168.3312883","9":"NAV3"},{"1":"blood vessel endothelial cell migration (GO:0043534)","2":"1/18","3":"2.403309e-02","4":"0.08143960","5":"0","6":"0","7":"45.1493213","8":"168.3312883","9":"ROBO1"},{"1":"positive regulation of synaptic transmission, glutamatergic (GO:0051968)","2":"1/18","3":"2.403309e-02","4":"0.08143960","5":"0","6":"0","7":"45.1493213","8":"168.3312883","9":"NRXN1"},{"1":"cellular response to lipid (GO:0071396)","2":"2/179","3":"2.414541e-02","4":"0.08143960","5":"0","6":"0","7":"8.9473446","8":"33.3168780","9":"RORA;BRINP3"},{"1":"positive regulation of excitatory postsynaptic potential (GO:2000463)","2":"1/19","3":"2.535182e-02","4":"0.08270513","5":"0","6":"0","7":"42.6388889","8":"156.6938488","9":"NRXN1"},{"1":"extrinsic apoptotic signaling pathway in absence of ligand (GO:0097192)","2":"1/19","3":"2.535182e-02","4":"0.08270513","5":"0","6":"0","7":"42.6388889","8":"156.6938488","9":"DCC"},{"1":"negative regulation of protein depolymerization (GO:1901880)","2":"1/20","3":"2.666885e-02","4":"0.08559839","5":"0","6":"0","7":"40.3927126","8":"146.3936612","9":"NAV3"},{"1":"regulation of interleukin-2 production (GO:0032663)","2":"1/21","3":"2.798415e-02","4":"0.08805168","5":"0","6":"0","7":"38.3711538","8":"137.2197311","9":"NAV3"},{"1":"regulation of microtubule depolymerization (GO:0031114)","2":"1/22","3":"2.929775e-02","4":"0.08805168","5":"0","6":"0","7":"36.5421245","8":"129.0026369","9":"NAV3"},{"1":"acylglycerol homeostasis (GO:0055090)","2":"1/22","3":"2.929775e-02","4":"0.08805168","5":"0","6":"0","7":"36.5421245","8":"129.0026369","9":"RORA"},{"1":"negative regulation of axonogenesis (GO:0050771)","2":"1/23","3":"3.060964e-02","4":"0.08805168","5":"0","6":"0","7":"34.8793706","8":"121.6048472","9":"DCC"},{"1":"membrane assembly (GO:0071709)","2":"1/23","3":"3.060964e-02","4":"0.08805168","5":"0","6":"0","7":"34.8793706","8":"121.6048472","9":"NRXN1"},{"1":"positive regulation of signal transduction (GO:0009967)","2":"2/207","3":"3.157882e-02","4":"0.08805168","5":"0","6":"0","7":"7.7143415","8":"26.6551232","9":"NRXN1;ROBO1"},{"1":"phospholipid dephosphorylation (GO:0046839)","2":"1/24","3":"3.191982e-02","4":"0.08805168","5":"0","6":"0","7":"33.3612040","8":"114.9136106","9":"PLPPR1"},{"1":"positive regulation of microtubule polymerization (GO:0031116)","2":"1/25","3":"3.322829e-02","4":"0.08805168","5":"0","6":"0","7":"31.9695513","8":"108.8356587","9":"NAV3"},{"1":"receptor clustering (GO:0043113)","2":"1/25","3":"3.322829e-02","4":"0.08805168","5":"0","6":"0","7":"31.9695513","8":"108.8356587","9":"NRXN1"},{"1":"positive regulation of microtubule polymerization or depolymerization (GO:0031112)","2":"1/25","3":"3.322829e-02","4":"0.08805168","5":"0","6":"0","7":"31.9695513","8":"108.8356587","9":"NAV3"},{"1":"negative regulation of developmental growth (GO:0048640)","2":"1/25","3":"3.322829e-02","4":"0.08805168","5":"0","6":"0","7":"31.9695513","8":"108.8356587","9":"DCC"},{"1":"positive regulation of vascular endothelial growth factor production (GO:0010575)","2":"1/25","3":"3.322829e-02","4":"0.08805168","5":"0","6":"0","7":"31.9695513","8":"108.8356587","9":"RORA"},{"1":"regulation of glucose metabolic process (GO:0010906)","2":"1/26","3":"3.453506e-02","4":"0.08805168","5":"0","6":"0","7":"30.6892308","8":"103.2932053","9":"RORA"},{"1":"inorganic cation import across plasma membrane (GO:0098659)","2":"1/26","3":"3.453506e-02","4":"0.08805168","5":"0","6":"0","7":"30.6892308","8":"103.2932053","9":"FAM155A"},{"1":"positive regulation of Rho protein signal transduction (GO:0035025)","2":"1/26","3":"3.453506e-02","4":"0.08805168","5":"0","6":"0","7":"30.6892308","8":"103.2932053","9":"ROBO1"},{"1":"modulation of excitatory postsynaptic potential (GO:0098815)","2":"1/26","3":"3.453506e-02","4":"0.08805168","5":"0","6":"0","7":"30.6892308","8":"103.2932053","9":"NRXN1"},{"1":"positive regulation of synapse assembly (GO:0051965)","2":"1/27","3":"3.584013e-02","4":"0.08805168","5":"0","6":"0","7":"29.5073964","8":"98.2208874","9":"NRXN1"},{"1":"lipid modification (GO:0030258)","2":"1/27","3":"3.584013e-02","4":"0.08805168","5":"0","6":"0","7":"29.5073964","8":"98.2208874","9":"PLPPR1"},{"1":"negative chemotaxis (GO:0050919)","2":"1/27","3":"3.584013e-02","4":"0.08805168","5":"0","6":"0","7":"29.5073964","8":"98.2208874","9":"ROBO1"},{"1":"regulation of vascular endothelial growth factor production (GO:0010574)","2":"1/28","3":"3.714350e-02","4":"0.08905491","5":"0","6":"0","7":"28.4131054","8":"93.5634015","9":"RORA"},{"1":"negative regulation of microtubule polymerization or depolymerization (GO:0031111)","2":"1/28","3":"3.714350e-02","4":"0.08905491","5":"0","6":"0","7":"28.4131054","8":"93.5634015","9":"NAV3"},{"1":"triglyceride homeostasis (GO:0070328)","2":"1/29","3":"3.844518e-02","4":"0.09107846","5":"0","6":"0","7":"27.3969780","8":"89.2736555","9":"RORA"},{"1":"striated muscle tissue development (GO:0014706)","2":"1/30","3":"3.974516e-02","4":"0.09305043","5":"0","6":"0","7":"26.4509284","8":"85.3113129","9":"SGCD"},{"1":"aminoglycan metabolic process (GO:0006022)","2":"1/31","3":"4.104345e-02","4":"0.09497263","5":"0","6":"0","7":"25.5679487","8":"81.6416334","9":"GPC5"},{"1":"aminoglycan catabolic process (GO:0006026)","2":"1/33","3":"4.363495e-02","4":"0.09867450","5":"0","6":"0","7":"23.9675481","8":"75.0638864","9":"GPC5"},{"1":"negative regulation of fat cell differentiation (GO:0045599)","2":"1/33","3":"4.363495e-02","4":"0.09867450","5":"0","6":"0","7":"23.9675481","8":"75.0638864","9":"RORA"},{"1":"regulation of microtubule polymerization (GO:0031113)","2":"1/36","3":"4.750957e-02","4":"0.10441669","5":"0","6":"0","7":"21.9098901","8":"66.7555801","9":"NAV3"},{"1":"regulation of synapse assembly (GO:0051963)","2":"1/36","3":"4.750957e-02","4":"0.10441669","5":"0","6":"0","7":"21.9098901","8":"66.7555801","9":"NRXN1"},{"1":"calcium ion transmembrane import into cytosol (GO:0097553)","2":"1/37","3":"4.879775e-02","4":"0.10441669","5":"0","6":"0","7":"21.3002137","8":"64.3281588","9":"FAM155A"},{"1":"cardiac muscle tissue development (GO:0048738)","2":"1/37","3":"4.879775e-02","4":"0.10441669","5":"0","6":"0","7":"21.3002137","8":"64.3281588","9":"SGCD"},{"1":"regulation of smoothened signaling pathway (GO:0008589)","2":"1/37","3":"4.879775e-02","4":"0.10441669","5":"0","6":"0","7":"21.3002137","8":"64.3281588","9":"RORA"},{"1":"positive regulation of axonogenesis (GO:0050772)","2":"1/38","3":"5.008425e-02","4":"0.10491333","5":"0","6":"0","7":"20.7234927","8":"62.0471460","9":"ROBO1"},{"1":"regulation of dendrite development (GO:0050773)","2":"1/38","3":"5.008425e-02","4":"0.10491333","5":"0","6":"0","7":"20.7234927","8":"62.0471460","9":"DCC"},{"1":"cellular response to oxygen-containing compound (GO:1901701)","2":"2/275","3":"5.277583e-02","4":"0.10939990","5":"0","6":"0","7":"5.7728938","8":"16.9821330","9":"RORA;BRINP3"},{"1":"negative regulation of cytokine-mediated signaling pathway (GO:0001960)","2":"1/41","3":"5.393371e-02","4":"0.10951844","5":"0","6":"0","7":"19.1663462","8":"55.9657243","9":"ROBO1"},{"1":"positive regulation of Ras protein signal transduction (GO:0046579)","2":"1/41","3":"5.393371e-02","4":"0.10951844","5":"0","6":"0","7":"19.1663462","8":"55.9657243","9":"ROBO1"},{"1":"negative regulation of cell projection organization (GO:0031345)","2":"1/42","3":"5.521352e-02","4":"0.11098474","5":"0","6":"0","7":"18.6979362","8":"54.1594605","9":"DCC"},{"1":"cellular response to retinoic acid (GO:0071300)","2":"1/43","3":"5.649166e-02","4":"0.11241840","5":"0","6":"0","7":"18.2518315","8":"52.4495999","9":"BRINP3"},{"1":"heart contraction (GO:0060047)","2":"1/45","3":"5.904295e-02","4":"0.11519164","5":"0","6":"0","7":"17.4204545","8":"49.2910039","9":"SGCD"},{"1":"sprouting angiogenesis (GO:0002040)","2":"1/45","3":"5.904295e-02","4":"0.11519164","5":"0","6":"0","7":"17.4204545","8":"49.2910039","9":"ROBO1"},{"1":"regulation of synaptic transmission, glutamatergic (GO:0051966)","2":"1/46","3":"6.031610e-02","4":"0.11653306","5":"0","6":"0","7":"17.0324786","8":"47.8298593","9":"NRXN1"},{"1":"aminoglycan biosynthetic process (GO:0006023)","2":"1/47","3":"6.158760e-02","4":"0.11784550","5":"0","6":"0","7":"16.6613712","8":"46.4401522","9":"GPC5"},{"1":"negative regulation of I-kappaB kinase/NF-kappaB signaling (GO:0043124)","2":"1/48","3":"6.285744e-02","4":"0.11912981","5":"0","6":"0","7":"16.3060556","8":"45.1169972","9":"RORA"},{"1":"negative regulation of mitotic cell cycle (GO:0045930)","2":"1/49","3":"6.412562e-02","4":"0.12038678","5":"0","6":"0","7":"15.9655449","8":"43.8559354","9":"BRINP3"},{"1":"blood circulation (GO:0008015)","2":"1/51","3":"6.665703e-02","4":"0.12396962","5":"0","6":"0","7":"15.3253846","8":"41.5041258","9":"SGCD"},{"1":"regulation of cell migration (GO:0030334)","2":"2/317","3":"6.779695e-02","4":"0.12400122","5":"0","6":"0","7":"4.9925079","8":"13.4360271","9":"NAV3;ROBO1"},{"1":"negative regulation of neuron projection development (GO:0010977)","2":"1/52","3":"6.792026e-02","4":"0.12400122","5":"0","6":"0","7":"15.0241327","8":"40.4062158","9":"DCC"},{"1":"positive regulation of nervous system development (GO:0051962)","2":"1/53","3":"6.918185e-02","4":"0.12515625","5":"0","6":"0","7":"14.7344675","8":"39.3560093","9":"NRXN1"},{"1":"action potential (GO:0001508)","2":"1/55","3":"7.170009e-02","4":"0.12626830","5":"0","6":"0","7":"14.1873219","8":"37.3873288","9":"KCND2"},{"1":"sensory perception (GO:0007600)","2":"1/55","3":"7.170009e-02","4":"0.12626830","5":"0","6":"0","7":"14.1873219","8":"37.3873288","9":"PCDH15"},{"1":"negative regulation of supramolecular fiber organization (GO:1902904)","2":"1/55","3":"7.170009e-02","4":"0.12626830","5":"0","6":"0","7":"14.1873219","8":"37.3873288","9":"NAV3"},{"1":"regulation of axonogenesis (GO:0050770)","2":"1/56","3":"7.295675e-02","4":"0.12735432","5":"0","6":"0","7":"13.9286713","8":"36.4637088","9":"ROBO1"},{"1":"glycosaminoglycan catabolic process (GO:0006027)","2":"1/57","3":"7.421177e-02","4":"0.12835526","5":"0","6":"0","7":"13.6792582","8":"35.5774602","9":"GPC5"},{"1":"response to retinoic acid (GO:0032526)","2":"1/58","3":"7.546515e-02","4":"0.12835526","5":"0","6":"0","7":"13.4385965","8":"34.7264663","9":"BRINP3"},{"1":"regulation of Rho protein signal transduction (GO:0035023)","2":"1/58","3":"7.546515e-02","4":"0.12835526","5":"0","6":"0","7":"13.4385965","8":"34.7264663","9":"ROBO1"},{"1":"positive regulation of cell morphogenesis involved in differentiation (GO:0010770)","2":"1/59","3":"7.671690e-02","4":"0.12937850","5":"0","6":"0","7":"13.2062334","8":"33.9087640","9":"ROBO1"},{"1":"muscle organ development (GO:0007517)","2":"1/61","3":"7.921551e-02","4":"0.13246963","5":"0","6":"0","7":"12.7647436","8":"32.3660696","9":"SGCD"},{"1":"glycosaminoglycan metabolic process (GO:0030203)","2":"1/62","3":"8.046237e-02","4":"0.13343342","5":"0","6":"0","7":"12.5548550","8":"31.6378041","9":"GPC5"},{"1":"diterpenoid metabolic process (GO:0016101)","2":"1/63","3":"8.170760e-02","4":"0.13403557","5":"0","6":"0","7":"12.3517370","8":"30.9362625","9":"GPC5"},{"1":"synapse assembly (GO:0007416)","2":"1/64","3":"8.295121e-02","4":"0.13403557","5":"0","6":"0","7":"12.1550672","8":"30.2600723","9":"NRXN1"},{"1":"positive regulation of cell projection organization (GO:0031346)","2":"1/64","3":"8.295121e-02","4":"0.13403557","5":"0","6":"0","7":"12.1550672","8":"30.2600723","9":"NAV3"},{"1":"positive regulation of protein polymerization (GO:0032273)","2":"1/65","3":"8.419320e-02","4":"0.13403557","5":"0","6":"0","7":"11.9645433","8":"29.6079512","9":"NAV3"},{"1":"negative regulation of neuron differentiation (GO:0045665)","2":"1/65","3":"8.419320e-02","4":"0.13403557","5":"0","6":"0","7":"11.9645433","8":"29.6079512","9":"DCC"},{"1":"negative regulation of epithelial cell proliferation (GO:0050680)","2":"1/67","3":"8.667232e-02","4":"0.13667172","5":"0","6":"0","7":"11.6008159","8":"28.3711961","9":"ROBO1"},{"1":"extrinsic apoptotic signaling pathway (GO:0097191)","2":"1/68","3":"8.790945e-02","4":"0.13667172","5":"0","6":"0","7":"11.4270953","8":"27.7843877","9":"DCC"},{"1":"positive regulation of synaptic transmission (GO:0050806)","2":"1/68","3":"8.790945e-02","4":"0.13667172","5":"0","6":"0","7":"11.4270953","8":"27.7843877","9":"NRXN1"},{"1":"positive regulation of neurogenesis (GO:0050769)","2":"1/70","3":"9.037888e-02","4":"0.13916507","5":"0","6":"0","7":"11.0947603","8":"26.6689709","9":"BRINP3"},{"1":"phospholipid metabolic process (GO:0006644)","2":"1/71","3":"9.161118e-02","4":"0.13916507","5":"0","6":"0","7":"10.9357143","8":"26.1385658","9":"PLPPR1"},{"1":"negative regulation of developmental process (GO:0051093)","2":"1/71","3":"9.161118e-02","4":"0.13916507","5":"0","6":"0","7":"10.9357143","8":"26.1385658","9":"ROBO1"},{"1":"regulation of neuron death (GO:1901214)","2":"1/73","3":"9.407096e-02","4":"0.13970239","5":"0","6":"0","7":"10.6308761","8":"25.1282647","9":"DCC"},{"1":"negative regulation of cell cycle (GO:0045786)","2":"1/73","3":"9.407096e-02","4":"0.13970239","5":"0","6":"0","7":"10.6308761","8":"25.1282647","9":"BRINP3"},{"1":"regulation of fat cell differentiation (GO:0045598)","2":"1/73","3":"9.407096e-02","4":"0.13970239","5":"0","6":"0","7":"10.6308761","8":"25.1282647","9":"RORA"},{"1":"C-terminal protein lipidation (GO:0006501)","2":"1/76","3":"9.774860e-02","4":"0.14302920","5":"0","6":"0","7":"10.2041026","8":"23.7281751","9":"NTM"},{"1":"retinoid metabolic process (GO:0001523)","2":"1/76","3":"9.774860e-02","4":"0.14302920","5":"0","6":"0","7":"10.2041026","8":"23.7281751","9":"GPC5"},{"1":"sensory perception of mechanical stimulus (GO:0050954)","2":"1/79","3":"1.014119e-01","4":"0.14623885","5":"0","6":"0","7":"9.8101578","8":"22.4511853","9":"PCDH15"},{"1":"negative regulation of inflammatory response (GO:0050728)","2":"1/79","3":"1.014119e-01","4":"0.14623885","5":"0","6":"0","7":"9.8101578","8":"22.4511853","9":"RORA"},{"1":"activation of cysteine-type endopeptidase activity involved in apoptotic process (GO:0006919)","2":"1/80","3":"1.026298e-01","4":"0.14693039","5":"0","6":"0","7":"9.6854917","8":"22.0502542","9":"ROBO1"},{"1":"negative regulation of defense response (GO:0031348)","2":"1/81","3":"1.038461e-01","4":"0.14723310","5":"0","6":"0","7":"9.5639423","8":"21.6608512","9":"RORA"},{"1":"sensory perception of sound (GO:0007605)","2":"1/82","3":"1.050608e-01","4":"0.14723310","5":"0","6":"0","7":"9.4453941","8":"21.2825128","9":"PCDH15"},{"1":"sensory perception of light stimulus (GO:0050953)","2":"1/82","3":"1.050608e-01","4":"0.14723310","5":"0","6":"0","7":"9.4453941","8":"21.2825128","9":"PCDH15"},{"1":"C-terminal protein amino acid modification (GO:0018410)","2":"1/83","3":"1.062739e-01","4":"0.14789172","5":"0","6":"0","7":"9.3297373","8":"20.9147997","9":"NTM"},{"1":"regulation of neuron differentiation (GO:0045664)","2":"1/85","3":"1.086955e-01","4":"0.14917517","5":"0","6":"0","7":"9.1066850","8":"20.2096016","9":"BRINP3"},{"1":"positive regulation of supramolecular fiber organization (GO:1902905)","2":"1/85","3":"1.086955e-01","4":"0.14917517","5":"0","6":"0","7":"9.1066850","8":"20.2096016","9":"NAV3"},{"1":"positive regulation of cellular component biogenesis (GO:0044089)","2":"1/88","3":"1.123159e-01","4":"0.15204674","5":"0","6":"0","7":"8.7913351","8":"19.2217228","9":"NRXN1"},{"1":"regulation of MAP kinase activity (GO:0043405)","2":"1/88","3":"1.123159e-01","4":"0.15204674","5":"0","6":"0","7":"8.7913351","8":"19.2217228","9":"ROBO1"},{"1":"cellular response to acid chemical (GO:0071229)","2":"1/92","3":"1.171212e-01","4":"0.15748051","5":"0","6":"0","7":"8.4032122","8":"18.0210763","9":"BRINP3"},{"1":"axon development (GO:0061564)","2":"1/95","3":"1.207087e-01","4":"0.16121493","5":"0","6":"0","7":"8.1337971","8":"17.1978999","9":"DCC"},{"1":"regulation of cell death (GO:0010941)","2":"1/97","3":"1.230925e-01","4":"0.16271146","5":"0","6":"0","7":"7.9635417","8":"16.6821782","9":"DCC"},{"1":"cell morphogenesis involved in neuron differentiation (GO:0048667)","2":"1/98","3":"1.242821e-01","4":"0.16271146","5":"0","6":"0","7":"7.8810468","8":"16.4335679","9":"DCC"},{"1":"positive regulation of neuron projection development (GO:0010976)","2":"1/98","3":"1.242821e-01","4":"0.16271146","5":"0","6":"0","7":"7.8810468","8":"16.4335679","9":"ROBO1"},{"1":"positive regulation of neuron differentiation (GO:0045666)","2":"1/99","3":"1.254702e-01","4":"0.16319321","5":"0","6":"0","7":"7.8002355","8":"16.1908501","9":"BRINP3"},{"1":"glycosaminoglycan biosynthetic process (GO:0006024)","2":"1/100","3":"1.266566e-01","4":"0.16366670","5":"0","6":"0","7":"7.7210567","8":"15.9538300","9":"GPC5"},{"1":"regulation of lipid metabolic process (GO:0019216)","2":"1/101","3":"1.278416e-01","4":"0.16413209","5":"0","6":"0","7":"7.6434615","8":"15.7223210","9":"RORA"},{"1":"positive regulation of cellular component organization (GO:0051130)","2":"1/107","3":"1.349188e-01","4":"0.17210790","5":"0","6":"0","7":"7.2086357","8":"14.4394921","9":"NRXN1"},{"1":"negative regulation of response to external stimulus (GO:0032102)","2":"1/108","3":"1.360929e-01","4":"0.17249991","5":"0","6":"0","7":"7.1409058","8":"14.2419482","9":"RORA"},{"1":"positive regulation of cysteine-type endopeptidase activity involved in apoptotic process (GO:0043280)","2":"1/110","3":"1.384366e-01","4":"0.17435998","5":"0","6":"0","7":"7.0091743","8":"13.8595421","9":"ROBO1"},{"1":"negative regulation of cytokine production (GO:0001818)","2":"1/111","3":"1.396061e-01","4":"0.17472714","5":"0","6":"0","7":"6.9451049","8":"13.6744276","9":"NAV3"},{"1":"negative regulation of cell growth (GO:0030308)","2":"1/116","3":"1.454309e-01","4":"0.18087971","5":"0","6":"0","7":"6.6414716","8":"12.8051163","9":"DCC"},{"1":"positive regulation of nucleic acid-templated transcription (GO:1903508)","2":"2/503","3":"1.470307e-01","4":"0.18173355","5":"0","6":"0","7":"3.1093014","8":"5.9608856","9":"RORA;NPAS3"},{"1":"positive regulation of protein serine/threonine kinase activity (GO:0071902)","2":"1/119","3":"1.489076e-01","4":"0.18291731","5":"0","6":"0","7":"6.4716428","8":"12.3247877","9":"ROBO1"},{"1":"regulation of neuron projection development (GO:0010975)","2":"1/120","3":"1.500634e-01","4":"0.18320625","5":"0","6":"0","7":"6.4169360","8":"12.1709853","9":"DCC"},{"1":"enzyme linked receptor protein signaling pathway (GO:0007167)","2":"1/121","3":"1.512178e-01","4":"0.18348983","5":"0","6":"0","7":"6.3631410","8":"12.0201924","9":"PTPRT"},{"1":"protein dephosphorylation (GO:0006470)","2":"1/126","3":"1.569668e-01","4":"0.18817109","5":"0","6":"0","7":"6.1070769","8":"11.3086009","9":"PTPRT"},{"1":"negative regulation of multicellular organismal process (GO:0051241)","2":"1/126","3":"1.569668e-01","4":"0.18817109","5":"0","6":"0","7":"6.1070769","8":"11.3086009","9":"ROBO1"},{"1":"protein lipidation (GO:0006497)","2":"1/128","3":"1.592560e-01","4":"0.18977207","5":"0","6":"0","7":"6.0102968","8":"11.0423732","9":"NTM"},{"1":"cellular response to organic cyclic compound (GO:0071407)","2":"1/136","3":"1.683527e-01","4":"0.19941781","5":"0","6":"0","7":"5.6518519","8":"10.0698704","9":"RORA"},{"1":"heart development (GO:0007507)","2":"1/150","3":"1.840445e-01","4":"0.21671509","5":"0","6":"0","7":"5.1171915","8":"8.6612451","9":"SGCD"},{"1":"negative regulation of cell differentiation (GO:0045596)","2":"1/151","3":"1.851543e-01","4":"0.21673949","5":"0","6":"0","7":"5.0828205","8":"8.5725100","9":"RORA"},{"1":"transcription initiation from RNA polymerase II promoter (GO:0006367)","2":"1/161","3":"1.961733e-01","4":"0.22691438","5":"0","6":"0","7":"4.7627404","8":"7.7573459","9":"RORA"},{"1":"protein localization to membrane (GO:0072657)","2":"1/161","3":"1.961733e-01","4":"0.22691438","5":"0","6":"0","7":"4.7627404","8":"7.7573459","9":"NRXN1"},{"1":"negative regulation of intracellular signal transduction (GO:1902532)","2":"1/162","3":"1.972673e-01","4":"0.22691438","5":"0","6":"0","7":"4.7329193","8":"7.6824545","9":"RORA"},{"1":"neuron projection morphogenesis (GO:0048812)","2":"1/164","3":"1.994509e-01","4":"0.22810765","5":"0","6":"0","7":"4.6743747","8":"7.5359667","9":"DCC"},{"1":"regulation of mitotic cell cycle (GO:0007346)","2":"1/166","3":"2.016288e-01","4":"0.22920691","5":"0","6":"0","7":"4.6172494","8":"7.3937249","9":"BRINP3"},{"1":"regulation of inflammatory response (GO:0050727)","2":"1/167","3":"2.027157e-01","4":"0.22920691","5":"0","6":"0","7":"4.5892030","8":"7.3241430","9":"RORA"},{"1":"positive regulation of MAP kinase activity (GO:0043406)","2":"1/178","3":"2.145770e-01","4":"0.24124763","5":"0","6":"0","7":"4.3016080","8":"6.6205466","9":"ROBO1"},{"1":"phosphate-containing compound metabolic process (GO:0006796)","2":"1/184","3":"2.209751e-01","4":"0.24704516","5":"0","6":"0","7":"4.1593106","8":"6.2793337","9":"PLPPR1"},{"1":"DNA-templated transcription, initiation (GO:0006352)","2":"1/187","3":"2.241552e-01","4":"0.24920051","5":"0","6":"0","7":"4.0916046","8":"6.1186529","9":"RORA"},{"1":"regulation of I-kappaB kinase/NF-kappaB signaling (GO:0043122)","2":"1/205","3":"2.429753e-01","4":"0.26862270","5":"0","6":"0","7":"3.7271870","8":"5.2732072","9":"RORA"},{"1":"central nervous system development (GO:0007417)","2":"1/218","3":"2.562934e-01","4":"0.28178110","5":"0","6":"0","7":"3.5015952","8":"4.7671857","9":"CNTN6"},{"1":"positive regulation of cytokine production (GO:0001819)","2":"1/221","3":"2.593346e-01","4":"0.28355812","5":"0","6":"0","7":"3.4533217","8":"4.6607283","9":"RORA"},{"1":"apoptotic process (GO:0006915)","2":"1/232","3":"2.703835e-01","4":"0.29402358","5":"0","6":"0","7":"3.2870463","8":"4.2991738","9":"DCC"},{"1":"regulation of signal transduction (GO:0009966)","2":"1/234","3":"2.723753e-01","4":"0.29457977","5":"0","6":"0","7":"3.2585012","8":"4.2379235","9":"RORA"},{"1":"positive regulation of gene expression (GO:0010628)","2":"2/772","3":"2.800241e-01","4":"0.30121509","5":"0","6":"0","7":"1.9951169","8":"2.5395437","9":"RORA;NPAS3"},{"1":"positive regulation of transcription from RNA polymerase II promoter (GO:0045944)","2":"2/849","3":"3.189710e-01","4":"0.34126469","5":"0","6":"0","7":"1.8064699","8":"2.0641719","9":"RORA;NPAS3"},{"1":"positive regulation of MAPK cascade (GO:0043410)","2":"1/290","3":"3.260665e-01","4":"0.34699051","5":"0","6":"0","7":"2.6196433","8":"2.9357139","9":"ROBO1"},{"1":"transmembrane receptor protein tyrosine kinase signaling pathway (GO:0007169)","2":"1/397","3":"4.182389e-01","4":"0.44271036","5":"0","6":"0","7":"1.9014180","8":"1.6574706","9":"PTPRT"},{"1":"vesicle-mediated transport (GO:0016192)","2":"1/411","3":"4.293607e-01","4":"0.45207815","5":"0","6":"0","7":"1.8351782","8":"1.5515662","9":"NCALD"},{"1":"positive regulation of transcription, DNA-templated (GO:0045893)","2":"2/1121","3":"4.516620e-01","4":"0.47305651","5":"0","6":"0","7":"1.3479178","8":"1.0713536","9":"RORA;NPAS3"},{"1":"cellular response to cytokine stimulus (GO:0071345)","2":"1/457","3":"4.644802e-01","4":"0.48393489","5":"0","6":"0","7":"1.6461707","8":"1.2623435","9":"RORA"},{"1":"transcription from RNA polymerase II promoter (GO:0006366)","2":"1/486","3":"4.855410e-01","4":"0.50324298","5":"0","6":"0","7":"1.5454401","8":"1.1165675","9":"RORA"},{"1":"regulation of nucleic acid-templated transcription (GO:1903506)","2":"1/608","3":"5.657279e-01","4":"0.58331528","5":"0","6":"0","7":"1.2270942","8":"0.6990045","9":"RORA"},{"1":"regulation of cellular macromolecule biosynthetic process (GO:2000112)","2":"1/632","3":"5.800178e-01","4":"0.59310972","5":"0","6":"0","7":"1.1789589","8":"0.6421749","9":"RORA"},{"1":"cytokine-mediated signaling pathway (GO:0019221)","2":"1/634","3":"5.811879e-01","4":"0.59310972","5":"0","6":"0","7":"1.1751124","8":"0.6377113","9":"RORA"},{"1":"regulation of transcription from RNA polymerase II promoter (GO:0006357)","2":"2/1479","3":"6.036413e-01","4":"0.61288070","5":"0","6":"0","7":"1.0018145","8":"0.5056911","9":"RORA;NPAS3"},{"1":"regulation of transcription, DNA-templated (GO:0006355)","2":"2/1599","3":"6.474395e-01","4":"0.65401250","5":"0","6":"0","7":"0.9205260","8":"0.4001802","9":"RORA;NPAS3"},{"1":"cellular protein modification process (GO:0006464)","2":"1/1002","3":"7.505966e-01","4":"0.75438752","5":"0","6":"0","7":"0.7289633","8":"0.2091300","9":"PTPRT"},{"1":"regulation of gene expression (GO:0010468)","2":"1/1038","3":"7.630557e-01","4":"0.76305574","5":"0","6":"0","7":"0.7023218","8":"0.1899248","9":"RORA"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
<div id="kegg_2016-1" class="section level4 unnumbered">
<h4 class="unnumbered">KEGG_2016</h4>
<pre class="r"><code>enrichr_type[[2]]$KEGG_2016</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["Term"],"name":[1],"type":["chr"],"align":["left"]},{"label":["Overlap"],"name":[2],"type":["chr"],"align":["left"]},{"label":["P.value"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["Adjusted.P.value"],"name":[4],"type":["dbl"],"align":["right"]},{"label":["Old.P.value"],"name":[5],"type":["int"],"align":["right"]},{"label":["Old.Adjusted.P.value"],"name":[6],"type":["int"],"align":["right"]},{"label":["Odds.Ratio"],"name":[7],"type":["dbl"],"align":["right"]},{"label":["Combined.Score"],"name":[8],"type":["dbl"],"align":["right"]},{"label":["Genes"],"name":[9],"type":["chr"],"align":["left"]}],"data":[{"1":"Axon guidance Homo sapiens hsa04360","2":"2/127","3":"0.01265807","4":"0.1798941","5":"0","6":"0","7":"12.7027200","8":"55.5040294","9":"DCC;ROBO1"},{"1":"Cell adhesion molecules (CAMs) Homo sapiens hsa04514","2":"2/142","3":"0.01564296","4":"0.1798941","5":"0","6":"0","7":"11.3331429","8":"47.1201936","9":"NRXN1;NRXN3"},{"1":"Pantothenate and CoA biosynthesis Homo sapiens hsa00770","2":"1/18","3":"0.02403309","4":"0.1842537","5":"0","6":"0","7":"45.1493213","8":"168.3312883","9":"DPYD"},{"1":"Circadian rhythm Homo sapiens hsa04710","2":"1/30","3":"0.03974516","4":"0.1880231","5":"0","6":"0","7":"26.4509284","8":"85.3113129","9":"RORA"},{"1":"beta-Alanine metabolism Homo sapiens hsa00410","2":"1/31","3":"0.04104345","4":"0.1880231","5":"0","6":"0","7":"25.5679487","8":"81.6416334","9":"DPYD"},{"1":"Drug metabolism - other enzymes Homo sapiens hsa00983","2":"1/46","3":"0.06031610","4":"0.1880231","5":"0","6":"0","7":"17.0324786","8":"47.8298593","9":"DPYD"},{"1":"Viral myocarditis Homo sapiens hsa05416","2":"1/59","3":"0.07671690","4":"0.1880231","5":"0","6":"0","7":"13.2062334","8":"33.9087640","9":"SGCD"},{"1":"Colorectal cancer Homo sapiens hsa05210","2":"1/62","3":"0.08046237","4":"0.1880231","5":"0","6":"0","7":"12.5548550","8":"31.6378041","9":"DCC"},{"1":"Inflammatory bowel disease (IBD) Homo sapiens hsa05321","2":"1/65","3":"0.08419320","4":"0.1880231","5":"0","6":"0","7":"11.9645433","8":"29.6079512","9":"RORA"},{"1":"Melanoma Homo sapiens hsa05218","2":"1/71","3":"0.09161118","4":"0.1880231","5":"0","6":"0","7":"10.9357143","8":"26.1385658","9":"FGF14"},{"1":"Arrhythmogenic right ventricular cardiomyopathy (ARVC) Homo sapiens hsa05412","2":"1/74","3":"0.09529844","4":"0.1880231","5":"0","6":"0","7":"10.4847208","8":"24.6468718","9":"SGCD"},{"1":"Pathways in cancer Homo sapiens hsa05200","2":"2/397","3":"0.09970768","4":"0.1880231","5":"0","6":"0","7":"3.9651646","8":"9.1417368","9":"FGF14;DCC"},{"1":"Hypertrophic cardiomyopathy (HCM) Homo sapiens hsa05410","2":"1/83","3":"0.10627395","4":"0.1880231","5":"0","6":"0","7":"9.3297373","8":"20.9147997","9":"SGCD"},{"1":"Dilated cardiomyopathy Homo sapiens hsa05414","2":"1/90","3":"0.11472170","4":"0.1884714","5":"0","6":"0","7":"8.5929127","8":"18.6057709","9":"SGCD"},{"1":"Pyrimidine metabolism Homo sapiens hsa00240","2":"1/105","3":"0.13256586","4":"0.2023628","5":"0","6":"0","7":"7.3480030","8":"14.8479309","9":"DPYD"},{"1":"Serotonergic synapse Homo sapiens hsa04726","2":"1/112","3":"0.14077412","4":"0.2023628","5":"0","6":"0","7":"6.8821899","8":"13.4932121","9":"KCND2"},{"1":"Rap1 signaling pathway Homo sapiens hsa04015","2":"1/211","3":"0.24915039","4":"0.3212508","5":"0","6":"0","7":"3.6195971","8":"5.0301489","9":"FGF14"},{"1":"Regulation of actin cytoskeleton Homo sapiens hsa04810","2":"1/214","3":"0.25221972","4":"0.3212508","5":"0","6":"0","7":"3.5680751","8":"4.9148617","9":"FGF14"},{"1":"Ras signaling pathway Homo sapiens hsa04014","2":"1/227","3":"0.26538107","4":"0.3212508","5":"0","6":"0","7":"3.3606195","8":"4.4581591","9":"FGF14"},{"1":"MAPK signaling pathway Homo sapiens hsa04010","2":"1/255","3":"0.29297522","4":"0.3369215","5":"0","6":"0","7":"2.9859176","8":"3.6657132","9":"FGF14"},{"1":"PI3K-Akt signaling pathway Homo sapiens hsa04151","2":"1/341","3":"0.37162833","4":"0.4070215","5":"0","6":"0","7":"2.2209276","8":"2.1984097","9":"FGF14"},{"1":"Olfactory transduction Homo sapiens hsa04740","2":"1/415","3":"0.43250049","4":"0.4521596","5":"0","6":"0","7":"1.8170754","8":"1.5230214","9":"NCALD"},{"1":"Metabolic pathways Homo sapiens hsa01100","2":"1/1239","3":"0.82233552","4":"0.8223355","5":"0","6":"0","7":"0.5820492","8":"0.1138528","9":"DPYD"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
</div>
</div>
<div id="split-matrix-into-constituent-parts-and-save"
class="section level1">
<h1>Split matrix into constituent parts and save</h1>
<pre class="r"><code># save cluster assignments, do not need whole array
cluster_file &lt;- cbind.data.frame(array$Tissue, array$Class, array$Type, array$Type.Marker)
colnames(cluster_file) &lt;- c(&quot;Tissue&quot;, &quot;Class&quot;, &quot;Type&quot;, &quot;Type.Marker&quot;)
write.csv(cluster_file, file = &quot;output/sequencing/cluster_lists/05P-NFOL_type_clusters.csv&quot;, col.names = TRUE)

# save DEGs
for (i in 1:length(names(marker_type))){
  write.csv(marker_type[[i]], 
            file = paste0(&quot;output/sequencing/marker_lists/cell_type/&quot;, names(marker_type)[i], &quot;.csv&quot;))
}

# save the overgroups/tissues
for (i in 1:length(names(marker_type_region_aplCoA))){
    write.csv(marker_type_region_aplCoA[[i]], 
              file = paste0(&quot;output/sequencing/marker_lists/region_in_celltype/&quot;, 
                            names(marker_type_region_aplCoA)[i], &quot;_aplCoA.csv&quot;)) 
}

for (i in 1:length(names(marker_type_region_pplCoA))){
  for (j in 1:length(names(marker_type_region_pplCoA[[i]]))){
    write.csv(marker_type_region_pplCoA[[i]], 
              file = paste0(&quot;output/sequencing/marker_lists/region_in_celltype/&quot;, 
                            names(marker_type_region_pplCoA)[i], &quot;_pplCoA.csv&quot;))
  }
}

# save enrichR, no region info due to lack of DEGs
saveRDS(enrichr_type, file = &quot;output/sequencing/enrichr/NFOL_subclusters_enrichr.rds&quot;)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.1 (2022-06-23)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur ... 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices datasets  utils     methods   base     

other attached packages:
 [1] EnhancedVolcano_1.14.0   ggrepel_0.9.1            enrichR_3.1             
 [4] Libra_1.0.0              colorRamps_2.3.1         heatmap3_1.1.9          
 [7] dendextend_1.16.0        pvclust_2.2-0            ggtree_3.4.4            
[10] speckle_0.99.1           chisq.posthoc.test_0.1.2 clustree_0.5.0          
[13] ggraph_2.1.0             scCustomize_0.7.0        knitr_1.40              
[16] forcats_0.5.2            stringr_1.4.1            dplyr_1.0.10            
[19] purrr_0.3.5              readr_2.1.3              tidyr_1.2.1             
[22] tibble_3.1.8             ggplot2_3.3.6            tidyverse_1.3.2         
[25] sp_1.5-0                 SeuratObject_4.1.2       Seurat_4.2.0            
[28] matrixStats_0.62.0       colorspace_2.0-3         farver_2.1.1            
[31] colorjam_0.0.23.900      jamba_0.0.87.900         viridis_0.6.2           
[34] viridisLite_0.4.1        workflowr_1.7.0         

loaded via a namespace (and not attached):
  [1] ica_1.0-3                   ps_1.7.1                   
  [3] lmtest_0.9-40               rprojroot_2.0.3            
  [5] crayon_1.5.2                spatstat.core_2.4-4        
  [7] MASS_7.3-58.1               dittoSeq_1.9.3             
  [9] nlme_3.1-160                backports_1.4.1            
 [11] reprex_2.0.2                rlang_1.0.6                
 [13] XVector_0.36.0              ROCR_1.0-11                
 [15] readxl_1.4.1                irlba_2.3.5.1              
 [17] nloptr_2.0.3                callr_3.7.2                
 [19] limma_3.52.4                rjson_0.2.21               
 [21] BiocParallel_1.30.4         bit64_4.0.5                
 [23] glue_1.6.2                  pheatmap_1.0.12            
 [25] sctransform_0.3.5           parallel_4.2.1             
 [27] processx_3.7.0              vipor_0.4.5                
 [29] spatstat.sparse_2.1-1       AnnotationDbi_1.58.0       
 [31] BiocGenerics_0.42.0         spatstat.geom_2.4-0        
 [33] haven_2.5.1                 tidyselect_1.2.0           
 [35] SummarizedExperiment_1.26.1 fitdistrplus_1.1-8         
 [37] XML_3.99-0.11               zoo_1.8-11                 
 [39] ggpubr_0.4.0                xtable_1.8-4               
 [41] magrittr_2.0.3              evaluate_0.17              
 [43] cli_3.4.1                   zlibbioc_1.42.0            
 [45] rstudioapi_0.14             miniUI_0.1.1.1             
 [47] whisker_0.4                 bslib_0.4.0                
 [49] rpart_4.1.16                pbmcapply_1.5.1            
 [51] treeio_1.21.2.001           shiny_1.7.2                
 [53] xfun_0.34                   cluster_2.1.4              
 [55] tidygraph_1.2.2             KEGGREST_1.36.3            
 [57] ape_5.6-2                   listenv_0.8.0              
 [59] Biostrings_2.64.1           png_0.1-7                  
 [61] future_1.28.0               withr_2.5.0                
 [63] bitops_1.0-7                ggforce_0.4.1              
 [65] plyr_1.8.7                  cellranger_1.1.0           
 [67] pillar_1.8.1                GlobalOptions_0.1.2        
 [69] cachem_1.0.6                multcomp_1.4-20            
 [71] fs_1.5.2                    paletteer_1.4.1            
 [73] tester_0.1.7                vctrs_0.5.0                
 [75] ellipsis_0.3.2              generics_0.1.3             
 [77] tools_4.2.1                 beeswarm_0.4.0             
 [79] munsell_0.5.0               tweenr_2.0.2               
 [81] emmeans_1.8.1-1             DelayedArray_0.22.0        
 [83] fastmap_1.1.0               compiler_4.2.1             
 [85] abind_1.4-5                 httpuv_1.6.6               
 [87] plotly_4.10.0               rgeos_0.5-9                
 [89] GenomeInfoDbData_1.2.8      gridExtra_2.3              
 [91] glmmTMB_1.1.4               edgeR_3.38.4               
 [93] colorway_0.2.0              lattice_0.20-45            
 [95] deldir_1.0-6                utf8_1.2.2                 
 [97] later_1.3.0                 prismatic_1.1.1            
 [99] jsonlite_1.8.3              scales_1.2.1               
[101] tidytree_0.4.1              pbapply_1.5-0              
[103] carData_3.0-5               estimability_1.4.1         
[105] renv_0.15.4                 genefilter_1.78.0          
[107] lazyeval_0.2.2              promises_1.2.0.1           
[109] car_3.1-0                   goftest_1.2-3              
[111] checkmate_2.1.0             spatstat.utils_2.3-1       
[113] reticulate_1.26             rmarkdown_2.17             
[115] sandwich_3.0-2              cowplot_1.1.1              
[117] blme_1.0-5                  Rtsne_0.16                 
[119] Biobase_2.56.0              uwot_0.1.14                
[121] igraph_1.3.5                survival_3.4-0             
[123] numDeriv_2016.8-1.1         yaml_2.3.6                 
[125] htmltools_0.5.3             memoise_2.0.1              
[127] locfit_1.5-9.6              graphlayouts_0.8.2         
[129] IRanges_2.30.1              digest_0.6.30              
[131] assertthat_0.2.1            mime_0.12                  
[133] RSQLite_2.2.18              yulab.utils_0.0.5          
[135] future.apply_1.9.1          data.table_1.14.4          
[137] blob_1.2.3                  S4Vectors_0.34.0           
[139] labeling_0.4.2              splines_4.2.1              
[141] rematch2_2.1.2              googledrive_2.0.0          
[143] RCurl_1.98-1.9              broom_1.0.1                
[145] hms_1.1.2                   modelr_0.1.9               
[147] BiocManager_1.30.18         ggbeeswarm_0.6.0           
[149] GenomicRanges_1.48.0        shape_1.4.6                
[151] aplot_0.1.8                 ggrastr_1.0.1              
[153] sass_0.4.2                  Rcpp_1.0.9                 
[155] RANN_2.6.1                  mvtnorm_1.1-3              
[157] circlize_0.4.15             fansi_1.0.3                
[159] tzdb_0.3.0                  parallelly_1.32.1          
[161] R6_2.5.1                    grid_4.2.1                 
[163] ggridges_0.5.4              lifecycle_1.0.3            
[165] curl_4.3.3                  ggsignif_0.6.4             
[167] googlesheets4_1.0.1         minqa_1.2.5                
[169] leiden_0.4.3                jquerylib_0.1.4            
[171] snakecase_0.11.0            fastcluster_1.2.3          
[173] Matrix_1.5-1                RcppAnnoy_0.0.19           
[175] TH.data_1.1-1               RColorBrewer_1.1-3         
[177] TMB_1.9.1                   htmlwidgets_1.5.4          
[179] polyclip_1.10-0             gridGraphics_0.5-1         
[181] rvest_1.0.3                 mgcv_1.8-41                
[183] globals_0.16.1              patchwork_1.1.2            
[185] spatstat.random_2.2-0       progressr_0.11.0           
[187] codetools_0.2-18            lubridate_1.8.0            
[189] getPass_0.2-2               SingleCellExperiment_1.18.1
[191] dbplyr_2.2.1                GenomeInfoDb_1.32.4        
[193] gtable_0.3.1                DBI_1.1.3                  
[195] git2r_0.30.1                stats4_4.2.1               
[197] highr_0.9                   ggfun_0.0.7                
[199] tensor_1.5                  httr_1.4.4                 
[201] KernSmooth_2.23-20          stringi_1.7.8              
[203] reshape2_1.4.4              annotate_1.74.0            
[205] xml2_1.3.3                  boot_1.3-28                
[207] lme4_1.1-30                 geneplotter_1.74.0         
[209] ggplotify_0.1.0             scattermore_0.8            
[211] DESeq2_1.36.0               bit_4.0.4                  
[213] MatrixGenerics_1.8.1        spatstat.data_2.2-0        
[215] janitor_2.1.0               pkgconfig_2.0.3            
[217] gargle_1.2.1                lmerTest_3.1-3             
[219] ggprism_1.0.3.9000          rstatix_0.7.0              </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
