<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="James Howe" />


<title>Spatial sequencing: merging</title>

<script src="site_libs/header-attrs-2.25/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="site_libs/pagedtable-1.1/js/pagedtable.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">CorticalAmygdala_pipelines</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jamesrhowe/CorticalAmygdala_pipelines">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Spatial sequencing: merging</h1>
<h4 class="author">James Howe</h4>
<h4 class="date">06 February, 2024</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-02-06
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong>
<code>CorticalAmygdala_pipelines/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R
Markdown file created these results, you’ll want to first commit it to
the Git repo. If you’re still working on the analysis, you can ignore
this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220906code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20220906)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220906code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220906)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomjamesrhoweCorticalAmygdalapipelinestree86c4464b24e834d51f55909bbe7c5b2945c83dcdtargetblank86c4464a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/jamesrhowe/CorticalAmygdala_pipelines/tree/86c4464b24e834d51f55909bbe7c5b2945c83dcd" target="_blank">86c4464</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomjamesrhoweCorticalAmygdalapipelinestree86c4464b24e834d51f55909bbe7c5b2945c83dcdtargetblank86c4464a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/jamesrhowe/CorticalAmygdala_pipelines/tree/86c4464b24e834d51f55909bbe7c5b2945c83dcd" target="_blank">86c4464</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/figure/
    Ignored:    data/behavior/
    Ignored:    data/projections/
    Ignored:    data/rnascope/
    Ignored:    data/sequencing/
    Ignored:    output/.DS_Store
    Ignored:    renv/library/
    Ignored:    renv/staging/

Untracked files:
    Untracked:  .renvignore
    Untracked:  analysis/08A-merge_rd1_ATAC.Rmd
    Untracked:  analysis/behavior_10-projections_silencing_TMT.Rmd
    Untracked:  analysis/behavior_11-projections_silencing_2PE.Rmd
    Untracked:  analysis/projections.Rmd
    Untracked:  analysis/sequencing-02B-initial_classification_rd1_RNA.Rmd
    Untracked:  analysis/sequencing-02C-reprocess_Neuron_rd1_RNA.Rmd
    Untracked:  analysis/sequencing-02D-Neuron_classification_rd1_RNA.Rmd
    Untracked:  analysis/sequencing-02E-reprocess_Glut_rd1_RNA.Rmd
    Untracked:  analysis/sequencing-02F-reprocess_GABA_rd1_RNA.Rmd
    Untracked:  analysis/sequencing-02G-Glut_classification_rd1_RNA.Rmd
    Untracked:  analysis/sequencing-03A-merge_filtered_RNA.Rmd
    Untracked:  analysis/sequencing-03B-initial_classification_RNA.Rmd
    Untracked:  analysis/sequencing-04-initial_deg.Rmd
    Untracked:  analysis/sequencing-04A-reprocess_Neuron_RNA.Rmd
    Untracked:  analysis/sequencing-04B-reprocess_Nonneuron_RNA.Rmd
    Untracked:  analysis/sequencing-04C-Neuron_classification_RNA.Rmd
    Untracked:  analysis/sequencing-04D-Nonneuron_classification_RNA.Rmd
    Untracked:  analysis/sequencing-05C-reprocess_Astro_RNA.Rmd
    Untracked:  analysis/sequencing-05L-Glut_classification_RNA.Rmd
    Untracked:  analysis/sequencing-05M-GABA_classification_RNA.Rmd
    Untracked:  analysis/sequencing-05N-Astro_classification_RNA.Rmd
    Untracked:  analysis/sequencing-05O-Macro_classification_RNA.Rmd
    Untracked:  analysis/sequencing-05P-NFOL_classification_RNA.Rmd
    Untracked:  analysis/sequencing-05Q-MOL_classification_RNA.Rmd
    Untracked:  analysis/sequencing-05R-ABC_classification_RNA.Rmd
    Untracked:  analysis/sequencing-05S-VLMC_classification_RNA.Rmd
    Untracked:  analysis/sequencing-05T-Immune_classification+labeling_RNA.Rmd
    Untracked:  analysis/sequencing-05X-reprocess_Astro.Rmd
    Untracked:  analysis/sequencing-05X-reprocess_GABAergic.Rmd
    Untracked:  analysis/sequencing-05X-reprocess_Glutamatergic.Rmd
    Untracked:  analysis/sequencing-05X-reprocess_Microglia.Rmd
    Untracked:  analysis/sequencing-05X-reprocess_Neuron.Rmd
    Untracked:  analysis/sequencing-05X-reprocess_Nonneuron.Rmd
    Untracked:  analysis/sequencing-05X-reprocess_OLG.Rmd
    Untracked:  analysis/sequencing-05X-reprocess_OPC.Rmd
    Untracked:  analysis/sequencing-05X-reprocess_VLMC.Rmd
    Untracked:  analysis/sequencing-06A-merged_classification_RNA.Rmd
    Untracked:  analysis/sequencing-06B-annotate_GABAergic.Rmd
    Untracked:  analysis/sequencing-06B-merged_deg_discovery_RNA.Rmd
    Untracked:  analysis/sequencing-07A-preprocess_aplCoA-1_P1_ATAC.Rmd
    Untracked:  analysis/sequencing-07B-preprocess_aplCoA-1_P2_ATAC.Rmd
    Untracked:  analysis/sequencing-07C-preprocess_aplCoA-2_ATAC.Rmd
    Untracked:  analysis/sequencing-07D-preprocess_pplCoA-1_P1_ATAC.Rmd
    Untracked:  analysis/sequencing-07E-preprocess_pplCoA-1_P2_ATAC.Rmd
    Untracked:  analysis/sequencing-07F-preprocess_pplCoA-2_ATAC.Rmd
    Untracked:  analysis/spatial-01A-preprocess_039-A_S18.Rmd
    Untracked:  analysis/spatial-01B-preprocess_039-B_S19.Rmd
    Untracked:  analysis/spatial-01C-preprocess_042-A_S22.Rmd
    Untracked:  analysis/spatial-01D-preprocess_061-D_S1.Rmd
    Untracked:  analysis/spatial-01E-preprocess_062-B_S2.Rmd
    Untracked:  analysis/spatial-01F-preprocess_063-A_S4.Rmd
    Untracked:  analysis/spatial-01G-preprocess_063-B_S5.Rmd
    Untracked:  analysis/spatial-01H-preprocess_063-D_S7.Rmd
    Untracked:  analysis/spatial-01I-preprocess_105-A_S8.Rmd
    Untracked:  analysis/spatial-01J-preprocess_105-B_S9.Rmd
    Untracked:  analysis/spatial-01K-preprocess_106-B_S11.Rmd
    Untracked:  analysis/spatial-01L-preprocess_106-C_S12.Rmd
    Untracked:  analysis/spatial-01M-preprocess_107-A_S15.Rmd
    Untracked:  analysis/spatial-01N-preprocess_107-B_S16.Rmd
    Untracked:  analysis/spatial-01O-preprocess_108-A_S13.Rmd
    Untracked:  analysis/spatial-01P-preprocess_122-A_S23.Rmd
    Untracked:  analysis/spatial-01Q-preprocess_122-B_S24.Rmd
    Untracked:  analysis/spatial-01R-preprocess_122-C_S25.Rmd
    Untracked:  analysis/spatial-01S-preprocess_123-A_S30.Rmd
    Untracked:  analysis/spatial-01T-preprocess_063-C_S6.Rmd
    Untracked:  analysis/spatial-01U-preprocess_108-C_S14.Rmd
    Untracked:  analysis/spatial-02-merge.Rmd
    Untracked:  analysis/spatial-03-classify.Rmd
    Untracked:  analysis/spatial-04A_project_Glut.Rmd
    Untracked:  output/sequencing/
    Untracked:  output/spatial/
    Untracked:  renv/settings.json

Unstaged changes:
    Modified:   analysis/sequencing-02A-merge_rd1_RNA.Rmd
    Modified:   analysis/sequencing-03-initial_classification.Rmd
    Modified:   analysis/sequencing-05A-reprocess_Glut_RNA.Rmd
    Modified:   code/sequencing-color_scales.R
    Modified:   code/sequencing_misc.R
    Modified:   renv.lock
    Modified:   renv/.gitignore
    Modified:   renv/activate.R

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with
<code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>This notebook is a basic workflow to go from raw CellRanger ATAC
pipeline output to a pre-processed, Signac-formatted dataset with
low-quality cells and outliers removed. Most filtering steps are
generally found in other papers, as listed in the methods.</p>
<pre class="r"><code># set up paths
input &lt;- params$input_path
output &lt;- params$output_path
id &lt;- params$dataset
path &lt;- paste0(input, id)
output_path &lt;- paste0(output, id)

# import helper scripts
source(&quot;code/sequencing_misc.R&quot;)
source(&quot;code/sequencing-color_scales.R&quot;)
source(&quot;code/sequencing-plot_functions.R&quot;)

# load packages
require(tidyverse)
require(Seurat)
require(SeuratData)
require(SeuratObject)
require(patchwork)
require(hdf5r)
require(Matrix.utils)
require(knitr)
require(future)
require(scCustomize)
require(clustree)

# all images are PDFs, avoids pixelation
knitr::opts_chunk$set(dev = &#39;pdf&#39;)</code></pre>
</div>
<div id="read-in-data-from-step-01" class="section level1">
<h1>Read in data from step 01</h1>
<p>Extracts all of the paths and their dataset names from the outputs of
step 1, then puts them into a list for merging.</p>
<pre class="r"><code>paths &lt;- Sys.glob(paste0(input, &quot;*&quot;))

array &lt;- rep(list(NULL), length(paths))

for (i in 1:length(paths)){
  array[[i]] &lt;- readRDS(paths[i])
  paths[i] &lt;- gsub(&quot;output/spatial/preprocessed/cropped/&quot;,&quot;&quot;, paths[i])
  paths[i] &lt;- gsub(&quot;_cropped_array.rds&quot;,&quot;&quot;, paths[i])
  names(array)[i] &lt;- paths[i]
}

# show datasets used
cat(names(array), sep = &quot;\t&quot;)</code></pre>
<pre><code>039-A_S18   039-B_S19   042-A_S22   061-D_S1    062-B_S2    063-A-S4    063-B_S5    063-C_S6    063-D_S7    105-A_S8    105-B_S9    106-B_S11   106-C_S12   107-A_S15   107-B_S16   108-A_S13   108-C_S14   122-A_S23   122-B_S24   122-C_S25   123-A_S30</code></pre>
</div>
<div id="merge-datasets-into-single-array"
class="section level1 tabset">
<h1 class="tabset">Merge datasets into single array</h1>
<p>This step merges all elements of the list into a single unified
Seurat array, with the region of origin and batch information saved.
Compare metadata across regions and batches.</p>
<pre class="r"><code>array2 &lt;- list()
for (i in 2:length(array)){
    array2[[i-1]] &lt;- array[[i]]
}
array &lt;- merge(x= array[[1]], 
               y = array2, 
               add.cell.ids = names(array))
rm(array2)

number_array &lt;- as.factor(str_sub(colnames(array), start = 1, end = 8))
levels(number_array) &lt;- as.character(c(1:21))
array &lt;- AddMetaData(array, metadata = str_sub(colnames(array), start = 1, end = 8), col.name = &quot;id&quot;)
array &lt;- AddMetaData(array, metadata = number_array, col.name = &quot;id_number&quot;)

plot_merge_array &lt;- cbind.data.frame(array$nCount_Spatial, 
                               array$nFeature_Spatial,
                               array$Mito_proportion,
                               array$Ribo_proportion,
                               array$id, array$id_number)
colnames(plot_merge_array) &lt;- c(&quot;UMIs&quot;, &quot;Genes&quot;, &quot;Mito_proportion&quot;, &quot;Ribo_proportion&quot;, &quot;Slide&quot;, &quot;Section&quot;)

table(plot_merge_array$Section)</code></pre>
<pre><code>
  1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17  18  19  20 
184 143 189 167 176 144 193 162 196 158 217 189 182 190 208 198 151  95 170 141 
 21 
163 </code></pre>
<div id="total-spots" class="section level2">
<h2>Total Spots</h2>
<pre class="r"><code>ggplot(plot_merge_array, aes(Section)) +
       ylab(&quot;Total spots&quot;) +
       geom_bar(aes(fill = Section)) + scale_y_continuous(expand = c(0,0)) +
       scale_fill_manual(values = spatial_colors_section) +
       theme_classic() +  theme(axis.ticks.x = element_blank(), 
                                axis.text.x = element_text(angle = 45, vjust = 1, hjust=1),
                                legend.position=&quot;none&quot;)</code></pre>
<embed src="figure/spatial-02-merge.Rmd/3-total_spots-1.pdf" width="576" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="genesumis-scatter" class="section level2">
<h2>Genes/UMIs Scatter</h2>
<pre class="r"><code>ggplot(plot_merge_array, aes(UMIs, Genes, colour = Section)) +
                 geom_point(alpha = 0.5, size = 0.3) +
                 geom_hline(yintercept = mean(plot_merge_array$Genes), 
                            colour = &#39;darkorchid&#39;, linetype = 2) +
                 geom_vline(xintercept = mean(plot_merge_array$UMIs), 
                            colour = &#39;forestgreen&#39;, linetype = 2) +
                 guides(colour = guide_legend(override.aes = list(size = 5, alpha = 1))) +
                 labs(colour = &#39;Section&#39;) + theme_minimal()</code></pre>
<embed src="figure/spatial-02-merge.Rmd/4-gene_umi_scatter-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="violin" class="section level2 tabset">
<h2 class="tabset">Violin</h2>
<div id="genes" class="section level3">
<h3>Genes</h3>
<pre class="r"><code>VlnPlot(array, features = &#39;nFeature_Spatial&#39;, group.by = &#39;id_number&#39;, pt.size = 0) +
    ggtitle(&#39;&#39;) + xlab(&#39;&#39;) + ylab(&quot;Features (thousands)&quot;) + 
    scale_fill_manual(values = spatial_colors_section) +
    NoLegend()</code></pre>
<embed src="figure/spatial-02-merge.Rmd/5-genes_violin-1.pdf" width="576" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="umis" class="section level3">
<h3>UMIs</h3>
<pre class="r"><code>VlnPlot(array, features = &#39;nCount_Spatial&#39;, group.by = &#39;id_number&#39;, pt.size = 0) +
    ggtitle(&#39;&#39;) + xlab(&#39;&#39;) + ylab(&quot;UMIs (thousands)&quot;) +
    scale_fill_manual(values = spatial_colors_section) +
    NoLegend()</code></pre>
<embed src="figure/spatial-02-merge.Rmd/6-umis_violin-1.pdf" width="576" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="mitochondrial-proportion" class="section level3">
<h3>Mitochondrial proportion</h3>
<pre class="r"><code>VlnPlot(array, features = &#39;Mito_proportion&#39;, group.by = &#39;id_number&#39;, pt.size = 0) +
    ggtitle(&#39;&#39;) + xlab(&#39;&#39;) + ylab(&quot;Percent Mitochondrial UMIs&quot;) +
    scale_fill_manual(values = spatial_colors_section) +
    NoLegend()</code></pre>
<embed src="figure/spatial-02-merge.Rmd/7-mito_violin-1.pdf" width="576" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="ribosomal-proportion" class="section level3">
<h3>Ribosomal proportion</h3>
<pre class="r"><code>VlnPlot(array, features = &#39;Ribo_proportion&#39;, group.by = &#39;id_number&#39;, pt.size = 0) +
    ggtitle(&#39;&#39;) + xlab(&#39;&#39;) + ylab(&quot;Percent Ribosomal UMIs&quot;) +
    scale_fill_manual(values = spatial_colors_section) +
    NoLegend()</code></pre>
<embed src="figure/spatial-02-merge.Rmd/8-ribo_violin-1.pdf" width="576" style="display: block; margin: auto;" type="application/pdf" />
</div>
</div>
</div>
<div id="transform-data" class="section level1">
<h1>Transform data</h1>
<p>Data needs to be transformed and then clustered to create a dataset
that can be subsetted.</p>
<p>SCTransform is the best performing normalization method. It uses a
variance stabilizing transform based on Pearson residuals to eliminate
count depth differences and robustly approximate log-normalization. 5000
genes are selected as highly variable genes to ensure maximum resolution
while removing non-informative features.</p>
<pre class="r"><code># needs conserve.memory = TRUE, or else it crashes R session on laptop
array &lt;- SCTransform(array, variable.features.n = 5000, residual_type = &quot;pearson&quot;, 
                     conserve.memory = TRUE, return.only.var.genes = FALSE, verbose = FALSE,
                     assay = &quot;Spatial&quot;, vst.flavor = &quot;v2&quot;,
                     vars.to.regress = &quot;Mito_proportion&quot;)</code></pre>
</div>
<div id="dimensionality-reduction" class="section level1">
<h1>Dimensionality Reduction</h1>
<div id="principal-component-analysis" class="section level2">
<h2>Principal Component Analysis</h2>
<p>Runs a principal component analysis on the imported array using
Seurat’s RunPCA function, returning results for the first 50 principal
components. We aim to use all 50 top principal components explaining for
downstream clustering. Adding principal components beyond the first 50
has strong diminishing returns and only serves to introduce noise into
the reduction in most cases.</p>
<pre class="r"><code>array &lt;- RunPCA(array, verbose = FALSE)

# create array for downstream plotting, with PCs, total variance, and cumulative variance
plot_var_array &lt;- cbind.data.frame(c(1:50),
                                   (array@reductions$pca@stdev)^2
                                   / array@reductions$pca@misc$total.variance,
                                   cumsum((array@reductions$pca@stdev)^2
                                           / array@reductions$pca@misc$total.variance))
colnames(plot_var_array) &lt;- c(&quot;PC&quot;, &quot;Proportion_variance&quot;, &quot;Cumulative_variance&quot;)

cat(paste0(&quot;First 50 PCs explain &quot;, round(plot_var_array$Cumulative_variance[50]*100, 1), &quot;% of the variance&quot;))</code></pre>
<pre><code>First 50 PCs explain 18.9% of the variance</code></pre>
<div id="variance-explained" class="section level3">
<h3>Variance Explained</h3>
<pre class="r"><code>variance_explained_plot(plot_var_array)</code></pre>
<embed src="figure/spatial-02-merge.Rmd/11-var_explained-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
</div>
<div id="visualization-non-linear-dr-tsne-and-umap"
class="section level2 tabset">
<h2 class="tabset">Visualization (non-linear DR): tSNE and UMAP</h2>
<p>The top two principal components are generally unsuitable for
visualizing single cell -omics data. As a result, non-linear
dimensionality reduction incorporating all relevant top PCs must be
used. This is implemented via tSNE or UMAP. Seurat’s tSNE implementation
is sufficient at default settings. via UMAP, using Seurat’s native
RunUMAP function, running uwot under the hood. The number of epochs has
been increased to 1000 from defaults (because it increases stability),
and the min.dist has been raised to 0.5 (because it makes global
structure more legible).</p>
<pre class="r"><code>array &lt;- RunTSNE(array, verbose = FALSE, dims.use = 1:50)
array &lt;- RunUMAP(array, verbose = FALSE, dims = 1:50, n.epochs = 1000, min.dist = 0.2)</code></pre>
<div id="umis-1" class="section level3 tabset">
<h3 class="tabset">UMIs</h3>
<pre class="r"><code>src_list &lt;- mapply(function(tab_name, dr_type) {
    src &lt;- c(
        &quot;#### {{tab_name}} {.unnumbered}&quot;,
        &quot;```{r umis-{{tab_name}}, message = FALSE, warning = FALSE}&quot;,
        &quot;feature_plot_formatted(array, &#39;{{dr_type}}&#39;, &#39;nCount_Spatial&#39;, c(&#39;lightgrey&#39;, &#39;blue&#39;), NA)&quot;,  
        &quot;```&quot;,
        &quot;&quot;
    )
    knit_expand(text = src)
}, dr_params$name, dr_params$type)
out &lt;- knit_child(text = unlist(src_list), options = list(cache = FALSE))</code></pre>
<div id="pca" class="section level4 unnumbered">
<h4 class="unnumbered">PCA</h4>
<pre class="r"><code>feature_plot_formatted(array, &#39;pca&#39;, &#39;nCount_Spatial&#39;, c(&#39;lightgrey&#39;, &#39;blue&#39;), NA)</code></pre>
<embed src="figure/spatial-02-merge.Rmd/umis-PCA-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="tsne" class="section level4 unnumbered">
<h4 class="unnumbered">tSNE</h4>
<pre class="r"><code>feature_plot_formatted(array, &#39;tsne&#39;, &#39;nCount_Spatial&#39;, c(&#39;lightgrey&#39;, &#39;blue&#39;), NA)</code></pre>
<embed src="figure/spatial-02-merge.Rmd/umis-tSNE-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="umap" class="section level4 unnumbered">
<h4 class="unnumbered">UMAP</h4>
<pre class="r"><code>feature_plot_formatted(array, &#39;umap&#39;, &#39;nCount_Spatial&#39;, c(&#39;lightgrey&#39;, &#39;blue&#39;), NA)</code></pre>
<embed src="figure/spatial-02-merge.Rmd/umis-UMAP-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
</div>
<div id="genes-1" class="section level3 tabset">
<h3 class="tabset">Genes</h3>
<pre class="r"><code>src_list &lt;- mapply(function(tab_name, dr_type) {
    src &lt;- c(
        &quot;#### {{tab_name}} {.unnumbered}&quot;,
        &quot;```{r genes-{{tab_name}}, message = FALSE, warning = FALSE}&quot;,
        &quot;feature_plot_formatted(array, &#39;{{dr_type}}&#39;, &#39;nFeature_Spatial&#39;, c(&#39;lightgrey&#39;, &#39;blue&#39;), NA)&quot;,  
        &quot;```&quot;,
        &quot;&quot;
    )
    knit_expand(text = src)
}, dr_params$name, dr_params$type)
out &lt;- knit_child(text = unlist(src_list), options = list(cache = FALSE))</code></pre>
<div id="pca-1" class="section level4 unnumbered">
<h4 class="unnumbered">PCA</h4>
<pre class="r"><code>feature_plot_formatted(array, &#39;pca&#39;, &#39;nFeature_Spatial&#39;, c(&#39;lightgrey&#39;, &#39;blue&#39;), NA)</code></pre>
<embed src="figure/spatial-02-merge.Rmd/genes-PCA-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="tsne-1" class="section level4 unnumbered">
<h4 class="unnumbered">tSNE</h4>
<pre class="r"><code>feature_plot_formatted(array, &#39;tsne&#39;, &#39;nFeature_Spatial&#39;, c(&#39;lightgrey&#39;, &#39;blue&#39;), NA)</code></pre>
<embed src="figure/spatial-02-merge.Rmd/genes-tSNE-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="umap-1" class="section level4 unnumbered">
<h4 class="unnumbered">UMAP</h4>
<pre class="r"><code>feature_plot_formatted(array, &#39;umap&#39;, &#39;nFeature_Spatial&#39;, c(&#39;lightgrey&#39;, &#39;blue&#39;), NA)</code></pre>
<embed src="figure/spatial-02-merge.Rmd/genes-UMAP-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
</div>
<div id="mito" class="section level3 tabset">
<h3 class="tabset">% Mito</h3>
<pre class="r"><code>src_list &lt;- mapply(function(tab_name, dr_type) {
    src &lt;- c(
        &quot;#### {{tab_name}} {.unnumbered}&quot;,
        &quot;```{r mito-{{tab_name}}, message = FALSE, warning = FALSE}&quot;,
        &quot;feature_plot_formatted(array, &#39;{{dr_type}}&#39;, &#39;Mito_proportion&#39;, c(&#39;lightgrey&#39;, &#39;blue&#39;), NA)&quot;,  
        &quot;```&quot;,
        &quot;&quot;
    )
    knit_expand(text = src)
}, dr_params$name, dr_params$type)
out &lt;- knit_child(text = unlist(src_list), options = list(cache = FALSE))</code></pre>
<div id="pca-2" class="section level4 unnumbered">
<h4 class="unnumbered">PCA</h4>
<pre class="r"><code>feature_plot_formatted(array, &#39;pca&#39;, &#39;Mito_proportion&#39;, c(&#39;lightgrey&#39;, &#39;blue&#39;), NA)</code></pre>
<embed src="figure/spatial-02-merge.Rmd/mito-PCA-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="tsne-2" class="section level4 unnumbered">
<h4 class="unnumbered">tSNE</h4>
<pre class="r"><code>feature_plot_formatted(array, &#39;tsne&#39;, &#39;Mito_proportion&#39;, c(&#39;lightgrey&#39;, &#39;blue&#39;), NA)</code></pre>
<embed src="figure/spatial-02-merge.Rmd/mito-tSNE-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="umap-2" class="section level4 unnumbered">
<h4 class="unnumbered">UMAP</h4>
<pre class="r"><code>feature_plot_formatted(array, &#39;umap&#39;, &#39;Mito_proportion&#39;, c(&#39;lightgrey&#39;, &#39;blue&#39;), NA)</code></pre>
<embed src="figure/spatial-02-merge.Rmd/mito-UMAP-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
</div>
<div id="ribo" class="section level3 tabset">
<h3 class="tabset">% Ribo</h3>
<pre class="r"><code>src_list &lt;- mapply(function(tab_name, dr_type) {
    src &lt;- c(
        &quot;#### {{tab_name}} {.unnumbered}&quot;,
        &quot;```{r ribo-{{tab_name}}, message = FALSE, warning = FALSE}&quot;,
        &quot;feature_plot_formatted(array, &#39;{{dr_type}}&#39;, &#39;Ribo_proportion&#39;, c(&#39;lightgrey&#39;, &#39;blue&#39;), NA)&quot;,  
        &quot;```&quot;,
        &quot;&quot;
    )
    knit_expand(text = src)
}, dr_params$name, dr_params$type)
out &lt;- knit_child(text = unlist(src_list), options = list(cache = FALSE))</code></pre>
<div id="pca-3" class="section level4 unnumbered">
<h4 class="unnumbered">PCA</h4>
<pre class="r"><code>feature_plot_formatted(array, &#39;pca&#39;, &#39;Ribo_proportion&#39;, c(&#39;lightgrey&#39;, &#39;blue&#39;), NA)</code></pre>
<embed src="figure/spatial-02-merge.Rmd/ribo-PCA-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="tsne-3" class="section level4 unnumbered">
<h4 class="unnumbered">tSNE</h4>
<pre class="r"><code>feature_plot_formatted(array, &#39;tsne&#39;, &#39;Ribo_proportion&#39;, c(&#39;lightgrey&#39;, &#39;blue&#39;), NA)</code></pre>
<embed src="figure/spatial-02-merge.Rmd/ribo-tSNE-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="umap-3" class="section level4 unnumbered">
<h4 class="unnumbered">UMAP</h4>
<pre class="r"><code>feature_plot_formatted(array, &#39;umap&#39;, &#39;Ribo_proportion&#39;, c(&#39;lightgrey&#39;, &#39;blue&#39;), NA)</code></pre>
<embed src="figure/spatial-02-merge.Rmd/ribo-UMAP-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
</div>
<div id="batch" class="section level3">
<h3>Batch</h3>
<pre class="r"><code>DimPlot_scCustom(array, reduction = &quot;umap&quot;, group.by = &quot;id&quot;) +
                  xlab(&quot;UMAP 1&quot;) + ylab(&quot;UMAP 2&quot;) +
                  scale_colour_manual(values = spatial_colors_section) +
                  theme_classic() +
                  theme(plot.title = element_blank(), legend.text=element_text(size=8)) + 
                  guides(colour = guide_legend(ncol = 1))</code></pre>
<embed src="figure/spatial-02-merge.Rmd/17-dr_batch-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
<p>Save pre-processed matrix</p>
<p>The output can now be passed to the next step of the pipeline, broad
cell type classification.</p>
<pre class="r"><code>saveRDS(array, file = &quot;output/spatial/merged/merged_array.rds&quot;)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.1 (2023-06-16)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS Sonoma 14.1.2

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: America/Los_Angeles
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices datasets  utils     methods   base     

other attached packages:
 [1] clustree_0.5.1        ggraph_2.1.0          scCustomize_2.0.1    
 [4] future_1.33.1         knitr_1.45            Matrix.utils_0.9.7   
 [7] Matrix_1.6-5          hdf5r_1.3.9           patchwork_1.2.0      
[10] SeuratData_0.2.2.9001 Seurat_5.0.1          SeuratObject_5.0.1   
[13] sp_2.1-3              lubridate_1.9.3       forcats_1.0.0        
[16] stringr_1.5.1         dplyr_1.1.4           purrr_1.0.2          
[19] readr_2.1.5           tidyr_1.3.1           tibble_3.2.1         
[22] ggplot2_3.4.4         tidyverse_2.0.0       matrixStats_1.2.0    
[25] farver_2.1.1          colorspace_2.1-0      colorjam_0.0.27.900  
[28] jamba_0.0.98.900      viridis_0.6.5         viridisLite_0.4.2    
[31] workflowr_1.7.1      

loaded via a namespace (and not attached):
  [1] RcppAnnoy_0.0.22       splines_4.3.1          later_1.3.2           
  [4] prismatic_1.1.1        polyclip_1.10-6        janitor_2.2.0         
  [7] fastDummies_1.7.3      lifecycle_1.0.4        rprojroot_2.0.4       
 [10] globals_0.16.2         processx_3.8.3         lattice_0.22-5        
 [13] MASS_7.3-60.0.1        magrittr_2.0.3         plotly_4.10.4         
 [16] sass_0.4.8             rmarkdown_2.25         jquerylib_0.1.4       
 [19] yaml_2.3.8             httpuv_1.6.14          sctransform_0.4.1     
 [22] spam_2.10-0            spatstat.sparse_3.0-3  reticulate_1.35.0     
 [25] cowplot_1.1.3          pbapply_1.7-2          RColorBrewer_1.1-3    
 [28] abind_1.4-5            Rtsne_0.17             tweenr_2.0.2          
 [31] rappdirs_0.3.3         git2r_0.33.0           circlize_0.4.15       
 [34] ggrepel_0.9.5          irlba_2.3.5.1          listenv_0.9.1         
 [37] spatstat.utils_3.0-4   goftest_1.2-3          RSpectra_0.16-1       
 [40] spatstat.random_3.2-2  fitdistrplus_1.1-11    parallelly_1.36.0     
 [43] leiden_0.4.3.1         codetools_0.2-19       ggforce_0.4.1         
 [46] tidyselect_1.2.0       shape_1.4.6            spatstat.explore_3.2-6
 [49] jsonlite_1.8.8         tidygraph_1.3.1        ellipsis_0.3.2        
 [52] progressr_0.14.0       ggridges_0.5.6         survival_3.5-7        
 [55] tools_4.3.1            ica_1.0-3              Rcpp_1.0.12           
 [58] glue_1.7.0             gridExtra_2.3          xfun_0.41             
 [61] withr_3.0.0            BiocManager_1.30.22    fastmap_1.1.1         
 [64] fansi_1.0.6            callr_3.7.3            digest_0.6.34         
 [67] timechange_0.3.0       R6_2.5.1               mime_0.12             
 [70] ggprism_1.0.4          scattermore_1.2        tensor_1.5            
 [73] spatstat.data_3.0-4    utf8_1.2.4             generics_0.1.3        
 [76] renv_1.0.3             data.table_1.15.0      graphlayouts_1.1.0    
 [79] httr_1.4.7             htmlwidgets_1.6.4      whisker_0.4.1         
 [82] uwot_0.1.16            pkgconfig_2.0.3        gtable_0.3.4          
 [85] lmtest_0.9-40          htmltools_0.5.7        dotCall64_1.1-1       
 [88] scales_1.3.0           png_0.1-8              snakecase_0.11.1      
 [91] rstudioapi_0.15.0      tzdb_0.4.0             reshape2_1.4.4        
 [94] nlme_3.1-164           cachem_1.0.8           zoo_1.8-12            
 [97] GlobalOptions_0.1.2    KernSmooth_2.23-22     parallel_4.3.1        
[100] miniUI_0.1.1.1         vipor_0.4.7            ggrastr_1.0.2         
[103] pillar_1.9.0           grid_4.3.1             vctrs_0.6.5           
[106] RANN_2.6.1             promises_1.2.1         xtable_1.8-4          
[109] cluster_2.1.6          paletteer_1.6.0        beeswarm_0.4.0        
[112] evaluate_0.23          cli_3.6.2              compiler_4.3.1        
[115] rlang_1.1.3            crayon_1.5.2           grr_0.9.5             
[118] future.apply_1.11.1    labeling_0.4.3         rematch2_2.1.2        
[121] ps_1.7.6               getPass_0.2-4          plyr_1.8.9            
[124] fs_1.6.3               ggbeeswarm_0.7.2       stringi_1.8.3         
[127] deldir_2.0-2           munsell_0.5.0          lazyeval_0.2.2        
[130] spatstat.geom_3.2-8    RcppHNSW_0.6.0         hms_1.1.3             
[133] bit64_4.0.5            shiny_1.8.0            highr_0.10            
[136] ROCR_1.0-11            igraph_2.0.1.1         bslib_0.6.1           
[139] bit_4.0.5             </code></pre>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
